{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Exclude the records that SATIVA has highligted, redo the alignment and rebuild the tree."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "from reprophylo import *"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "DEBUG:Cloud:Log file (/root/.picloud/cloud.log) opened\n"
     ]
    }
   ],
   "source": [
    "pj=Project('../fetch_clean_align_tree/target_locus.csv', pickle='amphib_post_SATIVA.pkl', git=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/reprophylo/reprophylo.py:1015: UserWarning: Version control off\n",
      "  warnings.warn('Version control off')\n"
     ]
    }
   ],
   "source": [
    "pj.read_embl_genbank(['../fetch_clean_align_tree/12S_preSATIVA_amphib.gb'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "pj.extract_by_locus()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Set the stage:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "to drop: 0\n",
      "to keep: 0\n",
      "TOTAL: 0\n"
     ]
    }
   ],
   "source": [
    "recs_to_keep = {'rrnS': []}\n",
    "recs_to_drop = {'rrnS': []}\n",
    "\n",
    "print \"to drop: %s\" %len(recs_to_drop['rrnS'])\n",
    "print \"to keep: %s\" %len(recs_to_keep['rrnS'])\n",
    "\n",
    "print \"TOTAL: %s\" %(len(recs_to_drop['rrnS']) + len(recs_to_keep['rrnS']))"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Record all record ids and assign them the status keep."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "for r in pj.records_by_locus['rrnS']:\n",
    "    recs_to_keep['rrnS'].append(\"_\".join(r.id.split(\"_\")[:-1]))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "to drop: 0\n",
      "to keep: 203\n",
      "TOTAL: 203\n"
     ]
    }
   ],
   "source": [
    "print \"to drop: %s\" %len(recs_to_drop['rrnS'])\n",
    "print \"to keep: %s\" %len(recs_to_keep['rrnS'])\n",
    "\n",
    "print \"TOTAL: %s\" %(len(recs_to_drop['rrnS']) + len(recs_to_keep['rrnS']))"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "How many do we want to drop? Parse SATIVA output and extract records ids putatively mislabeled."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "rec_mislabelled = {}\n",
    "sativa=open('../SATIVA/12S.mis', 'r')\n",
    "\n",
    "for line in sativa:\n",
    "    if not line.startswith(\";\"):\n",
    "        rec_mislabelled[line.split(\"\\t\")[0]] = line.split(\"\\t\")[4]\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "5\n"
     ]
    }
   ],
   "source": [
    "print len(rec_mislabelled)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Adjust lists accordingly."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "X95494.1\n",
      "\tadd to droplist\n",
      "\tremove from keeplist\n",
      "JN627424.1\n",
      "\tadd to droplist\n",
      "\tremove from keeplist\n",
      "HQ697272.1\n",
      "\tadd to droplist\n",
      "\tremove from keeplist\n",
      "AB023396.1\n",
      "\tadd to droplist\n",
      "\tremove from keeplist\n",
      "JN627425.1\n",
      "\tadd to droplist\n",
      "\tremove from keeplist\n"
     ]
    }
   ],
   "source": [
    "for r in rec_mislabelled:\n",
    "    print r\n",
    "    if not r in recs_to_drop['rrnS']:\n",
    "        print \"\\tadd to droplist\"\n",
    "        recs_to_drop['rrnS'].append(r)\n",
    "    for i in reversed(range(len(recs_to_keep['rrnS']))):\n",
    "#        print i\n",
    "        if r in recs_to_keep['rrnS'][i]:\n",
    "            print \"\\tremove from keeplist\"\n",
    "            del(recs_to_keep['rrnS'][i])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "to drop: 5\n",
      "to keep: 198\n",
      "TOTAL: 203\n"
     ]
    }
   ],
   "source": [
    "print \"to drop: %s\" %len(recs_to_drop['rrnS'])\n",
    "print \"to keep: %s\" %len(recs_to_keep['rrnS'])\n",
    "\n",
    "print \"TOTAL: %s\" %(len(recs_to_drop['rrnS']) + len(recs_to_keep['rrnS']))"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Write out good records in full Genbank format including all metadata. Adjust to valid species and taxids before doing it.\n",
    "\n",
    "Identify records with Subspecies."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "subspecies: Alytes obstetricans pertinax\n",
      "Pelophylax lessonae  .. add to records\n",
      "Bombina bombina  .. add to records\n",
      "Pelophylax ridibundus  .. add to records\n",
      "Epidalea calamita  .. add to records\n",
      "subspecies: Discoglossus pictus pictus\n",
      "subspecies: Bombina variegata variegata\n",
      "Rana temporaria  .. add to records\n",
      "Xenopus laevis  .. add to records\n",
      "subspecies: Bombina variegata scabra\n",
      "Discoglossus pictus  .. add to records\n",
      "Hyla arborea  .. add to records\n",
      "subspecies: Bombina variegata variegata\n",
      "Rana dalmatina  .. add to records\n",
      "Rana temporaria  .. already covered\n",
      "Pelophylax ridibundus  .. already covered\n",
      "Bombina variegata  .. add to records\n",
      "Rana catesbeiana  .. add to records\n",
      "Xenopus laevis  .. already covered\n",
      "subspecies: Salamandra salamandra alfredschmidti\n",
      "subspecies: Bombina variegata scabra\n",
      "subspecies: Discoglossus pictus auritus\n",
      "Rana dalmatina  .. already covered\n",
      "Rana temporaria  .. already covered\n",
      "Hydromantes genei  .. add to records\n",
      "Pelophylax lessonae  .. already covered\n",
      "subspecies: Discoglossus pictus auritus\n",
      "Xenopus laevis  .. already covered\n",
      "Pelophylax lessonae  .. already covered\n",
      "Rana catesbeiana  .. already covered\n",
      "Rana temporaria  .. already covered\n",
      "Bombina bombina  .. already covered\n",
      "Ichthyosaura alpestris  .. add to records\n",
      "Hyla arborea  .. already covered\n",
      "subspecies: Salamandra salamandra morenica\n",
      "Epidalea calamita  .. already covered\n",
      "subspecies: Bombina variegata kolombatovici\n",
      "Hyla arborea  .. already covered\n",
      "Salamandra salamandra  .. add to records\n",
      "Xenopus laevis  .. already covered\n",
      "Bombina variegata  .. already covered\n",
      "subspecies: Lissotriton vulgaris meridionalis\n",
      "Salamandra salamandra  .. already covered\n",
      "Bombina bombina  .. already covered\n",
      "Ichthyosaura alpestris  .. already covered\n",
      "Triturus carnifex  .. add to records\n",
      "subspecies: Salamandra salamandra bernardezi\n",
      "subspecies: Alytes obstetricans boscai\n",
      "Salamandra salamandra  .. already covered\n",
      "Bufo bufo  .. add to records\n",
      "Rana temporaria  .. already covered\n",
      "Pelophylax ridibundus  .. already covered\n",
      "subspecies: Alytes obstetricans boscai\n",
      "subspecies: Alytes obstetricans almogavarii\n",
      "Rana temporaria  .. already covered\n",
      "Rana catesbeiana  .. already covered\n",
      "subspecies: Discoglossus pictus auritus\n",
      "Rana temporaria  .. already covered\n",
      "Bombina variegata  .. already covered\n",
      "Discoglossus pictus  .. already covered\n",
      "Rana catesbeiana  .. already covered\n",
      "subspecies: Discoglossus pictus auritus\n",
      "subspecies: Salamandra salamandra almanzoris\n",
      "Pelophylax ridibundus  .. already covered\n",
      "subspecies: Salamandra salamandra gigliolii\n",
      "Bombina variegata  .. already covered\n",
      "Pelophylax ridibundus  .. already covered\n",
      "Discoglossus pictus  .. already covered\n",
      "Lissotriton vulgaris  .. add to records\n",
      "Pelophylax ridibundus  .. already covered\n",
      "subspecies: Alytes obstetricans pertinax\n",
      "Discoglossus pictus  .. already covered\n",
      "Rana temporaria  .. already covered\n",
      "Xenopus laevis  .. already covered\n",
      "Lissotriton vulgaris  .. already covered\n",
      "Hyla arborea  .. already covered\n",
      "subspecies: Alytes obstetricans boscai\n",
      "Rana temporaria  .. already covered\n",
      "Pelophylax lessonae  .. already covered\n",
      "subspecies: Alytes obstetricans obstetricans\n",
      "subspecies: Ichthyosaura alpestris inexpectatus\n",
      "Rana temporaria  .. already covered\n",
      "Rana temporaria  .. already covered\n",
      "Rana temporaria  .. already covered\n",
      "subspecies: Salamandra salamandra longirostris\n",
      "Xenopus laevis  .. already covered\n",
      "subspecies: Alytes obstetricans almogavarii\n",
      "subspecies: Salamandra salamandra gallaica\n",
      "Salamandra salamandra  .. already covered\n",
      "subspecies: Salamandra salamandra salamandra\n",
      "subspecies: Discoglossus pictus auritus\n",
      "subspecies: Bombina variegata variegata\n",
      "Discoglossus pictus  .. already covered\n",
      "subspecies: Discoglossus pictus auritus\n",
      "Pelophylax lessonae  .. already covered\n",
      "Bombina bombina  .. already covered\n",
      "Hydromantes genei  .. already covered\n",
      "Lissotriton helveticus  .. add to records\n",
      "subspecies: Bombina variegata variegata\n",
      "Bufo bufo  .. already covered\n",
      "Pelophylax ridibundus  .. already covered\n",
      "Hyla arborea  .. already covered\n",
      "subspecies: Salamandra salamandra alfredschmidti\n",
      "subspecies: Salamandra salamandra morenica\n",
      "Alytes obstetricans  .. add to records\n",
      "Bufo bufo  .. already covered\n",
      "Hyla arborea  .. already covered\n",
      "Pelophylax lessonae  .. already covered\n",
      "Salamandra salamandra  .. already covered\n",
      "subspecies: Salamandra salamandra bernardezi\n",
      "subspecies: Alytes obstetricans obstetricans\n",
      "subspecies: Salamandra salamandra bejarae\n",
      "Rana temporaria  .. already covered\n",
      "subspecies: Discoglossus pictus pictus\n",
      "Rana temporaria  .. already covered\n",
      "Pelophylax ridibundus  .. already covered\n",
      "Rana catesbeiana  .. already covered\n",
      "Rana temporaria  .. already covered\n",
      "Rana temporaria  .. already covered\n",
      "Hyla arborea  .. already covered\n",
      "Discoglossus pictus  .. already covered\n",
      "subspecies: Salamandra salamandra gallaica\n",
      "Bombina variegata  .. already covered\n",
      "subspecies: Xenopus laevis sudanensis\n",
      "subspecies: Salamandra salamandra corsica\n",
      "Discoglossus pictus  .. already covered\n",
      "subspecies: Salamandra salamandra fastuosa\n",
      "Xenopus laevis  .. already covered\n",
      "subspecies: Salamandra salamandra longirostris\n",
      "Hydromantes genei  .. already covered\n",
      "Hyla arborea  .. already covered\n",
      "Hyla arborea  .. already covered\n",
      "subspecies: Discoglossus pictus auritus\n",
      "Hydromantes genei  .. already covered\n",
      "Rana catesbeiana  .. already covered\n",
      "Xenopus laevis  .. already covered\n",
      "Pelophylax lessonae  .. already covered\n",
      "Hyla arborea  .. already covered\n",
      "subspecies: Salamandra salamandra terrestris\n",
      "Epidalea calamita  .. already covered\n",
      "subspecies: Discoglossus pictus auritus\n",
      "Rana temporaria  .. already covered\n",
      "Rana temporaria  .. already covered\n",
      "Rana temporaria  .. already covered\n",
      "Triturus cristatus  .. add to records\n",
      "subspecies: Alytes obstetricans boscai\n",
      "Gallus gallus  .. add to records\n",
      "Gallus gallus  .. already covered\n",
      "Gallus gallus  .. already covered\n",
      "subspecies: Gallus gallus murghi\n",
      "Gallus gallus  .. already covered\n",
      "Gallus gallus  .. already covered\n",
      "Gallus gallus  .. already covered\n",
      "Gallus gallus  .. already covered\n",
      "Gallus gallus  .. already covered\n",
      "Gallus gallus  .. already covered\n",
      "Gallus gallus  .. already covered\n",
      "subspecies: Gallus gallus spadiceus\n",
      "Gallus gallus  .. already covered\n",
      "Gallus gallus  .. already covered\n",
      "Chelydra serpentina  .. add to records\n",
      "Gallus gallus  .. already covered\n",
      "Gallus gallus  .. already covered\n",
      "Gallus gallus  .. already covered\n",
      "Gallus gallus  .. already covered\n",
      "Gallus gallus  .. already covered\n",
      "Cottus gobio  .. add to records\n",
      "Chelydra serpentina  .. already covered\n",
      "Gallus gallus  .. already covered\n",
      "Gallus gallus  .. already covered\n",
      "Gallus gallus  .. already covered\n",
      "Gallus gallus  .. already covered\n",
      "Gallus gallus  .. already covered\n",
      "Chelydra serpentina  .. already covered\n",
      "Gallus gallus  .. already covered\n",
      "Gallus gallus  .. already covered\n",
      "Gallus gallus  .. already covered\n",
      "Gallus gallus  .. already covered\n",
      "Gallus gallus  .. already covered\n",
      "Gallus gallus  .. already covered\n",
      "Chelydra serpentina  .. already covered\n",
      "Gallus gallus  .. already covered\n",
      "Gallus gallus  .. already covered\n",
      "Gallus gallus  .. already covered\n",
      "Gallus gallus  .. already covered\n",
      "Gallus gallus  .. already covered\n",
      "Gallus gallus  .. already covered\n",
      "Gallus gallus  .. already covered\n",
      "Gallus gallus  .. already covered\n",
      "Gallus gallus  .. already covered\n",
      "subspecies: Gallus gallus spadiceus\n",
      "Gallus gallus  .. already covered\n",
      "Gallus gallus  .. already covered\n",
      "Gallus gallus  .. already covered\n",
      "Gallus gallus  .. already covered\n",
      "Gallus gallus  .. already covered\n",
      "subspecies: Gallus gallus murghi\n",
      "subspecies: Gallus gallus jabouillei\n"
     ]
    }
   ],
   "source": [
    "from collections import defaultdict\n",
    "\n",
    "Seqs = []\n",
    "taxon_to_taxid = {}\n",
    "recs_to_adjust = defaultdict(list)\n",
    "taxon_to_recs = defaultdict(list)\n",
    "\n",
    "for r in pj.records:\n",
    "    if r.id in recs_to_keep['rrnS']:\n",
    "        Seqs.append(r)\n",
    "        \n",
    "        source = [f for f in r.features if f.type == 'source'][0]\n",
    "\n",
    "        #screen for subspecies and record taxids of valid species\n",
    "        #if valid species record the taxid\n",
    "        if (len(source.qualifiers['organism'][0].split(\" \")) == 2):\n",
    "            print source.qualifiers['organism'][0],\n",
    "            if 'db_xref' in source.qualifiers:\n",
    "#                print source.qualifiers['db_xref']\n",
    "                for t in source.qualifiers['db_xref']:\n",
    "#                    print t\n",
    "                    if 'taxon' in t:\n",
    "                        if not source.qualifiers['organism'][0] in taxon_to_taxid:\n",
    "                            print \" .. add to records\"\n",
    "                            taxon_to_taxid[source.qualifiers['organism'][0]] = t.split(\":\")[1]\n",
    "                        else:\n",
    "                            print \" .. already covered\"\n",
    "                        taxon_to_recs[source.qualifiers['organism'][0]].append(r.id)\n",
    "        #if the species status is 'sp.' but has a valid taxid\n",
    "        elif (len(source.qualifiers['organism'][0].split(\" \")) > 2 and source.qualifiers['organism'][0].split(\" \")[1] == 'sp.'):\n",
    "            print source.qualifiers['organism'][0],\n",
    "            if 'db_xref' in source.qualifiers:\n",
    "                for t in source.qualifiers['db_xref']:\n",
    "                    if 'taxon' in t:\n",
    "                        if not source.qualifiers['organism'][0] in taxon_to_taxid:\n",
    "                            print \" .. add to records\"\n",
    "                            taxon_to_taxid[source.qualifiers['organism'][0]] = t.split(\":\")[1]\n",
    "                        else:\n",
    "                            print \" .. already covered\" \n",
    "                        taxon_to_recs[source.qualifiers['organism'][0]].append(r.id)\n",
    "        #all other cases we consider dubious, e.g. supspecies\n",
    "        else:\n",
    "            print \"subspecies: %s\" %source.qualifiers['organism'][0]\n",
    "            recs_to_adjust[r.id] = []"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Check if any of the taxa that were considered dubious are not already covered in the list after removing subspecies."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Alytes obstetricans pertinax -> Alytes obstetricans\n",
      "Discoglossus pictus pictus -> Discoglossus pictus\n",
      "Bombina variegata variegata -> Bombina variegata\n",
      "Bombina variegata scabra -> Bombina variegata\n",
      "Bombina variegata variegata -> Bombina variegata\n",
      "Salamandra salamandra alfredschmidti -> Salamandra salamandra\n",
      "Bombina variegata scabra -> Bombina variegata\n",
      "Discoglossus pictus auritus -> Discoglossus pictus\n",
      "Discoglossus pictus auritus -> Discoglossus pictus\n",
      "Salamandra salamandra morenica -> Salamandra salamandra\n",
      "Bombina variegata kolombatovici -> Bombina variegata\n",
      "Lissotriton vulgaris meridionalis -> Lissotriton vulgaris\n",
      "Salamandra salamandra bernardezi -> Salamandra salamandra\n",
      "Alytes obstetricans boscai -> Alytes obstetricans\n",
      "Alytes obstetricans boscai -> Alytes obstetricans\n",
      "Alytes obstetricans almogavarii -> Alytes obstetricans\n",
      "Discoglossus pictus auritus -> Discoglossus pictus\n",
      "Discoglossus pictus auritus -> Discoglossus pictus\n",
      "Salamandra salamandra almanzoris -> Salamandra salamandra\n",
      "Salamandra salamandra gigliolii -> Salamandra salamandra\n",
      "Alytes obstetricans pertinax -> Alytes obstetricans\n",
      "Alytes obstetricans boscai -> Alytes obstetricans\n",
      "Alytes obstetricans obstetricans -> Alytes obstetricans\n",
      "Ichthyosaura alpestris inexpectatus -> Ichthyosaura alpestris\n",
      "Salamandra salamandra longirostris -> Salamandra salamandra\n",
      "Alytes obstetricans almogavarii -> Alytes obstetricans\n",
      "Salamandra salamandra gallaica -> Salamandra salamandra\n",
      "Salamandra salamandra salamandra -> Salamandra salamandra\n",
      "Discoglossus pictus auritus -> Discoglossus pictus\n",
      "Bombina variegata variegata -> Bombina variegata\n",
      "Discoglossus pictus auritus -> Discoglossus pictus\n",
      "Bombina variegata variegata -> Bombina variegata\n",
      "Salamandra salamandra alfredschmidti -> Salamandra salamandra\n",
      "Salamandra salamandra morenica -> Salamandra salamandra\n",
      "Salamandra salamandra bernardezi -> Salamandra salamandra\n",
      "Alytes obstetricans obstetricans -> Alytes obstetricans\n",
      "Salamandra salamandra bejarae -> Salamandra salamandra\n",
      "Discoglossus pictus pictus -> Discoglossus pictus\n",
      "Salamandra salamandra gallaica -> Salamandra salamandra\n",
      "Xenopus laevis sudanensis -> Xenopus laevis\n",
      "Salamandra salamandra corsica -> Salamandra salamandra\n",
      "Salamandra salamandra fastuosa -> Salamandra salamandra\n",
      "Salamandra salamandra longirostris -> Salamandra salamandra\n",
      "Discoglossus pictus auritus -> Discoglossus pictus\n",
      "Salamandra salamandra terrestris -> Salamandra salamandra\n",
      "Discoglossus pictus auritus -> Discoglossus pictus\n",
      "Alytes obstetricans boscai -> Alytes obstetricans\n",
      "Gallus gallus murghi -> Gallus gallus\n",
      "Gallus gallus spadiceus -> Gallus gallus\n",
      "Gallus gallus spadiceus -> Gallus gallus\n",
      "Gallus gallus murghi -> Gallus gallus\n",
      "Gallus gallus jabouillei -> Gallus gallus\n"
     ]
    }
   ],
   "source": [
    "from collections import defaultdict\n",
    "\n",
    "to_fetch = defaultdict(list)\n",
    "\n",
    "for r in Seqs:\n",
    "    if r.id in recs_to_adjust:\n",
    "        source = [f for f in r.features if f.type == 'source'][0]\n",
    "        adjust_from = source.qualifiers['organism'][0]\n",
    "        adjust_to = \" \".join(adjust_from.split(\" \")[:2])\n",
    "        print \"%s -> %s\" %(adjust_from,adjust_to)\n",
    "        if adjust_to in taxon_to_taxid:\n",
    "            taxon_to_recs[adjust_to].append(r.id)\n",
    "            recs_to_adjust[r.id].append(adjust_to)\n",
    "            recs_to_adjust[r.id].append(taxon_to_taxid[adjust_to])\n",
    "        else:\n",
    "            to_fetch[adjust_to].append(r.id)\n",
    "            recs_to_adjust[r.id].append(adjust_to)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "0\n"
     ]
    }
   ],
   "source": [
    "print len(to_fetch)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "adjusting AY585337.1\n",
      "Alytes obstetricans pertinax\t->\tAlytes obstetricans\n",
      "adjusting AY347451.1\n",
      "Discoglossus pictus pictus\t->\tDiscoglossus pictus\n",
      "adjusting JX893174.1\n",
      "Bombina variegata variegata\t->\tBombina variegata\n",
      "adjusting JX893176.1\n",
      "Bombina variegata scabra\t->\tBombina variegata\n",
      "adjusting JX893175.1\n",
      "Bombina variegata variegata\t->\tBombina variegata\n",
      "adjusting KF645375.1\n",
      "Salamandra salamandra alfredschmidti\t->\tSalamandra salamandra\n",
      "adjusting JX893178.1\n",
      "Bombina variegata scabra\t->\tBombina variegata\n",
      "adjusting AY347456.1\n",
      "Discoglossus pictus auritus\t->\tDiscoglossus pictus\n",
      "adjusting AY347452.1\n",
      "Discoglossus pictus auritus\t->\tDiscoglossus pictus\n",
      "adjusting AY222458.1\n",
      "Salamandra salamandra morenica\t->\tSalamandra salamandra\n",
      "adjusting AY333653.1\n",
      "Bombina variegata kolombatovici\t->\tBombina variegata\n",
      "adjusting AY147254.1\n",
      "Lissotriton vulgaris meridionalis\t->\tLissotriton vulgaris\n",
      "adjusting DQ092285.1\n",
      "Salamandra salamandra bernardezi\t->\tSalamandra salamandra\n",
      "adjusting KJ858774.1\n",
      "Alytes obstetricans boscai\t->\tAlytes obstetricans\n",
      "adjusting AY333669.1\n",
      "Alytes obstetricans boscai\t->\tAlytes obstetricans\n",
      "adjusting KJ858776.1\n",
      "Alytes obstetricans almogavarii\t->\tAlytes obstetricans\n",
      "adjusting AY347457.1\n",
      "Discoglossus pictus auritus\t->\tDiscoglossus pictus\n",
      "adjusting AY347458.1\n",
      "Discoglossus pictus auritus\t->\tDiscoglossus pictus\n",
      "adjusting DQ221219.1\n",
      "Salamandra salamandra almanzoris\t->\tSalamandra salamandra\n",
      "adjusting KF645371.1\n",
      "Salamandra salamandra gigliolii\t->\tSalamandra salamandra\n",
      "adjusting KJ858770.1\n",
      "Alytes obstetricans pertinax\t->\tAlytes obstetricans\n",
      "adjusting KJ858773.1\n",
      "Alytes obstetricans boscai\t->\tAlytes obstetricans\n",
      "adjusting KJ858772.1\n",
      "Alytes obstetricans obstetricans\t->\tAlytes obstetricans\n",
      "adjusting DQ092298.1\n",
      "Ichthyosaura alpestris inexpectatus\t->\tIchthyosaura alpestris\n",
      "adjusting KF645367.1\n",
      "Salamandra salamandra longirostris\t->\tSalamandra salamandra\n",
      "adjusting KJ858778.1\n",
      "Alytes obstetricans almogavarii\t->\tAlytes obstetricans\n",
      "adjusting DQ221215.1\n",
      "Salamandra salamandra gallaica\t->\tSalamandra salamandra\n",
      "adjusting KF645365.1\n",
      "Salamandra salamandra salamandra\t->\tSalamandra salamandra\n",
      "adjusting AY347455.1\n",
      "Discoglossus pictus auritus\t->\tDiscoglossus pictus\n",
      "adjusting JX893177.1\n",
      "Bombina variegata variegata\t->\tBombina variegata\n",
      "adjusting AY347454.1\n",
      "Discoglossus pictus auritus\t->\tDiscoglossus pictus\n",
      "adjusting AY333664.1\n",
      "Bombina variegata variegata\t->\tBombina variegata\n",
      "adjusting KF645373.1\n",
      "Salamandra salamandra alfredschmidti\t->\tSalamandra salamandra\n",
      "adjusting KF645370.1\n",
      "Salamandra salamandra morenica\t->\tSalamandra salamandra\n",
      "adjusting DQ221220.1\n",
      "Salamandra salamandra bernardezi\t->\tSalamandra salamandra\n",
      "adjusting KJ858771.1\n",
      "Alytes obstetricans obstetricans\t->\tAlytes obstetricans\n",
      "adjusting DQ221216.1\n",
      "Salamandra salamandra bejarae\t->\tSalamandra salamandra\n",
      "adjusting AY347450.1\n",
      "Discoglossus pictus pictus\t->\tDiscoglossus pictus\n",
      "adjusting KF645366.1\n",
      "Salamandra salamandra gallaica\t->\tSalamandra salamandra\n",
      "adjusting AY581640.1\n",
      "Xenopus laevis sudanensis\t->\tXenopus laevis\n",
      "adjusting DQ221225.1\n",
      "Salamandra salamandra corsica\t->\tSalamandra salamandra\n",
      "adjusting KF645364.1\n",
      "Salamandra salamandra fastuosa\t->\tSalamandra salamandra\n",
      "adjusting DQ221223.1\n",
      "Salamandra salamandra longirostris\t->\tSalamandra salamandra\n",
      "adjusting AY347453.1\n",
      "Discoglossus pictus auritus\t->\tDiscoglossus pictus\n",
      "adjusting AY222459.1\n",
      "Salamandra salamandra terrestris\t->\tSalamandra salamandra\n",
      "adjusting AY347459.1\n",
      "Discoglossus pictus auritus\t->\tDiscoglossus pictus\n",
      "adjusting AF224726.1\n",
      "Alytes obstetricans boscai\t->\tAlytes obstetricans\n",
      "adjusting DQ885561.1\n",
      "Gallus gallus murghi\t->\tGallus gallus\n",
      "adjusting GU261693.1\n",
      "Gallus gallus spadiceus\t->\tGallus gallus\n",
      "adjusting AP003321.1\n",
      "Gallus gallus spadiceus\t->\tGallus gallus\n",
      "adjusting GU261707.1\n",
      "Gallus gallus murghi\t->\tGallus gallus\n",
      "adjusting GU261674.1\n",
      "Gallus gallus jabouillei\t->\tGallus gallus\n"
     ]
    }
   ],
   "source": [
    "for r in Seqs:\n",
    "    if r.id in recs_to_adjust:\n",
    "        print \"adjusting %s\" %r.id\n",
    "        source = r.features[0]\n",
    "        print \"%s\\t->\\t%s\" %(source.qualifiers['organism'][0],recs_to_adjust[r.id][0])\n",
    "        source = r.features[0]\n",
    "        source.qualifiers['organism'] = [recs_to_adjust[r.id][0]]\n",
    "        for i in range(len(source.qualifiers['db_xref'])):\n",
    "            if 'taxon' in source.qualifiers['db_xref'][i]:\n",
    "                source.qualifiers['db_xref'][i] = \"taxon:\"+recs_to_adjust[r.id][1]\n",
    "        if not 'note' in source.qualifiers:\n",
    "            source.qualifiers['note'] = ['modified from original record: adjusted to species']\n",
    "        else:\n",
    "            source.qualifiers['note'].append('modified from original record: adjusted to species')\n",
    "#        print r.format('genbank')       \n",
    "#    print r,recs_to_adjust[r]"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Write only good records with all metadata in Genbank format."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "from Bio import SeqIO\n",
    "       \n",
    "out = open('../12S_UKamphibians_SATIVA_cleaned.gb','w')\n",
    "SeqIO.write(Seqs, out, \"genbank\")\n",
    "out.close()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Exclude the records that were highlighted by SATIVA. Then realign and rebuild phylogeny."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "pj.exclude(start_from_max=False, **recs_to_drop)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Total number of records: 198\n",
      "Number of species (see below for full list): 23\n",
      "['Alytes obstetricans', 'Bombina bombina', 'Bombina variegata', 'Bufo bufo', 'Chelydra serpentina', 'Cottus gobio', 'Discoglossus pictus', 'Epidalea calamita', 'Gallus gallus', 'Hydromantes genei', 'Hyla arborea', 'Ichthyosaura alpestris', 'Lissotriton helveticus', 'Lissotriton vulgaris', 'Pelophylax lessonae', 'Pelophylax ridibundus', 'Rana catesbeiana', 'Rana dalmatina', 'Rana temporaria', 'Salamandra salamandra', 'Triturus carnifex', 'Triturus cristatus', 'Xenopus laevis']\n"
     ]
    }
   ],
   "source": [
    "species=[]\n",
    "for r in pj.records:\n",
    "    if r.id in recs_to_keep['rrnS']:\n",
    "        source = [f for f in r.features if f.type == 'source'][0]\n",
    "        species.append(\" \".join(source.qualifiers['organism'][0].split()[0:2]))\n",
    "    \n",
    "#print set(species)\n",
    "total=len(species)\n",
    "species=list(set(species))\n",
    "\n",
    "print \"Total number of records: %i\" %total\n",
    "print \"Number of species (see below for full list): %i\" %len(species)\n",
    "print sorted(species)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Write short 12S rRNA sequences (< 500bp) to fasta file to build new alignment outside Reprophylo later in script using hmmbuild. The Hidden Markov Model alignment will include all 12S rRNA sequences - short and long."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Total number of records: 112\n"
     ]
    }
   ],
   "source": [
    "temp_seq = []\n",
    "\n",
    "for r in pj.records_by_locus['rrnS']:\n",
    "    if len(r.seq) < 500:\n",
    "        temp_seq.append(r)\n",
    "        \n",
    "total = len(temp_seq)\n",
    "print \"Total number of records: %i\" %total\n",
    "\n",
    "from Bio import SeqIO\n",
    "\n",
    "out=open('amphib_short_to_align_SATIVA.fasta','w')\n",
    "SeqIO.write(temp_seq, out, 'fasta')\n",
    "out.close()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Filter by Length"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Write cropped records (minimum length of 500bp) to fasta and Genbank file."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "from Bio import SeqIO\n",
    "\n",
    "temp=[]\n",
    "for r in pj.records:\n",
    "    if len(r.seq) >= 500:\n",
    "        temp.append(r)\n",
    "\n",
    "out = open('amphib12S_cropped_500bp_SATIVA.gb', 'w')\n",
    "SeqIO.write(temp, out, 'genbank')\n",
    "out.close()\n",
    "\n",
    "out = open('amphib12S_cropped_500bp_SATIVA.fa', 'w')\n",
    "SeqIO.write(temp, out, 'fasta')\n",
    "out.close()\n",
    "\n",
    "\n",
    "del(temp)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Now view records in cropped genbank file."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "index 0, ID = AY585337.1, length 937, with 2 features\n",
      "index 1, ID = EU115993.1, length 933, with 2 features\n",
      "index 2, ID = U52726.1, length 954, with 2 features\n",
      "index 3, ID = JX893174.1, length 933, with 2 features\n",
      "index 4, ID = AY581644.1, length 840, with 2 features\n",
      "index 5, ID = JX893176.1, length 932, with 2 features\n",
      "index 6, ID = JX893175.1, length 933, with 2 features\n",
      "index 7, ID = KC977215.1, length 663, with 2 features\n",
      "index 8, ID = AB761267.1, length 937, with 2 features\n",
      "index 9, ID = AY581648.1, length 840, with 2 features\n",
      "index 10, ID = JX893178.1, length 932, with 2 features\n",
      "index 11, ID = KC977204.1, length 663, with 2 features\n",
      "index 12, ID = HQ697272.1, length 924, with 2 features\n",
      "index 13, ID = M10217.1, length 819, with 2 features\n",
      "index 14, ID = KC977206.1, length 663, with 2 features\n",
      "index 15, ID = EU880335.1, length 926, with 2 features\n",
      "index 16, ID = DQ055835.1, length 936, with 2 features\n",
      "index 17, ID = JN627424.1, length 930, with 2 features\n",
      "index 18, ID = M27605.1, length 945, with 2 features\n",
      "index 19, ID = JX893172.1, length 933, with 2 features\n",
      "index 20, ID = KJ858774.1, length 524, with 2 features\n",
      "index 21, ID = AY325988.1, length 931, with 2 features\n",
      "index 22, ID = KC977218.1, length 663, with 2 features\n",
      "index 23, ID = KJ858776.1, length 524, with 2 features\n",
      "index 24, ID = D12694.1, length 620, with 2 features\n",
      "index 25, ID = KC977217.1, length 663, with 2 features\n",
      "index 26, ID = M57527.1, length 938, with 2 features\n",
      "index 27, ID = JN627421.1, length 932, with 2 features\n",
      "index 28, ID = KJ858770.1, length 524, with 2 features\n",
      "index 29, ID = JQ626644.1, length 525, with 2 features\n",
      "index 30, ID = Y10943.1, length 946, with 2 features\n",
      "index 31, ID = EU880339.1, length 927, with 2 features\n",
      "index 32, ID = KJ858773.1, length 524, with 2 features\n",
      "index 33, ID = JN627426.1, length 930, with 2 features\n",
      "index 34, ID = KJ858772.1, length 524, with 2 features\n",
      "index 35, ID = KC977225.1, length 663, with 2 features\n",
      "index 36, ID = KC977222.1, length 663, with 2 features\n",
      "index 37, ID = AY326063.1, length 930, with 2 features\n",
      "index 38, ID = AY581639.1, length 840, with 2 features\n",
      "index 39, ID = KJ858778.1, length 524, with 2 features\n",
      "index 40, ID = EU880331.1, length 921, with 2 features\n",
      "index 41, ID = JX893177.1, length 932, with 2 features\n",
      "index 42, ID = DQ158438.1, length 902, with 2 features\n",
      "index 43, ID = JN627422.1, length 930, with 2 features\n",
      "index 44, ID = AY928619.1, length 824, with 2 features\n",
      "index 45, ID = KJ858771.1, length 524, with 2 features\n",
      "index 46, ID = KC977226.1, length 663, with 2 features\n",
      "index 47, ID = KU840522.1, length 749, with 2 features\n",
      "index 48, ID = KC977224.1, length 663, with 2 features\n",
      "index 49, ID = JQ626643.1, length 525, with 2 features\n",
      "index 50, ID = AY971143.1, length 933, with 2 features\n",
      "index 51, ID = AY581640.1, length 840, with 2 features\n",
      "index 52, ID = AY581638.1, length 838, with 2 features\n",
      "index 53, ID = AY843601.1, length 925, with 2 features\n",
      "index 54, ID = KF049927.1, length 937, with 2 features\n",
      "index 55, ID = EU938400.1, length 932, with 2 features\n",
      "index 56, ID = AB685766.1, length 932, with 2 features\n",
      "index 57, ID = KC977220.1, length 663, with 2 features\n",
      "index 58, ID = EU880336.1, length 926, with 2 features\n",
      "index 59, ID = JN627425.1, length 930, with 2 features\n",
      "index 60, ID = AP003317.1, length 976, with 2 features\n",
      "index 61, ID = EF373884.1, length 576, with 2 features\n",
      "index 62, ID = EF373899.1, length 576, with 2 features\n",
      "index 63, ID = GU261680.1, length 976, with 2 features\n",
      "index 64, ID = FJ610338.1, length 872, with 2 features\n",
      "index 65, ID = GU261678.1, length 976, with 2 features\n",
      "index 66, ID = GU261693.1, length 976, with 2 features\n",
      "index 67, ID = EF373885.1, length 576, with 2 features\n",
      "index 68, ID = GU261698.1, length 976, with 2 features\n",
      "index 69, ID = KM886936.1, length 975, with 2 features\n",
      "index 70, ID = AB188189.1, length 783, with 2 features\n",
      "index 71, ID = EF122793.1, length 965, with 2 features\n",
      "index 72, ID = AY235570.1, length 975, with 2 features\n",
      "index 73, ID = GU261694.1, length 977, with 2 features\n",
      "index 74, ID = KF471517.1, length 516, with 2 features\n",
      "index 75, ID = EF373883.1, length 576, with 2 features\n",
      "index 76, ID = AB086102.1, length 976, with 2 features\n",
      "index 77, ID = JN860636.1, length 965, with 2 features\n",
      "index 78, ID = GU261683.1, length 976, with 2 features\n",
      "index 79, ID = KT626848.1, length 976, with 2 features\n",
      "index 80, ID = KP211419.1, length 976, with 2 features\n",
      "index 81, ID = KX512321.1, length 976, with 2 features\n",
      "index 82, ID = AP003321.1, length 976, with 2 features\n",
      "index 83, ID = KP211422.1, length 976, with 2 features\n",
      "index 84, ID = EF373904.1, length 576, with 2 features\n",
      "index 85, ID = FJ610339.1, length 875, with 2 features\n",
      "index 86, ID = AP003318.1, length 976, with 2 features\n",
      "index 87, ID = GU261707.1, length 976, with 2 features\n",
      "index 88, ID = GU261674.1, length 976, with 2 features\n"
     ]
    }
   ],
   "source": [
    "for index, record in enumerate(SeqIO.parse(\"amphib12S_cropped_500bp_SATIVA.gb\", \"genbank\")):\n",
    "        print(\"index %i, ID = %s, length %i, with %i features\"\n",
    "          % (index, record.id, len(record.seq), len(record.features)))"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Now need to filter all 12S rRNA sequences within pj.records dictionary by length to remove all sequences shorter than 500bp within Reprophylo."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Distribution Of Sequence Lengths\n",
      "Distribution Of Sequence Statistic \"Gc_Content\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/usr/lib/pymodules/python2.7/matplotlib/figure.py:371: UserWarning: matplotlib is currently using a non-GUI backend, so cannot show the figure\n",
      "  \"matplotlib is currently using a non-GUI backend, \"\n"
     ]
    },
    {
     "data": {
      "image/png": "iVBORw0KGgoAAAANSUhEUgAAAFEAAAIiCAYAAACwpiRwAAAABHNCSVQICAgIfAhkiAAAAAlwSFlz\nAAALEgAACxIB0t1+/AAAEotJREFUeJztnX20HGV9xz83b5IACQQISQVMeGtMrARoA1QtIQiCUpRj\ngZSC1FKlYovU0wbi8dSrUktD0aC2NR4EYq2obVqIx4ANPRCrpUQogcQkGCBXkwiJQnm78hK8t3/8\nZs/O3ezu3dn73d2Zu9/POXsy88zs7HM/mZfnbZ4fGGOMMcYYITcDu4ANqbSpwBrgx8B/AAekti0B\ntgJbgDNT6Scmx9gK3NjC/OaStwHHM1TiUmBxsnw1cF2yPAdYD4wHZgKPAT3JtnXA/GR5NXBWy3Kc\nU2YyVOIW4NBkeXqyDnEWXp3a7y7gZGAGsDmVvgj4UisyWo8x7f7BYTiUuMRJ/i0J/TVgR2q/HcDr\nq6TvTNLbSt4kphlMPhL23Xffx1LHbPazvtqx8yZxF3EZQ1yqu5PlncDhqf0OI87AnclyOn1ntQP3\n9/cfRdxHa30+Ocz2HuC4asfOm8RVwKXJ8qXA7an0RcAEYBZwDPFAeQp4HjiJ+CMvSX2nK7gN+Bnw\nKrAdeD9RxLmb6kWcjxFP5S3AO1LppSLOY8Dn6/zecLeG3gbyLLu9FJXhBCwQHGPUoxBQ9Rh5uycW\nEksUYIkCLFGAJQoY1+kM5IHe3t7B1HJPvX2r4TNRgCWmaOYsBEuUYIkCLFGAJQqwRAGWKMASBVii\nAEsUYIkCLFGAJQqwRAGWKMASBViiAEsUYIkCLFGAJQqwRAGWKMASBViiAEsUYIkCLFGAJQqwRAGW\nKMASBViiAEsUYIkCLFGAJQqwRAGWKMASBViiAEsUYIkCLFGAJQqwRAGWKMASBViiAEsUYIkp9ocf\ndToPeafmNFdTYdcgPDq1PAFmpmP4TBRgiQIsUYAlCrBEAZYowBIFWKIASxRgiQIsUYAlCrBEAZYo\nwBIFWKIASxRgiQIsUYAlCrBEAZYowBIFWKIASxRgiQIsUYAlCrBEAZYowBIF5FXiEmLA5Qbg68Dr\niGCxa6geLHYJEfN+C3BmW3OaU2YCTxDiAL5JhCleCixO0q4GrkuW5xDBq8cn332M6idHVw3yfB7Y\nA0wiQuJNIkIYnwusSPZZAbwnWX43EeZ4D9BHSJzfvuzmU+IzwA3ATwl5zxKX8aGUz5RdyTrsHfd+\nB22Oe5/H4IdHAVcRl+ZzwL8AF1fsU4pBX4uq2+bPn78ytbzp8ssv3wxwxokn7vOrCRP2n/Dqq/ts\nePDBi0r7LF++/I3r1q2bU1pft25dxj+lc1wI3JRavwT4e2AzMD1Jm0E8RACuST4l7iLCt1fSVffE\nLcDJwEQijv3bgU3At4kHDMm/pdj2q4BFwARgFnAMUJxTpoUsplzEWUE8eacCd1O9iPMx4oGyBXhH\njWO27EzsJrrqci4clijAEgVYogBLFGCJAixRgCUKsEQBlijAEgVYogBLFGCJAixRgCUKsEQBlijA\nEgVYogBLFGCJAixRgCUKsEQBlijAEgVYogBLFGCJAixRgCUKsEQBlijAEgVYogBLFGCJAixRgCUK\nsEQBlijAEgVYogBLFGCJAixRgCUKsEQBlijAEgVYogBLFGCJAixRgCUKsEQBlijAEgVYogBLFGCJ\nAixRgCUKsEQBlijAEgVYogBLFGCJAixRgCUKsEQBlijAEgVYooBmgtocCEwjghn8HPg/aY4KSCMS\nxxJxoM4HTqUcWKbEk8BaIoLPHcCAMoNFoJ7EccCHiKg7M4B+4AFgNfA0EXBmKnA0EYRrESH0b4B/\nBH7VslwXiEcJcV8BFhBnZC3GAguBW5LvbKmzb6foSFCbz1GOSJaF6cCyJr7XahwZSIAjA+WZZoo4\nJwHnEZHJIEJm3g7cr8rUaGYscDNRhKn2uZX6D59Ok4t74icIWf9GxNWbknxOAf492dY70h9pIbmQ\n+BMiXl41eoh4oz8Z6Y+0kFw8WKYRNZJaB7+D5opEhSeLxK3sXeVLM50ooJs6/D7R2DCvyrbjk22L\n2pqjbLTscs5SxDmWKM78kLj/bU7S5xBBWx9J9vmriu99KsNvlDiAiKw7l8j4+4kr4ZvAG4hg2BcQ\nsZ8hwrX/EVFfv5La9+6OU6toM9ynGVYQUiD+o6eQ43DtWc7EIzPsOxKmAG+jHIb4NSJY9rlEUxyE\n5HuJFqZa4dr/p035zSSxr1WZqGAW0dh7C3Ac8CARebxeuPa0sMKEa59E3Jsgyoa/1GQHiDydAPwp\ncf9dxtBI4lDwcO1zgTuJS6d0z3s1SXuT5BeiqLQttf5W4DuMknDtxwP3AWcC/wncmHzuSdL+O9ln\npDwFbCee9BBP/h8xSsK1300UKU6osu0E4uZ/t+i3jiMu5YeJuvoURkm49heBa+ts/2uiayCv5OJy\nfpnoiKrFk8BLGY43asgicTVRVqvF7yb7dB1ZJH4UOAj4V6Iwu3/yOQlYSdyz/lydwaIzQNRFB6os\n1/rkua+5I9W+rzaSs0Z+ZLRTT+IftisTRcddpgLqSTxwBMcdyXcLRz2JfUQP30EZjjcN+DRD675d\nzYeA3cArRJfoB4jq2H6pffYnqnwfJhoJ9hB138vbmtPG6Fi1bwrR9N7H0KLMK8knnfYY8JeE2DzS\nMok9DWZgDFHAPpVojj+E8kjZjURLzoMNHqtTDFLj750Ku56GZw+CA56p3+1b9RiNNsoOEK3HbWty\nLxIu4giwRAGWKMASBViiAEsUYIkCsnbejyG6MI8m6tTVCq/NDGDqGo4BNtGaAUztIBcDmr5ADGpa\nTFTzns7wXZPQD/xtpzMxAnLR7/wKMcjTVJBF4neBt7QqI0Uma7/zKcBfEIOHTEK99sRt7H0P2A84\nmOhf/hlD+5l7kv3bNaI2Kx1pT2zmxR73O1ewoF2ZKDpZ7om/Q3QL1OKQZJ+uI4vEe4Ez6mw/nSiE\ndx3KBoixdOk9USnxFOAXwuMVhuHqzh8h3iEpnWHLqD7keCowmXipvOsYTuJzlIs6M4kzbXfFPoPE\n6P77iBlMTB36iFfAikoumsJmZti3q3D3gIAsEktjsut9XiRav28kXlzsCrJI/CqwgaiAP0q8DrYq\nWe5Jtt1FyPwz4h3kvDZGdIyFxGtpZ1fZ9k7iSX56sv4u4sXJf25P1hoiF6+l3QdcX2f79QwdNfYl\n6r+B1W5y0T1wHPW7B7YBb06tryfbUOXCkkXic0Sfcy1OB55PrU9OvjPqySLxNmKiteXArxMNDmOA\n2Unaeck+JU4jntQmxSRicHupk34PQ9/AvzPZB2AfYlrUag+hTtHxMdtp3km8UVqa+q+PKOrk/Q3T\nlvWxdBO5eDqbGmQdFfYG4kWfeqPCFo40U0Uji8SziRlAxhN15Geq7NOV3QNZWA/sBH6z0xlpklzc\nE2cT3QMPZPhOV5BF4i+IkWGmgqxNYe9tVUaKTJYHy61EVW4V0ej6BNUnzvjpyLNVLLJITAdlOKfG\nPoPke67tlpBFYiNvBbiIM8rJRRHH1CCrxMnE5Bo/IGYbPiVJP5iY1Xi2LmvFIcs98RBC3izgceAo\nYGKy7WliYsgD6cL5wrJIvJZoazuZGJ+THpMzSBR9uq7xAbJdzucQAbxqTZjxBHDEiHNUQLJIPJi4\nD9ZigOgW6DqySNxF3AdrMY8urK1ANonfAS6j+hibk4D3UTvUiEmYQbQnPgV8mbh8VwDfIIaM9JHv\nzvqqBeXBCpo5RlaOIM62yhk+vw0cpviBFtIyic12/00hOvB7iDnCivDuc8enuarkOXI8q3q7cd1Z\nQL0zcYDGe/xL+7k9sQLPbtwgnt1YgO+JAixRgCUKsEQBlijAEgVYooAsdedmJ8n4XpPfKwxZJN6b\n/FutKliretgV1cAsEi8jXnw8hnhnLx1V9yIiFNwXpLkrCFkkTib6nmcTLdxpPk28+zcZ+Lwma6OT\nrcDH62z/OHE25pVcjMU5nPrBDX+J+52HpQ+4hOp9yxOTbX0jz9Lo5gNEQ+1GIuDNacnnCmJKlwHg\ng8LfGws8RHSCQcy9s4bqsUyXELebLURw2mrk4qVxiImG+tl7RuN+YpJKJR8lSgGrkvXchmtvhgOA\nCyjHVT6foWeFgsOICLqnUT4Tt1DuiZtOefjzEkJqibuIQVeV5GJenBLPAt9q4ntZ+BwRqmlyKm1U\nhWufRbyBP4243PqIOWanE3/cSN91OYcYtvcQtSfCLHS49qXAa5RHPpTGI04h7ouKAZ6fISKNbyMm\n4ugH/om4fHMZrj0LlxPylhFn4gBDB3V+nQjjruRUyvfEpZTvfdew94OlFK79cWrX46vSToml0OkQ\nYxUrJS5h7+rgSDmV8tN5VIRrfwn4k2S5msQ/Jt/v/uWi2vcysG+d7UcQT+6uI4vEHxLTtlRjH6La\n94MR56iAZJG4FPht4GuUZ2KaAZwFrCUaKP5OmrtRygeJy7qy2vcS+R92kpsay5eJIsfvAW8kihI/\nJmow6iezySG5ORPTjAd+i6inbiKaw0wVFhB9JpXjmGcBjzB08Pstbc1ZdjpW2L6V6v0mawl5/wXc\nQEyFOkC+Hy4dk7gJ+GJF2mxC2NpU2kRibll13VlJx2os09n7TFyQ/HtTKu0logHizXQhw0l8HSEo\nzfzk37UV6dvRt3C3hSt7e6d9srf32Ga/P5zE7cDcirS3Eo2mlS9DTqLgdec9TUbxGE7i94gXH38j\nWT+PmLHuzir7vomCF7hf2PuEaYjhJF5HXNLribNvJTEF6g0V+40FzgW+30wmis5wEp8gGkZXE1P9\nrU7WN1bstzDZ3pWv6jZSY3mAmEO2HmuIy7kr8UhZAZYowBIFWKIASxRgiQIsUYAlCrBEAZYowBIF\nWKIASxRgiQIsUYAlCrBEAZYowBIFWKIASxRgiQIsUYAlCrBEAZYowBIFWKIASxRgiQIsUYAlCrBE\nAZYowBIFWKIASxRgiQIsUYAlCrBEAZYowBIFWKIASxRgiQIsUYAlCrBEAZYowBIFWKIASxRgiQIs\nUYAlCrBEAZYowBIFWKIASxRgiQIsUYAlCrBEAZYowBIFWKIASxRgiQIsUYAlCrBEAZYoII8SDwfu\nIQKHbQSuTNJHGvO+ZeRR4h4iTvRcIuz6h4kYgdcQEo8lwjiVYjrPAS5M/j0L+Afa/HflUeJTROgS\ngBeBzURYu3OBFUn6CuA9yfK7gdsI+X1EiOJS9KK2kEeJaWYCxwP3Uz/m/Y7UdwoR875d7EcE0fkI\n8ELFNmnM+5UrI3nDhg0XpfdvVcz7djEe+C5wVSqtZTHve3t7B3t7exuJyTfiINntogf4ChEzcFkq\nfRVwabJ8KXB7Kn0R5Zj3xwBtPWXyeDm/BbiYCPP5UJK2hIiX9S3gMuIBckGybVOSvgl4DbgCQWz6\nLORR4vepfYW8vUb6Z5JPR8jj5Vw4LFGAJQqwRAGWKMASBViiAEsUYIkCLFGAJQqwRAGWKMASBVii\nAEsUYIkCLFGAJQqwRAGWKMASBViiAEsUYIkCLFGAJQqwRAGWKMASBViiAEsUYIkCLFGAJQqwRAGW\nKMASBViiAEsUYIkCLFGAJQqwRAGWKMASBViiAEsUYIkCLFGAJQqwRAGWKMASBViiAEsUYIkCLFGA\nJQqwRAGWKMASBVhiigZn89wLSzSZUMzuWZg5ZQuHJQqwRAGWKMASBViiAEsUYIkCLFGAJQqwRAGW\nKMASBViiAEsUYIkCLFGAJQqwRAGWKMASBViiAEsUYIkCLFHAaJJ4FhFpdytwdRPfXyDNTQEZS0QY\nn0mENV5PhHhPM9wwkt4GfmdUDyOZT0jsI8K2f4MI494WRovE1wPbU+ttDdmex9DEzdDIiK+HG9jv\nEw0cYy9Gi8SdwOGp9cOJszHNvPZlp5iMAx4nHiwTqP5gMQ1wNvAo8YBZ0uG8GGNMEZgHLKxIuxjY\nBuwGlhMPpkyMHXm+CsVNxN+8NlmfA9wJPJikvY+ogKyt+m0DRHny5NT6pxhagL4M2NjWHBWQlxla\nKL8XuDa1fjTwQtaDjpa6c6P8nHKdehxwInB/avsEYKDdmSoaXyPugUcCi4EXgf1S299LjfqxKTOL\nqNEMAK8BV1RsvwP4bLszVUTGE0Wdak1l84CD2psdA0BPpzPQIXqAC4HTgWnEA3YwSR8Ezs1ysNHS\nnpiVpcBVwD3AkwxtrFW80tsV7ALOVx2s28qJJcYAD6kO1m115xIHAnOJGsuI6dZ74hTgD4AzgEeI\nblYoP1iuzHKwbn0635taTj9IShJPa2tuCsg44F3AwZ3OSNF5hegZlNCtT+eHiWYvCd36dN4BXE+U\nF/uJS3xi6vNSloN164OlXpvhIBlPrm4s4kwAHgBuJsYzmibZDRyrOli33hMPBU4A1igO1o2XM8Ak\nor/5DKK7tD9Jb6rG0q0S5wD/mywfmUovSTTGGGM6x/8DkOG5cEJ/91QAAAAASUVORK5CYII=\n",
      "text/plain": [
       "<matplotlib.figure.Figure at 0x7f6affe5f490>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "image/png": "iVBORw0KGgoAAAANSUhEUgAAAEUAAAIiCAYAAACe4wXiAAAABHNCSVQICAgIfAhkiAAAAAlwSFlz\nAAALEgAACxIB0t1+/AAAFwBJREFUeJztnXt0HNV5wH9eySvZkpElY1sOBoxtBDbFPAzLuwZiAymU\nV8MjCcE82rqBQICYR0NTlEASQp1ASxsOJTxCk/bwMDEQGrBNgKYQXsY2AllQAwYMli38AgtrJWu3\nf3yz7Gj1rVazrxmNv985e7Rzd+bu3Z/mce+de78BwzAMwzAM9gGWu15bgcuBBmAJ8DawGBjtVwH9\nJgKsA3YHbgGucdKvBW72q1B+cwLwR+d9GzDeed/oLO+U3ANc4rzf7EoflrG80xAFOoCxznKmhE2Z\nG9TU1KwGkgW+VhT5dxSV04AnXcttyGEDMAH98EnmyLN5EN+r5hEZxIbl4GvAf7mWHwPmOu/nAovK\nXiKfqQE+AUa50hqApQx8SS7ZnjKUyfWDji1CHkOOYvygQJ9TAoVJUTApCiZFwaQomBQFk6JgUhRM\nioJJUTApCiZFwaQomBQFk6JgUhRMioJJUTApCiZFwaQomBQFk6JgUhRMioJJUTApCiZFwaQomBQF\nk6JgUhQq/S5AsZlRPbFpelXj7JpIVbQzEe9ujbcvfb1r7dte8giVlBnVE5vm1E67YP7YOd2ptAUd\nSy4A7vMiJlSHz/SqxtluIQDzx87pnlbVONtLPqGSUhOpimrptVnSsxEqKZ2JeLeWvi1LejZCJaU1\n3r50QceSPnvFgo4l0VXx9qVe8hlW3GKVlSRK+WdUT2yaVtU4uzZSFd2WiHevGvjqo+YxlLFxtOXE\npCiYFAWTomBSFEyKgklRMCkKJkXBpCiYFAWTohAEKaOBh4FVQCtwODJ9di3paBkn+VU4v/gVcJHz\nvhKoA24ArsqxXclayX53XNcBx5Cerb4DiaECPvZz+H347IUEfrgXeA24CxjpfHYZsBK4mzIHlfG7\n1+kQ4E/AkcArwG3Ap8DtyFR/gBuRIBAXZ2ybjMVij6QWYrFY67x581YN9GV33nnntJdffnl6avnl\nl18+E/8d9KMReM+1fDTwu4x1JgEtyrah7XlrBz4Empzl2cCbpMOEAJyBLiXUHIAcOiuBR5Dzx/3A\n607aItKhiNyUbE8ZyoT28AkkJkXBpCiYFAWTomBSFEyKgklRMCkKJkXBpCiYFAWTomBSFEyKgklR\nMCkKJkXBpCiYFAWTomBSFEyKgklRMCkKJkXBpCiYFAWTomBSFEyKgklRMCkKJkXBpCiYFAWTomBS\nFPyesFB0LPxQBhZ+SMHCDylY+CEFCz+kYOGHLPyQis3iKCcmRcGkKJgUBZOiYFIUTIpCEKRkRsU4\nDHkw/RIGfjB9qNGiYtwCXOOkXQvcrGwX2nnJdcC7Snob6Qnajc5yJqENFeKOinEAsAy4AhGy3lln\nPfoMdpUw9LxVAgcD3yYdFeO6jHWSZPmPxmKxha73rTMm77t5y+9aZp0z5tAdqfQHNr5y4uhT9n/u\niDmz2pWoGGqh/G4hNiKhQvZylo8G/h6YDByHBIiYADwD7Juxbb8W7rl1h1xy65fOHpf5JVd8/OCG\nB7a++gvl+9VWst9Xn2xRMR4nHX1nLhIEIifF6nnz+/ABiajzGyAKvANcCFQADyKBZNYAZw8mo85E\nvHv59g/GrO7umFxJRcUOenunRse+uy0R3+ClQEGQshI4VEn31NkM8GbXutVPf9b2jfnjTuhJpS3Y\nsPjg1q51j3vJJwhSisZ+1ROmnld/2Gtvdn08OcKwigTJ3vPqD3t37Y4tU1viHy0ebD6hklITqYo2\nDq/b2Di8bqM73XrzFaw333rzrTc/E+vNLycmRcGkKJgUBZOiYFIUTIqCSVEwKQomRcGkKJgUBZOi\nYFIUTIqCSVEwKQomRcGkKJgUBZOiYFIUTIqCSVEwKQomRcGkKJgUBZOiYFIUTIpCqMa8QTiGoRcV\nCwChYAEgFCwAhIINGVWwIaM2ZFTFhoyWE5OiYFIUTIqCSVEwKQomRcGkKJgUhSBIWQO8DiwHUvPs\nm4G1Ttpy4CQ/CuYn7yHxUtzcAFyVY7vQV/O1RplvDbUgSEkCS4FXgb9xpV+GzG6/mzJH2vH63xgJ\n7AOMQ35MB/AW8HkBZZgArAPGIiGHLnPy7HA+v9FZ5+KM7ZKxWOyR1EIsFmudN2/eqoG+SAkVciZ5\n7pENyPH9J6AbSGS84sALwJVAfT5f4OIG4LsZaZOAFmXdkp1TBrI0Gvg+cAlQhfz3XkQiV2x0tm0A\npgKHI+E+uoBfIP/drYMo1EgkAsZnQA0SlOoHyNWo3VnnSiRAxNeVH1ToecdzHp8AHyFi9sqxLsAU\n5D/9sbPtYNgLWOG83kACygDcj4hZicRO0cIP+RKo6jvIHuKVaiSuUqkJbfSuQgh9PSVQFCLlLOBJ\nJIbSYuCbRSlRAMhXyqVIJL8O4DHkUn0fEr0v9GST9gZwbkbarUjjrlz4dk55DZippNcgrVg3HwG1\n3ss19PgXpMb6M2BERvpqJBDmHKQWugUJd1gufL0kH4ZUpN4FTnTSRgB3IG2eBFKTvRcYVVgZPeF7\nPWU4cD0i4dfArk56BGkcVhRUtPzwXUqKJuBZpBp/fjEyLIDASAFpQP01sAmpnwymXVQKfJVyJnLI\nPALMR2I8gkQIfRDoBK6m/LVj36RcjpxInwV+i5xT/jtjnb9EIoW+hsSWLRe+SXkH+FfX8smIpMkZ\n640Cbkcu3+XCNylbgb9zLU9CpBySZf3D8/mSPPFNypNIj9ufA/sj55AtBKPm6puUKUjFLdUf+wlw\nWhEKUwx8vfpEkB78A5FetaAQqHpKUPCllVzI7YpCb3UElq1I7/wYD9uMQ25vbClJifriy+HzLWAD\nUvf4LXJL8wD6XnlGIRW2S4EngB7kfs28Ags7GHw7p9Qh92LW0P+uYDwjbTVS3S9X94EvdwjdRIAY\nMAuYjtz3Td1LfgN5AsKywsvoCV/uEAYdu+9TTkyKgklRMCkKJkXBpCh4kXIvcg8oGzHgnsKKEwy8\nVFwSwHnAf2b5/Fzns3LtfVkrXs3NzUnX+4F+Y8kfj1WDtH2GPLmm9e/pvFI2pyFdk5mMQRqQq4tX\ntMLJsZdkJZeUC4F/dC1f77w0EqSfPDmkySVlEekxJ/cA/44MG3WTBLYhkw0+LGbh/CKXlNRwTpDb\nGwvRB/qGCi+hQppLVYig4TV+SgR5FN5U5OSqnch+WGih/MaLlL2BR+n/OM1MdioptyP3kK9Beto2\nDrz60MWLlGOAfwYWlKgsgcFLjTaO/sDn0OFFylPAUaUqSJDwIuUq4Aj6jmYKJV7OKS8gjb5bgJ8g\n83p6XZ8PQ2q3mQN6hhxepLxP7vskobiL70XKsaUqRNCw7kiFfKTMAn4E3EW6dluL9LOEYgiGFymp\n56I/g9x0vwj4kvNZL9LNcElRS+cTXqRciww0vgrpgXOfcLcjwzW+Uryi+YcXKecD/wHcht7uaUNa\nz15ZQ/+oGA3IbPa3kaHuZZ3W70XKJKSuko0t5HdOSSJXtoOQ2yQA1yFSmoCnneWy4UXKZ/QP6eFm\nCun4BF7JrPucisxRxPl7ep755oUXKf+L3PfRtqlHTrzP5FEGLSrGeGC98349+gz2kuGl8vYj4Hng\nD8jMUpCxtU3I7l0L3JxHGY6ib1SMtozPk2SpKcdisYWu919ExVi4UJJbWlr6xEdQomLkUdz+nIwM\n9MuMjNEOnFCE/FNRMdqQqTMgYUIyRcEATYrm5uak+y7hAKjreO2jfQI54c4hfVl+G+lWyCeGSmZU\njBOQqBiPAXOBnzp/F+WRd954kbIHMjb/c+Bx5+VmJDK38AMPeY5H6jepsvwGuQS/ilQUL0Yu2Wd7\nyLNgvEhZw8A32E9FfpSXSZbvIeelTDYhdw18oZgNwtA0Lov5Q/alPMPPS06uw2eu80pxPTLTNJMx\nwJ+RPj8MaXJJqadv9+JY5CrhJnWD/W6yj0gYUuSScpvzAqmPXImcTEONl6tPaE6kudhpfqgXvEr5\nGtJ90EHfan6v6++Qx8vhczVS7f4EGc2kdTTtdLc4LgVeAo5Huh9Di5fDpxHpjgy1EPAm5R3K3Ffq\nF16kLEBqs+UMMeQLXs4pCaRrcBUyTv9d9KvN/UUol694keKOzJWtOp9kJ5NyfMlKETC8SHm2VIUI\nGoVU83clHdosVHiVshtyztiKTPnfAGxGbljtVtyi+YfXjuuXkM7mlUjIVZAZ7d9EevgPIwSTFrxI\nuRGpvJ1C/wheX0F63W6ib0/dkMTL4XMCEi8yUwjA75Eo6Ccqnw05vEipR258ZWM1O+FIpo+A4wb4\n/Bj6x6gdkniR8iAS1/pmJK5KijpkXO05wAPFK5p/eDnR3oTsDdcgN8E/dtJ3Q+Q+76wz5PGyp3Qi\nh888ZMjE587rKeBvnc8KeVBJYPA66qAHGSp6VwnKUnRGwZufwX5+l6OcZO0PboD1SXirIT0aylMe\ndotDwaQomBQFk6JgUhQGI+Vb5B5zdg7lCY1YFnJJOQP4N3KPUNqMtKCDEsC3pDzMwOPx3fwRCSNf\nLnyrpxyG3n+i8STpCQdDmlxSxjH47oCPnfWHPLmkfA7sMsi8diEkDcJcUlYjcwYHwzHA/xVWnGCQ\nS8rjyBXlyBzrHY7Myckcmh5K6pFpJ5uRPpPMhzVWI3N0NiPnlHL20apXjmQG+eQxGGLIpS2BDNhZ\nAfwPMudvO+nRCIfm+wV54qsUkFFMP6N/rOs1TnpZZ245lKyekg+jkH5ZvwfvlEyK1+5IkAlLn+Wx\n3ZBhMFImOH/XOX+rkZGSmZY/BB4qUrkCzT7ADvo+nnNX+s8hTCCP8ty7jGXzre1zITJL6zbls/nI\n6KbjgS8jMzkyn5M+JMl1+ByPTHLUHnu1gr6jmx4gJEPAcu0peyP1kcHQRnkPn5KRa0+pQQ4LN5uA\nGfQPb/Yp/SdIDUly7SlbSF99UiSQ529ktogbkWFfXqlA9sZUu6kZ6a5Y7rxOyiPPgsgl5Q0GPzN9\nDvmFZf0O0Er6SpAEfo5EyTgI6bwqK7mkPIyE8cjV93q6s97DHr9/IvAXwC9JR8YYRsCfpFKN/Be7\nkDFve2Z8PgkZfhFHBgZ6fdDaQ8jeMIv04XMD0qZaiUzWzDZJwrdqfhcy8O8JZOj595AT6qdIT1tq\n8E6bs15XjvzcnIIMOV1O33Bpd5AO33oj0uBU6z/ZomLMmTmzujcaHRXt7q5uWbbsi8gYxY6KMQK4\nDHgOmRHW4/x9Dvg2+T2K78dI0+A9pAnRSf9x/ZPIfp4KVCu5FLgPH/fV7kqyx1YIVCu5FKTiToLE\npjzAWX6PEN15LAc2aKecmBQFk6JgUhRMioJJUTApCqGUcnlz87gfNDc35bt9KKWk6JEIHp4JtZR8\nx+WHWkq+mBQFk6JgUhRMioJJUTApCiZFwaQomBQFk6JgUhRMioJJUTApCiZFwaQomBQFk6JgUhRM\nioJJUTApCiZFwaQomBQFk6JgUhRMioJJUTApCiZFwaQomBQFk6JgUhRMioJJUTApCiZFwaQomBQF\nk6IQBCmZUTEakIcVvI08pL7szz4MgpTMqBjXIVKagKed5bLitxQtKsapyOO2cP6eXu5C+S3lVuTp\nuglX2njSk6zX40O0QT+n9WeLiuEmyQAz1bNFxVi4UJJbWlq+7l5/sFEx/Aze8mPkAWw7kKgauyAx\nsg9FJLUjwSCeAfZVtk+SpfzNzc1J1/uBfqOah5+Hz/eA3YG9gHOBPyCSHiP9MLa5wCJfShcAZiEy\nQC7JS8l9SS5G7JNAx0/Jh5JJ8fvqE0hMioJJUTApCiZFwaQomBQFk6JgUhRMioJJUTApCiZFwaQo\nmBQFk6JgUhRMioJJUTApCkGJcV00ZlRPbJpe1Ti7JlIV7UzEu1vj7Utf71r7tpc8QiVlRvXEpjm1\n0y6YP3ZOdyptQceSC4D7vIgJ1eEzvapxtlsIwPyxc7qnVTXO9pJPqPaUmkhV9Lltb01+v2fTwcOp\nqOiht3fP4Q2v1Uaqcj3vuQ+hkrK2e/OU9Tu2HXt+/RFfjGJ4cMuyYz/s3uRJSqgOn5pIdPLMEXv0\nues3c8QeyZpI1RQv+YRqTxldObK7vmJk++r4hvoIwyIJkomGiprN9ZUju3NvnSZUUjoT3fGGyprt\nDZU1293p2xJxL0+TCdfh09q17p4FGxYPd6ct2LB4+Kqu9nu95BOqPaUl/tFigA93bL6wNlJVvS0R\n71rV1X5vKn1nwIZilJNQHT5gbZ9+WNtHwdo+CjWRqujy7R+MWd3dMbmSiood9PZOjY59d6du+7T3\nbB39fvemA86qm/lF22fR1hUHrOvZutFLPqE6fIZHKtm3qrHPYOF9qxqHDY9UeLp8h2pPGVNRswVY\n80LnOwdVDItU9CYTvQ0VNct3raj19CDaUEnZ2Ns5Gph0ZM2ULx5M0tbVPumT3m2rveQTqsOnJ7GD\ntnh7n0OlLd6e7En0epqDEKo9pXF43ZY9ow0rH9q6rM/VZ8Lwup336tOZiHcfNGKPjQeN2KPP1eZX\nm1/01J8SqsOnNd6+dEHHkqg7bUHHkuiqePtSL/kE+mHNOVDn6syontg0rapxdm2kKrotEe9eNXDb\nJ+ucoaGKdR2UE5OiYFIUTIqCSVEwKQomRcFvKdXAS8AKJAjET5z0ZmAtMrt9OXCSH4Xzk5HO30rg\nReBo4AbgqhzblazyFoQG4efO3ygSS2Wzs5xX9TsstzgiwGvAFOAO4E3gq8BlwPnAq8B3gZzN/2Ld\n4ghSY6gOeAoJItMKdDjpNyKBIC7OWD8Zi8UeSS3EYrHWMR/u2O2i6CF1mRnf0/Pq1q/eeOlSJSrG\nmSgOgiQF4PvAdmCBK20SEppo/4x1+7VwL64/6oqbGk/rFxvhH9of3XL35udvU74vcFExAHYlHeBh\nBDAHudo0utY5A2gZTGadibjambQtS3o2/JYyAQkRsgK5ND+OxGG6BXgdWIlEzLhyMJlZJ5N1MqlY\nJ1M5MSkKJkXBpCiYFAWTomBSFEyKgklRMCkKJkXBpCgEoTuyqISlj7Zo2DB0hWINQw+VlJpIVVRL\nr82Sno1QSQlLH21RsT5a66NVsT7acmJSFEyKgklRMCkKJkXBpCiYFAWTomBSFEyKQqh63sC6I/th\n3ZEK1h2pYN2RCtYdqWDdkdYdqWLdkeXEpCiYFAWTomBSFEyKgklRMCkKJkXBpCiYFAWTomBSFPyU\nki0iRgOwhNwPps/FsQWWzze0iBi3ANc46dcCN2fZNlfXQfMgvr8Y3Q8lYyTwCrAf0AaMd9IbnWWN\nkknx+5wSQQ6f9cAzSESM8c4yzt/x+qbhpw45fI4jHT8lxaYs26xA/tOFvFZoGQflZthW4AlgJrJ3\nNALtSCyEDVm2ObBUhfHz8MkWEeMxYK6TPhdYVP6i+cf+SISdFUgEjKud9AZgKYVfkg3DMLxyIHB8\nRtp5wHvIVetOJFSaJyoKL5ev/BL5Dc85y9OB3wPLnLTzkSvsc+rWIeUj4HDX8g+ROE4pLgbeKGuJ\nAkAXsLtr+VngJtfyVOAzr5n63fYplA5gN+d9JVIjfsn1eRRIZG4Udn6NnEMmI90N24Ba1+d/Rd/D\naadgL2A1sjfsAC7J+PxR4OflLlQQGI5cmndTPjsQGFPe4oSUsAxtGgacA3wZGIdcQFJDt5LAqV4y\nC0p/SqHcAlyB9N6to283Y6D7YUvJeuCsYmU21OspKSJIB1VRGOptnxT1yJ2AZ4uRWVjOKXXAN5Au\nzdeBHic9daK93EtmYbn6POt67z6xpqQcV9bSBIBK4GSkI9xwEUcChBeFsFx9ViLdBEUhLFeftcA/\nIfWVTuSQGuF6bfeSWVhOtAP1mSTx+M8PwyU5ijya4h6yj1DYKdkANBUrs7CcU8YDByMjoAomDIcP\nyKCf85Aa7TLkZAt51mjDImU6crMepL82RUqKYRiGEST+H2G9ByUhknSpAAAAAElFTkSuQmCC\n",
      "text/plain": [
       "<matplotlib.figure.Figure at 0x7f6affd2be10>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "%matplotlib inline\n",
    "pj.report_seq_stats()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "# Define minima and maxima\n",
    "\n",
    "gc_inliers = {\n",
    "    'rrnS': [34,55]\n",
    "}\n",
    "len_inliers = {\n",
    "    'rrnS': [500,1000]\n",
    "}\n",
    "\n",
    "\n",
    "# Apply to loci data\n",
    "for locus in gc_inliers:\n",
    "    # trim GC outliers\n",
    "    pj.filter_by_gc_content(locus,\n",
    "                            min_percent_gc=gc_inliers[locus][0],\n",
    "                            max_percent_gc=gc_inliers[locus][1])\n",
    "\n",
    "    # trim length outlier\n",
    "    pj.filter_by_seq_length(locus,\n",
    "                            min_length=len_inliers[locus][0],\n",
    "                            max_length=len_inliers[locus][1])"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Confirm that the length filter has worked."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Distribution Of Sequence Lengths\n",
      "Distribution Of Sequence Statistic \"Gc_Content\"\n"
     ]
    },
    {
     "data": {
      "image/png": "iVBORw0KGgoAAAANSUhEUgAAAFEAAAIiCAYAAACwpiRwAAAABHNCSVQICAgIfAhkiAAAAAlwSFlz\nAAALEgAACxIB0t1+/AAAEsNJREFUeJztnX2UXHV9h59NSAwJBEgghBYw4TUEWt7aAFVKeEe0IqVA\npCilFKm0Iu1pwXA8dVW0GGoL2hdjKRKPL9WKFji8WOKR4LGUCBoIDYtEWIUACYYSJYbX3f7x/c3Z\nu7N3ZvfOfmbmzs7nOeeevXPvnTu/PLkvv/cvGGOMMcYYITcCG4G1mW2zgLuBHwP/Beyc2bcUeBzo\nA07JbD8yneNx4PompreUHAscznCJy4Ar0vqVwDVpfSGwBpgCzAPWAz1p32pgUVq/AzitaSkuKfMY\nLrEP2D2tz02fIa7CKzPH3QUcDewBPJrZvgT4XDMSWo9Jrf7BUdiduMVJfytCfw14OnPc08Cv52zf\nkLa3lLJJzDKYFgkzZsxYnzlno8uavHOXTeJG4jaGuFU3pfUNwF6Z4/YkrsANaT27fUPeibdu3bov\n8RyttXx0lP09wKF55y6bxFuBC9L6BcB/ZrYvAaYC84H9iRfKc8AvgKOIf+R7Mt/pCr4KPAO8CjwF\nXEhkcVaSn8W5ingr9wGnZrZXsjjrgc/U+b3RHg29Y0iz7PHSqYwmYLHgHBMehYDcc5TtmdiRWKIA\nSxRgiQIsUYAlCrBEAZYowBIFWKIASxRgiQIsUYAlCrBEAZYoYLt2J6AM9Pb2DmbWe+odm4evRAGW\nmKGRqxAsUYIlCrBEAZYowBIFWKIASxRgiQIsUYAlCrBEAZYowBIFWKIASxRgiQIsUYAlCrBEAZYo\nwBIFWKIASxRgiQIsUYAlCrBEAZYowBIFWGKGHeF/252GslNz5P0s2DgIj80ampOn0Dl8JQqwRAGW\nKMASBViiAEsUYIkCLFGAJQqwRAGWKMASBViiAEsU0NCoyg5lkJx/7+Dg4LDqrZ6ennpOcs/hK1FA\n14+87+np6ZkFGzfDi7OHTwQ8ZnwlCrBEAZYowBIFWKIASxRgiQIsUYAlCrBEAZYowBIFWKIASxRg\niQIsUYAlCrBEAZYowBIFWKIASxRgiQLKKvGDRJDXR9I6RLDYu8kPFruUiHnfB5zSumSWl0MIgdOA\nyYS4fYFlwBXpmCuBa9L6QiJ49RQi/Pt68i+OrhpRtQC4H3gZeANYBZwFvBNYkY5ZAbwrrZ9BhDl+\nDegnJC5qXXLLKfER4Fji9p0OnE5EEN+doStlY/oMI+PeP02L496XsS9OH/Ap4rm3lbhV36g6phKD\nvha5+xYtWnRzZn3dJZdc8ijAyUceOe2NqVN3nPrqq9PWPvjgeZVjli9fftDq1asXVj6vXr266L+l\nNHwCeD8hd27atkf6DPChtFS4iwjfXk1XPRMB5qS/ewO/D3yFiHt/Qdp+AUOx7W8FlgBTgfnA/kDn\nXjJC7iUGcK8Bjk/bZgEryc/iXEW8UPqAU2ucs2lXYjfRdbdzR2GJAixRgCUKsEQBlijAEgVYogBL\nFGCJAixRgCUKsEQBlijAEgVYogBLFGCJAixRgCUKsEQBlijAEgVYogBLFGCJAixRgCUKsEQBlijA\nEgVYogBLFGCJAixRgCUKsEQBlijAEgVYogBLFGCJAixRgCUKsEQBlijAEgVYogBLFGCJAixRgCUK\nsEQBlijAEgVYogBLFGCJAixRgCUKsEQBlijAEgVYogBLFGCJAixRgCUKsEQBlijAEgVYogBLFGCJ\nAixRgCUKaCSozS5EqI9B4Hng/6Qp6kDGInEyEQfqbOA4hgLLVHiWiCP1H8AtwIAygZ1APYnbERF5\nPkRE4tkKPADcAWwGeogYKfsRQbiWEEL/FvgXRoZE6koeI8T9G7CYuCJrMRk4AfhC+k5fnWPbRVuC\n2vwDQxHJijAXuK6B7zUbRwYS4MhAZaaRLM5RwJlEZDKAJ4jIZferEjWRmQzcSGRh8pabqP/yaTel\neCZ+hJD1TeBoYKe0HAN8K+3rHe+PNJFSSPwpES8vjx4i+u1Px/sjTaQUL5Y5RImk1slvobEsUcdT\nROLjjCzyZZlLZNBNHd5NVDYclrPv8LRvSUtTVIym3c5FsjgHENmZHxDPv0fT9oXAScDD6Zi/qfre\nxwr8RoWlwPnEy2otcCEwA/ga8GYiGPY5wIuZ4/+YKK9fRu1nd9uplbUZbSnKPOI/603p89eIKLql\nDdde5Ercp8Cx4+EXROj16cSVNR14hrjajkvHrADuIWqYaoVr/58WpbeQxP5mJaKKF4BPAz8DtgHf\nJh4f9cK1Z4V1TLj26cSzCSJv+CtNcgDYF7icuDW3EJW951cd09Hh2g8G7iRuncoz79W07RDJL8C5\nwA2Zz+8B/ol4kXV8uPbDgfuAU4DvANen5btp23+nY8ZLH1Gs3J4oCZ0ErANuYwKEa19JZCmOyNl3\nBHHrrRT91hVEuPa1xEtkChMkXPtLwNV19n+CaBooK6W4nV8mGqJq8SzxNu06iki8g2jVq8XvpWO6\njiIS/xKYDXyDyMzumJajgJuJZ9ZfqBPY6QwQJYaBnPVaS5nbmttS7PviWFI2lh+Z6NST+EetSkSn\n4yZTAfUk7jKO847nux1HPYn9RAvf7ALnmwN8HHhyHGmaULwf2AS8QjSJXgwcCuyQOWZHosj3Z8Dt\nRMXEc8AlLU3p2GhbsW8nojK0n+FZmVfSkt22HvhrQmwZaZrEnjEmYBKRwT6OqI7fjaGeso8QNTkP\njvFc7WKQGv/eWbBxM7w4G3Z+oX6zb+45xlopO0DUHresyr2TcBZHgCUKsEQBlijAEgVYogBLFFC0\n8X4S0YS5H1Gmzsu8NtKBqWvYn2j/VXdgahWl6ND0WaJT0xVEMW9zge+axFbgU+1OxDgoRbvzK0S/\nQVNFEYnfBt7SrIR0MkXbnY8B/oroPGQS9eoTn2TkM2AHYFeiffkZhrcz96TjW9WjtihtqU9sZGCP\n252rWNyqRHQ6RZ6Jv0s0C9Rit3RM11FE4j3AyXX2n0hkwrsOZQXEZLr0maiUeAzwc+H5OobRys4f\nJIZDVK6w68jvcjwLmEkMKu86RpO4haGszjziSttUdcwg0Un9PmIGE1OHfmIIWKdSiqqweQWO7Src\nPCCgiMRKn+x6y0tE7ff1xMDFrqCIxC8SI5x6iGkKbk3LY2nbWmJc3RvAB4gxyGWtjGgbJxDD0t6W\ns+904k1+Yvr8dmLg5Jdbk7QxUYphafcB19bZfy3De419jvojsFpNKZoHDqV+88CTwG9mPq+hWFfl\njqWIxC1Em3MtTiSmHqgwM31nwlNE4leJidaWAwcSFQ6TgAVp25npmArHE29qk2E60bm90kj/GsNH\n4N+ZjgGYRkyLmvcSahdt77Od5XRiRGll6r9+IqtT9hGmTWtj6SZK8XY2NSjaK+zNxECfer3CThhv\nojqNIhLfRswAMoUoI7+Qc0xXNg8UYQ2wAfitdiekQUrxTFxANA88UOA7XUERiT8neoaZKopWhZ3V\nrIR0MkVeLDcRRblbiUrXJ8ifOONn409WZ1FEYjYowztqHDNIuefabgpFJI5lVICzOBOcUmRxTA2K\nSpxJTK7xfWLe7WPS9l2JWY0X6JLWORR5Ju5GyJsP/ISYtnT7tG8zMTHkLnThfGFFJF5N1LUdTfTP\nyfbJGSSyPl1X+QDFbud3EAG8ak2Y8QSw97hT1IEUkbgr8RysxQDRLNB1FJG4kXgO1uIwurC0AsUk\n3g5cRH4fm6OA91I71IhJ7EHUJz4HfJ64fVcA/050Gemn3I31pehGAvHiuIWRM3zeBuyp+IEmUopO\nnhDPvDOIOcQOJNpY1tPlY58bjT2whRLPqt5qXHYWUO9KHGDsLf6V41yfWIVnNx4jnt1YgJ+JAixR\ngCUKsEQBlijAEgVYooAiZedGJ8m4t8HvdQxFJN6T/uYVBWsVD7uiGFhE4kXEwMf9iTF72ai65xGh\n4D4rTV2HUETiTKLteQFRw53l48TYv5nAZzRJm5g8Dny4zv4PE1djWSlFX5y9qB/c8Fe43XlU+ong\nrHlty9unff3jT9LE5mKiovYRIuDN8Wm5lJjSZQB4n+B3DgR+lFm2ECHYZxFxnvNimS4lHjd9RHDa\nPErT2nc5cUtXz2i8lZikUs0kYuD5XpQ4XHsj7Aycw1Bc5bMZflUoOQX4XlrvY2jw4lyGuj8vJaRW\nuIvodFVNaZpMIeaB+HoD32uEJQyNoZ5Q4drnEyPw5xCZ7n5ijtm5xD9ONdZlKjEk+MqcfR0drn0Z\n8DpDPR8q/RF3Ip6Lyg6eZxC3ZoU+JkC49kuImY3/kXhWZcvKW4juJbWGZjTCuxk+HcKtTIBw7Q8B\n30zruxJXY7Zn7FJGFgcbZQYxDC4bvm5ChGvfBvxpWs+T+CeUe+xfKW7nl4krpBZ7E2/urqOIxB8Q\n07bkMY0o9n1/3CnqQIpIXAb8DvAlhmZi2gM4DVhFlCr+Tpq6Ccr7iNu6uti3jfJ3OylNieXzRK/Y\nPwAOIrI5PyZKMKo3sykxpbkSs0wBfpsop64jqsNMDouJNpPqqZ/mAw8zvPP7F1qasuK0LbN9E/nt\nJqsIed8DPk1MhTpAuV8ubZO4jigrZ1lACFuV2bY9Mbfsdxr5kRbRthLLXEZeiYvT3xsy27YBX2H4\nTJ5dw2gS30QIyrIo/V1Vtf0pmlfDXWpGk/gUcHDVtrcSY52rB0NOx2XnXO4lBj7+Rvp8JjFj3Z05\nxx6CM9y57EPUWL9BXH0DRHXXIVXHTSaip1W/hMpE214sTwDHEVOdvpD+Hke0PWc5Ie33UN0JTikq\nZU0NLFGAJQqwRAGWKMASBViiAEsUYIkCLFGAJQqwRAGWKMASBVgicFlv75yP9vYe0Oj3LTHDaw1G\nSrfEDL8c2Sg3JixRgCUKsEQBlijAEgVYogBLFGCJAixRgCUKsEQBlijAEgVYogBLFGCJAixRgCUK\nsEQBlijAEgVYogBLFGCJAixRgCUKsEQBlijAEgVYogBLFGCJAixRgCUKsEQBlijAEgVYogBLFGCJ\nAixRgCUKsEQBlijAEgVYogBLFGCJAixRgCUKsEQBlijAEgVYogBLFGCJAixRQFkl7gx8A3iUiE50\nFOOPed80yirxeiJYxEFEtKE+Iobz3cABRBinSkznhcC56e9pwD/T4n9XGSXuBBwL3Jg+v07Ej34n\nsCJtWwG8K62fQcSBfo2Ier6eoehFLaGMEucDzxMh7H4I/CsRzbdezPunM9/viJj3zWY74Ajgz4lI\nvtcxPBw7iGPe33xzbF67du152eObFfO+FcwFnsx8fitwO/GSaUrM+97e3sHe3t6xxOTrmHgszxEB\nxioza55EhPi8jZLGvC/j7QzwAeDLhJifABcScbC+DlxEvEDOSceuS9vXES+hSxHEpi9CWSU+RASb\nreakGsd/Mi1toYy3c8dhiQIsUYAlCrBEAZYowBIFWKIASxRgiQIsUYAlCrBEAZYowBIFWKIASxRg\niQIsUYAlCrBEAZYowBIFWKIASxRgiQIsUYAlCrBEAZYowBIFWKIASxRgiQIsUYAlCrBEAZYowBIF\nWKIASxRgiQIsUYAlCrBEAZYowBIFWKIASxRgiQIsUYAlCrBEAZYowBIFWKIASxRgiQIsUYAlCrBE\nAZYowBIFWKIASxRgiQIsUYAlCrBEAZYowBIFWKIASxRgiQIsUYAlCrBEAZaYYYyBbUZgiaYQikA3\nHRNeqeOwRAGWKMASBViiAEsUYIkCLFGAJQqwRAFlldgPPAz8iKHgrs0O1764we+VlicJaVmWAVek\n9SuBa9L6QmANMAWYR0Qaz7s4Ris7944hXR1Xdu6p+uxw7QUZBFYCDwAXp20O116QtwDPArsRz8G+\nqv2NhGt/aJTvAHxklP0P5W0sq8Rn09/ngW8Rt+dGIlz7c0S49k3pmA3AXpnv7pm2VXNYU1JKOW/n\n6cCOaX0G8bZdS4RlL2W49jIyn3jbrgEeIbIvEG/rleRnca4iXih9wKktS6kxxnQ2hwEnVG07nygh\nbQKWEy+oQkwef7o6ihuIf/Oq9HkhcCfwYNr2XiLHsir32waI/OPRmc8fY3gG+iIiR2Dq8DLDM+b3\nAFdnPu8H/LLoScuY2W4mzzNUrt4OOBK4P7N/KjDQ6kR1Gl8inoH7ENVqLwE7ZPafRY3ysRliPlGy\nGQBeBy6t2n8L8PetTlQnMoXI6uRVlx0GzG5tcgwwsva4W+gBzgVOBOYQL9jBtH2QqEUfM2WtT2w2\ny4DLge8SdZfZylpFP8auYCNwtupk3ZZPrDCJaI6V0G1l5wq7AAcTJZZx063PxJ2APwROJjoJvJa2\nV14slxU5Wbe+ne/JrGdfJBWJx7c0NR3IdsDbgV3bnZBO5xWiy4mEbn07P0RUe0no1rfz08C1RH5x\nK3GLb59ZthU5Wbe+WOrVGQ5S8OLqxizOVKKj1I2M7ONjCrAJOEB1sm59Ju4OHEH0OBs33Xg7Q3Sa\nOp8osTxIvFygwRJLt0pcCPwwre+T2V6RaIwxxrSP/weON8th/MKzbAAAAABJRU5ErkJggg==\n",
      "text/plain": [
       "<matplotlib.figure.Figure at 0x7f6affffc8d0>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "image/png": "iVBORw0KGgoAAAANSUhEUgAAAEUAAAIiCAYAAACe4wXiAAAABHNCSVQICAgIfAhkiAAAAAlwSFlz\nAAALEgAACxIB0t1+/AAAEgtJREFUeJztnXu0XFV9xz8JkASSAEmAXArKS6LELkqhDdguGogJrcWK\n1iIWLFCpUmQVLPIy1HKraGlI1a5UWqriu12hxFpYFCEUQmurpoJAaAg0EFReIbwEAtyE3Ns/fmfW\nnDv5zj1z5s7MOXPy/aw1686cx559P3POPvvsvc/+gTHGGGNKwISiMwA8CrwIbAO2AvOAmcBy4IBk\n/fuAF4rJXjFsICSkWQJcnLy/BLiypzkqARuAWQ3L1gGzk/cDyecdikeAHwM/Aj6ULHs+tX5Cw+cd\ngn2Tv3sD9wDHsr2E5xp3mjp16npgZJyvezr7r3SHy4GPEafLQLJsX/TpM5KR1mAL3yfTmNjCjt1k\nN2B68n4qcAKwBrgBOCNZfgbwnV5maudefplgNvAvyfudgW8BtxLly3XAWdQvyaYFsk6f4zqQRt/R\niX+olGVKKbEUgaUILEVgKQJLEViKwFIEliKwFIGlCCxFYCkCSxFYisBSBJYisBSBpQgsRWApAksR\nFN0Z1hUGBwdHUu9zj8HxkSKotJR2jhKouJR2sRSBpQgsRWApAksRWIrAUgSWIrAUgaUILEVgKQJL\nEViKwFIEliKwFIGlCCxFYCkCSxFYisBSBJYisBSBpQgsRWApAksRWIrAUgSWIrAUgaUILEVgKQJL\nEViKwFIEliKwFIGlCCxFYCkCSxFYisBSBJYisBSBpQgsRWApAksRWIrAUgSWIrAUgaUILEVgKQJL\nEViKwFIEliIog5SdiKBHNyaf5wGrk2X/A/xqrzNUBinnA2upx8VYAnwC+GXgz5PPPaVoKfsDvw18\niXqkuyeBPZL3ewKP9zpTRU9p9jngImD31LJLge8BS4kf7W29zlSRUt4JPE2UHcelln8ZOI8IhnQy\ncC2wSCUwb968Fan3a88+++wHAFasiMVr1qw5Nb39Nddcc9jq1avn1j6vXr26A/9GZ/kM8DMi5N6T\nwGbgG0SgxhoTgJ832b9pfJ/BwcGR9AR4Y1C6+D6LgTcABwHvB24H/gBYD8xPtlkAPNTrjBVdpqSp\n/WofBr4ATAZeTT73lLJIuTN5QQRRO7rAvBR+SS4lliKwFIGlCCxFYCkCSxFYisBSBJYisBSBpQgs\nRWApAksRWIrAUgSWIrAUgaUILEVgKQJLEViKwFIEliKwFIGlCCotZSZsnA7/m3e/tuL0lYQRmuR/\nJmx8Fl4AmAV7Pgez86RR6SOlXSxFYCkCSxFYisBSBJYisBSBpQgsRWApAksRWIrAUgSWIrAUgaUI\nLEVgKQJLEViKwFIEliKwFIGlCCxFYCkCSxFYisBSBJYisBSBpQgsRWApAksRWIrAUgSWIrAUgaUI\nLEVgKQJLEViKwFIEliKwFIGlCCxFYCmCyj1DODIyst1k3hMmTGj2f8o08k7nvBvwZmCfJMFNwIPA\nKznT6XtmAhcA3we2AMMNryHgv4E/BWb0MF9Np4CfCRtH4MEReHAmbMybxlinz55ETIyPEDOTPwj8\nAHgYeDbZdybwJuAYYA7wGnA18Cmaz3ffKbr2CO5YPEPEwfgEMbd9FocAlwNPJPt2m64dKWNxPnGE\n5GUK8NE29stLIadP2fET7L1kPFJOBr5LTLBwKxFHoxK0K+Vc4GtEPeUG4lL9VeCSNtJqDI8F8CfA\nA8D9wF+1mceu0Uza/USklTSfAx5t4zsuAL5FyAU4HlgJ7JJ83rvJfl0raLOOlLuBo8TyqcBjDcse\nB6ZlpNeICo91DvCXwNbk86acaY6bLCn/QdRW/xrYNbX8RuJ0+SARpeljwJ8x+hRohVp4rOHUskOB\n3yAqiquAX8mZ5rjJuvc5jzi0vwi8h/gVbyHKjl2AvyXqJVuAf0q2b5Vm4bF2Jm4XjiHC7V0HHKwS\naBYea9FRR03ZNmnSdIBJW7ZMWXPXXadC58Nj7QJcRtz4fRPYK1k+kbg53KmNNJuFx7qZeiQoiMhQ\ns8T+hdRoFXOIQ/oZ4PROJJgwn/qpdzbwF6nv+2mTfUojBaJA/CPgOaJ+0sp9URbzqV99diGOmDXA\nXYw+tdIUKuV3iVPm28CFwKRk+QBxvm8mCste144Lk3IecWVYRcQFfAX4t4ZtfocoG+4GjmznS9qk\nMCkPE1eYGicSkhqvBtOBZUSDU68orPK2F1F7rVGbSG5mw3YvEVXz+VSArHrK94lmxrXA80SD04vA\nuibb/6BzWSsvhwD3UW+PfQY4qdAc1ena6ZN1pDwMHEFUvXcljpDXMvbpe1rp4hgmGq13GMYqaMfT\nXdHLro6OM5aUR4nWeXXf0Yx9iO6NDePIU6k5h7iLHSIqbh8CfonRbSbTiQrbucBNRBvIU8T9S7cp\nrPK2B/Bx4qhp7BUcali2nqjuT2/ni9qg8C6OicA8onI2l2girPUl3w/cQdy89ZKudXG02sE+TFTM\ndojKmft9BJYisBSBpQgsRWApgjxSvgIcPcb6ecC148tOOcgj5QyifaUZBwNnjis3JaGTp89U6v2/\nfU1WjfaA5FWrCh9G9PM2Mou4gVzfuayVl0G2HyLa7PU6ne01zKKw5sjvUB9zci3wD2x//zMCvAys\nJvp/+p4sKfckL4ADgRVEd2alyTMMfbBbmSgbecfmTwQWEqOsZ6HbMz453kz1E4cSnWJZBW6vKKyg\nTbOMqKBdTLS0PZtj374ij5Rjgb8BlnYpL6UhT412CHikWxkpE3mk3AL8ercyUibySLkAeBujRzNV\nkjxPcWwgbvr2ArYRz/Vsa0hrhCbDO7tA4V0cAD8ZKyOpL+l78kg5rluZKBtujhS0I2U+8GliaPpb\nkmXTiHaWvh6C0Q47EeNma9X5bcCCZN2uxGDjy3qYn8JGR6a5hBhofAHRApcucF8lhmu8I0d6pSWP\nlNOJ4eGfR9/3rCPunvuePFIOJJ79acYLVKRMySPlJbYfVJzmEAp4iqsb5JHyPeADTfaZQTwldkcn\nMlU0eaR8mnj+5nbiqS6IMbZ/TDzdNQ24sqO56xNOJAb6Nba2PQWc0OO8lKLlDWIE5IHEw5S1y/JD\nRLPCDjeHCsAbiUllmrFbsk2vKEXl7VHg3WOsfxcVGVTcyRvCytxcdvIfeQtJw06/k1XQnpG8alxG\nPGnayCzgF4n7n74nS8oMRjcv7k00SaapdbB/md7eJZeCYeC0ojORounVZ3BwcGRwcHCkF/WUvitI\nZ8GeW9uYNCtv5a2vGKMVf0zy/vq/TzQfbGJ0NX9b6m/fk+dIuYiYCugZYjSTamja4bo4zgV+SLTL\nvtqd7JSDPKfPANEcWWkhkE/Kw8R8kpUnj5SlRG22V88IFkaeMmWYqAg9QIzTfwR9tfl6B/JVKHmk\nfCX1vll1foQdTMqC7E2qQR4pq7qVibIxnvuZvahPbVYp8krZjygzfk488v80MdnM15J1lSDP6fNG\nokY7G7iX+lREc4lpVxcRT471/UMLeaR8iqi8vZPtZ/B6B9HqdgWjW+r6kjynzwnA37G9EIipDa8G\nfrMTmSqaPFJmEB1fzVjPDjjq4HFiAt1mHMv2c9T2JXmkXEfMa30lMa9KjT2ISXVPAZZ3LmvFkaeg\nvYI4Gi4mJuV9Ilm+HyH3v5Jt+p48R8pm4vQ5m5iD+pXkdQvw4WRdJTrZ8zZcbyWGin6xC3kpDX3X\nbdELLEVgKQJLEViKwFIErUg5B3hfxjan0JupEXtClpT3AF8ge4TS88QddFkm8B0XWVJOI/qNb83Y\n7laimt/3bSmQLeVodPuJ4rvEvEx9T5aUfWi9OeCJZPu+J0vKK8DuLaa1OxW5IcySsp7W58I/Fvi/\nNvKgwmNBNE8MM/bjNF0hS8qNxBXl1zK2O4YYjZ036BFEvMO1jB7w8waid+AnbaQ3brKkLCOGct1E\ntJk0BmucQkzzfDPRB7Qs5/er8FgAnyUaswohS8rzxJGyBfh7or5yDxE268fJ+muS9Scln/OgwmOd\nRBTu9+VMq2O00si0mpjw+yLgvcDhqXU/JSavWsLY41UVKjzWbsBi4tSp0XQagWbhsVasiMVr1qw5\nNb19p8NjpZlOtMuOd/COCo91PSF3Q/LaSjw9oi71mYOLW8hDRx6CgnjA8qU29mtkcfKCuMJdCPxe\nwzYbiJB/z3Xg+1qmFSn7Jn+fTP5OIUZKNlr+GfDP48iL+tUKGYKaJeXNREf6ZdTDaU4DrhLbvk4U\nwu3UVe5MXo30ai6WUWRdff6QOHQ/L9ZdSIxuWgC8nXiS46yO5q4gso6UBUTUNxX26h5Gj25aTkWG\ngGUdKYcSl8xWWJds3/dkHSlTidMizXNEXaVxerMX2f4Bqb4kS8oL1K8+NYYZHVytxgAx7KvvyTp9\n7qf1J9MXUZFpWbOkXE9UwbPaXt+dbHf9+LNUPFlSriUK0OXEmLcDGtYfSAy/WE4MT6/EFPFZZcpr\nxI3bTUQFbjFRoL5ItLTVBu+sS7arROS5Vvp9HiFCYJ1PzKEyDPxC8vc/ifinR1KRR/qh9RvCV4kG\npLyNSH2Ju00FliKwFIGlCCxFYCkCSxFYisBSBJYisBSBpQgsRWApAksRWIrAUgSWIrAUgaUILEVg\nKQJLEViKwFIEliKwFIGlCCxFYCkCSxFYisBSBJYisBSBpQgsRWApAksRWIrAUgSWIrAUgaUILEVg\nKQJLEViKwFIEliKwFIGlCCxFYCkCSxFYisBSBJYisBSBpQgsRWApAksRWIrAUgSWIrAUgaUILEVg\nKQJLEViKwFIEliKwFIGlCCxFYCkCSxFYiqAsUhrjhl1FzJl9L/BtRsd87zplkdIYN+xW4K1EsKWH\ngI/3MjNlkKLihq2kHjLrh8k2PaMMUlTcsDQfpPWomR2hnUhQnUTFDUtzGRGk7R/VzmWKGdZJVNyw\nryfrziSCyE5psm/XYoYVffosJgIxHgS8H7gdOB34LeKUOokCAhQULSXNBOq/3DIiDNdK4tS6upcZ\nKbpMSbOKemSpQoMnlelIKQ2WIrAUgaUILEVgKQJLEViKwFIEliKwFIGlCCxFYCkCSxFYisBSBJYi\nsBSBpQgsRWApAksRWIqg0lJa7E/ejkpL2RFp6yhoJQ0fKQJLEViKwFIEliKwFIGlCCxFYCkCSxFY\nisBSBJYisBSBpQgsRWApAksRWIrAUgSWIrAUgaUILEVgKQJLEViKwFIEZXouuSMcPmX/OXMnDyyc\nOnHypM3DQ1vWDj11232vPfZQnjQqJeXwKfvPWTTtsDMv3HvRltqypZtWngl8NY+YSp0+cycPLEwL\nAbhw70VbDps8sDBPOpWSMnXi5Elq+bQmy5tRKSmbh4e2qOUvN1nejEpJWTv01G1LN60cdVQs3bRy\n0gNDT92WJ50J2ZuUlhFE/g+fsv+cwyYPLJw2cfKkl4eHtjww9tVHptHPeMxbL7EUgaUILEVgKQJL\nEViKwFIEliKwFIGlCCxFYCkCSxFYisBSBJYisBSBpQgsRWApAksRWIrAUgSWIrAUgaUIyiClMTTW\nTCIGx0NERKg920z3uHYzVAYpjaGxLiWkzAH+PfncDseNO2cFsT9wG3A89SNlHTA7eT+QfFZkjToY\nbOH7SznqQIXGmg1sTN5vpC6oZxQpJR0aq9nAmRGaHxH3ptar1+UZ60eSNEqFCo31DeJ0GUi22Zfm\np0/lmU+9TFkCXJK8vxS4spAclYD5wA3J+5lE4TveS7IxxuTlCGBBw7IPEFe0p4FrgFxD0CHuO/qZ\nLxH/w53J57nAzcBdybLTibrYnXLvivI4cEzq8ycZXSE7C7i/pzkqAa8RYUBrrAKuSH1+E/BS3kSL\nvvcZL5uA/ZL3OwNHEfGVa0yiecjhyvJNogw5GLgYeJmIiVrjvZTw/qbbHASsJ46G14GPNKz/V+Cz\nvc5UGdiFuDTvJ9YdAczqbXYqSlUeAJoAnAK8HdiHuIDUHnAaAd6VJ7GqPIK7BPgocAfRNpNumOrE\nw1J9yUbg5E4l1u/1lBoTiWbNjtDv9z41ZgBvpR6ufFxUpUzZAzgNWATcB2xNltcK2vPyJFaVq8+q\n1Pt0wVqTcnxPc1MCdgZOBPYqOiNlYwg4sFOJVeXqcy/RTNARqnL1eQy4iqivbCZOqV1Tr1fzJFaV\ngnasNpMRcv74VbgkTwJ+BFzLDtzFqniaGM/SEapSpswGjiQG+4ybKpw+ALsR/T2LiO6Nzcnytmq0\nVZEyF7g7eX9wanlNijHGmDLx/wI1FTmsi8AKAAAAAElFTkSuQmCC\n",
      "text/plain": [
       "<matplotlib.figure.Figure at 0x7f6affc87350>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "pj.report_seq_stats()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Pickle the project."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'amphib_post_SATIVA.pkl'"
      ]
     },
     "execution_count": 25,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "pickle_pj(pj, 'amphib_post_SATIVA.pkl')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Long Alignment"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Now, need to produce a sequence alignment file based on an alignment of all 12S sequences longer than 500bp. This will be done with the AlnConf class."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "pj = unpickle_pj('amphib_post_SATIVA.pkl', git=False)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "muscle default configuration:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "muscle -in 750651477486369.36_rrnS.fasta\n"
     ]
    }
   ],
   "source": [
    "muscle_default = AlnConf(pj,\n",
    "                         method_name=\"muscleDefault-500bp-SATIVA\",\n",
    "                         program_name=\"muscle\",\n",
    "                         loci=['rrnS'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "pj.align([muscle_default])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "AlnConf named muscleDefault-500bp-SATIVA with ID 750651477486369.36\n",
      "Loci: rrnS \n",
      "Created on: Wed Oct 26 12:52:49 2016\n",
      "Commands:\n",
      "rrnS: muscle -in 750651477486369.36_rrnS.fasta\n",
      "\n",
      "Environment:\n",
      "Platform: Linux-4.4.0-38-generic-x86_64-with-Ubuntu-14.04-trusty\n",
      " Processor: x86_64\n",
      " Python build: defaultMar 22 2014 22:59:56\n",
      " Python compiler: GCC 4.8.2\n",
      " Python implementation: CPython\n",
      " Python version: 2.7.6\n",
      " ete2 version: 2.2.1072\n",
      " biopython version: 1.68\n",
      " dendropy version: 3.12.2\n",
      " cloud version: 2.8.5\n",
      " reprophylo version 1.3\n",
      " User: biopc4\n",
      " Program and version: MUSCLE v3.8.31\\nPal2Nal v14\n",
      " Program reference:\\nEdgar 2004: MUSCLE: multiple sequence alignment with high accuracy and high throughput. Nucleic Acids Research 32(5):1792-1797\\nMikita Suyama\n",
      " David Torrents\n",
      " and Peer Bork (2006) PAL2NAL: robust conversion of protein sequence alignments into the corresponding codon alignments.Nucleic Acids Res. 34\n",
      " W609-W612.\n",
      "execution time:\n",
      "14.8714580536\n",
      "\n",
      "\n",
      "==============================\n",
      "Core Methods section sentence:\n",
      "==============================\n",
      "The dataset(s) rrnS were first aligned at the protein level using the program MUSCLE v3.8.31 [1].\n",
      "The resulting alignments served as guides to codon-align the DNA sequences using Pal2Nal v14 [2].\n",
      "\n",
      "Reference:\n",
      "[1]Edgar 2004: MUSCLE: multiple sequence alignment with high accuracy and high throughput. Nucleic Acids Research 32(5):1792-1797\n",
      "[2]Mikita Suyama, David Torrents, and Peer Bork (2006) PAL2NAL: robust conversion of protein sequence alignments into the corresponding codon alignments.Nucleic Acids Res. 34, W609-W612.\n",
      "\n"
     ]
    }
   ],
   "source": [
    "print pj.used_methods['muscleDefault-500bp-SATIVA']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'rrnS@muscleDefault-500bp-SATIVA': <<class 'Bio.Align.MultipleSeqAlignment'> instance (86 records of length 1291, IUPACProtein()) at 7f6affe6bb10>}"
      ]
     },
     "execution_count": 30,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "pj.alignments"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['rrnS@muscleDefault-500bp-SATIVA_aln.fasta']"
      ]
     },
     "execution_count": 31,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# record_id and source_organism are feature qualifiers in the SeqRecord object\n",
    "files = pj.write_alns(id=['record_id','source_organism'],\n",
    "                      format='fasta')\n",
    "files"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Pickle the project."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'amphib_post_SATIVA.pkl'"
      ]
     },
     "execution_count": 32,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "pickle_pj(pj, 'amphib_post_SATIVA.pkl')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Build tree from long alignment"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Trim the alignment."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "trimal -in 536331477486453.01_rrnS@muscleDefault-500bp-SATIVA.fasta -gappyout\n"
     ]
    }
   ],
   "source": [
    "muscle_gappyout = TrimalConf(pj,                       # The Project\n",
    "\n",
    "                            method_name='gappyout',  # Any unique string \n",
    "                                                     # ('gappyout' is default)\n",
    "                            \n",
    "                            program_name='trimal',   # No alternatives in \n",
    "                                                     # this ReproPhylo version\n",
    "                            \n",
    "                            cmd='default',           # the default is trimal. \n",
    "                                                     # Change it here\n",
    "                                                     # or in pj.defaults['trimal']\n",
    "                            \n",
    "                            alns=['rrnS@muscleDefault-500bp-SATIVA'],  # 'all' by default\n",
    "                            \n",
    "                            trimal_commands={'gappyout': True} # By default, the gappyout algorithm is used.\n",
    "                           )"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "pj.trim([muscle_gappyout])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "TrimalConf named gappyout with ID 536331477486453.01\n",
      "Alignments: rrnS@muscleDefault-500bp-SATIVA \n",
      "Created on: Wed Oct 26 12:54:13 2016\n",
      "Commands:\n",
      "rrnS@muscleDefault-500bp-SATIVA@gappyout: trimal -in 536331477486453.01_rrnS@muscleDefault-500bp-SATIVA.fasta -gappyout\n",
      "\n",
      "Environment:Platform: Linux-4.4.0-38-generic-x86_64-with-Ubuntu-14.04-trusty\n",
      " Processor: x86_64\n",
      " Python build: defaultMar 22 2014 22:59:56\n",
      " Python compiler: GCC 4.8.2\n",
      " Python implementation: CPython\n",
      " Python version: 2.7.6\n",
      " ete2 version: 2.2.1072\n",
      " biopython version: 1.68\n",
      " dendropy version: 3.12.2\n",
      " cloud version: 2.8.5\n",
      " reprophylo version 1.3\n",
      " User: biopc4\n",
      " Program and version: trimAl 1.2rev59\n",
      " Program reference: Salvador Capella-Gutierrez; Jose M. Silla-Martinez; Toni Gabaldon. trimAl: a tool for automated alignment trimming in large-scale phylogenetic analyses. Bioinformatics 2009 25: 1972-1973.\n",
      "execution time:\n",
      "0.859714984894\n",
      "\n",
      "==============================\n",
      "Core Methods section sentence:\n",
      "==============================\n",
      "The alignment(s) rrnS@muscleDefault-500bp-SATIVA were trimmed using the program trimAl 1.2rev59 [1].\n",
      "\n",
      "Reference:\n",
      "Salvador Capella-Gutierrez; Jose M. Silla-Martinez; Toni Gabaldon. trimAl: a tool for automated alignment trimming in large-scale phylogenetic analyses. Bioinformatics 2009 25: 1972-1973.\n",
      "\n"
     ]
    }
   ],
   "source": [
    "print pj.used_methods['gappyout']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'rrnS@muscleDefault-500bp-SATIVA@gappyout': <<class 'Bio.Align.MultipleSeqAlignment'> instance (86 records of length 1182, IUPACAmbiguousDNA()) at 7f6affac5d50>}"
      ]
     },
     "execution_count": 36,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "pj.trimmed_alignments"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 37,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['rrnS@muscleDefault-500bp-SATIVA@gappyout_trimmed_aln.fasta']"
      ]
     },
     "execution_count": 37,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "files = pj.write_trimmed_alns(id=['record_id','source_organism'],\n",
    "                                   format='fasta')\n",
    "files"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Infer Maximum Likelihood tree based on the clipped alignment. We are going to use RAxML. Configure RAxML for trimmed muscle alignment."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 38,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "raxmlHPC-PTHREADS-SSE3 -f a -m GTRGAMMA -n 899801477486484.35_rrnS@muscleDefault-500bp-SATIVA@gappyout0 -p 619 -s 899801477486484.35_rrnS@muscleDefault-500bp-SATIVA@gappyout.fasta -T 5 -x 417 -N 100\n"
     ]
    }
   ],
   "source": [
    "raxml = RaxmlConf(pj, method_name='raxml-muscleDefault-500bp-SATIVA-trimmed',\n",
    "                  program_name='raxmlHPC-PTHREADS-SSE3',\n",
    "                  keepfiles=True,\n",
    "                  cmd='raxmlHPC-PTHREADS-SSE3',\n",
    "                  preset='fa',\n",
    "                  alns=['rrnS@muscleDefault-500bp-SATIVA@gappyout'],\n",
    "                  model='GAMMA',\n",
    "                  threads=5,\n",
    "                  cline_args={'-#': 100})"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Infer tree."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 39,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "pj.tree([raxml])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 40,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "RaxmlConf named raxml-muscleDefault-500bp-SATIVA-trimmed with ID 899801477486484.35\n",
      "Alignments: rrnS@muscleDefault-500bp-SATIVA@gappyout \n",
      "Created on: Wed Oct 26 12:54:44 2016\n",
      "Commands:\n",
      "rrnS@muscleDefault-500bp-SATIVA@gappyout:\n",
      "raxmlHPC-PTHREADS-SSE3 -f a -m GTRGAMMA -n 899801477486484.35_rrnS@muscleDefault-500bp-SATIVA@gappyout0 -p 619 -s 899801477486484.35_rrnS@muscleDefault-500bp-SATIVA@gappyout.fasta -T 5 -x 417 -N 100\n",
      "\n",
      "Environment:\n",
      "Platform: Linux-4.4.0-38-generic-x86_64-with-Ubuntu-14.04-trusty\n",
      " Processor: x86_64\n",
      " Python build: defaultMar 22 2014 22:59:56\n",
      " Python compiler: GCC 4.8.2\n",
      " Python implementation: CPython\n",
      " Python version: 2.7.6\n",
      " ete2 version: 2.2.1072\n",
      " biopython version: 1.68\n",
      " dendropy version: 3.12.2\n",
      " cloud version: 2.8.5\n",
      " reprophylo version 1.3\n",
      " User: biopc4\n",
      " Program and version: raxmlHPC-PTHREADS-SSE3RAxML version 8.0.12\n",
      " Program reference: A. Stamatakis: RAxML Version 8: A tool for Phylogenetic Analysis and Post-Analysis of Large Phylogenies. In Bioinformatics\n",
      " 2014.\n",
      "execution time:\n",
      "330.802323103\n",
      "\n",
      "==============================\n",
      "Core Methods section sentence:\n",
      "==============================\n",
      "Phylogenetic trees were reconstructed from the dataset(s) rrnS@muscleDefault-500bp-SATIVA@gappyout using the program raxmlHPC-PTHREADS-SSE3RAxML version 8.0.12 [1].\n",
      "\n",
      "Reference:\n",
      "A. Stamatakis: RAxML Version 8: A tool for Phylogenetic Analysis and Post-Analysis of Large Phylogenies. In Bioinformatics, 2014.\n",
      "\n"
     ]
    }
   ],
   "source": [
    "print pj.used_methods['raxml-muscleDefault-500bp-SATIVA-trimmed']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 41,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'amphib_post_SATIVA.pkl'"
      ]
     },
     "execution_count": 41,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "pickle_pj(pj, 'amphib_post_SATIVA.pkl')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Visualise Tree"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Annotate final tree."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 42,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "pj.add_qualifier_from_source('organism')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 43,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "supports = {'black': [100,95],\n",
    "            'dimgray': [95,75],\n",
    "            'silver': [75,50]}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 44,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "for species in ['Chelydra','Cottus','Gallus']:\n",
    "    pj.if_this_then_that(species, 'organism', 'outgroup', 'outgroup',\n",
    "                         mode='part')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 45,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "pj.clear_tree_annotations()\n",
    "pj.annotate('.', 'outgroup', 'outgroup', \n",
    "            ['source_organism', 'record_id'], \n",
    "            node_support_dict=supports, multifurc=None,\n",
    "            html='muscleDefault-trimmed-amphib-raxml-post_SATIVA.html')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Create a static copy of the tree (the last *.png with the expected alignment id) for later."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 46,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "%%bash\n",
    "\n",
    "last_png=$(ls -hlrt *rrnS@muscleDefault-500bp-SATIVA@gappyout.png | tail -n 1 | perl -ne 'chomp; @a=split(\" \"); print \"$a[-1]\"')\n",
    "cp $last_png rrnS_muscleDefault_trimmed_amphib_500bp_SATIVA_raxml.png"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Pickle the project."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 47,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'amphib_post_SATIVA.pkl'"
      ]
     },
     "execution_count": 47,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "pickle_pj(pj, 'amphib_post_SATIVA.pkl')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "**=======================================================================================================**"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 48,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "from reprophylo import *"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 49,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "pj = unpickle_pj('amphib_post_SATIVA.pkl', git=False)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "- Identify the record IDs of the sequences shorter than 500bp\n",
    "- Write these sequences to new fasta file (make sure the sequence IDs are as Reprophylo expects them: 'id_f0')\n",
    "- Prepare a hmm profile for the alignment the tree is built on (either muscle 'or mafft)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 50,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'rrnS@muscleDefault-500bp-SATIVA': <<class 'Bio.Align.MultipleSeqAlignment'> instance (86 records of length 1291, IUPACProtein()) at 7f6afbc6c590>}"
      ]
     },
     "execution_count": 50,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "pj.alignments"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 51,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "returning trimmed alignment object rrnS@muscleDefault-500bp-SATIVA@gappyout\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "1"
      ]
     },
     "execution_count": 51,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "from Bio import AlignIO\n",
    "\n",
    "out=open('500bp_amphib_align_SATIVA.fasta','w')\n",
    "\n",
    "AlignIO.write(pj.fta('rrnS@muscleDefault-500bp-SATIVA'), out, 'fasta')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Check number of records in file containing short 12S rRNA sequences."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 53,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Total records: 112\n"
     ]
    }
   ],
   "source": [
    "from Bio import SeqIO\n",
    "\n",
    "records = list(SeqIO.parse(\"amphib_short_to_align_SATIVA.fasta\", \"fasta\"))\n",
    "print(\"Total records: %i\" % len(records))"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Check that record IDs are in format that reprophylo expects for alignment."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 54,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "AY322321.1_f0\n",
      "FJ595208.1_f0\n",
      "AY347451.1_f0\n",
      "AF161036.1_f0\n",
      "AJ440767.1_f0\n",
      "KP109553.1_f0\n",
      "KX269198.1_f0\n",
      "AJ222711.1_f0\n",
      "AY333688.1_f0\n",
      "KF645375.1_f0\n",
      "AY347456.1_f0\n",
      "AY043038.1_f0\n",
      "EU116959.1_f0\n",
      "X95493.1_f0\n",
      "AY347452.1_f0\n",
      "AB023395.1_f0\n",
      "AF218718.1_f0\n",
      "AY333663.1_f0\n",
      "AY222458.1_f0\n",
      "KF664850.1_f0\n",
      "AY333653.1_f0\n",
      "KP109559.1_f0\n",
      "KF645417.1_f0\n",
      "AY333665.1_f0\n",
      "AY147254.1_f0\n",
      "KF645409.1_f0\n",
      "AY147256.1_f0\n",
      "U04702.1_f0\n",
      "DQ092285.1_f0\n",
      "KF645408.1_f0\n",
      "AJ222710.1_f0\n",
      "AY333669.1_f0\n",
      "AY880481.1_f0\n",
      "AY347457.1_f0\n",
      "AY333687.1_f0\n",
      "X86235.1_f0\n",
      "AY347458.1_f0\n",
      "DQ221219.1_f0\n",
      "FJ595204.1_f0\n",
      "KF645371.1_f0\n",
      "AY333655.1_f0\n",
      "AB023397.1_f0\n",
      "AY333683.1_f0\n",
      "U04704.1_f0\n",
      "AY880461.1_f0\n",
      "KP109558.1_f0\n",
      "AY043041.1_f0\n",
      "DQ092298.1_f0\n",
      "KF645367.1_f0\n",
      "DQ221215.1_f0\n",
      "KF645365.1_f0\n",
      "AY347455.1_f0\n",
      "AY333684.1_f0\n",
      "AY347454.1_f0\n",
      "AY043034.1_f0\n",
      "AJ440762.1_f0\n",
      "FJ602097.1_f0\n",
      "DQ092286.1_f0\n",
      "AY333664.1_f0\n",
      "AY043032.1_f0\n",
      "KP109557.1_f0\n",
      "KF645373.1_f0\n",
      "KF645370.1_f0\n",
      "AY333666.1_f0\n",
      "AB159478.1_f0\n",
      "KP109555.1_f0\n",
      "DQ221220.1_f0\n",
      "DQ221216.1_f0\n",
      "AY347450.1_f0\n",
      "AF124103.1_f0\n",
      "AJ309142.1_f0\n",
      "AY779191.1_f0\n",
      "KP109552.1_f0\n",
      "KF645366.1_f0\n",
      "DQ221225.1_f0\n",
      "AY333686.1_f0\n",
      "KF645364.1_f0\n",
      "DQ221223.1_f0\n",
      "EU116957.1_f0\n",
      "KP109556.1_f0\n",
      "AY347453.1_f0\n",
      "EU116960.1_f0\n",
      "M13046.1_f0\n",
      "AB023394.1_f0\n",
      "KP109554.1_f0\n",
      "AY222459.1_f0\n",
      "AY347459.1_f0\n",
      "KT074476.1_f0\n",
      "AF224726.1_f0\n",
      "DQ885561.1_f0\n",
      "JN695757.1_f0\n",
      "EF362674.1_f0\n",
      "KF484729.1_f0\n",
      "AJ849444.2_f0\n",
      "KF471521.1_f0\n",
      "EF432760.1_f0\n",
      "U81335.1_f0\n",
      "EF362642.1_f0\n",
      "KF471523.1_f0\n",
      "EF362717.1_f0\n",
      "HM637771.1_f0\n",
      "EU851896.1_f0\n",
      "GU477775.1_f0\n",
      "KF471516.1_f0\n",
      "EF362643.1_f0\n",
      "EF362639.1_f0\n",
      "KF471518.1_f0\n",
      "KP234022.1_f0\n",
      "KF471524.1_f0\n",
      "KF471520.1_f0\n",
      "KP171705.1_f0\n",
      "EF362708.1_f0\n"
     ]
    }
   ],
   "source": [
    "from Bio import SeqIO\n",
    "\n",
    "short_seq = open(\"amphib_short_to_align_SATIVA.fasta\", \"rU\")\n",
    "sequences = []\n",
    "\n",
    "for r in SeqIO.parse(short_seq, \"fasta\"):\n",
    "    print(r.id)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "If record IDs were not in the format Reprophylo expects for tree construction, the code cells below could be used to format them to be 'id_f0'."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 55,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "#temp_seq = []\n",
    "\n",
    "#for r in pj.records:\n",
    "    #if len(r.seq) < 500:\n",
    "        #temp_seq.append(r)\n",
    "\n",
    "#from Bio import SeqIO\n",
    "\n",
    "#out=open('short_to_align.fasta','w')\n",
    "#SeqIO.write(temp_seq, out, 'fasta')\n",
    "#out.close()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 56,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "#from Bio import SeqIO\n",
    "\n",
    "#short_seq = open(\"short_to_align.fasta\", \"rU\")\n",
    "#sequences = []\n",
    "\n",
    "#for r in SeqIO.parse(short_seq, \"fasta\"):\n",
    "    #print(r.id)\n",
    "    #r.description = r.id+'_f0'\n",
    "    #r.id = r.description\n",
    "    #print r.description\n",
    "    #sequences.append(r)\n",
    "\n",
    "\n",
    "#out = open(\"seq_to_align.fasta\", \"w\")\n",
    "#SeqIO.write(sequences, out, \"fasta\")\n",
    "#out.close()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Check that sequences in reprophylo project have not been altered."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 57,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "AY322321.1\n",
      "Rana lessonae 12S ribosomal RNA gene, partial sequence; tRNA-Val (trnV) gene, complete sequence; and 16S ribosomal RNA gene, partial sequence; mitochondrial genes for mitochondrial products.\n",
      "FJ595208.1\n",
      "Rana ridibunda 12S ribosomal RNA gene, partial sequence; mitochondrial.\n",
      "AY347451.1\n",
      "Discoglossus pictus pictus haplotype ph3 12S ribosomal RNA gene, partial sequence; mitochondrial.\n",
      "AF161036.1\n",
      "Rana temporaria 12S ribosomal RNA gene, partial sequence; mitochondrial gene for mitochondrial product.\n",
      "AJ440767.1\n",
      "Discoglossus pictus mitochondrion partial 12S rRNA gene.\n",
      "KP109553.1\n",
      "Hyla arborea isolate H523 12S ribosomal RNA gene, partial sequence; mitochondrial.\n",
      "KX269198.1\n",
      "Rana dalmatina isolate MSUZP-NPUA-R-21-1 12S ribosomal RNA gene, partial sequence; tRNA-Val gene, complete sequence; and 16S ribosomal RNA gene, partial sequence; mitochondrial.\n",
      "AJ222711.1\n",
      "Rana ridibunda mitochondrial 12S rRNA gene, isolated from Ashkhabad (Turkmenistan) and Uzbekistan.\n",
      "AY333688.1\n",
      "Bombina variegata isolate Bvs06 12S ribosomal RNA gene, partial sequence; mitochondrial gene for mitochondrial product.\n",
      "KF645375.1\n",
      "Salamandra salamandra alfredschmidti isolate SS 56 12S ribosomal RNA gene, partial sequence; tRNA-Val gene, complete sequence; and 16S ribosomal RNA gene, partial sequence; mitochondrial.\n",
      "AY347456.1\n",
      "Discoglossus pictus auritus haplotype ph10 12S ribosomal RNA gene, partial sequence; mitochondrial.\n",
      "AY043038.1\n",
      "Rana dalmatina 12S ribosomal RNA gene, partial sequence; mitochondrial gene for mitochondrial product.\n",
      "EU116959.1\n",
      "Hydromantes genei isolate E3056.8 12S ribosomal RNA gene, partial sequence; mitochondrial.\n",
      "X95493.1\n",
      "R.lessonae mitochondrial 12S ribosomal RNA.\n",
      "AY347452.1\n",
      "Discoglossus pictus auritus haplotype ph4 12S ribosomal RNA gene, partial sequence; mitochondrial.\n",
      "AB023395.1\n",
      "Rana lessonae mitochondrial gene for 12S rRNA, partial sequence, strain:Luxembourg.\n",
      "AF218718.1\n",
      "Rana catesbeiana 12S ribosomal RNA gene, partial sequence; mitochondrial gene for mitochondrial product.\n",
      "AY333663.1\n",
      "Bombina bombina isolate Bb02 12S ribosomal RNA gene, partial sequence; mitochondrial gene for mitochondrial product.\n",
      "AB023396.1\n",
      "Rana esculenta mitochondrial gene for 12S rRNA, partial sequence.\n",
      "AY222458.1\n",
      "Salamandra salamandra morenica E171243 12S ribosomal RNA gene, partial sequence; mitochondrial gene for mitochondrial product.\n",
      "KF664850.1\n",
      "Epidalea calamita voucher vg07119 12S ribosomal RNA gene, partial sequence; mitochondrial.\n",
      "AY333653.1\n",
      "Bombina variegata kolombatovici 12S ribosomal RNA gene, partial sequence; mitochondrial gene for mitochondrial product.\n",
      "KP109559.1\n",
      "Hyla arborea isolate HJ57 12S ribosomal RNA gene, partial sequence; mitochondrial.\n",
      "KF645417.1\n",
      "Salamandra salamandra isolate FS2 12S ribosomal RNA gene, partial sequence; mitochondrial.\n",
      "AY333665.1\n",
      "Bombina variegata isolate Bvs03 12S ribosomal RNA gene, partial sequence; mitochondrial gene for mitochondrial product.\n",
      "AY147254.1\n",
      "Triturus vulgaris meridionalis 12S ribosomal RNA gene, partial sequence; mitochondrial gene for mitochondrial product.\n",
      "KF645409.1\n",
      "Salamandra salamandra isolate FS12 12S ribosomal RNA gene, partial sequence; tRNA-Val gene, complete sequence; and 16S ribosomal RNA gene, partial sequence; mitochondrial.\n",
      "AY147256.1\n",
      "Triturus alpestris 12S ribosomal RNA gene, partial sequence; mitochondrial gene for mitochondrial product.\n",
      "U04702.1\n",
      "Triturus carnifex mitochondrion 12S rRNA gene, partial sequence.\n",
      "DQ092285.1\n",
      "Salamandra salamandra bernardezi isolate E1712.19 12S ribosomal RNA gene, partial sequence; mitochondrial.\n",
      "KF645408.1\n",
      "Salamandra salamandra isolate FS10 12S ribosomal RNA gene, partial sequence; tRNA-Val gene, complete sequence; and 16S ribosomal RNA gene, partial sequence; mitochondrial.\n",
      "AJ222710.1\n",
      "Rana ridibunda mitochondrial 12S rRNA gene, isolated from Georgia.\n",
      "AY333669.1\n",
      "Alytes obstetricans boscai 12S ribosomal RNA gene, partial sequence; mitochondrial gene for mitochondrial product.\n",
      "AY880481.1\n",
      "Rana temporaria 12S ribosomal RNA gene, partial sequence; mitochondrial.\n",
      "AY347457.1\n",
      "Discoglossus pictus auritus haplotype ph9 12S ribosomal RNA gene, partial sequence; mitochondrial.\n",
      "AY333687.1\n",
      "Bombina variegata isolate Bvs05 12S ribosomal RNA gene, partial sequence; mitochondrial gene for mitochondrial product.\n",
      "X86235.1\n",
      "Discoglossus pictus mitochondrial partial 12S rRNA gene.\n",
      "AY347458.1\n",
      "Discoglossus pictus auritus haplotype ph8 12S ribosomal RNA gene, partial sequence; mitochondrial.\n",
      "DQ221219.1\n",
      "Salamandra salamandra almanzoris isolate E1712.17 12S ribosomal RNA gene, partial sequence; mitochondrial.\n",
      "FJ595204.1\n",
      "Rana ridibunda 12S ribosomal RNA gene, partial sequence; mitochondrial.\n",
      "KF645371.1\n",
      "Salamandra salamandra gigliolii isolate SPM 780 12S ribosomal RNA gene, partial sequence; tRNA-Val gene, complete sequence; and 16S ribosomal RNA gene, partial sequence; mitochondrial.\n",
      "AY333655.1\n",
      "Bombina variegata isolate Bvs02 12S ribosomal RNA gene, partial sequence; mitochondrial gene for mitochondrial product.\n",
      "AB023397.1\n",
      "Rana ridibunda mitochondrial gene for 12S rRNA, partial sequence.\n",
      "AY333683.1\n",
      "Discoglossus pictus isolate Dpa01 12S ribosomal RNA gene, partial sequence; mitochondrial gene for mitochondrial product.\n",
      "X95494.1\n",
      "R.ridibunda mitochondrial 12S ribosomal RNA.\n",
      "U04704.1\n",
      "Triturus vulgaris mitochondrion 12S rRNA gene, partial sequence.\n",
      "AY880461.1\n",
      "Rana temporaria 12S ribosomal RNA gene, partial sequence; mitochondrial.\n",
      "KP109558.1\n",
      "Hyla arborea isolate H463 12S ribosomal RNA gene, partial sequence; mitochondrial.\n",
      "AY043041.1\n",
      "Rana temporaria 12S ribosomal RNA gene, partial sequence; mitochondrial gene for mitochondrial product.\n",
      "DQ092298.1\n",
      "Triturus alpestris inexpectatus isolate E1806.1 12S ribosomal RNA gene, partial sequence; mitochondrial.\n",
      "KF645367.1\n",
      "Salamandra salamandra longirostris isolate Sei3 12S ribosomal RNA gene, partial sequence; tRNA-Val gene, complete sequence; and 16S ribosomal RNA gene, partial sequence; mitochondrial.\n",
      "DQ221215.1\n",
      "Salamandra salamandra gallaica isolate E7110.11 12S ribosomal RNA gene, partial sequence; mitochondrial.\n",
      "KF645365.1\n",
      "Salamandra salamandra salamandra isolate GAR 02 12S ribosomal RNA gene, partial sequence; tRNA-Val gene, complete sequence; and 16S ribosomal RNA gene, partial sequence; mitochondrial.\n",
      "AY347455.1\n",
      "Discoglossus pictus auritus haplotype ph7 12S ribosomal RNA gene, partial sequence; mitochondrial.\n",
      "AY333684.1\n",
      "Discoglossus pictus isolate Dpa02 12S ribosomal RNA gene, partial sequence; mitochondrial gene for mitochondrial product.\n",
      "AY347454.1\n",
      "Discoglossus pictus auritus haplotype ph6 12S ribosomal RNA gene, partial sequence; mitochondrial.\n",
      "AY043034.1\n",
      "Rana lessonae 12S ribosomal RNA gene, partial sequence; mitochondrial gene for mitochondrial product.\n",
      "AJ440762.1\n",
      "Bombina bombina mitochondrion partial 12S rRNA gene.\n",
      "FJ602097.1\n",
      "Hydromantes genei isolate 10 12S ribosomal RNA gene, partial sequence; mitochondrial.\n",
      "DQ092286.1\n",
      "Triturus helveticus isolate E1806.17 12S ribosomal RNA gene, partial sequence; mitochondrial.\n",
      "AY333664.1\n",
      "Bombina variegata variegata 12S ribosomal RNA gene, partial sequence; mitochondrial gene for mitochondrial product.\n",
      "AY043032.1\n",
      "Rana ridibunda 12S ribosomal RNA gene, partial sequence; mitochondrial gene for mitochondrial product.\n",
      "KP109557.1\n",
      "Hyla arborea isolate H459 12S ribosomal RNA gene, partial sequence; mitochondrial.\n",
      "KF645373.1\n",
      "Salamandra salamandra alfredschmidti isolate SS 57 12S ribosomal RNA gene, partial sequence; tRNA-Val gene, complete sequence; and 16S ribosomal RNA gene, partial sequence; mitochondrial.\n",
      "KF645370.1\n",
      "Salamandra salamandra morenica isolate SS 34 12S ribosomal RNA gene, partial sequence; tRNA-Val gene, complete sequence; and 16S ribosomal RNA gene, partial sequence; mitochondrial.\n",
      "AY333666.1\n",
      "Alytes obstetricans 12S ribosomal RNA gene, partial sequence; mitochondrial gene for mitochondrial product.\n",
      "AB159478.1\n",
      "Bufo bufo mitochondrial genes for tRNA-Leu(CUN), tRNA-Thr, tRNA-Pro, tRNA-Phe, and 12S rRNA, complete and partial sequence, haplotype:B.bufo.\n",
      "KP109555.1\n",
      "Hyla arborea isolate H524 12S ribosomal RNA gene, partial sequence; mitochondrial.\n",
      "DQ221220.1\n",
      "Salamandra salamandra bernardezi isolate E1712.20 12S ribosomal RNA gene, partial sequence; mitochondrial.\n",
      "DQ221216.1\n",
      "Salamandra salamandra bejarae isolate E1712.44 12S ribosomal RNA gene, partial sequence; mitochondrial.\n",
      "AY347450.1\n",
      "Discoglossus pictus pictus haplotype ph1 12S ribosomal RNA gene, partial sequence; mitochondrial.\n",
      "AF124103.1\n",
      "Rana temporaria 12S ribosomal RNA gene, partial sequence; mitochondrial gene for mitochondrial product.\n",
      "AJ309142.1\n",
      "Rana ridibunda mitochondrial partial 12S rRNA gene (Albania:Durres).\n",
      "AY779191.1\n",
      "Rana temporaria voucher DMH 84-R1 12S ribosomal RNA gene, partial sequence; tRNA-Val gene, complete sequence; and 16S ribosomal RNA gene, partial sequence; mitochondrial.\n",
      "KP109552.1\n",
      "Hyla arborea isolate H211 12S ribosomal RNA gene, partial sequence; mitochondrial.\n",
      "KF645366.1\n",
      "Salamandra salamandra gallaica isolate Sei 6 12S ribosomal RNA gene, partial sequence; tRNA-Val gene, complete sequence; and 16S ribosomal RNA gene, partial sequence; mitochondrial.\n",
      "DQ221225.1\n",
      "Salamandra salamandra corsica isolate E3009.18 12S ribosomal RNA gene, partial sequence; mitochondrial.\n",
      "AY333686.1\n",
      "Discoglossus pictus isolate Dpp01 12S ribosomal RNA gene, partial sequence; mitochondrial gene for mitochondrial product.\n",
      "KF645364.1\n",
      "Salamandra salamandra fastuosa isolate SS 59 12S ribosomal RNA gene, partial sequence; tRNA-Val gene, complete sequence; and 16S ribosomal RNA gene, partial sequence; mitochondrial.\n",
      "DQ221223.1\n",
      "Salamandra salamandra longirostris isolate E1712.37 12S ribosomal RNA gene, partial sequence; mitochondrial.\n",
      "EU116957.1\n",
      "Hydromantes genei isolate E10076.5 12S ribosomal RNA gene, partial sequence; mitochondrial.\n",
      "KP109556.1\n",
      "Hyla arborea isolate H489 12S ribosomal RNA gene, partial sequence; mitochondrial.\n",
      "AY347453.1\n",
      "Discoglossus pictus auritus haplotype ph5 12S ribosomal RNA gene, partial sequence; mitochondrial.\n",
      "EU116960.1\n",
      "Hydromantes genei isolate E1037.1 12S ribosomal RNA gene, partial sequence; mitochondrial.\n",
      "M13046.1\n",
      "X.laevis mitochondrial origin of replication region containing the D-loop, Phe-tRNA, Pro-tRNA, 12S rRNA genes.\n",
      "AB023394.1\n",
      "Rana lessonae mitochondrial gene for 12S rRNA, partial sequence, strain:Italy.\n",
      "KP109554.1\n",
      "Hyla arborea isolate H288 12S ribosomal RNA gene, partial sequence; mitochondrial.\n",
      "AY222459.1\n",
      "Salamandra salamandra terrestris E171249 12S ribosomal RNA gene, partial sequence; mitochondrial gene for mitochondrial product.\n",
      "AY347459.1\n",
      "Discoglossus pictus auritus haplotype ph2 12S ribosomal RNA gene, partial sequence; mitochondrial.\n",
      "KT074476.1\n",
      "Rana temporaria voucher TE_OIA05 12S ribosomal RNA gene, partial sequence; mitochondrial.\n",
      "AF224726.1\n",
      "Alytes obstetricans boscai 12S ribosomal RNA gene, partial sequence; mitochondrial gene for mitochondrial product.\n",
      "DQ885561.1\n",
      "Gallus gallus murghi 12S ribosomal RNA gene, partial sequence; mitochondrial.\n",
      "JN695757.1\n",
      "Gallus gallus isolate S28 12S ribosomal RNA gene, partial sequence; mitochondrial.\n",
      "EF362674.1\n",
      "Gallus gallus isolate bar1-38 D-loop, partial sequence; tRNA-Phe gene, complete sequence; and 12S ribosomal RNA gene, partial sequence; mitochondrial.\n",
      "KF484729.1\n",
      "Gallus gallus clone 1 12S ribosomal RNA gene, partial sequence; mitochondrial.\n",
      "AJ849444.2\n",
      "Gallus gallus mitochondrial partial 12S rRNA gene.\n",
      "KF471521.1\n",
      "Gallus gallus 12S ribosomal RNA gene, partial sequence; mitochondrial.\n",
      "EF432760.1\n",
      "Gallus gallus isolate bar1-37 D-loop, partial sequence; tRNA-Phe gene, complete sequence; and 12S ribosomal RNA gene, partial sequence; mitochondrial.\n",
      "U81335.1\n",
      "Chelydra serpentina 12S mitochondrial ribosomal RNA, small subunit, mitochondrial gene, partial sequence.\n",
      "EF362642.1\n",
      "Gallus gallus isolate bar1-5 D-loop, partial sequence; tRNA-Phe gene, complete sequence; and 12S ribosomal RNA gene, partial sequence; mitochondrial.\n",
      "KF471523.1\n",
      "Gallus gallus 12S ribosomal RNA gene, partial sequence; mitochondrial.\n",
      "EF362717.1\n",
      "Gallus gallus isolate bar3-34 12S ribosomal RNA and tRNA-Val genes, partial sequence; mitochondrial.\n",
      "HM637771.1\n",
      "Gallus gallus 12S ribosomal RNA gene, partial sequence; mitochondrial.\n",
      "EU851896.1\n",
      "Gallus gallus isolate Gallus1 12S ribosomal RNA gene, partial sequence; mitochondrial.\n",
      "GU477775.1\n",
      "Chelydra serpentina 12S ribosomal RNA gene, partial sequence; mitochondrial.\n",
      "KF471516.1\n",
      "Gallus gallus 12S ribosomal RNA gene, partial sequence; mitochondrial.\n",
      "EF362643.1\n",
      "Gallus gallus isolate bar1-26 D-loop, partial sequence; tRNA-Phe gene, complete sequence; and 12S ribosomal RNA gene, partial sequence; mitochondrial.\n",
      "EF362639.1\n",
      "Gallus gallus isolate bar1-1 D-loop, partial sequence; tRNA-Phe gene, complete sequence; and 12S ribosomal RNA gene, partial sequence; mitochondrial.\n",
      "KF471518.1\n",
      "Gallus gallus 12S ribosomal RNA gene, partial sequence; mitochondrial.\n",
      "KP234022.1\n",
      "Gallus gallus isolate GG clone J3 12S ribosomal RNA gene, partial sequence; mitochondrial.\n",
      "KF471524.1\n",
      "Gallus gallus 12S ribosomal RNA gene, partial sequence; mitochondrial.\n",
      "KF471520.1\n",
      "Gallus gallus 12S ribosomal RNA gene, partial sequence; mitochondrial.\n",
      "KP171705.1\n",
      "Gallus gallus 12S ribosomal RNA gene, partial sequence; mitochondrial.\n",
      "EF362708.1\n",
      "Gallus gallus isolate bar3-21 12S ribosomal RNA and tRNA-Val genes, partial sequence; mitochondrial.\n"
     ]
    }
   ],
   "source": [
    "for r in pj.records:\n",
    "    if len(r.seq) < 500:\n",
    "        print r.id\n",
    "        print r.description"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Pickle the project."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 58,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'amphib_post_SATIVA.pkl'"
      ]
     },
     "execution_count": 58,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "pickle_pj(pj, 'amphib_post_SATIVA.pkl')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Add short sequences to reference alignment"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "# hmmbuild :: profile HMM construction from multiple sequence alignments\n",
      "# HMMER 3.1b1 (May 2013); http://hmmer.org/\n",
      "# Copyright (C) 2013 Howard Hughes Medical Institute.\n",
      "# Freely distributed under the GNU General Public License (GPLv3).\n",
      "# - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -\n",
      "# input alignment file:             500bp_amphib_align_SATIVA.fasta\n",
      "# output HMM file:                  12S_amphib_ref_SATIVA.hmm\n",
      "# - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -\n",
      "\n",
      "# idx name                  nseq  alen  mlen     W eff_nseq re/pos description\n",
      "#---- -------------------- ----- ----- ----- ----- -------- ------ -----------\n",
      "1     500bp_amphib_align_SATIVA    86  1182   943  1143     2.61  0.450 \n",
      "\n",
      "# CPU time: 0.37u 0.00s 00:00:00.37 Elapsed: 00:00:00.38\n"
     ]
    }
   ],
   "source": [
    "!hmmbuild 12S_amphib_ref_SATIVA.hmm 500bp_amphib_align_SATIVA.fasta"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Now use hmmalign to align the short 12S rRNA sequences to the longer reference alignment."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "%%bash\n",
    "\n",
    "hmmalign -o 12S_amphib500bpref_plus_short_query_SATIVA.sto \\\n",
    "--mapali 500bp_amphib_align_SATIVA.fasta \\\n",
    "12S_amphib_ref_SATIVA.hmm amphib_short_to_align_SATIVA.fasta"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "The hmmalign command has produced a file containing a new alignment of all 12S rRNA sequences (long and short) in stockholm output format. This must be converted to a more useful file format such as fasta or GenBank using the Seq command in BioPython."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Converted 198 records\n"
     ]
    }
   ],
   "source": [
    "from Bio import AlignIO\n",
    "\n",
    "input_file = open(\"12S_amphib500bpref_plus_short_query_SATIVA.sto\", \"rU\")\n",
    "output_file = open(\"12S_AmphibAlign_ALL_SATIVA.fasta\", \"w\")\n",
    "\n",
    "alignment = AlignIO.read(input_file, \"stockholm\")\n",
    "count = len(alignment)\n",
    "AlignIO.write(alignment, output_file, \"fasta\")\n",
    "\n",
    "output_file.close()\n",
    "input_file.close()\n",
    "print(\"Converted %i records\" % count)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Tree Building"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "The fasta file containing the 12S rRNA amphibian sequence alignment must now be converted into a format which can be used for alignment trimming. The alignment must be parsed to an alignment object, aln."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "DEBUG:Cloud:Log file (/root/.picloud/cloud.log) opened\n"
     ]
    }
   ],
   "source": [
    "pj=unpickle_pj('amphib_post_SATIVA.pkl', git=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "AB188189.1_f0 is 1559bp\n",
      "EF122793.1_f0 is 1559bp\n",
      "JN860636.1_f0 is 1559bp\n",
      "JN627421.1_f0 is 1559bp\n",
      "JN627426.1_f0 is 1559bp\n",
      "JN627422.1_f0 is 1559bp\n",
      "KC977226.1_f0 is 1559bp\n",
      "KC977215.1_f0 is 1559bp\n",
      "AY326063.1_f0 is 1559bp\n",
      "KC977206.1_f0 is 1559bp\n",
      "KC977218.1_f0 is 1559bp\n",
      "KC977222.1_f0 is 1559bp\n",
      "KC977217.1_f0 is 1559bp\n",
      "KC977220.1_f0 is 1559bp\n",
      "KC977225.1_f0 is 1559bp\n",
      "KC977204.1_f0 is 1559bp\n",
      "AB685766.1_f0 is 1559bp\n",
      "KC977224.1_f0 is 1559bp\n",
      "KF049927.1_f0 is 1559bp\n",
      "M57527.1_f0 is 1559bp\n",
      "D12694.1_f0 is 1559bp\n",
      "KU840522.1_f0 is 1559bp\n",
      "AB761267.1_f0 is 1559bp\n",
      "DQ055835.1_f0 is 1559bp\n",
      "AY843601.1_f0 is 1559bp\n",
      "AY325988.1_f0 is 1559bp\n",
      "DQ158438.1_f0 is 1559bp\n",
      "U52726.1_f0 is 1559bp\n",
      "EU938400.1_f0 is 1559bp\n",
      "EU880331.1_f0 is 1559bp\n",
      "AY928619.1_f0 is 1559bp\n",
      "EU880335.1_f0 is 1559bp\n",
      "EU880336.1_f0 is 1559bp\n",
      "EU880339.1_f0 is 1559bp\n",
      "KF471517.1_f0 is 1559bp\n",
      "EF373884.1_f0 is 1559bp\n",
      "EF373885.1_f0 is 1559bp\n",
      "EF373883.1_f0 is 1559bp\n",
      "KP211422.1_f0 is 1559bp\n",
      "EF373899.1_f0 is 1559bp\n",
      "AP003321.1_f0 is 1559bp\n",
      "FJ610338.1_f0 is 1559bp\n",
      "FJ610339.1_f0 is 1559bp\n",
      "KX512321.1_f0 is 1559bp\n",
      "EF373904.1_f0 is 1559bp\n",
      "KM886936.1_f0 is 1559bp\n",
      "AY235570.1_f0 is 1559bp\n",
      "GU261698.1_f0 is 1559bp\n",
      "GU261674.1_f0 is 1559bp\n",
      "AP003317.1_f0 is 1559bp\n",
      "GU261680.1_f0 is 1559bp\n",
      "GU261678.1_f0 is 1559bp\n",
      "GU261693.1_f0 is 1559bp\n",
      "AB086102.1_f0 is 1559bp\n",
      "GU261683.1_f0 is 1559bp\n",
      "KT626848.1_f0 is 1559bp\n",
      "KP211419.1_f0 is 1559bp\n",
      "GU261707.1_f0 is 1559bp\n",
      "GU261694.1_f0 is 1559bp\n",
      "AP003318.1_f0 is 1559bp\n",
      "KJ858774.1_f0 is 1559bp\n",
      "KJ858773.1_f0 is 1559bp\n",
      "AY585337.1_f0 is 1559bp\n",
      "KJ858770.1_f0 is 1559bp\n",
      "KJ858771.1_f0 is 1559bp\n",
      "KJ858772.1_f0 is 1559bp\n",
      "KJ858776.1_f0 is 1559bp\n",
      "KJ858778.1_f0 is 1559bp\n",
      "M10217.1_f0 is 1559bp\n",
      "Y10943.1_f0 is 1559bp\n",
      "M27605.1_f0 is 1559bp\n",
      "AY581639.1_f0 is 1559bp\n",
      "AY581648.1_f0 is 1559bp\n",
      "AY581644.1_f0 is 1559bp\n",
      "AY581640.1_f0 is 1559bp\n",
      "AY581638.1_f0 is 1559bp\n",
      "JQ626644.1_f0 is 1559bp\n",
      "JQ626643.1_f0 is 1559bp\n",
      "EU115993.1_f0 is 1559bp\n",
      "JX893172.1_f0 is 1559bp\n",
      "AY971143.1_f0 is 1559bp\n",
      "JX893174.1_f0 is 1559bp\n",
      "JX893175.1_f0 is 1559bp\n",
      "JX893178.1_f0 is 1559bp\n",
      "JX893176.1_f0 is 1559bp\n",
      "JX893177.1_f0 is 1559bp\n",
      "AY322321.1_f0 is 1559bp\n",
      "FJ595208.1_f0 is 1559bp\n",
      "AY347451.1_f0 is 1559bp\n",
      "AF161036.1_f0 is 1559bp\n",
      "AJ440767.1_f0 is 1559bp\n",
      "KP109553.1_f0 is 1559bp\n",
      "KX269198.1_f0 is 1559bp\n",
      "AJ222711.1_f0 is 1559bp\n",
      "AY333688.1_f0 is 1559bp\n",
      "KF645375.1_f0 is 1559bp\n",
      "AY347456.1_f0 is 1559bp\n",
      "AY043038.1_f0 is 1559bp\n",
      "EU116959.1_f0 is 1559bp\n",
      "X95493.1_f0 is 1559bp\n",
      "AY347452.1_f0 is 1559bp\n",
      "AB023395.1_f0 is 1559bp\n",
      "AF218718.1_f0 is 1559bp\n",
      "AY333663.1_f0 is 1559bp\n",
      "AY222458.1_f0 is 1559bp\n",
      "KF664850.1_f0 is 1559bp\n",
      "AY333653.1_f0 is 1559bp\n",
      "KP109559.1_f0 is 1559bp\n",
      "KF645417.1_f0 is 1559bp\n",
      "AY333665.1_f0 is 1559bp\n",
      "AY147254.1_f0 is 1559bp\n",
      "KF645409.1_f0 is 1559bp\n",
      "AY147256.1_f0 is 1559bp\n",
      "U04702.1_f0 is 1559bp\n",
      "DQ092285.1_f0 is 1559bp\n",
      "KF645408.1_f0 is 1559bp\n",
      "AJ222710.1_f0 is 1559bp\n",
      "AY333669.1_f0 is 1559bp\n",
      "AY880481.1_f0 is 1559bp\n",
      "AY347457.1_f0 is 1559bp\n",
      "AY333687.1_f0 is 1559bp\n",
      "X86235.1_f0 is 1559bp\n",
      "AY347458.1_f0 is 1559bp\n",
      "DQ221219.1_f0 is 1559bp\n",
      "FJ595204.1_f0 is 1559bp\n",
      "KF645371.1_f0 is 1559bp\n",
      "AY333655.1_f0 is 1559bp\n",
      "AB023397.1_f0 is 1559bp\n",
      "AY333683.1_f0 is 1559bp\n",
      "U04704.1_f0 is 1559bp\n",
      "AY880461.1_f0 is 1559bp\n",
      "KP109558.1_f0 is 1559bp\n",
      "AY043041.1_f0 is 1559bp\n",
      "DQ092298.1_f0 is 1559bp\n",
      "KF645367.1_f0 is 1559bp\n",
      "DQ221215.1_f0 is 1559bp\n",
      "KF645365.1_f0 is 1559bp\n",
      "AY347455.1_f0 is 1559bp\n",
      "AY333684.1_f0 is 1559bp\n",
      "AY347454.1_f0 is 1559bp\n",
      "AY043034.1_f0 is 1559bp\n",
      "AJ440762.1_f0 is 1559bp\n",
      "FJ602097.1_f0 is 1559bp\n",
      "DQ092286.1_f0 is 1559bp\n",
      "AY333664.1_f0 is 1559bp\n",
      "AY043032.1_f0 is 1559bp\n",
      "KP109557.1_f0 is 1559bp\n",
      "KF645373.1_f0 is 1559bp\n",
      "KF645370.1_f0 is 1559bp\n",
      "AY333666.1_f0 is 1559bp\n",
      "AB159478.1_f0 is 1559bp\n",
      "KP109555.1_f0 is 1559bp\n",
      "DQ221220.1_f0 is 1559bp\n",
      "DQ221216.1_f0 is 1559bp\n",
      "AY347450.1_f0 is 1559bp\n",
      "AF124103.1_f0 is 1559bp\n",
      "AJ309142.1_f0 is 1559bp\n",
      "AY779191.1_f0 is 1559bp\n",
      "KP109552.1_f0 is 1559bp\n",
      "KF645366.1_f0 is 1559bp\n",
      "DQ221225.1_f0 is 1559bp\n",
      "AY333686.1_f0 is 1559bp\n",
      "KF645364.1_f0 is 1559bp\n",
      "DQ221223.1_f0 is 1559bp\n",
      "EU116957.1_f0 is 1559bp\n",
      "KP109556.1_f0 is 1559bp\n",
      "AY347453.1_f0 is 1559bp\n",
      "EU116960.1_f0 is 1559bp\n",
      "M13046.1_f0 is 1559bp\n",
      "AB023394.1_f0 is 1559bp\n",
      "KP109554.1_f0 is 1559bp\n",
      "AY222459.1_f0 is 1559bp\n",
      "AY347459.1_f0 is 1559bp\n",
      "KT074476.1_f0 is 1559bp\n",
      "AF224726.1_f0 is 1559bp\n",
      "DQ885561.1_f0 is 1559bp\n",
      "JN695757.1_f0 is 1559bp\n",
      "EF362674.1_f0 is 1559bp\n",
      "KF484729.1_f0 is 1559bp\n",
      "AJ849444.2_f0 is 1559bp\n",
      "KF471521.1_f0 is 1559bp\n",
      "EF432760.1_f0 is 1559bp\n",
      "U81335.1_f0 is 1559bp\n",
      "EF362642.1_f0 is 1559bp\n",
      "KF471523.1_f0 is 1559bp\n",
      "EF362717.1_f0 is 1559bp\n",
      "HM637771.1_f0 is 1559bp\n",
      "EU851896.1_f0 is 1559bp\n",
      "GU477775.1_f0 is 1559bp\n",
      "KF471516.1_f0 is 1559bp\n",
      "EF362643.1_f0 is 1559bp\n",
      "EF362639.1_f0 is 1559bp\n",
      "KF471518.1_f0 is 1559bp\n",
      "KP234022.1_f0 is 1559bp\n",
      "KF471524.1_f0 is 1559bp\n",
      "KF471520.1_f0 is 1559bp\n",
      "KP171705.1_f0 is 1559bp\n",
      "EF362708.1_f0 is 1559bp\n"
     ]
    }
   ],
   "source": [
    "from Bio import AlignIO\n",
    "from Bio.Alphabet import generic_dna\n",
    "\n",
    "aln = AlignIO.read(open(\"12S_amphib500bpref_plus_short_query_SATIVA.sto\"), \"stockholm\", alphabet=generic_dna)\n",
    "for r in aln:\n",
    "    print(r.id + \" is \" + str(len(r)) + \"bp\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "pj.alignments['rrnS@amphib_align_allseq_SATIVA'] = aln"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'rrnS@amphib_align_allseq_SATIVA': <<class 'Bio.Align.MultipleSeqAlignment'> instance (198 records of length 1559, DNAAlphabet()) at 7f3b048f3f50>,\n",
       " 'rrnS@muscleDefault-500bp-SATIVA': <<class 'Bio.Align.MultipleSeqAlignment'> instance (86 records of length 1291, IUPACProtein()) at 7f3b04916790>}"
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "pj.alignments"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['rrnS@amphib_align_allseq_SATIVA', 'rrnS@muscleDefault-500bp-SATIVA']"
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "pj.alignments.keys()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Trim the alignment."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "trimal -in 506711477489096.71_rrnS@amphib_align_allseq_SATIVA.fasta -gappyout\n"
     ]
    }
   ],
   "source": [
    "gappyout = TrimalConf(pj,                      # The Project\n",
    "\n",
    "                      method_name='gappyout',  # Any unique string \n",
    "                                               # ('gappyout' is default)\n",
    "\n",
    "                      program_name='trimal',   # No alternatives in \n",
    "                                               # this ReproPhylo version\n",
    "\n",
    "                      cmd='default',           # the default is trimal. \n",
    "                                               # Change it here\n",
    "                                               # or in pj.defaults['trimal']\n",
    "\n",
    "                      alns=['rrnS@amphib_align_allseq_SATIVA'],# 'all' by default\n",
    "\n",
    "                      trimal_commands={'gappyout': True} # By default, the gappyout algorithm is used.\n",
    "                      )"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "pj.trim([gappyout])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "TrimalConf named gappyout with ID 506711477489096.71\n",
      "Alignments: rrnS@amphib_align_allseq_SATIVA \n",
      "Created on: Wed Oct 26 13:38:16 2016\n",
      "Commands:\n",
      "rrnS@amphib_align_allseq_SATIVA@gappyout: trimal -in 506711477489096.71_rrnS@amphib_align_allseq_SATIVA.fasta -gappyout\n",
      "\n",
      "Environment:Platform: Linux-4.4.0-38-generic-x86_64-with-Ubuntu-14.04-trusty\n",
      " Processor: x86_64\n",
      " Python build: defaultMar 22 2014 22:59:56\n",
      " Python compiler: GCC 4.8.2\n",
      " Python implementation: CPython\n",
      " Python version: 2.7.6\n",
      " ete2 version: 2.2.1072\n",
      " biopython version: 1.68\n",
      " dendropy version: 3.12.2\n",
      " cloud version: 2.8.5\n",
      " reprophylo version 1.3\n",
      " User: biopc4\n",
      " Program and version: trimAl 1.2rev59\n",
      " Program reference: Salvador Capella-Gutierrez; Jose M. Silla-Martinez; Toni Gabaldon. trimAl: a tool for automated alignment trimming in large-scale phylogenetic analyses. Bioinformatics 2009 25: 1972-1973.\n",
      "execution time:\n",
      "1.96037101746\n",
      "\n",
      "==============================\n",
      "Core Methods section sentence:\n",
      "==============================\n",
      "The alignment(s) rrnS@amphib_align_allseq_SATIVA were trimmed using the program trimAl 1.2rev59 [1].\n",
      "\n",
      "Reference:\n",
      "Salvador Capella-Gutierrez; Jose M. Silla-Martinez; Toni Gabaldon. trimAl: a tool for automated alignment trimming in large-scale phylogenetic analyses. Bioinformatics 2009 25: 1972-1973.\n",
      "\n"
     ]
    }
   ],
   "source": [
    "print pj.used_methods['gappyout']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'rrnS@amphib_align_allseq_SATIVA@gappyout': <<class 'Bio.Align.MultipleSeqAlignment'> instance (198 records of length 1189, IUPACAmbiguousDNA()) at 7f3b04e9ec90>,\n",
       " 'rrnS@muscleDefault-500bp-SATIVA@gappyout': <<class 'Bio.Align.MultipleSeqAlignment'> instance (86 records of length 1182, IUPACAmbiguousDNA()) at 7f3b054f39d0>}"
      ]
     },
     "execution_count": 13,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "pj.trimmed_alignments"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['rrnS@muscleDefault-500bp-SATIVA@gappyout_trimmed_aln.fasta',\n",
       " 'rrnS@amphib_align_allseq_SATIVA@gappyout_trimmed_aln.fasta']"
      ]
     },
     "execution_count": 14,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "files = pj.write_trimmed_alns(id=['record_id','source_organism'],\n",
    "                                   format='fasta')\n",
    "files"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Infer Maximum Likelihood tree based on the clipped alignment. We are going to use RAxML.\n",
    "\n",
    "Configure RAxML."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "raxmlHPC-PTHREADS-SSE3 -f a -m GTRGAMMA -n 280851477489120.51_rrnS@amphib_align_allseq_SATIVA@gappyout0 -p 987 -s 280851477489120.51_rrnS@amphib_align_allseq_SATIVA@gappyout.fasta -T 5 -x 315 -N 100\n"
     ]
    }
   ],
   "source": [
    "raxml = RaxmlConf(pj, method_name='raxml-muscleDefault-trimmed',\n",
    "                  program_name='raxmlHPC-PTHREADS-SSE3',\n",
    "                  keepfiles=True,\n",
    "                  cmd='raxmlHPC-PTHREADS-SSE3',\n",
    "                  preset='fa',\n",
    "                  alns=['rrnS@amphib_align_allseq_SATIVA@gappyout'],\n",
    "                  model='GAMMA',\n",
    "                  threads=5,\n",
    "                  cline_args={'-#': 100})"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "pj.tree([raxml])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "RaxmlConf named raxml-muscleDefault-trimmed with ID 280851477489120.51\n",
      "Alignments: rrnS@amphib_align_allseq_SATIVA@gappyout \n",
      "Created on: Wed Oct 26 13:38:40 2016\n",
      "Commands:\n",
      "rrnS@amphib_align_allseq_SATIVA@gappyout:\n",
      "raxmlHPC-PTHREADS-SSE3 -f a -m GTRGAMMA -n 280851477489120.51_rrnS@amphib_align_allseq_SATIVA@gappyout0 -p 987 -s 280851477489120.51_rrnS@amphib_align_allseq_SATIVA@gappyout.fasta -T 5 -x 315 -N 100\n",
      "\n",
      "Environment:\n",
      "Platform: Linux-4.4.0-38-generic-x86_64-with-Ubuntu-14.04-trusty\n",
      " Processor: x86_64\n",
      " Python build: defaultMar 22 2014 22:59:56\n",
      " Python compiler: GCC 4.8.2\n",
      " Python implementation: CPython\n",
      " Python version: 2.7.6\n",
      " ete2 version: 2.2.1072\n",
      " biopython version: 1.68\n",
      " dendropy version: 3.12.2\n",
      " cloud version: 2.8.5\n",
      " reprophylo version 1.3\n",
      " User: biopc4\n",
      " Program and version: raxmlHPC-PTHREADS-SSE3RAxML version 8.0.12\n",
      " Program reference: A. Stamatakis: RAxML Version 8: A tool for Phylogenetic Analysis and Post-Analysis of Large Phylogenies. In Bioinformatics\n",
      " 2014.\n",
      "execution time:\n",
      "1863.336128\n",
      "\n",
      "==============================\n",
      "Core Methods section sentence:\n",
      "==============================\n",
      "Phylogenetic trees were reconstructed from the dataset(s) rrnS@amphib_align_allseq_SATIVA@gappyout using the program raxmlHPC-PTHREADS-SSE3RAxML version 8.0.12 [1].\n",
      "\n",
      "Reference:\n",
      "A. Stamatakis: RAxML Version 8: A tool for Phylogenetic Analysis and Post-Analysis of Large Phylogenies. In Bioinformatics, 2014.\n",
      "\n"
     ]
    }
   ],
   "source": [
    "print pj.used_methods['raxml-muscleDefault-trimmed']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['rrnS@muscleDefault-500bp-SATIVA@gappyout@raxml-muscleDefault-500bp-SATIVA-trimmed',\n",
       " 'rrnS@amphib_align_allseq_SATIVA@gappyout@raxml-muscleDefault-trimmed']"
      ]
     },
     "execution_count": 18,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "pj.trees.keys()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Pickle the project."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'amphib_post_SATIVA.pkl'"
      ]
     },
     "execution_count": 19,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "pickle_pj(pj, 'amphib_post_SATIVA.pkl')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Visualise tree"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Now let's visualize the tree containing all 12S rRNA sequences with outliers identified by SATIVA removed.\n",
    "\n",
    "- Prepare tree annotation:\n",
    "- define outgroups\n",
    "- specify support coloration"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "pj.add_qualifier_from_source('organism')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "for species in ['Chelydra','Cottus','Gallus']:\n",
    "    pj.if_this_then_that(species, 'organism', 'outgroup', 'outgroup',\n",
    "                         mode='part')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "supports = {'black': [100,95],\n",
    "            'dimgray': [95,75],\n",
    "            'silver': [75,50]}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "pj.clear_tree_annotations()\n",
    "pj.annotate('.', 'outgroup', 'outgroup', \n",
    "            ['source_organism', 'record_id'], \n",
    "            node_support_dict=supports, multifurc=10,\n",
    "            html='muscleDefault-trimmed-amphib-all-SATIVA-raxml.html')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Create a static copy of the tree (the last *.png with the expected alignment id) for later."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "%%bash\n",
    "\n",
    "last_png=$(ls -hlrt *rrnS@amphib_align_allseq_SATIVA@gappyout.png | tail -n 1 | perl -ne 'chomp; @a=split(\" \"); print \"$a[-1]\"')\n",
    "cp $last_png rrnS_muscleDefault-trimmed-amphib-all-SATIVA-raxml.png"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Pickle the project."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'amphib_post_SATIVA.pkl'"
      ]
     },
     "execution_count": 25,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "pickle_pj(pj, 'amphib_post_SATIVA.pkl')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Write out alignment for pplacer input. Need to modify record id - remove the '_f0' that is appended by Reprophylo."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "returning trimmed alignment object rrnS@amphib_align_allseq_SATIVA@gappyout\n"
     ]
    }
   ],
   "source": [
    "from Bio import AlignIO\n",
    "\n",
    "aln = pj.fta('rrnS@amphib_align_allseq_SATIVA@gappyout')\n",
    "\n",
    "for r in aln:\n",
    "    r.id = \"_\".join(r.id.split(\"_\")[:-1])\n",
    "    r.description = r.id\n",
    "    \n",
    "out=open('../12S_UKamphibians_SATIVA_cleaned.alignment.fasta', 'w')\n",
    "AlignIO.write(aln, out, 'fasta')\n",
    "out.close()\n",
    "\n",
    "out=open('../12S_UKamphibians_SATIVA_cleaned.alignment.phylip', 'w')\n",
    "AlignIO.write(aln, out, 'phylip-relaxed')\n",
    "out.close()"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 2",
   "language": "python",
   "name": "python2"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 2
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython2",
   "version": "2.7.6"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 0
}
