{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Great crested newt eDNA metabarcoding data processing"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "I will use [metaBEAT](https://github.com/HullUni-bioinformatics/metaBEAT), a tool tailored towards reproducible and efficient analyses of metabarcoding data that was developed by Christoph Hahn (University of Graz) for the EvoHull group at University of Hull. The pipeline is still under active development and will likely be extended further in the future. The pipeline is available in a Docker container with all necessary dependencies. The Docker image builds on [ReproPhylo](https://github.com/HullUni-bioinformatics/ReproPhylo).\n",
    "\n",
    "The data I will be analyzing are 12S rRNA sequences amplified from eDNA samples collected from ponds across the UK. The experiment was designed determine the sensitivity of eDNA metabarcoding for rare species, such as great crested newt, and assess the potential of eDNA metabarcoding to assess vertebrate community composition in freshwater ponds. \n",
    "\n",
    "The metaBEAT tool is designed for complete bioinformatic analysis from raw data, and performs (optionally) de-multiplexing, quality filtering, chimera detection, clustering, and taxononomic assignment (outputs in biom and tsv formats). It currently supports BLAST, Kraken and phylogenetic placement (pplacer). Further approaches will be included in the future to allow for efficient and standardized comparative assessments of all approaches. A large number of options are offered at each step of the pipeline to tailor bioinformatic analysis to different projects."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Data input"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "This notebook will perform basic processing (trimming, merging, chimera removal) of the eDNA data. Clustering and taxonomic assignment will be performed in a separate notebook.\n",
    "\n",
    "Minimum input for an analysis is a set of query sequences in one or several files (accepted are a number of file formats, e.g. fasta, fastq). These will be run through the pipeline sequentially.\n",
    "\n",
    "Information on the nature and location of the query sequence files must be provided in a separate tab-delimited text file via the -Q flags.\n",
    "\n",
    "Each line in this text file should look as follows: unique sample_ID, format, file1, file2\n",
    "\n",
    "The required text files can be generated in any text editor. So theoretically, nano could be used in the terminal to construct the text file. For reproducibility and ease, a simple program can be used to generate the required file.\n",
    "\n",
    "In the cell below, it is produced using a simple python script. The script will list all files in the location to which you've downloaded your Illumina data (specified via the 'datadir' variable). It assumes that there is a file ending in _R1.fastq for each sample. For each such file, it will extract the sample name from the filename and format the required line for the text file accordingly. The resulting file is called Querymap.txt (specified in the 'to' variable)."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/home/working/Harper_et_al_2018\n"
     ]
    }
   ],
   "source": [
    "cd .."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/home/working/Harper_et_al_2018\r\n"
     ]
    }
   ],
   "source": [
    "!pwd"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "!mkdir 1-trimming"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/home/working/Harper_et_al_2018/1-trimming\n"
     ]
    }
   ],
   "source": [
    "cd 1-trimming"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "F100-12S_R1.fastq.gz\r\n",
      "F100-12S_R2.fastq.gz\r\n",
      "F101-12S_R1.fastq.gz\r\n",
      "F101-12S_R2.fastq.gz\r\n",
      "F10-12S_R1.fastq.gz\r\n",
      "F10-12S_R2.fastq.gz\r\n",
      "F102-12S_R1.fastq.gz\r\n",
      "F102-12S_R2.fastq.gz\r\n",
      "F103-12S_R1.fastq.gz\r\n",
      "F103-12S_R2.fastq.gz\r\n",
      "F104-12S_R1.fastq.gz\r\n",
      "F104-12S_R2.fastq.gz\r\n",
      "F105-12S_R1.fastq.gz\r\n",
      "F105-12S_R2.fastq.gz\r\n",
      "F106-12S_R1.fastq.gz\r\n",
      "F106-12S_R2.fastq.gz\r\n",
      "F107-12S_R1.fastq.gz\r\n",
      "F107-12S_R2.fastq.gz\r\n",
      "F108-12S_R1.fastq.gz\r\n",
      "F108-12S_R2.fastq.gz\r\n",
      "F109-12S_R1.fastq.gz\r\n",
      "F109-12S_R2.fastq.gz\r\n",
      "F110-12S_R1.fastq.gz\r\n",
      "F110-12S_R2.fastq.gz\r\n",
      "F111-12S_R1.fastq.gz\r\n",
      "F111-12S_R2.fastq.gz\r\n",
      "F11-12S_R1.fastq.gz\r\n",
      "F11-12S_R2.fastq.gz\r\n",
      "F112-12S_R1.fastq.gz\r\n",
      "F112-12S_R2.fastq.gz\r\n",
      "F1-12S_R1.fastq.gz\r\n",
      "F1-12S_R2.fastq.gz\r\n",
      "F113-12S_R1.fastq.gz\r\n",
      "F113-12S_R2.fastq.gz\r\n",
      "F114-12S_R1.fastq.gz\r\n",
      "F114-12S_R2.fastq.gz\r\n",
      "F115-12S_R1.fastq.gz\r\n",
      "F115-12S_R2.fastq.gz\r\n",
      "F116-12S_R1.fastq.gz\r\n",
      "F116-12S_R2.fastq.gz\r\n",
      "F117-12S_R1.fastq.gz\r\n",
      "F117-12S_R2.fastq.gz\r\n",
      "F118-12S_R1.fastq.gz\r\n",
      "F118-12S_R2.fastq.gz\r\n",
      "F119-12S_R1.fastq.gz\r\n",
      "F119-12S_R2.fastq.gz\r\n",
      "F120-12S_R1.fastq.gz\r\n",
      "F120-12S_R2.fastq.gz\r\n",
      "F121-12S_R1.fastq.gz\r\n",
      "F121-12S_R2.fastq.gz\r\n",
      "F12-12S_R1.fastq.gz\r\n",
      "F12-12S_R2.fastq.gz\r\n",
      "F122-12S_R1.fastq.gz\r\n",
      "F122-12S_R2.fastq.gz\r\n",
      "F123-12S_R1.fastq.gz\r\n",
      "F123-12S_R2.fastq.gz\r\n",
      "F124-12S_R1.fastq.gz\r\n",
      "F124-12S_R2.fastq.gz\r\n",
      "F125-12S_R1.fastq.gz\r\n",
      "F125-12S_R2.fastq.gz\r\n",
      "F126-12S_R1.fastq.gz\r\n",
      "F126-12S_R2.fastq.gz\r\n",
      "F127-12S_R1.fastq.gz\r\n",
      "F127-12S_R2.fastq.gz\r\n",
      "F128-12S_R1.fastq.gz\r\n",
      "F128-12S_R2.fastq.gz\r\n",
      "F129-12S_R1.fastq.gz\r\n",
      "F129-12S_R2.fastq.gz\r\n",
      "F130-12S_R1.fastq.gz\r\n",
      "F130-12S_R2.fastq.gz\r\n",
      "F131-12S_R1.fastq.gz\r\n",
      "F131-12S_R2.fastq.gz\r\n",
      "F13-12S_R1.fastq.gz\r\n",
      "F13-12S_R2.fastq.gz\r\n",
      "F132-12S_R1.fastq.gz\r\n",
      "F132-12S_R2.fastq.gz\r\n",
      "F133-12S_R1.fastq.gz\r\n",
      "F133-12S_R2.fastq.gz\r\n",
      "F134-12S_R1.fastq.gz\r\n",
      "F134-12S_R2.fastq.gz\r\n",
      "F135-12S_R1.fastq.gz\r\n",
      "F135-12S_R2.fastq.gz\r\n",
      "F136-12S_R1.fastq.gz\r\n",
      "F136-12S_R2.fastq.gz\r\n",
      "F137-12S_R1.fastq.gz\r\n",
      "F137-12S_R2.fastq.gz\r\n",
      "F138-12S_R1.fastq.gz\r\n",
      "F138-12S_R2.fastq.gz\r\n",
      "F139-12S_R1.fastq.gz\r\n",
      "F139-12S_R2.fastq.gz\r\n",
      "F140-12S_R1.fastq.gz\r\n",
      "F140-12S_R2.fastq.gz\r\n",
      "F141-12S_R1.fastq.gz\r\n",
      "F141-12S_R2.fastq.gz\r\n",
      "F14-12S_R1.fastq.gz\r\n",
      "F14-12S_R2.fastq.gz\r\n",
      "F142-12S_R1.fastq.gz\r\n",
      "F142-12S_R2.fastq.gz\r\n",
      "F143-12S_R1.fastq.gz\r\n",
      "F143-12S_R2.fastq.gz\r\n",
      "F144-12S_R1.fastq.gz\r\n",
      "F144-12S_R2.fastq.gz\r\n",
      "F145-12S_R1.fastq.gz\r\n",
      "F145-12S_R2.fastq.gz\r\n",
      "F146-12S_R1.fastq.gz\r\n",
      "F146-12S_R2.fastq.gz\r\n",
      "F147-12S_R1.fastq.gz\r\n",
      "F147-12S_R2.fastq.gz\r\n",
      "F148-12S_R1.fastq.gz\r\n",
      "F148-12S_R2.fastq.gz\r\n",
      "F149-12S_R1.fastq.gz\r\n",
      "F149-12S_R2.fastq.gz\r\n",
      "F150-12S_R1.fastq.gz\r\n",
      "F150-12S_R2.fastq.gz\r\n",
      "F151-12S_R1.fastq.gz\r\n",
      "F151-12S_R2.fastq.gz\r\n",
      "F15-12S_R1.fastq.gz\r\n",
      "F15-12S_R2.fastq.gz\r\n",
      "F152-12S_R1.fastq.gz\r\n",
      "F152-12S_R2.fastq.gz\r\n",
      "F153-12S_R1.fastq.gz\r\n",
      "F153-12S_R2.fastq.gz\r\n",
      "F154-12S_R1.fastq.gz\r\n",
      "F154-12S_R2.fastq.gz\r\n",
      "F155-12S_R1.fastq.gz\r\n",
      "F155-12S_R2.fastq.gz\r\n",
      "F156-12S_R1.fastq.gz\r\n",
      "F156-12S_R2.fastq.gz\r\n",
      "F157-12S_R1.fastq.gz\r\n",
      "F157-12S_R2.fastq.gz\r\n",
      "F158-12S_R1.fastq.gz\r\n",
      "F158-12S_R2.fastq.gz\r\n",
      "F159-12S_R1.fastq.gz\r\n",
      "F159-12S_R2.fastq.gz\r\n",
      "F160-12S_R1.fastq.gz\r\n",
      "F160-12S_R2.fastq.gz\r\n",
      "F161-12S_R1.fastq.gz\r\n",
      "F161-12S_R2.fastq.gz\r\n",
      "F16-12S_R1.fastq.gz\r\n",
      "F16-12S_R2.fastq.gz\r\n",
      "F162-12S_R1.fastq.gz\r\n",
      "F162-12S_R2.fastq.gz\r\n",
      "F163-12S_R1.fastq.gz\r\n",
      "F163-12S_R2.fastq.gz\r\n",
      "F164-12S_R1.fastq.gz\r\n",
      "F164-12S_R2.fastq.gz\r\n",
      "F165-12S_R1.fastq.gz\r\n",
      "F165-12S_R2.fastq.gz\r\n",
      "F166-12S_R1.fastq.gz\r\n",
      "F166-12S_R2.fastq.gz\r\n",
      "F167-12S_R1.fastq.gz\r\n",
      "F167-12S_R2.fastq.gz\r\n",
      "F168-12S_R1.fastq.gz\r\n",
      "F168-12S_R2.fastq.gz\r\n",
      "F169-12S_R1.fastq.gz\r\n",
      "F169-12S_R2.fastq.gz\r\n",
      "F170-12S_R1.fastq.gz\r\n",
      "F170-12S_R2.fastq.gz\r\n",
      "F171-12S_R1.fastq.gz\r\n",
      "F171-12S_R2.fastq.gz\r\n",
      "F17-12S_R1.fastq.gz\r\n",
      "F17-12S_R2.fastq.gz\r\n",
      "F172-12S_R1.fastq.gz\r\n",
      "F172-12S_R2.fastq.gz\r\n",
      "F173-12S_R1.fastq.gz\r\n",
      "F173-12S_R2.fastq.gz\r\n",
      "F174-12S_R1.fastq.gz\r\n",
      "F174-12S_R2.fastq.gz\r\n",
      "F175-12S_R1.fastq.gz\r\n",
      "F175-12S_R2.fastq.gz\r\n",
      "F176-12S_R1.fastq.gz\r\n",
      "F176-12S_R2.fastq.gz\r\n",
      "F177-12S_R1.fastq.gz\r\n",
      "F177-12S_R2.fastq.gz\r\n",
      "F178-12S_R1.fastq.gz\r\n",
      "F178-12S_R2.fastq.gz\r\n",
      "F179-12S_R1.fastq.gz\r\n",
      "F179-12S_R2.fastq.gz\r\n",
      "F180-12S_R1.fastq.gz\r\n",
      "F180-12S_R2.fastq.gz\r\n",
      "F181-12S_R1.fastq.gz\r\n",
      "F181-12S_R2.fastq.gz\r\n",
      "F18-12S_R1.fastq.gz\r\n",
      "F18-12S_R2.fastq.gz\r\n",
      "F182-12S_R1.fastq.gz\r\n",
      "F182-12S_R2.fastq.gz\r\n",
      "F183-12S_R1.fastq.gz\r\n",
      "F183-12S_R2.fastq.gz\r\n",
      "F184-12S_R1.fastq.gz\r\n",
      "F184-12S_R2.fastq.gz\r\n",
      "F185-12S_R1.fastq.gz\r\n",
      "F185-12S_R2.fastq.gz\r\n",
      "F186-12S_R1.fastq.gz\r\n",
      "F186-12S_R2.fastq.gz\r\n",
      "F187-12S_R1.fastq.gz\r\n",
      "F187-12S_R2.fastq.gz\r\n",
      "F188-12S_R1.fastq.gz\r\n",
      "F188-12S_R2.fastq.gz\r\n",
      "F189-12S_R1.fastq.gz\r\n",
      "F189-12S_R2.fastq.gz\r\n",
      "F190-12S_R1.fastq.gz\r\n",
      "F190-12S_R2.fastq.gz\r\n",
      "F191-12S_R1.fastq.gz\r\n",
      "F191-12S_R2.fastq.gz\r\n",
      "F19-12S_R1.fastq.gz\r\n",
      "F19-12S_R2.fastq.gz\r\n",
      "F192-12S_R1.fastq.gz\r\n",
      "F192-12S_R2.fastq.gz\r\n",
      "F193-12S_R1.fastq.gz\r\n",
      "F193-12S_R2.fastq.gz\r\n",
      "F194-12S_R1.fastq.gz\r\n",
      "F194-12S_R2.fastq.gz\r\n",
      "F195-12S_R1.fastq.gz\r\n",
      "F195-12S_R2.fastq.gz\r\n",
      "F196-12S_R1.fastq.gz\r\n",
      "F196-12S_R2.fastq.gz\r\n",
      "F197-12S_R1.fastq.gz\r\n",
      "F197-12S_R2.fastq.gz\r\n",
      "F198-12S_R1.fastq.gz\r\n",
      "F198-12S_R2.fastq.gz\r\n",
      "F199-12S_R1.fastq.gz\r\n",
      "F199-12S_R2.fastq.gz\r\n",
      "F200-12S_R1.fastq.gz\r\n",
      "F200-12S_R2.fastq.gz\r\n",
      "F201-12S_R1.fastq.gz\r\n",
      "F201-12S_R2.fastq.gz\r\n",
      "F20-12S_R1.fastq.gz\r\n",
      "F20-12S_R2.fastq.gz\r\n",
      "F202-12S_R1.fastq.gz\r\n",
      "F202-12S_R2.fastq.gz\r\n",
      "F203-12S_R1.fastq.gz\r\n",
      "F203-12S_R2.fastq.gz\r\n",
      "F204-12S_R1.fastq.gz\r\n",
      "F204-12S_R2.fastq.gz\r\n",
      "F205-12S_R1.fastq.gz\r\n",
      "F205-12S_R2.fastq.gz\r\n",
      "F206-12S_R1.fastq.gz\r\n",
      "F206-12S_R2.fastq.gz\r\n",
      "F207-12S_R1.fastq.gz\r\n",
      "F207-12S_R2.fastq.gz\r\n",
      "F208-12S_R1.fastq.gz\r\n",
      "F208-12S_R2.fastq.gz\r\n",
      "F209-12S_R1.fastq.gz\r\n",
      "F209-12S_R2.fastq.gz\r\n",
      "F210-12S_R1.fastq.gz\r\n",
      "F210-12S_R2.fastq.gz\r\n",
      "F211-12S_R1.fastq.gz\r\n",
      "F211-12S_R2.fastq.gz\r\n",
      "F21-12S_R1.fastq.gz\r\n",
      "F21-12S_R2.fastq.gz\r\n",
      "F212-12S_R1.fastq.gz\r\n",
      "F212-12S_R2.fastq.gz\r\n",
      "F2-12S_R1.fastq.gz\r\n",
      "F2-12S_R2.fastq.gz\r\n",
      "F213-12S_R1.fastq.gz\r\n",
      "F213-12S_R2.fastq.gz\r\n",
      "F214-12S_R1.fastq.gz\r\n",
      "F214-12S_R2.fastq.gz\r\n",
      "F215-12S_R1.fastq.gz\r\n",
      "F215-12S_R2.fastq.gz\r\n",
      "F216-12S_R1.fastq.gz\r\n",
      "F216-12S_R2.fastq.gz\r\n",
      "F217-12S_R1.fastq.gz\r\n",
      "F217-12S_R2.fastq.gz\r\n",
      "F218-12S_R1.fastq.gz\r\n",
      "F218-12S_R2.fastq.gz\r\n",
      "F219-12S_R1.fastq.gz\r\n",
      "F219-12S_R2.fastq.gz\r\n",
      "F220-12S_R1.fastq.gz\r\n",
      "F220-12S_R2.fastq.gz\r\n",
      "F221-12S_R1.fastq.gz\r\n",
      "F221-12S_R2.fastq.gz\r\n",
      "F22-12S_R1.fastq.gz\r\n",
      "F22-12S_R2.fastq.gz\r\n",
      "F222-12S_R1.fastq.gz\r\n",
      "F222-12S_R2.fastq.gz\r\n",
      "F223-12S_R1.fastq.gz\r\n",
      "F223-12S_R2.fastq.gz\r\n",
      "F224-12S_R1.fastq.gz\r\n",
      "F224-12S_R2.fastq.gz\r\n",
      "F225-12S_R1.fastq.gz\r\n",
      "F225-12S_R2.fastq.gz\r\n",
      "F226-12S_R1.fastq.gz\r\n",
      "F226-12S_R2.fastq.gz\r\n",
      "F227-12S_R1.fastq.gz\r\n",
      "F227-12S_R2.fastq.gz\r\n",
      "F228-12S_R1.fastq.gz\r\n",
      "F228-12S_R2.fastq.gz\r\n",
      "F229-12S_R1.fastq.gz\r\n",
      "F229-12S_R2.fastq.gz\r\n",
      "F230-12S_R1.fastq.gz\r\n",
      "F230-12S_R2.fastq.gz\r\n",
      "F231-12S_R1.fastq.gz\r\n",
      "F231-12S_R2.fastq.gz\r\n",
      "F23-12S_R1.fastq.gz\r\n",
      "F23-12S_R2.fastq.gz\r\n",
      "F232-12S_R1.fastq.gz\r\n",
      "F232-12S_R2.fastq.gz\r\n",
      "F233-12S_R1.fastq.gz\r\n",
      "F233-12S_R2.fastq.gz\r\n",
      "F234-12S_R1.fastq.gz\r\n",
      "F234-12S_R2.fastq.gz\r\n",
      "F235-12S_R1.fastq.gz\r\n",
      "F235-12S_R2.fastq.gz\r\n",
      "F236-12S_R1.fastq.gz\r\n",
      "F236-12S_R2.fastq.gz\r\n",
      "F237-12S_R1.fastq.gz\r\n",
      "F237-12S_R2.fastq.gz\r\n",
      "F238-12S_R1.fastq.gz\r\n",
      "F238-12S_R2.fastq.gz\r\n",
      "F239-12S_R1.fastq.gz\r\n",
      "F239-12S_R2.fastq.gz\r\n",
      "F240-12S_R1.fastq.gz\r\n",
      "F240-12S_R2.fastq.gz\r\n",
      "F241-12S_R1.fastq.gz\r\n",
      "F241-12S_R2.fastq.gz\r\n",
      "F24-12S_R1.fastq.gz\r\n",
      "F24-12S_R2.fastq.gz\r\n",
      "F242-12S_R1.fastq.gz\r\n",
      "F242-12S_R2.fastq.gz\r\n",
      "F243-12S_R1.fastq.gz\r\n",
      "F243-12S_R2.fastq.gz\r\n",
      "F244-12S_R1.fastq.gz\r\n",
      "F244-12S_R2.fastq.gz\r\n",
      "F245-12S_R1.fastq.gz\r\n",
      "F245-12S_R2.fastq.gz\r\n",
      "F246-12S_R1.fastq.gz\r\n",
      "F246-12S_R2.fastq.gz\r\n",
      "F247-12S_R1.fastq.gz\r\n",
      "F247-12S_R2.fastq.gz\r\n",
      "F248-12S_R1.fastq.gz\r\n",
      "F248-12S_R2.fastq.gz\r\n",
      "F249-12S_R1.fastq.gz\r\n",
      "F249-12S_R2.fastq.gz\r\n",
      "F250-12S_R1.fastq.gz\r\n",
      "F250-12S_R2.fastq.gz\r\n",
      "F251-12S_R1.fastq.gz\r\n",
      "F251-12S_R2.fastq.gz\r\n",
      "F25-12S_R1.fastq.gz\r\n",
      "F25-12S_R2.fastq.gz\r\n",
      "F252-12S_R1.fastq.gz\r\n",
      "F252-12S_R2.fastq.gz\r\n",
      "F253-12S_R1.fastq.gz\r\n",
      "F253-12S_R2.fastq.gz\r\n",
      "F254-12S_R1.fastq.gz\r\n",
      "F254-12S_R2.fastq.gz\r\n",
      "F255-12S_R1.fastq.gz\r\n",
      "F255-12S_R2.fastq.gz\r\n",
      "F256-12S_R1.fastq.gz\r\n",
      "F256-12S_R2.fastq.gz\r\n",
      "F257-12S_R1.fastq.gz\r\n",
      "F257-12S_R2.fastq.gz\r\n",
      "F258-12S_R1.fastq.gz\r\n",
      "F258-12S_R2.fastq.gz\r\n",
      "F259-12S_R1.fastq.gz\r\n",
      "F259-12S_R2.fastq.gz\r\n",
      "F260-12S_R1.fastq.gz\r\n",
      "F260-12S_R2.fastq.gz\r\n",
      "F261-12S_R1.fastq.gz\r\n",
      "F261-12S_R2.fastq.gz\r\n",
      "F26-12S_R1.fastq.gz\r\n",
      "F26-12S_R2.fastq.gz\r\n",
      "F262-12S_R1.fastq.gz\r\n",
      "F262-12S_R2.fastq.gz\r\n",
      "F263-12S_R1.fastq.gz\r\n",
      "F263-12S_R2.fastq.gz\r\n",
      "F264-12S_R1.fastq.gz\r\n",
      "F264-12S_R2.fastq.gz\r\n",
      "F265-12S_R1.fastq.gz\r\n",
      "F265-12S_R2.fastq.gz\r\n",
      "F266-12S_R1.fastq.gz\r\n",
      "F266-12S_R2.fastq.gz\r\n",
      "F267-12S_R1.fastq.gz\r\n",
      "F267-12S_R2.fastq.gz\r\n",
      "F268-12S_R1.fastq.gz\r\n",
      "F268-12S_R2.fastq.gz\r\n",
      "F269-12S_R1.fastq.gz\r\n",
      "F269-12S_R2.fastq.gz\r\n",
      "F270-12S_R1.fastq.gz\r\n",
      "F270-12S_R2.fastq.gz\r\n",
      "F271-12S_R1.fastq.gz\r\n",
      "F271-12S_R2.fastq.gz\r\n",
      "F27-12S_R1.fastq.gz\r\n",
      "F27-12S_R2.fastq.gz\r\n",
      "F272-12S_R1.fastq.gz\r\n",
      "F272-12S_R2.fastq.gz\r\n",
      "F273-12S_R1.fastq.gz\r\n",
      "F273-12S_R2.fastq.gz\r\n",
      "F274-12S_R1.fastq.gz\r\n",
      "F274-12S_R2.fastq.gz\r\n",
      "F275-12S_R1.fastq.gz\r\n",
      "F275-12S_R2.fastq.gz\r\n",
      "F276-12S_R1.fastq.gz\r\n",
      "F276-12S_R2.fastq.gz\r\n",
      "F277-12S_R1.fastq.gz\r\n",
      "F277-12S_R2.fastq.gz\r\n",
      "F278-12S_R1.fastq.gz\r\n",
      "F278-12S_R2.fastq.gz\r\n",
      "F279-12S_R1.fastq.gz\r\n",
      "F279-12S_R2.fastq.gz\r\n",
      "F280-12S_R1.fastq.gz\r\n",
      "F280-12S_R2.fastq.gz\r\n",
      "F281-12S_R1.fastq.gz\r\n",
      "F281-12S_R2.fastq.gz\r\n",
      "F28-12S_R1.fastq.gz\r\n",
      "F28-12S_R2.fastq.gz\r\n",
      "F282-12S_R1.fastq.gz\r\n",
      "F282-12S_R2.fastq.gz\r\n",
      "F283-12S_R1.fastq.gz\r\n",
      "F283-12S_R2.fastq.gz\r\n",
      "F284-12S_R1.fastq.gz\r\n",
      "F284-12S_R2.fastq.gz\r\n",
      "F285-12S_R1.fastq.gz\r\n",
      "F285-12S_R2.fastq.gz\r\n",
      "F286-12S_R1.fastq.gz\r\n",
      "F286-12S_R2.fastq.gz\r\n",
      "F287-12S_R1.fastq.gz\r\n",
      "F287-12S_R2.fastq.gz\r\n",
      "F288-12S_R1.fastq.gz\r\n",
      "F288-12S_R2.fastq.gz\r\n",
      "F289-12S_R1.fastq.gz\r\n",
      "F289-12S_R2.fastq.gz\r\n",
      "F290-12S_R1.fastq.gz\r\n",
      "F290-12S_R2.fastq.gz\r\n",
      "F291-12S_R1.fastq.gz\r\n",
      "F291-12S_R2.fastq.gz\r\n",
      "F29-12S_R1.fastq.gz\r\n",
      "F29-12S_R2.fastq.gz\r\n",
      "F292-12S_R1.fastq.gz\r\n",
      "F292-12S_R2.fastq.gz\r\n",
      "F293-12S_R1.fastq.gz\r\n",
      "F293-12S_R2.fastq.gz\r\n",
      "F294-12S_R1.fastq.gz\r\n",
      "F294-12S_R2.fastq.gz\r\n",
      "F295-12S_R1.fastq.gz\r\n",
      "F295-12S_R2.fastq.gz\r\n",
      "F296-12S_R1.fastq.gz\r\n",
      "F296-12S_R2.fastq.gz\r\n",
      "F297-12S_R1.fastq.gz\r\n",
      "F297-12S_R2.fastq.gz\r\n",
      "F298-12S_R1.fastq.gz\r\n",
      "F298-12S_R2.fastq.gz\r\n",
      "F299-12S_R1.fastq.gz\r\n",
      "F299-12S_R2.fastq.gz\r\n",
      "F300-12S_R1.fastq.gz\r\n",
      "F300-12S_R2.fastq.gz\r\n",
      "F301-12S_R1.fastq.gz\r\n",
      "F301-12S_R2.fastq.gz\r\n",
      "F30-12S_R1.fastq.gz\r\n",
      "F30-12S_R2.fastq.gz\r\n",
      "F302-12S_R1.fastq.gz\r\n",
      "F302-12S_R2.fastq.gz\r\n",
      "F303-12S_R1.fastq.gz\r\n",
      "F303-12S_R2.fastq.gz\r\n",
      "F304-12S_R1.fastq.gz\r\n",
      "F304-12S_R2.fastq.gz\r\n",
      "F305-12S_R1.fastq.gz\r\n",
      "F305-12S_R2.fastq.gz\r\n",
      "F306-12S_R1.fastq.gz\r\n",
      "F306-12S_R2.fastq.gz\r\n",
      "F307-12S_R1.fastq.gz\r\n",
      "F307-12S_R2.fastq.gz\r\n",
      "F308-12S_R1.fastq.gz\r\n",
      "F308-12S_R2.fastq.gz\r\n",
      "F309-12S_R1.fastq.gz\r\n",
      "F309-12S_R2.fastq.gz\r\n",
      "F310-12S_R1.fastq.gz\r\n",
      "F310-12S_R2.fastq.gz\r\n",
      "F311-12S_R1.fastq.gz\r\n",
      "F311-12S_R2.fastq.gz\r\n",
      "F31-12S_R1.fastq.gz\r\n",
      "F31-12S_R2.fastq.gz\r\n",
      "F312-12S_R1.fastq.gz\r\n",
      "F312-12S_R2.fastq.gz\r\n",
      "F3-12S_R1.fastq.gz\r\n",
      "F3-12S_R2.fastq.gz\r\n",
      "F313-12S_R1.fastq.gz\r\n",
      "F313-12S_R2.fastq.gz\r\n",
      "F314-12S_R1.fastq.gz\r\n",
      "F314-12S_R2.fastq.gz\r\n",
      "F315-12S_R1.fastq.gz\r\n",
      "F315-12S_R2.fastq.gz\r\n",
      "F316-12S_R1.fastq.gz\r\n",
      "F316-12S_R2.fastq.gz\r\n",
      "F317-12S_R1.fastq.gz\r\n",
      "F317-12S_R2.fastq.gz\r\n",
      "F318-12S_R1.fastq.gz\r\n",
      "F318-12S_R2.fastq.gz\r\n",
      "F319-12S_R1.fastq.gz\r\n",
      "F319-12S_R2.fastq.gz\r\n",
      "F320-12S_R1.fastq.gz\r\n",
      "F320-12S_R2.fastq.gz\r\n",
      "F321-12S_R1.fastq.gz\r\n",
      "F321-12S_R2.fastq.gz\r\n",
      "F32-12S_R1.fastq.gz\r\n",
      "F32-12S_R2.fastq.gz\r\n",
      "F322-12S_R1.fastq.gz\r\n",
      "F322-12S_R2.fastq.gz\r\n",
      "F323-12S_R1.fastq.gz\r\n",
      "F323-12S_R2.fastq.gz\r\n",
      "F324-12S_R1.fastq.gz\r\n",
      "F324-12S_R2.fastq.gz\r\n",
      "F325-12S_R1.fastq.gz\r\n",
      "F325-12S_R2.fastq.gz\r\n",
      "F326-12S_R1.fastq.gz\r\n",
      "F326-12S_R2.fastq.gz\r\n",
      "F327-12S_R1.fastq.gz\r\n",
      "F327-12S_R2.fastq.gz\r\n",
      "F328-12S_R1.fastq.gz\r\n",
      "F328-12S_R2.fastq.gz\r\n",
      "F329-12S_R1.fastq.gz\r\n",
      "F329-12S_R2.fastq.gz\r\n",
      "F330-12S_R1.fastq.gz\r\n",
      "F330-12S_R2.fastq.gz\r\n",
      "F331-12S_R1.fastq.gz\r\n",
      "F331-12S_R2.fastq.gz\r\n",
      "F33-12S_R1.fastq.gz\r\n",
      "F33-12S_R2.fastq.gz\r\n",
      "F332-12S_R1.fastq.gz\r\n",
      "F332-12S_R2.fastq.gz\r\n",
      "F333-12S_R1.fastq.gz\r\n",
      "F333-12S_R2.fastq.gz\r\n",
      "F334-12S_R1.fastq.gz\r\n",
      "F334-12S_R2.fastq.gz\r\n",
      "F335-12S_R1.fastq.gz\r\n",
      "F335-12S_R2.fastq.gz\r\n",
      "F336-12S_R1.fastq.gz\r\n",
      "F336-12S_R2.fastq.gz\r\n",
      "F337-12S_R1.fastq.gz\r\n",
      "F337-12S_R2.fastq.gz\r\n",
      "F338-12S_R1.fastq.gz\r\n",
      "F338-12S_R2.fastq.gz\r\n",
      "F339-12S_R1.fastq.gz\r\n",
      "F339-12S_R2.fastq.gz\r\n",
      "F340-12S_R1.fastq.gz\r\n",
      "F340-12S_R2.fastq.gz\r\n",
      "F341-12S_R1.fastq.gz\r\n",
      "F341-12S_R2.fastq.gz\r\n",
      "F34-12S_R1.fastq.gz\r\n",
      "F34-12S_R2.fastq.gz\r\n",
      "F342-12S_R1.fastq.gz\r\n",
      "F342-12S_R2.fastq.gz\r\n",
      "F343-12S_R1.fastq.gz\r\n",
      "F343-12S_R2.fastq.gz\r\n",
      "F344-12S_R1.fastq.gz\r\n",
      "F344-12S_R2.fastq.gz\r\n",
      "F345-12S_R1.fastq.gz\r\n",
      "F345-12S_R2.fastq.gz\r\n",
      "F346-12S_R1.fastq.gz\r\n",
      "F346-12S_R2.fastq.gz\r\n",
      "F347-12S_R1.fastq.gz\r\n",
      "F347-12S_R2.fastq.gz\r\n",
      "F348-12S_R1.fastq.gz\r\n",
      "F348-12S_R2.fastq.gz\r\n",
      "F349-12S_R1.fastq.gz\r\n",
      "F349-12S_R2.fastq.gz\r\n",
      "F350-12S_R1.fastq.gz\r\n",
      "F350-12S_R2.fastq.gz\r\n",
      "F351-12S_R1.fastq.gz\r\n",
      "F351-12S_R2.fastq.gz\r\n",
      "F35-12S_R1.fastq.gz\r\n",
      "F35-12S_R2.fastq.gz\r\n",
      "F352-12S_R1.fastq.gz\r\n",
      "F352-12S_R2.fastq.gz\r\n",
      "F353-12S_R1.fastq.gz\r\n",
      "F353-12S_R2.fastq.gz\r\n",
      "F354-12S_R1.fastq.gz\r\n",
      "F354-12S_R2.fastq.gz\r\n",
      "F355-12S_R1.fastq.gz\r\n",
      "F355-12S_R2.fastq.gz\r\n",
      "F356-12S_R1.fastq.gz\r\n",
      "F356-12S_R2.fastq.gz\r\n",
      "F357-12S_R1.fastq.gz\r\n",
      "F357-12S_R2.fastq.gz\r\n",
      "F358-12S_R1.fastq.gz\r\n",
      "F358-12S_R2.fastq.gz\r\n",
      "F359-12S_R1.fastq.gz\r\n",
      "F359-12S_R2.fastq.gz\r\n",
      "F360-12S_R1.fastq.gz\r\n",
      "F360-12S_R2.fastq.gz\r\n",
      "F361-12S_R1.fastq.gz\r\n",
      "F361-12S_R2.fastq.gz\r\n",
      "F36-12S_R1.fastq.gz\r\n",
      "F36-12S_R2.fastq.gz\r\n",
      "F362-12S_R1.fastq.gz\r\n",
      "F362-12S_R2.fastq.gz\r\n",
      "F363-12S_R1.fastq.gz\r\n",
      "F363-12S_R2.fastq.gz\r\n",
      "F364-12S_R1.fastq.gz\r\n",
      "F364-12S_R2.fastq.gz\r\n",
      "F365-12S_R1.fastq.gz\r\n",
      "F365-12S_R2.fastq.gz\r\n",
      "F366-12S_R1.fastq.gz\r\n",
      "F366-12S_R2.fastq.gz\r\n",
      "F367-12S_R1.fastq.gz\r\n",
      "F367-12S_R2.fastq.gz\r\n",
      "F368-12S_R1.fastq.gz\r\n",
      "F368-12S_R2.fastq.gz\r\n",
      "F369-12S_R1.fastq.gz\r\n",
      "F369-12S_R2.fastq.gz\r\n",
      "F370-12S_R1.fastq.gz\r\n",
      "F370-12S_R2.fastq.gz\r\n",
      "F371-12S_R1.fastq.gz\r\n",
      "F371-12S_R2.fastq.gz\r\n",
      "F37-12S_R1.fastq.gz\r\n",
      "F37-12S_R2.fastq.gz\r\n",
      "F372-12S_R1.fastq.gz\r\n",
      "F372-12S_R2.fastq.gz\r\n",
      "F373-12S_R1.fastq.gz\r\n",
      "F373-12S_R2.fastq.gz\r\n",
      "F374-12S_R1.fastq.gz\r\n",
      "F374-12S_R2.fastq.gz\r\n",
      "F375-12S_R1.fastq.gz\r\n",
      "F375-12S_R2.fastq.gz\r\n",
      "F376-12S_R1.fastq.gz\r\n",
      "F376-12S_R2.fastq.gz\r\n",
      "F377-12S_R1.fastq.gz\r\n",
      "F377-12S_R2.fastq.gz\r\n",
      "F378-12S_R1.fastq.gz\r\n",
      "F378-12S_R2.fastq.gz\r\n",
      "F379-12S_R1.fastq.gz\r\n",
      "F379-12S_R2.fastq.gz\r\n",
      "F380-12S_R1.fastq.gz\r\n",
      "F380-12S_R2.fastq.gz\r\n",
      "F381-12S_R1.fastq.gz\r\n",
      "F381-12S_R2.fastq.gz\r\n",
      "F38-12S_R1.fastq.gz\r\n",
      "F38-12S_R2.fastq.gz\r\n",
      "F382-12S_R1.fastq.gz\r\n",
      "F382-12S_R2.fastq.gz\r\n",
      "F383-12S_R1.fastq.gz\r\n",
      "F383-12S_R2.fastq.gz\r\n",
      "F384-12S_R1.fastq.gz\r\n",
      "F384-12S_R2.fastq.gz\r\n",
      "F385-12S_R1.fastq.gz\r\n",
      "F385-12S_R2.fastq.gz\r\n",
      "F386-12S_R1.fastq.gz\r\n",
      "F386-12S_R2.fastq.gz\r\n",
      "F387-12S_R1.fastq.gz\r\n",
      "F387-12S_R2.fastq.gz\r\n",
      "F388-12S_R1.fastq.gz\r\n",
      "F388-12S_R2.fastq.gz\r\n",
      "F389-12S_R1.fastq.gz\r\n",
      "F389-12S_R2.fastq.gz\r\n",
      "F390-12S_R1.fastq.gz\r\n",
      "F390-12S_R2.fastq.gz\r\n",
      "F391-12S_R1.fastq.gz\r\n",
      "F391-12S_R2.fastq.gz\r\n",
      "F39-12S_R1.fastq.gz\r\n",
      "F39-12S_R2.fastq.gz\r\n",
      "F392-12S_R1.fastq.gz\r\n",
      "F392-12S_R2.fastq.gz\r\n",
      "F393-12S_R1.fastq.gz\r\n",
      "F393-12S_R2.fastq.gz\r\n",
      "F394-12S_R1.fastq.gz\r\n",
      "F394-12S_R2.fastq.gz\r\n",
      "F395-12S_R1.fastq.gz\r\n",
      "F395-12S_R2.fastq.gz\r\n",
      "F396-12S_R1.fastq.gz\r\n",
      "F396-12S_R2.fastq.gz\r\n",
      "F397-12S_R1.fastq.gz\r\n",
      "F397-12S_R2.fastq.gz\r\n",
      "F398-12S_R1.fastq.gz\r\n",
      "F398-12S_R2.fastq.gz\r\n",
      "F399-12S_R1.fastq.gz\r\n",
      "F399-12S_R2.fastq.gz\r\n",
      "F400-12S_R1.fastq.gz\r\n",
      "F400-12S_R2.fastq.gz\r\n",
      "F401-12S_R1.fastq.gz\r\n",
      "F401-12S_R2.fastq.gz\r\n",
      "F40-12S_R1.fastq.gz\r\n",
      "F40-12S_R2.fastq.gz\r\n",
      "F402-12S_R1.fastq.gz\r\n",
      "F402-12S_R2.fastq.gz\r\n",
      "F403-12S_R1.fastq.gz\r\n",
      "F403-12S_R2.fastq.gz\r\n",
      "F404-12S_R1.fastq.gz\r\n",
      "F404-12S_R2.fastq.gz\r\n",
      "F405-12S_R1.fastq.gz\r\n",
      "F405-12S_R2.fastq.gz\r\n",
      "F406-12S_R1.fastq.gz\r\n",
      "F406-12S_R2.fastq.gz\r\n",
      "F407-12S_R1.fastq.gz\r\n",
      "F407-12S_R2.fastq.gz\r\n",
      "F408-12S_R1.fastq.gz\r\n",
      "F408-12S_R2.fastq.gz\r\n",
      "F409-12S_R1.fastq.gz\r\n",
      "F409-12S_R2.fastq.gz\r\n",
      "F410-12S_R1.fastq.gz\r\n",
      "F410-12S_R2.fastq.gz\r\n",
      "F411-12S_R1.fastq.gz\r\n",
      "F411-12S_R2.fastq.gz\r\n",
      "F41-12S_R1.fastq.gz\r\n",
      "F41-12S_R2.fastq.gz\r\n",
      "F412-12S_R1.fastq.gz\r\n",
      "F412-12S_R2.fastq.gz\r\n",
      "F4-12S_R1.fastq.gz\r\n",
      "F4-12S_R2.fastq.gz\r\n",
      "F413-12S_R1.fastq.gz\r\n",
      "F413-12S_R2.fastq.gz\r\n",
      "F414-12S_R1.fastq.gz\r\n",
      "F414-12S_R2.fastq.gz\r\n",
      "F415-12S_R1.fastq.gz\r\n",
      "F415-12S_R2.fastq.gz\r\n",
      "F416-12S_R1.fastq.gz\r\n",
      "F416-12S_R2.fastq.gz\r\n",
      "F417-12S_R1.fastq.gz\r\n",
      "F417-12S_R2.fastq.gz\r\n",
      "F418-12S_R1.fastq.gz\r\n",
      "F418-12S_R2.fastq.gz\r\n",
      "F419-12S_R1.fastq.gz\r\n",
      "F419-12S_R2.fastq.gz\r\n",
      "F420-12S_R1.fastq.gz\r\n",
      "F420-12S_R2.fastq.gz\r\n",
      "F421-12S_R1.fastq.gz\r\n",
      "F421-12S_R2.fastq.gz\r\n",
      "F42-12S_R1.fastq.gz\r\n",
      "F42-12S_R2.fastq.gz\r\n",
      "F422-12S_R1.fastq.gz\r\n",
      "F422-12S_R2.fastq.gz\r\n",
      "F423-12S_R1.fastq.gz\r\n",
      "F423-12S_R2.fastq.gz\r\n",
      "F424-12S_R1.fastq.gz\r\n",
      "F424-12S_R2.fastq.gz\r\n",
      "F425-12S_R1.fastq.gz\r\n",
      "F425-12S_R2.fastq.gz\r\n",
      "F426-12S_R1.fastq.gz\r\n",
      "F426-12S_R2.fastq.gz\r\n",
      "F427-12S_R1.fastq.gz\r\n",
      "F427-12S_R2.fastq.gz\r\n",
      "F428-12S_R1.fastq.gz\r\n",
      "F428-12S_R2.fastq.gz\r\n",
      "F429-12S_R1.fastq.gz\r\n",
      "F429-12S_R2.fastq.gz\r\n",
      "F430-12S_R1.fastq.gz\r\n",
      "F430-12S_R2.fastq.gz\r\n",
      "F431-12S_R1.fastq.gz\r\n",
      "F431-12S_R2.fastq.gz\r\n",
      "F43-12S_R1.fastq.gz\r\n",
      "F43-12S_R2.fastq.gz\r\n",
      "F432-12S_R1.fastq.gz\r\n",
      "F432-12S_R2.fastq.gz\r\n",
      "F433-12S_R1.fastq.gz\r\n",
      "F433-12S_R2.fastq.gz\r\n",
      "F434-12S_R1.fastq.gz\r\n",
      "F434-12S_R2.fastq.gz\r\n",
      "F435-12S_R1.fastq.gz\r\n",
      "F435-12S_R2.fastq.gz\r\n",
      "F436-12S_R1.fastq.gz\r\n",
      "F436-12S_R2.fastq.gz\r\n",
      "F437-12S_R1.fastq.gz\r\n",
      "F437-12S_R2.fastq.gz\r\n",
      "F438-12S_R1.fastq.gz\r\n",
      "F438-12S_R2.fastq.gz\r\n",
      "F439-12S_R1.fastq.gz\r\n",
      "F439-12S_R2.fastq.gz\r\n",
      "F440-12S_R1.fastq.gz\r\n",
      "F440-12S_R2.fastq.gz\r\n",
      "F441-12S_R1.fastq.gz\r\n",
      "F441-12S_R2.fastq.gz\r\n",
      "F44-12S_R1.fastq.gz\r\n",
      "F44-12S_R2.fastq.gz\r\n",
      "F442-12S_R1.fastq.gz\r\n",
      "F442-12S_R2.fastq.gz\r\n",
      "F443-12S_R1.fastq.gz\r\n",
      "F443-12S_R2.fastq.gz\r\n",
      "F444-12S_R1.fastq.gz\r\n",
      "F444-12S_R2.fastq.gz\r\n",
      "F445-12S_R1.fastq.gz\r\n",
      "F445-12S_R2.fastq.gz\r\n",
      "F446-12S_R1.fastq.gz\r\n",
      "F446-12S_R2.fastq.gz\r\n",
      "F447-12S_R1.fastq.gz\r\n",
      "F447-12S_R2.fastq.gz\r\n",
      "F448-12S_R1.fastq.gz\r\n",
      "F448-12S_R2.fastq.gz\r\n",
      "F449-12S_R1.fastq.gz\r\n",
      "F449-12S_R2.fastq.gz\r\n",
      "F450-12S_R1.fastq.gz\r\n",
      "F450-12S_R2.fastq.gz\r\n",
      "F451-12S_R1.fastq.gz\r\n",
      "F451-12S_R2.fastq.gz\r\n",
      "F45-12S_R1.fastq.gz\r\n",
      "F45-12S_R2.fastq.gz\r\n",
      "F452-12S_R1.fastq.gz\r\n",
      "F452-12S_R2.fastq.gz\r\n",
      "F453-12S_R1.fastq.gz\r\n",
      "F453-12S_R2.fastq.gz\r\n",
      "F454-12S_R1.fastq.gz\r\n",
      "F454-12S_R2.fastq.gz\r\n",
      "F455-12S_R1.fastq.gz\r\n",
      "F455-12S_R2.fastq.gz\r\n",
      "F456-12S_R1.fastq.gz\r\n",
      "F456-12S_R2.fastq.gz\r\n",
      "F457-12S_R1.fastq.gz\r\n",
      "F457-12S_R2.fastq.gz\r\n",
      "F458-12S_R1.fastq.gz\r\n",
      "F458-12S_R2.fastq.gz\r\n",
      "F459-12S_R1.fastq.gz\r\n",
      "F459-12S_R2.fastq.gz\r\n",
      "F460-12S_R1.fastq.gz\r\n",
      "F460-12S_R2.fastq.gz\r\n",
      "F461-12S_R1.fastq.gz\r\n",
      "F461-12S_R2.fastq.gz\r\n",
      "F46-12S_R1.fastq.gz\r\n",
      "F46-12S_R2.fastq.gz\r\n",
      "F462-12S_R1.fastq.gz\r\n",
      "F462-12S_R2.fastq.gz\r\n",
      "F463-12S_R1.fastq.gz\r\n",
      "F463-12S_R2.fastq.gz\r\n",
      "F464-12S_R1.fastq.gz\r\n",
      "F464-12S_R2.fastq.gz\r\n",
      "F465-12S_R1.fastq.gz\r\n",
      "F465-12S_R2.fastq.gz\r\n",
      "F466-12S_R1.fastq.gz\r\n",
      "F466-12S_R2.fastq.gz\r\n",
      "F467-12S_R1.fastq.gz\r\n",
      "F467-12S_R2.fastq.gz\r\n",
      "F468-12S_R1.fastq.gz\r\n",
      "F468-12S_R2.fastq.gz\r\n",
      "F469-12S_R1.fastq.gz\r\n",
      "F469-12S_R2.fastq.gz\r\n",
      "F470-12S_R1.fastq.gz\r\n",
      "F470-12S_R2.fastq.gz\r\n",
      "F471-12S_R1.fastq.gz\r\n",
      "F471-12S_R2.fastq.gz\r\n",
      "F47-12S_R1.fastq.gz\r\n",
      "F47-12S_R2.fastq.gz\r\n",
      "F472-12S_R1.fastq.gz\r\n",
      "F472-12S_R2.fastq.gz\r\n",
      "F473-12S_R1.fastq.gz\r\n",
      "F473-12S_R2.fastq.gz\r\n",
      "F474-12S_R1.fastq.gz\r\n",
      "F474-12S_R2.fastq.gz\r\n",
      "F475-12S_R1.fastq.gz\r\n",
      "F475-12S_R2.fastq.gz\r\n",
      "F476-12S_R1.fastq.gz\r\n",
      "F476-12S_R2.fastq.gz\r\n",
      "F477-12S_R1.fastq.gz\r\n",
      "F477-12S_R2.fastq.gz\r\n",
      "F478-12S_R1.fastq.gz\r\n",
      "F478-12S_R2.fastq.gz\r\n",
      "F479-12S_R1.fastq.gz\r\n",
      "F479-12S_R2.fastq.gz\r\n",
      "F480-12S_R1.fastq.gz\r\n",
      "F480-12S_R2.fastq.gz\r\n",
      "F481-12S_R1.fastq.gz\r\n",
      "F481-12S_R2.fastq.gz\r\n",
      "F48-12S_R1.fastq.gz\r\n",
      "F48-12S_R2.fastq.gz\r\n",
      "F482-12S_R1.fastq.gz\r\n",
      "F482-12S_R2.fastq.gz\r\n",
      "F483-12S_R1.fastq.gz\r\n",
      "F483-12S_R2.fastq.gz\r\n",
      "F484-12S_R1.fastq.gz\r\n",
      "F484-12S_R2.fastq.gz\r\n",
      "F485-12S_R1.fastq.gz\r\n",
      "F485-12S_R2.fastq.gz\r\n",
      "F486-12S_R1.fastq.gz\r\n",
      "F486-12S_R2.fastq.gz\r\n",
      "F487-12S_R1.fastq.gz\r\n",
      "F487-12S_R2.fastq.gz\r\n",
      "F488-12S_R1.fastq.gz\r\n",
      "F488-12S_R2.fastq.gz\r\n",
      "F489-12S_R1.fastq.gz\r\n",
      "F489-12S_R2.fastq.gz\r\n",
      "F490-12S_R1.fastq.gz\r\n",
      "F490-12S_R2.fastq.gz\r\n",
      "F491-12S_R1.fastq.gz\r\n",
      "F491-12S_R2.fastq.gz\r\n",
      "F49-12S_R1.fastq.gz\r\n",
      "F49-12S_R2.fastq.gz\r\n",
      "F492-12S_R1.fastq.gz\r\n",
      "F492-12S_R2.fastq.gz\r\n",
      "F493-12S_R1.fastq.gz\r\n",
      "F493-12S_R2.fastq.gz\r\n",
      "F494-12S_R1.fastq.gz\r\n",
      "F494-12S_R2.fastq.gz\r\n",
      "F495-12S_R1.fastq.gz\r\n",
      "F495-12S_R2.fastq.gz\r\n",
      "F496-12S_R1.fastq.gz\r\n",
      "F496-12S_R2.fastq.gz\r\n",
      "F497-12S_R1.fastq.gz\r\n",
      "F497-12S_R2.fastq.gz\r\n",
      "F498-12S_R1.fastq.gz\r\n",
      "F498-12S_R2.fastq.gz\r\n",
      "F499-12S_R1.fastq.gz\r\n",
      "F499-12S_R2.fastq.gz\r\n",
      "F500-12S_R1.fastq.gz\r\n",
      "F500-12S_R2.fastq.gz\r\n",
      "F501-12S_R1.fastq.gz\r\n",
      "F501-12S_R2.fastq.gz\r\n",
      "F50-12S_R1.fastq.gz\r\n",
      "F50-12S_R2.fastq.gz\r\n",
      "F502-12S_R1.fastq.gz\r\n",
      "F502-12S_R2.fastq.gz\r\n",
      "F503-12S_R1.fastq.gz\r\n",
      "F503-12S_R2.fastq.gz\r\n",
      "F504-12S_R1.fastq.gz\r\n",
      "F504-12S_R2.fastq.gz\r\n",
      "F505-12S_R1.fastq.gz\r\n",
      "F505-12S_R2.fastq.gz\r\n",
      "F506-12S_R1.fastq.gz\r\n",
      "F506-12S_R2.fastq.gz\r\n",
      "F507-12S_R1.fastq.gz\r\n",
      "F507-12S_R2.fastq.gz\r\n",
      "F508-12S_R1.fastq.gz\r\n",
      "F508-12S_R2.fastq.gz\r\n",
      "F51-12S_R1.fastq.gz\r\n",
      "F51-12S_R2.fastq.gz\r\n",
      "F5-12S_R1.fastq.gz\r\n",
      "F5-12S_R2.fastq.gz\r\n",
      "F52-12S_R1.fastq.gz\r\n",
      "F52-12S_R2.fastq.gz\r\n",
      "F53-12S_R1.fastq.gz\r\n",
      "F53-12S_R2.fastq.gz\r\n",
      "F54-12S_R1.fastq.gz\r\n",
      "F54-12S_R2.fastq.gz\r\n",
      "F55-12S_R1.fastq.gz\r\n",
      "F55-12S_R2.fastq.gz\r\n",
      "F56-12S_R1.fastq.gz\r\n",
      "F56-12S_R2.fastq.gz\r\n",
      "F57-12S_R1.fastq.gz\r\n",
      "F57-12S_R2.fastq.gz\r\n",
      "F58-12S_R1.fastq.gz\r\n",
      "F58-12S_R2.fastq.gz\r\n",
      "F59-12S_R1.fastq.gz\r\n",
      "F59-12S_R2.fastq.gz\r\n",
      "F60-12S_R1.fastq.gz\r\n",
      "F60-12S_R2.fastq.gz\r\n",
      "F61-12S_R1.fastq.gz\r\n",
      "F61-12S_R2.fastq.gz\r\n",
      "F6-12S_R1.fastq.gz\r\n",
      "F6-12S_R2.fastq.gz\r\n",
      "F62-12S_R1.fastq.gz\r\n",
      "F62-12S_R2.fastq.gz\r\n",
      "F63-12S_R1.fastq.gz\r\n",
      "F63-12S_R2.fastq.gz\r\n",
      "F64-12S_R1.fastq.gz\r\n",
      "F64-12S_R2.fastq.gz\r\n",
      "F65-12S_R1.fastq.gz\r\n",
      "F65-12S_R2.fastq.gz\r\n",
      "F66-12S_R1.fastq.gz\r\n",
      "F66-12S_R2.fastq.gz\r\n",
      "F67-12S_R1.fastq.gz\r\n",
      "F67-12S_R2.fastq.gz\r\n",
      "F68-12S_R1.fastq.gz\r\n",
      "F68-12S_R2.fastq.gz\r\n",
      "F69-12S_R1.fastq.gz\r\n",
      "F69-12S_R2.fastq.gz\r\n",
      "F70-12S_R1.fastq.gz\r\n",
      "F70-12S_R2.fastq.gz\r\n",
      "F71-12S_R1.fastq.gz\r\n",
      "F71-12S_R2.fastq.gz\r\n",
      "F7-12S_R1.fastq.gz\r\n",
      "F7-12S_R2.fastq.gz\r\n",
      "F72-12S_R1.fastq.gz\r\n",
      "F72-12S_R2.fastq.gz\r\n",
      "F73-12S_R1.fastq.gz\r\n",
      "F73-12S_R2.fastq.gz\r\n",
      "F74-12S_R1.fastq.gz\r\n",
      "F74-12S_R2.fastq.gz\r\n",
      "F75-12S_R1.fastq.gz\r\n",
      "F75-12S_R2.fastq.gz\r\n",
      "F76-12S_R1.fastq.gz\r\n",
      "F76-12S_R2.fastq.gz\r\n",
      "F77-12S_R1.fastq.gz\r\n",
      "F77-12S_R2.fastq.gz\r\n",
      "F78-12S_R1.fastq.gz\r\n",
      "F78-12S_R2.fastq.gz\r\n",
      "F79-12S_R1.fastq.gz\r\n",
      "F79-12S_R2.fastq.gz\r\n",
      "F80-12S_R1.fastq.gz\r\n",
      "F80-12S_R2.fastq.gz\r\n",
      "F81-12S_R1.fastq.gz\r\n",
      "F81-12S_R2.fastq.gz\r\n",
      "F8-12S_R1.fastq.gz\r\n",
      "F8-12S_R2.fastq.gz\r\n",
      "F82-12S_R1.fastq.gz\r\n",
      "F82-12S_R2.fastq.gz\r\n",
      "F83-12S_R1.fastq.gz\r\n",
      "F83-12S_R2.fastq.gz\r\n",
      "F84-12S_R1.fastq.gz\r\n",
      "F84-12S_R2.fastq.gz\r\n",
      "F85-12S_R1.fastq.gz\r\n",
      "F85-12S_R2.fastq.gz\r\n",
      "F86-12S_R1.fastq.gz\r\n",
      "F86-12S_R2.fastq.gz\r\n",
      "F87-12S_R1.fastq.gz\r\n",
      "F87-12S_R2.fastq.gz\r\n",
      "F88-12S_R1.fastq.gz\r\n",
      "F88-12S_R2.fastq.gz\r\n",
      "F89-12S_R1.fastq.gz\r\n",
      "F89-12S_R2.fastq.gz\r\n",
      "F90-12S_R1.fastq.gz\r\n",
      "F90-12S_R2.fastq.gz\r\n",
      "F91-12S_R1.fastq.gz\r\n",
      "F91-12S_R2.fastq.gz\r\n",
      "F9-12S_R1.fastq.gz\r\n",
      "F9-12S_R2.fastq.gz\r\n",
      "F92-12S_R1.fastq.gz\r\n",
      "F92-12S_R2.fastq.gz\r\n",
      "F93-12S_R1.fastq.gz\r\n",
      "F93-12S_R2.fastq.gz\r\n",
      "F94-12S_R1.fastq.gz\r\n",
      "F94-12S_R2.fastq.gz\r\n",
      "F95-12S_R1.fastq.gz\r\n",
      "F95-12S_R2.fastq.gz\r\n",
      "F96-12S_R1.fastq.gz\r\n",
      "F96-12S_R2.fastq.gz\r\n",
      "F97-12S_R1.fastq.gz\r\n",
      "F97-12S_R2.fastq.gz\r\n",
      "F98-12S_R1.fastq.gz\r\n",
      "F98-12S_R2.fastq.gz\r\n",
      "F99-12S_R1.fastq.gz\r\n",
      "F99-12S_R2.fastq.gz\r\n",
      "How_to_download_from_SRA.ipynb\r\n",
      "N10-1-12S_R1.fastq.gz\r\n",
      "N10-1-12S_R2.fastq.gz\r\n",
      "N10-2-12S_R1.fastq.gz\r\n",
      "N10-2-12S_R2.fastq.gz\r\n",
      "N10-3-12S_R1.fastq.gz\r\n",
      "N10-3-12S_R2.fastq.gz\r\n",
      "N10-4-12S_R1.fastq.gz\r\n",
      "N10-4-12S_R2.fastq.gz\r\n",
      "N10-5-12S_R1.fastq.gz\r\n",
      "N10-5-12S_R2.fastq.gz\r\n",
      "N10-6-12S_R1.fastq.gz\r\n",
      "N10-6-12S_R2.fastq.gz\r\n",
      "N11-1-12S_R1.fastq.gz\r\n",
      "N11-1-12S_R2.fastq.gz\r\n",
      "N1-1-12S_R1.fastq.gz\r\n",
      "N1-1-12S_R2.fastq.gz\r\n",
      "N11-2-12S_R1.fastq.gz\r\n",
      "N11-2-12S_R2.fastq.gz\r\n",
      "N11-3-12S_R1.fastq.gz\r\n",
      "N11-3-12S_R2.fastq.gz\r\n",
      "N11-4-12S_R1.fastq.gz\r\n",
      "N11-4-12S_R2.fastq.gz\r\n",
      "N11-5-12S_R1.fastq.gz\r\n",
      "N11-5-12S_R2.fastq.gz\r\n",
      "N11-6-12S_R1.fastq.gz\r\n",
      "N11-6-12S_R2.fastq.gz\r\n",
      "N12-1-12S_R1.fastq.gz\r\n",
      "N12-1-12S_R2.fastq.gz\r\n",
      "N1-2-12S_R1.fastq.gz\r\n",
      "N1-2-12S_R2.fastq.gz\r\n",
      "N12-2-12S_R1.fastq.gz\r\n",
      "N12-2-12S_R2.fastq.gz\r\n",
      "N12-3-12S_R1.fastq.gz\r\n",
      "N12-3-12S_R2.fastq.gz\r\n",
      "N12-4-12S_R1.fastq.gz\r\n",
      "N12-4-12S_R2.fastq.gz\r\n",
      "N12-5-12S_R1.fastq.gz\r\n",
      "N12-5-12S_R2.fastq.gz\r\n",
      "N12-6-12S_R1.fastq.gz\r\n",
      "N12-6-12S_R2.fastq.gz\r\n",
      "N13-1-12S_R1.fastq.gz\r\n",
      "N13-1-12S_R2.fastq.gz\r\n",
      "N1-3-12S_R1.fastq.gz\r\n",
      "N1-3-12S_R2.fastq.gz\r\n",
      "N13-2-12S_R1.fastq.gz\r\n",
      "N13-2-12S_R2.fastq.gz\r\n",
      "N13-3-12S_R1.fastq.gz\r\n",
      "N13-3-12S_R2.fastq.gz\r\n",
      "N13-4-12S_R1.fastq.gz\r\n",
      "N13-4-12S_R2.fastq.gz\r\n",
      "N13-5-12S_R1.fastq.gz\r\n",
      "N13-5-12S_R2.fastq.gz\r\n",
      "N13-6-12S_R1.fastq.gz\r\n",
      "N13-6-12S_R2.fastq.gz\r\n",
      "N14-1-12S_R1.fastq.gz\r\n",
      "N14-1-12S_R2.fastq.gz\r\n",
      "N1-4-12S_R1.fastq.gz\r\n",
      "N1-4-12S_R2.fastq.gz\r\n",
      "N14-2-12S_R1.fastq.gz\r\n",
      "N14-2-12S_R2.fastq.gz\r\n",
      "N14-3-12S_R1.fastq.gz\r\n",
      "N14-3-12S_R2.fastq.gz\r\n",
      "N14-4-12S_R1.fastq.gz\r\n",
      "N14-4-12S_R2.fastq.gz\r\n",
      "N14-5-12S_R1.fastq.gz\r\n",
      "N14-5-12S_R2.fastq.gz\r\n",
      "N14-6-12S_R1.fastq.gz\r\n",
      "N14-6-12S_R2.fastq.gz\r\n",
      "N15-1-12S_R1.fastq.gz\r\n",
      "N15-1-12S_R2.fastq.gz\r\n",
      "N1-5-12S_R1.fastq.gz\r\n",
      "N1-5-12S_R2.fastq.gz\r\n",
      "N15-2-12S_R1.fastq.gz\r\n",
      "N15-2-12S_R2.fastq.gz\r\n",
      "N15-3-12S_R1.fastq.gz\r\n",
      "N15-3-12S_R2.fastq.gz\r\n",
      "N15-4-12S_R1.fastq.gz\r\n",
      "N15-4-12S_R2.fastq.gz\r\n",
      "N15-5-12S_R1.fastq.gz\r\n",
      "N15-5-12S_R2.fastq.gz\r\n",
      "N15-6-12S_R1.fastq.gz\r\n",
      "N15-6-12S_R2.fastq.gz\r\n",
      "N16-1-12S_R1.fastq.gz\r\n",
      "N16-1-12S_R2.fastq.gz\r\n",
      "N1-6-12S_R1.fastq.gz\r\n",
      "N1-6-12S_R2.fastq.gz\r\n",
      "N16-2-12S_R1.fastq.gz\r\n",
      "N16-2-12S_R2.fastq.gz\r\n",
      "N16-3-12S_R1.fastq.gz\r\n",
      "N16-3-12S_R2.fastq.gz\r\n",
      "N16-4-12S_R1.fastq.gz\r\n",
      "N16-4-12S_R2.fastq.gz\r\n",
      "N16-5-12S_R1.fastq.gz\r\n",
      "N16-5-12S_R2.fastq.gz\r\n",
      "N16-6-12S_R1.fastq.gz\r\n",
      "N16-6-12S_R2.fastq.gz\r\n",
      "N17-1-12S_R1.fastq.gz\r\n",
      "N17-1-12S_R2.fastq.gz\r\n",
      "N17-2-12S_R1.fastq.gz\r\n",
      "N17-2-12S_R2.fastq.gz\r\n",
      "N17-3-12S_R1.fastq.gz\r\n",
      "N17-3-12S_R2.fastq.gz\r\n",
      "N17-4-12S_R1.fastq.gz\r\n",
      "N17-4-12S_R2.fastq.gz\r\n",
      "N17-5-12S_R1.fastq.gz\r\n",
      "N17-5-12S_R2.fastq.gz\r\n",
      "N17-6-12S_R1.fastq.gz\r\n",
      "N17-6-12S_R2.fastq.gz\r\n",
      "N18-1-12S_R1.fastq.gz\r\n",
      "N18-1-12S_R2.fastq.gz\r\n",
      "N18-2-12S_R1.fastq.gz\r\n",
      "N18-2-12S_R2.fastq.gz\r\n",
      "N18-3-12S_R1.fastq.gz\r\n",
      "N18-3-12S_R2.fastq.gz\r\n",
      "N18-4-12S_R1.fastq.gz\r\n",
      "N18-4-12S_R2.fastq.gz\r\n",
      "N18-5-12S_R1.fastq.gz\r\n",
      "N18-5-12S_R2.fastq.gz\r\n",
      "N18-6-12S_R1.fastq.gz\r\n",
      "N18-6-12S_R2.fastq.gz\r\n",
      "N19-1-12S_R1.fastq.gz\r\n",
      "N19-1-12S_R2.fastq.gz\r\n",
      "N19-2-12S_R1.fastq.gz\r\n",
      "N19-2-12S_R2.fastq.gz\r\n",
      "N19-3-12S_R1.fastq.gz\r\n",
      "N19-3-12S_R2.fastq.gz\r\n",
      "N19-4-12S_R1.fastq.gz\r\n",
      "N19-4-12S_R2.fastq.gz\r\n",
      "N19-5-12S_R1.fastq.gz\r\n",
      "N19-5-12S_R2.fastq.gz\r\n",
      "N19-6-12S_R1.fastq.gz\r\n",
      "N19-6-12S_R2.fastq.gz\r\n",
      "N2-1-12S_R1.fastq.gz\r\n",
      "N2-1-12S_R2.fastq.gz\r\n",
      "N2-2-12S_R1.fastq.gz\r\n",
      "N2-2-12S_R2.fastq.gz\r\n",
      "N2-3-12S_R1.fastq.gz\r\n",
      "N2-3-12S_R2.fastq.gz\r\n",
      "N2-4-12S_R1.fastq.gz\r\n",
      "N2-4-12S_R2.fastq.gz\r\n",
      "N2-5-12S_R1.fastq.gz\r\n",
      "N2-5-12S_R2.fastq.gz\r\n",
      "N2-6-12S_R1.fastq.gz\r\n",
      "N2-6-12S_R2.fastq.gz\r\n",
      "N3-1-12S_R1.fastq.gz\r\n",
      "N3-1-12S_R2.fastq.gz\r\n",
      "N3-2-12S_R1.fastq.gz\r\n",
      "N3-2-12S_R2.fastq.gz\r\n",
      "N3-3-12S_R1.fastq.gz\r\n",
      "N3-3-12S_R2.fastq.gz\r\n",
      "N3-4-12S_R1.fastq.gz\r\n",
      "N3-4-12S_R2.fastq.gz\r\n",
      "N3-5-12S_R1.fastq.gz\r\n",
      "N3-5-12S_R2.fastq.gz\r\n",
      "N3-6-12S_R1.fastq.gz\r\n",
      "N3-6-12S_R2.fastq.gz\r\n",
      "N4-1-12S_R1.fastq.gz\r\n",
      "N4-1-12S_R2.fastq.gz\r\n",
      "N4-2-12S_R1.fastq.gz\r\n",
      "N4-2-12S_R2.fastq.gz\r\n",
      "N4-3-12S_R1.fastq.gz\r\n",
      "N4-3-12S_R2.fastq.gz\r\n",
      "N4-4-12S_R1.fastq.gz\r\n",
      "N4-4-12S_R2.fastq.gz\r\n",
      "N4-5-12S_R1.fastq.gz\r\n",
      "N4-5-12S_R2.fastq.gz\r\n",
      "N4-6-12S_R1.fastq.gz\r\n",
      "N4-6-12S_R2.fastq.gz\r\n",
      "N5-1-12S_R1.fastq.gz\r\n",
      "N5-1-12S_R2.fastq.gz\r\n",
      "N5-2-12S_R1.fastq.gz\r\n",
      "N5-2-12S_R2.fastq.gz\r\n",
      "N5-3-12S_R1.fastq.gz\r\n",
      "N5-3-12S_R2.fastq.gz\r\n",
      "N5-4-12S_R1.fastq.gz\r\n",
      "N5-4-12S_R2.fastq.gz\r\n",
      "N5-5-12S_R1.fastq.gz\r\n",
      "N5-5-12S_R2.fastq.gz\r\n",
      "N5-6-12S_R1.fastq.gz\r\n",
      "N5-6-12S_R2.fastq.gz\r\n",
      "N6-1-12S_R1.fastq.gz\r\n",
      "N6-1-12S_R2.fastq.gz\r\n",
      "N6-2-12S_R1.fastq.gz\r\n",
      "N6-2-12S_R2.fastq.gz\r\n",
      "N6-3-12S_R1.fastq.gz\r\n",
      "N6-3-12S_R2.fastq.gz\r\n",
      "N6-4-12S_R1.fastq.gz\r\n",
      "N6-4-12S_R2.fastq.gz\r\n",
      "N6-5-12S_R1.fastq.gz\r\n",
      "N6-5-12S_R2.fastq.gz\r\n",
      "N6-6-12S_R1.fastq.gz\r\n",
      "N6-6-12S_R2.fastq.gz\r\n",
      "N7-1-12S_R1.fastq.gz\r\n",
      "N7-1-12S_R2.fastq.gz\r\n",
      "N7-2-12S_R1.fastq.gz\r\n",
      "N7-2-12S_R2.fastq.gz\r\n",
      "N7-3-12S_R1.fastq.gz\r\n",
      "N7-3-12S_R2.fastq.gz\r\n",
      "N7-4-12S_R1.fastq.gz\r\n",
      "N7-4-12S_R2.fastq.gz\r\n",
      "N7-5-12S_R1.fastq.gz\r\n",
      "N7-5-12S_R2.fastq.gz\r\n",
      "N7-6-12S_R1.fastq.gz\r\n",
      "N7-6-12S_R2.fastq.gz\r\n",
      "N8-1-12S_R1.fastq.gz\r\n",
      "N8-1-12S_R2.fastq.gz\r\n",
      "N8-2-12S_R1.fastq.gz\r\n",
      "N8-2-12S_R2.fastq.gz\r\n",
      "N8-3-12S_R1.fastq.gz\r\n",
      "N8-3-12S_R2.fastq.gz\r\n",
      "N8-4-12S_R1.fastq.gz\r\n",
      "N8-4-12S_R2.fastq.gz\r\n",
      "N8-5-12S_R1.fastq.gz\r\n",
      "N8-5-12S_R2.fastq.gz\r\n",
      "N8-6-12S_R1.fastq.gz\r\n",
      "N8-6-12S_R2.fastq.gz\r\n",
      "N9-1-12S_R1.fastq.gz\r\n",
      "N9-1-12S_R2.fastq.gz\r\n",
      "N9-2-12S_R1.fastq.gz\r\n",
      "N9-2-12S_R2.fastq.gz\r\n",
      "N9-3-12S_R1.fastq.gz\r\n",
      "N9-3-12S_R2.fastq.gz\r\n",
      "N9-4-12S_R1.fastq.gz\r\n",
      "N9-4-12S_R2.fastq.gz\r\n",
      "N9-5-12S_R1.fastq.gz\r\n",
      "N9-5-12S_R2.fastq.gz\r\n",
      "N9-6-12S_R1.fastq.gz\r\n",
      "N9-6-12S_R2.fastq.gz\r\n",
      "NU10-12S_R1.fastq.gz\r\n",
      "NU10-12S_R2.fastq.gz\r\n",
      "NU11-12S_R1.fastq.gz\r\n",
      "NU11-12S_R2.fastq.gz\r\n",
      "NU1-12S_R1.fastq.gz\r\n",
      "NU1-12S_R2.fastq.gz\r\n",
      "NU12-12S_R1.fastq.gz\r\n",
      "NU12-12S_R2.fastq.gz\r\n",
      "NU13-12S_R1.fastq.gz\r\n",
      "NU13-12S_R2.fastq.gz\r\n",
      "NU14-12S_R1.fastq.gz\r\n",
      "NU14-12S_R2.fastq.gz\r\n",
      "NU15-12S_R1.fastq.gz\r\n",
      "NU15-12S_R2.fastq.gz\r\n",
      "NU16-12S_R1.fastq.gz\r\n",
      "NU16-12S_R2.fastq.gz\r\n",
      "NU17-12S_R1.fastq.gz\r\n",
      "NU17-12S_R2.fastq.gz\r\n",
      "NU18-12S_R1.fastq.gz\r\n",
      "NU18-12S_R2.fastq.gz\r\n",
      "NU19-12S_R1.fastq.gz\r\n",
      "NU19-12S_R2.fastq.gz\r\n",
      "NU20-12S_R1.fastq.gz\r\n",
      "NU20-12S_R2.fastq.gz\r\n",
      "NU21-12S_R1.fastq.gz\r\n",
      "NU21-12S_R2.fastq.gz\r\n",
      "NU2-12S_R1.fastq.gz\r\n",
      "NU2-12S_R2.fastq.gz\r\n",
      "NU22-12S_R1.fastq.gz\r\n",
      "NU22-12S_R2.fastq.gz\r\n",
      "NU23-12S_R1.fastq.gz\r\n",
      "NU23-12S_R2.fastq.gz\r\n",
      "NU24-12S_R1.fastq.gz\r\n",
      "NU24-12S_R2.fastq.gz\r\n",
      "NU3-12S_R1.fastq.gz\r\n",
      "NU3-12S_R2.fastq.gz\r\n",
      "NU4-12S_R1.fastq.gz\r\n",
      "NU4-12S_R2.fastq.gz\r\n",
      "NU5-12S_R1.fastq.gz\r\n",
      "NU5-12S_R2.fastq.gz\r\n",
      "NU6-12S_R1.fastq.gz\r\n",
      "NU6-12S_R2.fastq.gz\r\n",
      "NU7-12S_R1.fastq.gz\r\n",
      "NU7-12S_R2.fastq.gz\r\n",
      "NU8-12S_R1.fastq.gz\r\n",
      "NU8-12S_R2.fastq.gz\r\n",
      "NU9-12S_R1.fastq.gz\r\n",
      "NU9-12S_R2.fastq.gz\r\n",
      "P10-1-12S_R1.fastq.gz\r\n",
      "P10-1-12S_R2.fastq.gz\r\n",
      "P10-2-12S_R1.fastq.gz\r\n",
      "P10-2-12S_R2.fastq.gz\r\n",
      "P10-3-12S_R1.fastq.gz\r\n",
      "P10-3-12S_R2.fastq.gz\r\n",
      "P10-4-12S_R1.fastq.gz\r\n",
      "P10-4-12S_R2.fastq.gz\r\n",
      "P10-5-12S_R1.fastq.gz\r\n",
      "P10-5-12S_R2.fastq.gz\r\n",
      "P10-6-12S_R1.fastq.gz\r\n",
      "P10-6-12S_R2.fastq.gz\r\n",
      "P11-1-12S_R1.fastq.gz\r\n",
      "P11-1-12S_R2.fastq.gz\r\n",
      "P1-1-12S_R1.fastq.gz\r\n",
      "P1-1-12S_R2.fastq.gz\r\n",
      "P11-2-12S_R1.fastq.gz\r\n",
      "P11-2-12S_R2.fastq.gz\r\n",
      "P11-3-12S_R1.fastq.gz\r\n",
      "P11-3-12S_R2.fastq.gz\r\n",
      "P11-4-12S_R1.fastq.gz\r\n",
      "P11-4-12S_R2.fastq.gz\r\n",
      "P11-5-12S_R1.fastq.gz\r\n",
      "P11-5-12S_R2.fastq.gz\r\n",
      "P11-6-12S_R1.fastq.gz\r\n",
      "P11-6-12S_R2.fastq.gz\r\n",
      "P12-1-12S_R1.fastq.gz\r\n",
      "P12-1-12S_R2.fastq.gz\r\n",
      "P1-2-12S_R1.fastq.gz\r\n",
      "P1-2-12S_R2.fastq.gz\r\n",
      "P12-2-12S_R1.fastq.gz\r\n",
      "P12-2-12S_R2.fastq.gz\r\n",
      "P12-3-12S_R1.fastq.gz\r\n",
      "P12-3-12S_R2.fastq.gz\r\n",
      "P12-4-12S_R1.fastq.gz\r\n",
      "P12-4-12S_R2.fastq.gz\r\n",
      "P12-5-12S_R1.fastq.gz\r\n",
      "P12-5-12S_R2.fastq.gz\r\n",
      "P12-6-12S_R1.fastq.gz\r\n",
      "P12-6-12S_R2.fastq.gz\r\n",
      "P13-1-12S_R1.fastq.gz\r\n",
      "P13-1-12S_R2.fastq.gz\r\n",
      "P1-3-12S_R1.fastq.gz\r\n",
      "P1-3-12S_R2.fastq.gz\r\n",
      "P13-2-12S_R1.fastq.gz\r\n",
      "P13-2-12S_R2.fastq.gz\r\n",
      "P13-3-12S_R1.fastq.gz\r\n",
      "P13-3-12S_R2.fastq.gz\r\n",
      "P13-4-12S_R1.fastq.gz\r\n",
      "P13-4-12S_R2.fastq.gz\r\n",
      "P13-5-12S_R1.fastq.gz\r\n",
      "P13-5-12S_R2.fastq.gz\r\n",
      "P13-6-12S_R1.fastq.gz\r\n",
      "P13-6-12S_R2.fastq.gz\r\n",
      "P14-1-12S_R1.fastq.gz\r\n",
      "P14-1-12S_R2.fastq.gz\r\n",
      "P1-4-12S_R1.fastq.gz\r\n",
      "P1-4-12S_R2.fastq.gz\r\n",
      "P14-2-12S_R1.fastq.gz\r\n",
      "P14-2-12S_R2.fastq.gz\r\n",
      "P14-3-12S_R1.fastq.gz\r\n",
      "P14-3-12S_R2.fastq.gz\r\n",
      "P14-4-12S_R1.fastq.gz\r\n",
      "P14-4-12S_R2.fastq.gz\r\n",
      "P14-5-12S_R1.fastq.gz\r\n",
      "P14-5-12S_R2.fastq.gz\r\n",
      "P14-6-12S_R1.fastq.gz\r\n",
      "P14-6-12S_R2.fastq.gz\r\n",
      "P15-1-12S_R1.fastq.gz\r\n",
      "P15-1-12S_R2.fastq.gz\r\n",
      "P1-5-12S_R1.fastq.gz\r\n",
      "P1-5-12S_R2.fastq.gz\r\n",
      "P15-2-12S_R1.fastq.gz\r\n",
      "P15-2-12S_R2.fastq.gz\r\n",
      "P15-3-12S_R1.fastq.gz\r\n",
      "P15-3-12S_R2.fastq.gz\r\n",
      "P15-4-12S_R1.fastq.gz\r\n",
      "P15-4-12S_R2.fastq.gz\r\n",
      "P15-5-12S_R1.fastq.gz\r\n",
      "P15-5-12S_R2.fastq.gz\r\n",
      "P15-6-12S_R1.fastq.gz\r\n",
      "P15-6-12S_R2.fastq.gz\r\n",
      "P16-1-12S_R1.fastq.gz\r\n",
      "P16-1-12S_R2.fastq.gz\r\n",
      "P1-6-12S_R1.fastq.gz\r\n",
      "P1-6-12S_R2.fastq.gz\r\n",
      "P16-2-12S_R1.fastq.gz\r\n",
      "P16-2-12S_R2.fastq.gz\r\n",
      "P16-3-12S_R1.fastq.gz\r\n",
      "P16-3-12S_R2.fastq.gz\r\n",
      "P16-4-12S_R1.fastq.gz\r\n",
      "P16-4-12S_R2.fastq.gz\r\n",
      "P16-5-12S_R1.fastq.gz\r\n",
      "P16-5-12S_R2.fastq.gz\r\n",
      "P16-6-12S_R1.fastq.gz\r\n",
      "P16-6-12S_R2.fastq.gz\r\n",
      "P17-1-12S_R1.fastq.gz\r\n",
      "P17-1-12S_R2.fastq.gz\r\n",
      "P17-2-12S_R1.fastq.gz\r\n",
      "P17-2-12S_R2.fastq.gz\r\n",
      "P17-3-12S_R1.fastq.gz\r\n",
      "P17-3-12S_R2.fastq.gz\r\n",
      "P17-4-12S_R1.fastq.gz\r\n",
      "P17-4-12S_R2.fastq.gz\r\n",
      "P17-5-12S_R1.fastq.gz\r\n",
      "P17-5-12S_R2.fastq.gz\r\n",
      "P17-6-12S_R1.fastq.gz\r\n",
      "P17-6-12S_R2.fastq.gz\r\n",
      "P18-1-12S_R1.fastq.gz\r\n",
      "P18-1-12S_R2.fastq.gz\r\n",
      "P18-2-12S_R1.fastq.gz\r\n",
      "P18-2-12S_R2.fastq.gz\r\n",
      "P18-3-12S_R1.fastq.gz\r\n",
      "P18-3-12S_R2.fastq.gz\r\n",
      "P18-4-12S_R1.fastq.gz\r\n",
      "P18-4-12S_R2.fastq.gz\r\n",
      "P18-5-12S_R1.fastq.gz\r\n",
      "P18-5-12S_R2.fastq.gz\r\n",
      "P18-6-12S_R1.fastq.gz\r\n",
      "P18-6-12S_R2.fastq.gz\r\n",
      "P19-1-12S_R1.fastq.gz\r\n",
      "P19-1-12S_R2.fastq.gz\r\n",
      "P19-2-12S_R1.fastq.gz\r\n",
      "P19-2-12S_R2.fastq.gz\r\n",
      "P19-3-12S_R1.fastq.gz\r\n",
      "P19-3-12S_R2.fastq.gz\r\n",
      "P19-4-12S_R1.fastq.gz\r\n",
      "P19-4-12S_R2.fastq.gz\r\n",
      "P19-5-12S_R1.fastq.gz\r\n",
      "P19-5-12S_R2.fastq.gz\r\n",
      "P19-6-12S_R1.fastq.gz\r\n",
      "P19-6-12S_R2.fastq.gz\r\n",
      "P2-1-12S_R1.fastq.gz\r\n",
      "P2-1-12S_R2.fastq.gz\r\n",
      "P2-2-12S_R1.fastq.gz\r\n",
      "P2-2-12S_R2.fastq.gz\r\n",
      "P2-3-12S_R1.fastq.gz\r\n",
      "P2-3-12S_R2.fastq.gz\r\n",
      "P2-4-12S_R1.fastq.gz\r\n",
      "P2-4-12S_R2.fastq.gz\r\n",
      "P2-5-12S_R1.fastq.gz\r\n",
      "P2-5-12S_R2.fastq.gz\r\n",
      "P2-6-12S_R1.fastq.gz\r\n",
      "P2-6-12S_R2.fastq.gz\r\n",
      "P3-1-12S_R1.fastq.gz\r\n",
      "P3-1-12S_R2.fastq.gz\r\n",
      "P3-2-12S_R1.fastq.gz\r\n",
      "P3-2-12S_R2.fastq.gz\r\n",
      "P3-3-12S_R1.fastq.gz\r\n",
      "P3-3-12S_R2.fastq.gz\r\n",
      "P3-4-12S_R1.fastq.gz\r\n",
      "P3-4-12S_R2.fastq.gz\r\n",
      "P3-5-12S_R1.fastq.gz\r\n",
      "P3-5-12S_R2.fastq.gz\r\n",
      "P3-6-12S_R1.fastq.gz\r\n",
      "P3-6-12S_R2.fastq.gz\r\n",
      "P4-1-12S_R1.fastq.gz\r\n",
      "P4-1-12S_R2.fastq.gz\r\n",
      "P4-2-12S_R1.fastq.gz\r\n",
      "P4-2-12S_R2.fastq.gz\r\n",
      "P4-3-12S_R1.fastq.gz\r\n",
      "P4-3-12S_R2.fastq.gz\r\n",
      "P4-4-12S_R1.fastq.gz\r\n",
      "P4-4-12S_R2.fastq.gz\r\n",
      "P4-5-12S_R1.fastq.gz\r\n",
      "P4-5-12S_R2.fastq.gz\r\n",
      "P4-6-12S_R1.fastq.gz\r\n",
      "P4-6-12S_R2.fastq.gz\r\n",
      "P5-1-12S_R1.fastq.gz\r\n",
      "P5-1-12S_R2.fastq.gz\r\n",
      "P5-2-12S_R1.fastq.gz\r\n",
      "P5-2-12S_R2.fastq.gz\r\n",
      "P5-3-12S_R1.fastq.gz\r\n",
      "P5-3-12S_R2.fastq.gz\r\n",
      "P5-4-12S_R1.fastq.gz\r\n",
      "P5-4-12S_R2.fastq.gz\r\n",
      "P5-5-12S_R1.fastq.gz\r\n",
      "P5-5-12S_R2.fastq.gz\r\n",
      "P5-6-12S_R1.fastq.gz\r\n",
      "P5-6-12S_R2.fastq.gz\r\n",
      "P6-1-12S_R1.fastq.gz\r\n",
      "P6-1-12S_R2.fastq.gz\r\n",
      "P6-2-12S_R1.fastq.gz\r\n",
      "P6-2-12S_R2.fastq.gz\r\n",
      "P6-3-12S_R1.fastq.gz\r\n",
      "P6-3-12S_R2.fastq.gz\r\n",
      "P6-4-12S_R1.fastq.gz\r\n",
      "P6-4-12S_R2.fastq.gz\r\n",
      "P6-5-12S_R1.fastq.gz\r\n",
      "P6-5-12S_R2.fastq.gz\r\n",
      "P6-6-12S_R1.fastq.gz\r\n",
      "P6-6-12S_R2.fastq.gz\r\n",
      "P7-1-12S_R1.fastq.gz\r\n",
      "P7-1-12S_R2.fastq.gz\r\n",
      "P7-2-12S_R1.fastq.gz\r\n",
      "P7-2-12S_R2.fastq.gz\r\n",
      "P7-3-12S_R1.fastq.gz\r\n",
      "P7-3-12S_R2.fastq.gz\r\n",
      "P7-4-12S_R1.fastq.gz\r\n",
      "P7-4-12S_R2.fastq.gz\r\n",
      "P7-5-12S_R1.fastq.gz\r\n",
      "P7-5-12S_R2.fastq.gz\r\n",
      "P7-6-12S_R1.fastq.gz\r\n",
      "P7-6-12S_R2.fastq.gz\r\n",
      "P8-1-12S_R1.fastq.gz\r\n",
      "P8-1-12S_R2.fastq.gz\r\n",
      "P8-2-12S_R1.fastq.gz\r\n",
      "P8-2-12S_R2.fastq.gz\r\n",
      "P8-3-12S_R1.fastq.gz\r\n",
      "P8-3-12S_R2.fastq.gz\r\n",
      "P8-4-12S_R1.fastq.gz\r\n",
      "P8-4-12S_R2.fastq.gz\r\n",
      "P8-5-12S_R1.fastq.gz\r\n",
      "P8-5-12S_R2.fastq.gz\r\n",
      "P8-6-12S_R1.fastq.gz\r\n",
      "P8-6-12S_R2.fastq.gz\r\n",
      "P9-1-12S_R1.fastq.gz\r\n",
      "P9-1-12S_R2.fastq.gz\r\n",
      "P9-2-12S_R1.fastq.gz\r\n",
      "P9-2-12S_R2.fastq.gz\r\n",
      "P9-3-12S_R1.fastq.gz\r\n",
      "P9-3-12S_R2.fastq.gz\r\n",
      "P9-4-12S_R1.fastq.gz\r\n",
      "P9-4-12S_R2.fastq.gz\r\n",
      "P9-5-12S_R1.fastq.gz\r\n",
      "P9-5-12S_R2.fastq.gz\r\n",
      "P9-6-12S_R1.fastq.gz\r\n",
      "P9-6-12S_R2.fastq.gz\r\n"
     ]
    }
   ],
   "source": [
    "!ls -1 ../raw_reads/"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Prepare a text file specifying the samples to be processed including the format and location of the reads i.e. the querymap.\n",
    "\n",
    "The next command expects two fastq files (forward and reverse reads) per sample in the directory ```../raw_reads/```. It expects the files to be named 'SampleID', followed by '_R1' or '_R2' to identify the forward/reverse read file respectively.\n",
    "\n",
    "The raw data need to be downloaded with [this notebook](https://github.com/HullUni-bioinformatics/Harper_et_al_2018/blob/master/raw_reads/How_to_download_from_SRA.ipynb).\n",
    "\n",
    "SampleID must correspond to the first column in the file Sample_accessions.tsv [(here)](https://github.com/HullUni-bioinformatics/Harper_et_al_2018/blob/master/Data/Sample_accessions.tsv). The marker is '12S'."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "%%bash\n",
    "\n",
    "for a in $(cat ../Data/Sample_accessions.tsv | grep \"12S\" | cut -f 1 | grep \"SampleID\" -v)\n",
    "do\n",
    "    R1=$(ls -1 ../raw_reads/$a-12S_* | grep \"_R1.fastq\")\n",
    "    R2=$(ls -1 ../raw_reads/$a-12S_* | grep \"_R2.fastq\")\n",
    "\n",
    "    echo -e \"$a\\tfastq\\t$R1\\t$R2\"\n",
    "done > Querymap.txt"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "F1\tfastq\t../raw_reads/F1-12S_R1.fastq.gz\t../raw_reads/F1-12S_R2.fastq.gz\r\n",
      "F10\tfastq\t../raw_reads/F10-12S_R1.fastq.gz\t../raw_reads/F10-12S_R2.fastq.gz\r\n",
      "F100\tfastq\t../raw_reads/F100-12S_R1.fastq.gz\t../raw_reads/F100-12S_R2.fastq.gz\r\n",
      "F101\tfastq\t../raw_reads/F101-12S_R1.fastq.gz\t../raw_reads/F101-12S_R2.fastq.gz\r\n",
      "F102\tfastq\t../raw_reads/F102-12S_R1.fastq.gz\t../raw_reads/F102-12S_R2.fastq.gz\r\n",
      "F103\tfastq\t../raw_reads/F103-12S_R1.fastq.gz\t../raw_reads/F103-12S_R2.fastq.gz\r\n",
      "F104\tfastq\t../raw_reads/F104-12S_R1.fastq.gz\t../raw_reads/F104-12S_R2.fastq.gz\r\n",
      "F105\tfastq\t../raw_reads/F105-12S_R1.fastq.gz\t../raw_reads/F105-12S_R2.fastq.gz\r\n",
      "F106\tfastq\t../raw_reads/F106-12S_R1.fastq.gz\t../raw_reads/F106-12S_R2.fastq.gz\r\n",
      "F107\tfastq\t../raw_reads/F107-12S_R1.fastq.gz\t../raw_reads/F107-12S_R2.fastq.gz\r\n"
     ]
    }
   ],
   "source": [
    "!head -n 10 Querymap.txt"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Add two columns to the text file containing the value 25 - this specifies the number of bases to be removed from the forward and reverse read."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "%%bash\n",
    "\n",
    "sed 's/$/&\\t25/' Querymap.txt > Querymap_new.txt"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "F1\tfastq\t../raw_reads/F1-12S_R1.fastq.gz\t../raw_reads/F1-12S_R2.fastq.gz\t25\r\n",
      "F10\tfastq\t../raw_reads/F10-12S_R1.fastq.gz\t../raw_reads/F10-12S_R2.fastq.gz\t25\r\n",
      "F100\tfastq\t../raw_reads/F100-12S_R1.fastq.gz\t../raw_reads/F100-12S_R2.fastq.gz\t25\r\n",
      "F101\tfastq\t../raw_reads/F101-12S_R1.fastq.gz\t../raw_reads/F101-12S_R2.fastq.gz\t25\r\n"
     ]
    }
   ],
   "source": [
    "!head -n 4 Querymap_new.txt"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "%%bash\n",
    "\n",
    "sed 's/$/&\\t25/' Querymap_new.txt > Querymap_final.txt"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "F1\tfastq\t../raw_reads/F1-12S_R1.fastq.gz\t../raw_reads/F1-12S_R2.fastq.gz\t25\t25\r\n",
      "F10\tfastq\t../raw_reads/F10-12S_R1.fastq.gz\t../raw_reads/F10-12S_R2.fastq.gz\t25\t25\r\n",
      "F100\tfastq\t../raw_reads/F100-12S_R1.fastq.gz\t../raw_reads/F100-12S_R2.fastq.gz\t25\t25\r\n",
      "F101\tfastq\t../raw_reads/F101-12S_R1.fastq.gz\t../raw_reads/F101-12S_R2.fastq.gz\t25\t25\r\n"
     ]
    }
   ],
   "source": [
    "!head -n 4 Querymap_final.txt"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Raw read processing"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Now, perform basic quality trimming and clipping (Trimmomatic) and paired-end read merging (flash). metaBEAT will be used to process all 768 samples in one go."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "usage: metaBEAT.py [-h] [-Q <FILE>] [-B <FILE>] [--g_queries <FILE>] [-v] [-s]\r\n",
      "                   [-f] [-p] [-k] [-t] [-b] [-m <string>] [-n <INT>] [-E] [-e]\r\n",
      "                   [--read_stats_off] [--PCR_primer <FILE>] [--bc_dist <INT>]\r\n",
      "                   [--trim_adapter <FILE>] [--trim_qual <INT>] [--phred <INT>]\r\n",
      "                   [--trim_window <INT>] [--read_crop <INT>]\r\n",
      "                   [--trim_minlength <INT>] [--merge] [--product_length <INT>]\r\n",
      "                   [--merged_only] [--forward_only] [--length_filter <INT>]\r\n",
      "                   [--length_deviation <FLOAT>] [-R <FILE>] [--gb_out <FILE>]\r\n",
      "                   [--rec_check] [--gb_to_taxid <FILE>] [--cluster]\r\n",
      "                   [--clust_match <FLOAT>] [--clust_cov <INT>]\r\n",
      "                   [--blast_db <PATH>] [--blast_xml <PATH>]\r\n",
      "                   [--update_taxonomy] [--taxonomy_db <FILE>]\r\n",
      "                   [--min_ident <FLOAT>] [--min_ali_length <FLOAT>]\r\n",
      "                   [--bitscore_skim_LCA <FLOAT>] [--bitscore_skim_adjust_off]\r\n",
      "                   [--min_bit <INT>] [--refpkg <DIR>] [--jplace <FILE>]\r\n",
      "                   [--jellyfish_hash_size <STR>] [--kraken_db <DIR>]\r\n",
      "                   [--kraken_score_threshold <FLOAT>] [--rm_kraken_db]\r\n",
      "                   [-o OUTPUT_PREFIX] [--metadata METADATA]\r\n",
      "                   [-@ <email-address>] [--version]\r\n",
      "\r\n",
      "metaBEAT - metaBarcoding and Environmental DNA Analyses tool version:\r\n",
      "v.0.97.10-global\r\n",
      "\r\n",
      "optional arguments:\r\n",
      "  -h, --help            show this help message and exit\r\n",
      "  -Q <FILE>, --querylist <FILE>\r\n",
      "                        file containing a list of query files\r\n",
      "  -B <FILE>, --BIOM_input <FILE>\r\n",
      "                        OTU table in BIOM format\r\n",
      "  --g_queries <FILE>    fasta file containing query sequences (in combination\r\n",
      "                        with '-B' sequence headers are expected to match)\r\n",
      "  -v, --verbose         turn verbose output on\r\n",
      "  -s, --seqinfo         write out seq_info.csv file\r\n",
      "  -f, --fasta           write out ref.fasta file\r\n",
      "  -p, --pplace          perform phylogenetic placement\r\n",
      "  -k, --kraken          perform phylogenetic placement\r\n",
      "  -t, --taxids          write out taxid.txt file\r\n",
      "  -b, --blast           compile local blast db and blast queries\r\n",
      "  -m <string>, --marker <string>\r\n",
      "                        marker ID (default: marker)\r\n",
      "  -n <INT>, --n_threads <INT>\r\n",
      "                        Number of threads (default: 1)\r\n",
      "  -E, --extract_centroid_reads\r\n",
      "                        extract centroid reads to files\r\n",
      "  -e, --extract_all_reads\r\n",
      "                        extract reads to files\r\n",
      "  --read_stats_off      ommit writing read stats to file\r\n",
      "  --version             show program's version number and exit\r\n",
      "\r\n",
      "Query preprocessing:\r\n",
      "  The parameters in this group affect how the query sequences are processed\r\n",
      "\r\n",
      "  --PCR_primer <FILE>   PCR primers (provided in fasta file) to be clipped\r\n",
      "                        from reads\r\n",
      "  --bc_dist <INT>       Number of mismatches allowed in barcode sequences\r\n",
      "  --trim_adapter <FILE>\r\n",
      "                        trim adapters provided in file\r\n",
      "  --trim_qual <INT>     minimum phred quality score (default: 30)\r\n",
      "  --phred <INT>         phred quality score offset - 33 or 64 (default: 33)\r\n",
      "  --trim_window <INT>   sliding window size (default: 5) for trimming; if\r\n",
      "                        average quality drops below the specified minimum\r\n",
      "                        quality all subsequent bases are removed from the\r\n",
      "                        reads\r\n",
      "  --read_crop <INT>     Crop reads to this length if they are longer than that\r\n",
      "                        (default: off)\r\n",
      "  --trim_minlength <INT>\r\n",
      "                        minimum length of reads to be retained after trimming\r\n",
      "                        (default: 50)\r\n",
      "  --merge               attempt to merge paired-end reads\r\n",
      "  --product_length <INT>\r\n",
      "                        estimated length of PCR product (specifying this\r\n",
      "                        option increases merging efficiency)\r\n",
      "  --merged_only         only process successfully merged read-pairs\r\n",
      "  --forward_only        only process sequences that contain forward reads\r\n",
      "                        (i.e. unmerged forward reads and merged reads)\r\n",
      "  --length_filter <INT>\r\n",
      "                        only process reads, which are within +/- 10 percent of\r\n",
      "                        this length\r\n",
      "  --length_deviation <FLOAT>\r\n",
      "                        allowed deviation (in percent) from length specified\r\n",
      "                        by --length_filter (default=0.1)\r\n",
      "\r\n",
      "Reference:\r\n",
      "  The parameters in this group affect the reference to be used in the\r\n",
      "  analyses\r\n",
      "\r\n",
      "  -R <FILE>, --REFlist <FILE>\r\n",
      "                        file containing a list of files to be used as\r\n",
      "                        reference sequences\r\n",
      "  --gb_out <FILE>       output the corrected gb file\r\n",
      "  --rec_check           check records to be used as reference\r\n",
      "  --gb_to_taxid <FILE>  comma delimited file containing 'gb accession,taxid'\r\n",
      "                        for a list of taxa\r\n",
      "\r\n",
      "Query clustering options:\r\n",
      "  The parameters in this group affect read clustering\r\n",
      "\r\n",
      "  --cluster             perform clustering of query sequences using vsearch\r\n",
      "  --clust_match <FLOAT>\r\n",
      "                        identity threshold for clustering in percent (default:\r\n",
      "                        1)\r\n",
      "  --clust_cov <INT>     minimum number of records in cluster (default: 1)\r\n",
      "\r\n",
      "BLAST search:\r\n",
      "  The parameters in this group affect BLAST search and BLAST based taxonomic\r\n",
      "  assignment\r\n",
      "\r\n",
      "  --blast_db <PATH>     path to precompiled blast database\r\n",
      "  --blast_xml <PATH>    path to Blast result in xml format\r\n",
      "  --update_taxonomy     Download/update taxonomy database. Database will be\r\n",
      "                        called 'taxonomy.db' and will be compiled in the same\r\n",
      "                        location as the metaBEAT.py script.\r\n",
      "  --taxonomy_db <FILE>  taxonomy database file location. In case it's not the\r\n",
      "                        default, which is 'taxonomy.db' in the same directory\r\n",
      "                        as the metaBEAT.py script.\r\n",
      "  --min_ident <FLOAT>   minimum identity threshold in percent (default: 0.80)\r\n",
      "  --min_ali_length <FLOAT>\r\n",
      "                        minimum alignment length in percent of total query\r\n",
      "                        length (default: 0.95)\r\n",
      "  --bitscore_skim_LCA <FLOAT>\r\n",
      "                        Only BLAST hits with bitscores differing by less than\r\n",
      "                        this factor from the top hit (bitscore skim window)\r\n",
      "                        will be considered for LCA (0-1; default: 0.1)\r\n",
      "  --bitscore_skim_adjust_off\r\n",
      "                        Per default a 100% identity BLAST top hit across the\r\n",
      "                        minimum alignment length triggers an adjustment of the\r\n",
      "                        bitscore skim window to '0', i.e. only hits with\r\n",
      "                        bitscores as good as the top hit are considered for\r\n",
      "                        LCA. This flag switches this behaviour off.\r\n",
      "  --min_bit <INT>       minimum bitscore (default: 80)\r\n",
      "\r\n",
      "Phylogenetic placement:\r\n",
      "  The parameters in this group affect phylogenetic placement\r\n",
      "\r\n",
      "  --refpkg <DIR>        PATH to refpkg for pplacer\r\n",
      "  --jplace <FILE>       phylogenetic placement result from prefious pplacer\r\n",
      "                        run in *.jplace format\r\n",
      "\r\n",
      "Kraken:\r\n",
      "  The parameters in this group affect taxonomic assignment using Kraken\r\n",
      "\r\n",
      "  --jellyfish_hash_size <STR>\r\n",
      "                        jellyfish hash size to control memory usage during\r\n",
      "                        kraken database building. A table size of '6400M' will\r\n",
      "                        require ~44G of RAM. '2700M' -> 20G RAM.\r\n",
      "  --kraken_db <DIR>     PATH to a Kraken database\r\n",
      "  --kraken_score_threshold <FLOAT>\r\n",
      "                        minimum proportion of k-mers to support assignment\r\n",
      "                        (0-1; default: 0)\r\n",
      "  --rm_kraken_db        Remove Kraken database after successful completion\r\n",
      "\r\n",
      "BIOM OUTPUT:\r\n",
      "  The arguments in this groups affect the output in BIOM format\r\n",
      "\r\n",
      "  -o OUTPUT_PREFIX, --output_prefix OUTPUT_PREFIX\r\n",
      "                        prefix for BIOM output files (default='metaBEAT')\r\n",
      "  --metadata METADATA   comma delimited file containing metadata (optional)\r\n",
      "\r\n",
      "Entrez identification:\r\n",
      "  metaBEAT is querying the NCBI Entrez databases, please provide an email\r\n",
      "  address for identification\r\n",
      "\r\n",
      "  -@ <email-address>, --email <email-address>\r\n",
      "                        provide your email address for identification to NCBI\r\n"
     ]
    }
   ],
   "source": [
    "!metaBEAT_global.py -h"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Command to trim:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "%%bash\n",
    "\n",
    "metaBEAT_global.py \\\n",
    "-Q Querymap_final.txt \\\n",
    "--trim_qual 30 --read_crop 110 --trim_minlength 90 \\\n",
    "--merge --forward_only --product_length 110 --length_filter 100 --length_deviation 0.2 \\\n",
    "-@ L.Harper@2015.hull.ac.uk \\\n",
    "-n 5 -v &> log "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Read processing will take several hours.\n",
    "\n",
    "\n",
    "# Visualise query survival after trimming\n",
    "\n",
    "metaBEAT will generate a directory with all temporary files that were created during the processing for each sample and will record useful stats summarizing the data processing in the file `metaBEAT_read_stats.csv`. You can explore the table manually or quickly plot out some of these stats here:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "<matplotlib.axes.AxesSubplot at 0x7efc849c03d0>"
      ]
     },
     "execution_count": 14,
     "metadata": {},
     "output_type": "execute_result"
    },
    {
     "data": {
      "image/png": "iVBORw0KGgoAAAANSUhEUgAAAXkAAAEACAYAAABWLgY0AAAABHNCSVQICAgIfAhkiAAAAAlwSFlz\nAAALEgAACxIB0t1+/AAAFLlJREFUeJzt3X2sJXV9x/E3epdadAE3JLsImItPrZjURaPQVvRri4pN\nI037R2tbs0iTNmlaNX1ggf5R/qmlNFHTaPuHlkJMpA9qCLSUskRva23FPrC4AisP5Ta7qKu2rKAx\nhc1u/5i53rOHc849j/P9zZz3Kzne+c3Mmfnccfneud+ZMxckSZIkSZIkSZIkSZIkSZLU50bgCHBg\nwLLfBo4DO3rmXQM8DBwE3rrwdJKkmVwCXMizi/x5wJ3AY2wW+QuA/cA2YBV4BHhOIyklSQNtVYQ/\nBzwxYP4HgKv65l0O3AI8A6xTFfnXz5hPkjSDac60LwcOA1/qm/+iev6Gw8A5U+aSJM3ByoTrnwZc\nC7ylZ94pI9Y/MXEiSdLcTFrkX0rVb7+vHp8L/AdwEfA4Va+enmWPD9jG41Rn/ZKk8T0KvGwRG15l\n8N01MPjC66nA+XWgQWf5nt1vui47QEGuyw5QkOuyAxTkuuwABZmqdm7Vk78F+BfgFcAh4N0jdvoA\n8Nf1178Hfn3aUEtkNTtAQVazAxRkNTtAQVazA7TdVu2ad26x/CV94/fXL0nSkvLsflNkByhIZAco\nSGQHKEhkByhIa2pna4JKUkEW0pPXYkV2gIJEdoCCRHaAgkR2gLazyEuS5sp2jSRNznaNJOlkFvlc\nkR2gIJEdoCCRHaAgkR2g7SzykpbcypNUrZC+18qTqbHmZNTDxRblRNJ+JWmQE4Pb3ad8/38KMVXt\n9ExekjrMIp8rsgMUJLIDFCSyAxQksgO0nUVekjrMnrykZWdPXpLUThb5XJEdoCCRHaAgkR2gIJEd\noO0s8pLUYfbkJS07e/KSpHayyOeK7AAFiewABYnsAAWJ7ABtZ5GXpA6zJy9p2dmTlyS1k0U+V2QH\nKEhkByhIZAcoSGQHaLutivyNwBHgQM+8PwYeBO4DPg2c0bPsGuBh4CDw1vnFlCQtwiXAhZxc5N/C\n5g+H6+sXwAXAfmAbsAo8wuAfIv6NV0klOQEnBryKq1UL+RuvnwOe6Ju3DzheT98DnFtPXw7cAjwD\nrFMV+ddPE0qSNB+z9uSvBO6op18EHO5Zdhg4Z8btd11kByhIZAcoSGQHKEjk7XoFOvBnAVdmeO/v\nAU8DnxixzrBfL26iOtsHOErV5lmrx1F/dbxcY7ZYvkzj3YXlyRzvbmh/A2YfAz474G1v3t7zhkXl\nWaunr6jH6yzQKif35Kl3/HngeT3zrq5fG+4ELhqwvdL6XJKW24iefFG9+oXtd5WTi/xlwP3AWX3r\nbVx4PRU4H3iUwTfuW+QllWSpi/wtwFep2jKHqHrwDwP/Ddxbv/60Z/1rqS64HgTe1mTQlorsAAWJ\n7AAFiewABYkG9rHURX4RWhO0AZEdoCCRHaAgkR2gINHAPjpd5H12jaRlN+LZNUU908Zn10iSTmaR\nzxXZAQoS2QEKEtkBChLZAdrOIi9JHdaavpIkLYg9eUlSO1nkc0V2gIJEdoCCRHaAgkR2gLazyEtS\nh7WmryRJC2JPXpLUThb5XJEdoCCRHaAgkR2gIJEdoO0s8pLUYa3pK0nSgtiTlyS1k0U+V2QHKEhk\nByhIZAcoSGQHaDuLvCR1WGv6SpK0IPbkJUntZJHPFdkBChLZAQoS2QEKEtkB2s4iL0kd1pq+kiQt\niD15SVI7WeRzRXaAgkR2gIJEdoCCRHaAttuqyN8IHAEO9MzbAewDHgLuAs7sWXYN8DBwEHjr/GJK\nkhbhEuBCTi7yNwBX1dN7gevr6QuA/cA2YBV4hME/RAY1uSQpywk4MeA1an5WzsVY5eQifxDYWU/v\nqsdQncXv7VnvTuDiAduzyEsqSaeL/DQ9+Z1ULRzqrxsF/0XA4Z71DgPnTBNqiUR2gIJEdoCCRHaA\ngkR2gLZbmfH9W/1UG7bsJmC9nj5K1eZZq8dRf3W8XGO2WL5M492F5ckc725ofxO+7fsWlWetnr6i\nHq/373ieVnl2u2ZXPX02m+2aq+vXhjuBiwZsz3aNpJLYrulzG7Cnnt4D3Noz/xeAU4HzgZcDX5wm\nlCSpGbcAXwWeBg4B76a6hfJuBt9CeS3VXTUHgbcN2aZn8psiO0BBIjtAQSI7QEGigX10+kw+Q2uC\nNiCyAxQksgMUJLIDFCTms5mVJ9m8hjjg1d0i35rnL0jSDIY8nwZGP6PGZ9dIkgpmkc8V2QEKEtkB\nChLZAQoS2QHaziIvSR3Wmr6SJM3AnrwkqXss8rkiO0BBIjtAQSI7QEEiO0DbWeQlqcNa01eSpBnY\nk5ckdY9FPldkByhIZAcoSGQHKEhkB2g7i7wkdVhr+kqSNAN78pKk7rHI54rsAAWJ7AAFiewABYns\nAG1nkZekDmtNX0mSZmBPXpLUPRb5XJEdoCCRHaAgkR2gIJEdoO0s8pLUYa3pK0nSDOzJS5K6xyKf\nK7IDFCSyAxQksgMUJLIDtN0sRf4a4H7gAPAJ4AeAHcA+4CHgLuDMWQNKkqY3bV9pFfgM8Erg/4C/\nAu4AXgV8C7gB2Au8ELi677325CU1bWl78tPaAXyFqoivALcDbwEOAjvrdXbV437DjrQkLcoJODHk\nNWzZqPlZ30OzfhV4CvgG8PF63hM9y0/pG2+wyG+K7AAFiewABYnsAAWJOW1naYv8ypQ7eynwPqq2\nzbeBvwF+eUCgYaFuAtbr6aPAfmCtHkf91fFyjdli+TKNdxeWJ3O8e07b22K1rZYPW3+h338AV9Tj\n9f4dL9rPAx/rGb8L+AjwIFWbBuBsbNdIKsPSnslPe3fNQeBi4Aep2jKXAg9Q9eb31OvsAW6dcvuS\npGRXsXkL5c3ANqoLsncz+hZKz+Q3RXaAgkR2gIJEdoCCxJy2s7Rn8hlaE7QBkR2gIJEdoCCRHaAg\nMaftLG2R7/y9npLEEt8n72MNJKnDLPK5IjtAQSI7QEEiO0BBIjtA21nkJanDWtNXkqQZ2JOXJHWP\nRT5XZAcoSGQHKEhkByhIZAdoO4u8JHVYa/pKkjQDe/KSpO6xyOeK7AAFiewABYnsAAWJ7ABtZ5GX\npA5rTV9JkmZgT16S1D0W+VyRHaAgkR2gIJEdoCCRHaDtLPKS1GGt6StJ0gzsyUuSuscinyuyAxQk\nsgMUJLIDFCSyA7SdRV6SOqw1fSVJmoE9eUlS91jkc0V2gIJEdoCCRHaAgkR2gLabpcifCXwSeBB4\nALgI2AHsAx4C7qrXkSS10M3AlfX0CnAGcANwVT1vL3D9gPcNa4xJ0qKcgBNDXsOWjZqf9T005wzg\nvwbMPwjsrKd31eN+FnlJTVvaIj9tu+Z84JvAXwD/CXwUeD5VgT9Sr3OEzYKvwSI7QEEiO0BBIjtA\nQSI7QNutzPC+1wC/Afwb8CHg6r51Rv3EuwlYr6ePAvuBtXoc9VfHyzVmi+XLNN5dWJ7M8e45bW+L\n1bZaPmz9hX7/AVxRj9f7d7xou4DHesZvAP6O6iLsrnre2diukVQG2zUT+jpwCHhFPb4UuB+4HdhT\nz9sD3Drl9iVJyV5N1aq5D/g01cXYHcDdjL6F0jP5TZEdoCCRHaAgkR2gIDGn7Sztmfy0PXmoivvr\nBsy/dIZtSpLmqDXPX5CkGfjsGklS91jkc0V2gIJEdoCCRHaAgkR2gLazyEtSh7WmryRJM7AnL0nt\nt/Ikm5+2730tLYt8rsgOUJDIDlCQyA5QkJhs9WPbrfEns8hLUoe1pq8kSWMY0nsf1l8ftcyevCSp\ncBb5XJEdoCCRHaAgkR2gIJEdoO0s8pLUYa3pK0nSGOzJ9/FMXpI6zCKfK7IDFCSyAxQksgMUJLID\ntJ1FXpI6rDV9JUkagz35Pp7JS1KHWeRzRXaAgkR2gIJEdoCCRHaAtrPIS1KHtaavJEljsCffxzN5\nSeowi3yuyA5QkMgOUJDIDlCQyA7QdrMW+ecC9wK31+MdwD7gIeAu4MwZty9JmsGsfaXfAl4LbAfe\nAdwAfKv+uhd4IXB133vsyUtaFHvyc3QucDfwZjbP5A8CO+vpXfW433L/LS5Ji3QCTgx4DZs/atmo\n+Vnf2+Rmadd8EPhd4HjPvJ3AkXr6CJsFX4NFdoCCRHaAgkR2gIJEdoC2W5nyfT8NfIOqHx9D1hn1\nE+8mYL2ePgrsB9bq8cb2HC/XmC2WL9N4d2F5Mse7J1y/nhwwe6u3Tbqb8VacZRzAFfV4vX/Hi/Z+\n4BDwGPA14LvAx6naM7vqdc7Gdo2kZtmuWYA3sdmT37jgCtUF1+sHrG+Rl7QoFvkFeBNwWz29g+pi\n7KhbKC3ymyI7QEEiO0BBIjtAQWLC9S3yfabtyff6x/oF8L/ApXPYpiRpDpbiXk9JS6OB++S3AccG\nzF95Co6dPlbK6UxVOy3ykrqkgSKf9iGpqWqnz67JFdkBChLZAQoS2QEKEtkB2s4iL0kdZrtGUpfY\nrunjmbwkdZhFPldkByhIZAcoSGQHKEhkB2i7edwnP60Xwxkfh22nPnvRMwfh2+9uPpIkdUtmT/6N\n8JI74CPPP3nxUeDK78D3tidkk9Ru9uT7ZJ7JA6cfg8v65n0jJYkkdZE9+VyRHaAgkR2gIJEdoCCR\nHaDtCi3yz7yAzefR971WnsxMJkkabaOZ9UbYffTZT3g7MsYT4yRpoIaeQplSm6bafqFn8pKkebDI\n54rsAAWJ7AAFiewABYnsAG1nkZckzZU9eUkzWnmSoTdn2JPvlXyfvCRN49j2ER9IUg/bNbkiO0BB\nIjtAQSI7QEEiO0DbWeQlSXNlT17SrObUR+9+T94zeUnqMIt8rsgOUJDIDlCQyA7QvGF3yzznu6mx\nOmDaIn8e8FngfuDLwHvq+TuAfcBDwF3AmbMGlLQMNu6W6X8dPy011hLbBeyup18AfAV4JXADcFU9\nfy9w/YD32pOX1G/S/97tyY9p2jP5rwP76+nvAA8C5wDvAG6u598M/MyU25ck6o/yDDrF15jm0ZNf\nBS4E7gF2Akfq+UfqsYaL7AAFiewABYnsAOU4hjV+NrN+4vUFwKeA9wJP9S0b9f/GTdWyrz0PPkTV\n+Yl60ef7Vl2rv0bvzBiwwHG7x2yxfJnGuwvL08R4xOy1AW+b1/rjLh97/1u9cZJxAFfU43USbAP+\nAXhfz7yDVP16gLPrcT978pL6NdAvtyc/iVOAPwceoDoV33AbsKee3gPcOuX2JUmJ3gAcp7r4em/9\nuozqFsq7GX0LpWfymyI7QEEiO0BBIjtAAs/kxzpGk5u2J//PDP8t4NIptylJmjM/8ZprLTtAQday\nAxRkLTuAusMiL0kdZpHPFdkBChLZAQoS2QHUHRZ5Seowi3yutewABVnLDlCQtewAizP0b7NqQfwb\nr5Ia5N9mbZpn8rkiO0BBIjtAQSJnt8POsleezMmjefBMXlJt6Fn29sajaG48k8+1lh2gIGvZAQqy\nlh1gPEP76579F8QzeUlTGnbmD579l8Mz+VyRHaAgkR2gIJEdQN1hkZe0AP5Fp1JY5HOtZQcoyFp2\ngPE0cgfK2pwyDeuXD5k/dPtMtj74F53KYU9emkiJd6CMuvd80vkDtz/h+iqJZ/K5IjtAQWL2TXTm\nPu/IDqDu8ExeHVLiWbaUq4Vn8sP6g607W4PW9KEbsZYdoCBr2QHUHS0s8kMv6GwfvKCVxV+NyGzv\ndKa1pMK1sMgPM6z4Hyv5V/XIDlCQaH6XG+2djH8zI/cdi9+/lkWHirw0TOY92/N6tO6oRwhIw3nh\nNddadoCCrC1u05m3AE71aN218bez5ba05DyTl6QOW0SRvww4CDwM7F3A9pPM84l787ro1sTFu67s\noy18HIDma95F/rnAh6kK/QXAO4FXznkfSYZdKJvmQt3Gtj44p+0s8sJhU/voPxbFXzRfkGMMPhbS\ndOZd5F8PPAKsA88AfwlcPud9LNg8/wblVts62r8+g9ef+Ax/yHZGbWvS73ven1foPxYj97HgC5Cj\njl8TBh0LaTrzvvB6DnCoZ3wYuGjO+1iwaS6UrQDHBrxp0gt+w9bftn3Iggm3M2pbi866Ur9nEqMy\nLfJC6qjjN2wfw/4NSLnmXeQn+Ud+HB56HvzEt0+e/fRzqD7Y1CLT3r2xvuDtL3pbk25/VOFcn9P+\ns8zzuK7PFkXqMe97ry4GrqPqyQNcAxwH/qhnnUeAl855v5LUdY8CL8sOsVIHWQVOBfbTmQuvkiSA\ntwNfoTpjvyY5iyRJkqRJjfOhqD+pl98HXNhQrgxbHYtfojoGXwI+D/xIc9EaN+6H5V5HdTXzZ5sI\nlWScYxHAvcCX6fZjMLY6FmcBd1K1gL8MXNFYsmbdCBwBDoxYp4i6+Vyqds0qsI3BvfmfAu6opy8C\nvtBUuIaNcyx+FDijnr6M5T4WG+t9Bvhb4OeaCtewcY7FmcD9wLn1+KymwjVsnGNxHfCH9fRZwP/Q\nzWdvXUJVuIcV+Ynr5qKeXTPOh6LeAdxcT99D9Q9654LyZBrnWPwrsHEr6T1s/kfdNeN+WO43gU8C\n32wsWfPGORa/CHyK6vMmAN9qKlzDxjkWXwNOr6dPpyryk37wog0+BzwxYvnEdXNRRX7Qh6LOGWOd\nLha3cY5Fr19h8yd114z77+Jy4M/qcVc/YDTOsXg5sAP4LPDvwLuaida4cY7FR4FXAV+lalO8t5lo\nxZm4bi7q151x/8Psv0+/i/9BT/I9vRm4EvjxBWXJNs6x+BBwdb3uKXT3ObrjHIttwGuAnwROo/qN\n7wtU/dguGedYXEvVxgmqz9nsA14NPLW4WMWaqG4uqsg/DpzXMz6PzV85h61zbj2va8Y5FlBdbP0o\nVU9+1K9rbTbOsXgt1a/rUPVe3071K/xtC0/XrHGOxSGqFs336tc/URW2rhX5cY7FjwF/UE8/CjwG\n/BDVbzjLpJi6Oc6HonovIFxMdy82jnMsXkzVk7y40WTNm/TDcn9Bd++uGedY/DBwN9WFydOoLsZd\n0FzExoxzLD4A/H49vZPqh8COhvI1bZXxLrym181BH4r6tfq14cP18vuofi3tqq2OxceoLiTdW7++\n2HTABo3z72JDl4s8jHcsfofqDpsDwHsaTdesrY7FWcDtVLXiANVF6S66heq6w9NUv8ldyfLWTUmS\nJEmSJEmSJEmSJEmSJEmSJEnL5P8BBQ8mPQERqVcAAAAASUVORK5CYII=\n",
      "text/plain": [
       "<matplotlib.figure.Figure at 0x7efc849c0090>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "%matplotlib inline\n",
    "import pandas as pd\n",
    "\n",
    "df = pd.read_csv('metaBEAT_read_stats.csv',index_col=0)\n",
    "df['fraction'] = df['queries']/(df['total']*0.5)\n",
    "df.fraction.hist(bins=50)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "The final step in the processing will be global clustering of the centroids from all clusters from all samples to produce denovo OTUs. The temporary files from the global clustering and the final OTU table were written to the directory `./GLOBAL`."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "global_centroids.fasta\tglobal.uc\t\t  metaBEAT-OTU-denovo.tsv\r\n",
      "global_queries.fasta\tmetaBEAT-OTU-denovo.biom\r\n"
     ]
    }
   ],
   "source": [
    "!ls GLOBAL/"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "The denovo OTU table (numbers are reads) can be briefly to see how OTUs are distributed across your samples. \n",
    "\n",
    "Detailed information on what metaBEAT did to each sample is contained in the `log` file. It contains the exact commands that were run for each sample during each step of the process.\n",
    "\n",
    "It's a large text file - look at the first 100 lines."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\r\n",
      "metaBEAT - metaBarcoding and Environmental DNA Analyses tool\r\n",
      "version: v.0.97.10-global\r\n",
      "\r\n",
      "\r\n",
      "Sat Feb 17 18:04:27 2018\r\n",
      "\r\n",
      "/usr/bin/metaBEAT_global.py -Q Querymap_final.txt --trim_qual 30 --read_crop 110 --trim_minlength 90 --merge --forward_only --product_length 110 --length_filter 100 --length_deviation 0.2 -@ L.Harper@2015.hull.ac.uk -n 5 -v\r\n",
      "\r\n",
      "\r\n",
      "metaBEAT may be querying NCBI's Entrez databases to fetch/verify taxonomic ids. Entrez User requirements state that you need to identify yourself by providing an email address so that NCBI can contact you in case there is a problem.\r\n",
      "\r\n",
      "You have specified: 'L.Harper@2015.hull.ac.uk'\r\n",
      "\r\n",
      "\r\n",
      "Parsing querylist file\r\n",
      "\r\n",
      "Number of samples to process: 760\r\n",
      "Sequence input format: defaultdict(<type 'int'>, {'fastq': 760})\r\n",
      "Barcodes for demultiplexing provided for 0 samples\r\n",
      "Cropping instructions provided for 760 samples\r\n",
      "\r\n",
      "\r\n",
      "Sat Feb 17 18:04:27 2018\r\n",
      "\r\n",
      "\r\n",
      "##### processing query ID: F1 #####\r\n",
      "\r\n",
      "\r\n",
      "### READ QUALITY TRIMMING ###\r\n",
      "\r\n",
      "\r\n",
      "trimming PE reads with trimmomatic\r\n",
      "java -jar /usr/bin/trimmomatic-0.32.jar PE -threads 5 -phred33 -trimlog trimmomatic.log /home/working/Harper_et_al_2018/raw_reads/F1-12S_R1.fastq.gz /home/working/Harper_et_al_2018/raw_reads/F1-12S_R2.fastq.gz F1_forward.paired.fastq.gz F1_forward.singletons.fastq.gz F1_reverse.paired.fastq.gz F1_reverse.singletons.fastq.gz ILLUMINACLIP:None:5:5:5 TRAILING:30 LEADING:30 SLIDINGWINDOW:5:30 CROP:110 MINLEN:90\r\n",
      "\r\n",
      "TrimmomaticPE: Started with arguments: -threads 5 -phred33 -trimlog trimmomatic.log /home/working/Harper_et_al_2018/raw_reads/F1-12S_R1.fastq.gz /home/working/Harper_et_al_2018/raw_reads/F1-12S_R2.fastq.gz F1_forward.paired.fastq.gz F1_forward.singletons.fastq.gz F1_reverse.paired.fastq.gz F1_reverse.singletons.fastq.gz ILLUMINACLIP:None:5:5:5 TRAILING:30 LEADING:30 SLIDINGWINDOW:5:30 CROP:110 MINLEN:90\r\n",
      "Feb 17, 2018 6:04:33 PM org.usadellab.trimmomatic.trim.IlluminaClippingTrimmer makeIlluminaClippingTrimmer\r\n",
      "SEVERE: null\r\n",
      "java.io.FileNotFoundException: /home/working/Harper_et_al_2018/1-trimming/F1/None (No such file or directory)\r\n",
      "\tat java.io.FileInputStream.open(Native Method)\r\n",
      "\tat java.io.FileInputStream.<init>(FileInputStream.java:146)\r\n",
      "\tat org.usadellab.trimmomatic.fasta.FastaParser.parse(FastaParser.java:54)\r\n",
      "\tat org.usadellab.trimmomatic.trim.IlluminaClippingTrimmer.loadSequences(IlluminaClippingTrimmer.java:107)\r\n",
      "\tat org.usadellab.trimmomatic.trim.IlluminaClippingTrimmer.makeIlluminaClippingTrimmer(IlluminaClippingTrimmer.java:70)\r\n",
      "\tat org.usadellab.trimmomatic.trim.TrimmerFactory.makeTrimmer(TrimmerFactory.java:27)\r\n",
      "\tat org.usadellab.trimmomatic.TrimmomaticPE.run(TrimmomaticPE.java:495)\r\n",
      "\tat org.usadellab.trimmomatic.Trimmomatic.main(Trimmomatic.java:35)\r\n",
      "\r\n",
      "Input Read Pairs: 75335 Both Surviving: 57166 (75.88%) Forward Only Surviving: 5693 (7.56%) Reverse Only Surviving: 1860 (2.47%) Dropped: 10616 (14.09%)\r\n",
      "TrimmomaticPE: Completed successfully\r\n",
      "\r\n",
      "\r\n",
      "### READ CLIPPING ###\r\n",
      "\r\n",
      "Cropping the first 25 bases off reads in file: F1_forward.paired.fastq.gz\r\n",
      "Cropping the first 25 bases off reads in file: F1_forward.singletons.fastq.gz\r\n",
      "Cropping the first 25 bases off reads in file: F1_reverse.paired.fastq.gz\r\n",
      "Cropping the first 25 bases off reads in file: F1_reverse.singletons.fastq.gz\r\n",
      "\r\n",
      "### MERGING READ PAIRS ###\r\n",
      "\r\n",
      "merging paired-end reads with flash\r\n",
      "\r\n",
      "flash F1_forward.paired.headclipped.fastq.gz F1_reverse.paired.headclipped.fastq.gz -M 110 -t 5 -p 33 -o F1 -z\r\n",
      "[FLASH] Starting FLASH v1.2.11\r\n",
      "[FLASH] Fast Length Adjustment of SHort reads\r\n",
      "[FLASH]  \r\n",
      "[FLASH] Input files:\r\n",
      "[FLASH]     F1_forward.paired.headclipped.fastq.gz\r\n",
      "[FLASH]     F1_reverse.paired.headclipped.fastq.gz\r\n",
      "[FLASH]  \r\n",
      "[FLASH] Output files:\r\n",
      "[FLASH]     ./F1.extendedFrags.fastq.gz\r\n",
      "[FLASH]     ./F1.notCombined_1.fastq.gz\r\n",
      "[FLASH]     ./F1.notCombined_2.fastq.gz\r\n",
      "[FLASH]     ./F1.hist\r\n",
      "[FLASH]     ./F1.histogram\r\n",
      "[FLASH]  \r\n",
      "[FLASH] Parameters:\r\n",
      "[FLASH]     Min overlap:           10\r\n",
      "[FLASH]     Max overlap:           110\r\n",
      "[FLASH]     Max mismatch density:  0.250000\r\n",
      "[FLASH]     Allow \"outie\" pairs:   false\r\n",
      "[FLASH]     Cap mismatch quals:    false\r\n",
      "[FLASH]     Combiner threads:      5\r\n",
      "[FLASH]     Input format:          FASTQ, phred_offset=33\r\n",
      "[FLASH]     Output format:         FASTQ, phred_offset=33, gzip\r\n",
      "[FLASH]  \r\n",
      "[FLASH] Starting reader and writer threads\r\n",
      "[FLASH] Starting 5 combiner threads\r\n",
      "[FLASH] Processed 25000 read pairs\r\n",
      "[FLASH] Processed 50000 read pairs\r\n",
      "[FLASH] Processed 57166 read pairs\r\n",
      "[FLASH]  \r\n",
      "[FLASH] Read combination statistics:\r\n",
      "[FLASH]     Total pairs:      57166\r\n",
      "[FLASH]     Combined pairs:   57150\r\n",
      "[FLASH]     Uncombined pairs: 16\r\n",
      "[FLASH]     Percent combined: 99.97%\r\n",
      "[FLASH]  \r\n"
     ]
    }
   ],
   "source": [
    "!head -n 100 log"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Chimera detection"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Some stats on the read counts before/after trimming, merging etc. are summarized for you in `read_stats.csv`.\n",
    "\n",
    "Next stage of the processing is chimera detection and removal of putative chimeric sequences. We'll do that using uchime as implemented in vsearch."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/home/working/Harper_et_al_2018/Jupyter notebooks\r\n"
     ]
    }
   ],
   "source": [
    "!pwd"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/home/working/Harper_et_al_2018\n"
     ]
    }
   ],
   "source": [
    "cd ../"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "!mkdir 2-chimera_detection"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/home/working/Harper_et_al_2018/2-chimera_detection\n"
     ]
    }
   ],
   "source": [
    "cd 2-chimera_detection"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Convert reference database from GenBank to fasta format to be used in chimera detection.\n",
    "\n",
    "Prepare `REFmap.txt` file, i.e. text file that specifies the location and the format of the reference to be used.\n",
    "The reference sequences in GenBank format are present in subdirectories for each vertebrate group in the `../Reference_database` directory."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "!echo '../Reference_database/Amphibians/12S_UKamphibians_SATIVA_cleaned.gb\\tgb\\n' \\\n",
    "'../Reference_database/Reptiles/12S_UKreptiles_SATIVA_cleaned.gb\\tgb\\n' \\\n",
    "'../Reference_database/Mammals/12S_UKmammals_SATIVA_cleaned.gb\\tgb\\n' \\\n",
    "'../Reference_database/Birds/12S_UKbirds_SATIVA_cleaned.gb\\tgb\\n' \\\n",
    "'../Reference_database/Fish/20161026_INBO_12S_fishrefs_hfj_edit.fasta\\tfasta\\n' \\\n",
    "'../Reference_database/Fish/custom_extended_12S_edit_10_2016.gb\\tgb\\n' \\\n",
    "'../Reference_database/Fish/RhamphochromisEsox_mt.gb\\tgb' > REFmap.txt"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "../Reference_database/Amphibians/12S_UKamphibians_SATIVA_cleaned.gb\tgb\r\n",
      " ../Reference_database/Reptiles/12S_UKreptiles_SATIVA_cleaned.gb\tgb\r\n",
      " ../Reference_database/Mammals/12S_UKmammals_SATIVA_cleaned.gb\tgb\r\n",
      " ../Reference_database/Birds/12S_UKbirds_SATIVA_cleaned.gb\tgb\r\n",
      " ../Reference_database/Fish/20161026_INBO_12S_fishrefs_hfj_edit.fasta\tfasta\r\n",
      " ../Reference_database/Fish/custom_extended_12S_edit_10_2016.gb\tgb\r\n",
      " ../Reference_database/Fish/RhamphochromisEsox_mt.gb\tgb\r\n"
     ]
    }
   ],
   "source": [
    "!cat REFmap.txt"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "usage: metaBEAT.py [-h] [-Q <FILE>] [-B <FILE>] [--g_queries <FILE>] [-v] [-s]\r\n",
      "                   [-f] [-p] [-k] [-t] [-b] [-m <string>] [-n <INT>] [-E] [-e]\r\n",
      "                   [--read_stats_off] [--PCR_primer <FILE>] [--bc_dist <INT>]\r\n",
      "                   [--trim_adapter <FILE>] [--trim_qual <INT>] [--phred <INT>]\r\n",
      "                   [--trim_window <INT>] [--read_crop <INT>]\r\n",
      "                   [--trim_minlength <INT>] [--merge] [--product_length <INT>]\r\n",
      "                   [--merged_only] [--forward_only] [--length_filter <INT>]\r\n",
      "                   [--length_deviation <FLOAT>] [-R <FILE>] [--gb_out <FILE>]\r\n",
      "                   [--rec_check] [--gb_to_taxid <FILE>] [--cluster]\r\n",
      "                   [--clust_match <FLOAT>] [--clust_cov <INT>]\r\n",
      "                   [--blast_db <PATH>] [--blast_xml <PATH>]\r\n",
      "                   [--update_taxonomy] [--taxonomy_db <FILE>]\r\n",
      "                   [--min_ident <FLOAT>] [--min_ali_length <FLOAT>]\r\n",
      "                   [--bitscore_skim_LCA <FLOAT>] [--bitscore_skim_adjust_off]\r\n",
      "                   [--min_bit <INT>] [--refpkg <DIR>] [--jplace <FILE>]\r\n",
      "                   [--jellyfish_hash_size <STR>] [--kraken_db <DIR>]\r\n",
      "                   [--kraken_score_threshold <FLOAT>] [--rm_kraken_db]\r\n",
      "                   [-o OUTPUT_PREFIX] [--metadata METADATA]\r\n",
      "                   [-@ <email-address>] [--version]\r\n",
      "\r\n",
      "metaBEAT - metaBarcoding and Environmental DNA Analyses tool version:\r\n",
      "v.0.97.10-global\r\n",
      "\r\n",
      "optional arguments:\r\n",
      "  -h, --help            show this help message and exit\r\n",
      "  -Q <FILE>, --querylist <FILE>\r\n",
      "                        file containing a list of query files\r\n",
      "  -B <FILE>, --BIOM_input <FILE>\r\n",
      "                        OTU table in BIOM format\r\n",
      "  --g_queries <FILE>    fasta file containing query sequences (in combination\r\n",
      "                        with '-B' sequence headers are expected to match)\r\n",
      "  -v, --verbose         turn verbose output on\r\n",
      "  -s, --seqinfo         write out seq_info.csv file\r\n",
      "  -f, --fasta           write out ref.fasta file\r\n",
      "  -p, --pplace          perform phylogenetic placement\r\n",
      "  -k, --kraken          perform phylogenetic placement\r\n",
      "  -t, --taxids          write out taxid.txt file\r\n",
      "  -b, --blast           compile local blast db and blast queries\r\n",
      "  -m <string>, --marker <string>\r\n",
      "                        marker ID (default: marker)\r\n",
      "  -n <INT>, --n_threads <INT>\r\n",
      "                        Number of threads (default: 1)\r\n",
      "  -E, --extract_centroid_reads\r\n",
      "                        extract centroid reads to files\r\n",
      "  -e, --extract_all_reads\r\n",
      "                        extract reads to files\r\n",
      "  --read_stats_off      ommit writing read stats to file\r\n",
      "  --version             show program's version number and exit\r\n",
      "\r\n",
      "Query preprocessing:\r\n",
      "  The parameters in this group affect how the query sequences are processed\r\n",
      "\r\n",
      "  --PCR_primer <FILE>   PCR primers (provided in fasta file) to be clipped\r\n",
      "                        from reads\r\n",
      "  --bc_dist <INT>       Number of mismatches allowed in barcode sequences\r\n",
      "  --trim_adapter <FILE>\r\n",
      "                        trim adapters provided in file\r\n",
      "  --trim_qual <INT>     minimum phred quality score (default: 30)\r\n",
      "  --phred <INT>         phred quality score offset - 33 or 64 (default: 33)\r\n",
      "  --trim_window <INT>   sliding window size (default: 5) for trimming; if\r\n",
      "                        average quality drops below the specified minimum\r\n",
      "                        quality all subsequent bases are removed from the\r\n",
      "                        reads\r\n",
      "  --read_crop <INT>     Crop reads to this length if they are longer than that\r\n",
      "                        (default: off)\r\n",
      "  --trim_minlength <INT>\r\n",
      "                        minimum length of reads to be retained after trimming\r\n",
      "                        (default: 50)\r\n",
      "  --merge               attempt to merge paired-end reads\r\n",
      "  --product_length <INT>\r\n",
      "                        estimated length of PCR product (specifying this\r\n",
      "                        option increases merging efficiency)\r\n",
      "  --merged_only         only process successfully merged read-pairs\r\n",
      "  --forward_only        only process sequences that contain forward reads\r\n",
      "                        (i.e. unmerged forward reads and merged reads)\r\n",
      "  --length_filter <INT>\r\n",
      "                        only process reads, which are within +/- 10 percent of\r\n",
      "                        this length\r\n",
      "  --length_deviation <FLOAT>\r\n",
      "                        allowed deviation (in percent) from length specified\r\n",
      "                        by --length_filter (default=0.1)\r\n",
      "\r\n",
      "Reference:\r\n",
      "  The parameters in this group affect the reference to be used in the\r\n",
      "  analyses\r\n",
      "\r\n",
      "  -R <FILE>, --REFlist <FILE>\r\n",
      "                        file containing a list of files to be used as\r\n",
      "                        reference sequences\r\n",
      "  --gb_out <FILE>       output the corrected gb file\r\n",
      "  --rec_check           check records to be used as reference\r\n",
      "  --gb_to_taxid <FILE>  comma delimited file containing 'gb accession,taxid'\r\n",
      "                        for a list of taxa\r\n",
      "\r\n",
      "Query clustering options:\r\n",
      "  The parameters in this group affect read clustering\r\n",
      "\r\n",
      "  --cluster             perform clustering of query sequences using vsearch\r\n",
      "  --clust_match <FLOAT>\r\n",
      "                        identity threshold for clustering in percent (default:\r\n",
      "                        1)\r\n",
      "  --clust_cov <INT>     minimum number of records in cluster (default: 1)\r\n",
      "\r\n",
      "BLAST search:\r\n",
      "  The parameters in this group affect BLAST search and BLAST based taxonomic\r\n",
      "  assignment\r\n",
      "\r\n",
      "  --blast_db <PATH>     path to precompiled blast database\r\n",
      "  --blast_xml <PATH>    path to Blast result in xml format\r\n",
      "  --update_taxonomy     Download/update taxonomy database. Database will be\r\n",
      "                        called 'taxonomy.db' and will be compiled in the same\r\n",
      "                        location as the metaBEAT.py script.\r\n",
      "  --taxonomy_db <FILE>  taxonomy database file location. In case it's not the\r\n",
      "                        default, which is 'taxonomy.db' in the same directory\r\n",
      "                        as the metaBEAT.py script.\r\n",
      "  --min_ident <FLOAT>   minimum identity threshold in percent (default: 0.80)\r\n",
      "  --min_ali_length <FLOAT>\r\n",
      "                        minimum alignment length in percent of total query\r\n",
      "                        length (default: 0.95)\r\n",
      "  --bitscore_skim_LCA <FLOAT>\r\n",
      "                        Only BLAST hits with bitscores differing by less than\r\n",
      "                        this factor from the top hit (bitscore skim window)\r\n",
      "                        will be considered for LCA (0-1; default: 0.1)\r\n",
      "  --bitscore_skim_adjust_off\r\n",
      "                        Per default a 100% identity BLAST top hit across the\r\n",
      "                        minimum alignment length triggers an adjustment of the\r\n",
      "                        bitscore skim window to '0', i.e. only hits with\r\n",
      "                        bitscores as good as the top hit are considered for\r\n",
      "                        LCA. This flag switches this behaviour off.\r\n",
      "  --min_bit <INT>       minimum bitscore (default: 80)\r\n",
      "\r\n",
      "Phylogenetic placement:\r\n",
      "  The parameters in this group affect phylogenetic placement\r\n",
      "\r\n",
      "  --refpkg <DIR>        PATH to refpkg for pplacer\r\n",
      "  --jplace <FILE>       phylogenetic placement result from prefious pplacer\r\n",
      "                        run in *.jplace format\r\n",
      "\r\n",
      "Kraken:\r\n",
      "  The parameters in this group affect taxonomic assignment using Kraken\r\n",
      "\r\n",
      "  --jellyfish_hash_size <STR>\r\n",
      "                        jellyfish hash size to control memory usage during\r\n",
      "                        kraken database building. A table size of '6400M' will\r\n",
      "                        require ~44G of RAM. '2700M' -> 20G RAM.\r\n",
      "  --kraken_db <DIR>     PATH to a Kraken database\r\n",
      "  --kraken_score_threshold <FLOAT>\r\n",
      "                        minimum proportion of k-mers to support assignment\r\n",
      "                        (0-1; default: 0)\r\n",
      "  --rm_kraken_db        Remove Kraken database after successful completion\r\n",
      "\r\n",
      "BIOM OUTPUT:\r\n",
      "  The arguments in this groups affect the output in BIOM format\r\n",
      "\r\n",
      "  -o OUTPUT_PREFIX, --output_prefix OUTPUT_PREFIX\r\n",
      "                        prefix for BIOM output files (default='metaBEAT')\r\n",
      "  --metadata METADATA   comma delimited file containing metadata (optional)\r\n",
      "\r\n",
      "Entrez identification:\r\n",
      "  metaBEAT is querying the NCBI Entrez databases, please provide an email\r\n",
      "  address for identification\r\n",
      "\r\n",
      "  -@ <email-address>, --email <email-address>\r\n",
      "                        provide your email address for identification to NCBI\r\n"
     ]
    }
   ],
   "source": [
    "!metaBEAT_global.py -h"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "metaBEAT - metaBarcoding and Environmental DNA Analyses tool\n",
      "version: v.0.97.10-global\n",
      "\n",
      "\n",
      "Tue Feb 20 16:54:07 2018\n",
      "\n",
      "/usr/bin/metaBEAT_global.py -R REFmap.txt -f -@ L.Harper@2015.hull.ac.uk\n",
      "\n",
      "\n",
      "metaBEAT may be querying NCBI's Entrez databases to fetch/verify taxonomic ids. Entrez User requirements state that you need to identify yourself by providing an email address so that NCBI can contact you in case there is a problem.\n",
      "\n",
      "You have specified: 'L.Harper@2015.hull.ac.uk'\n",
      "\n",
      "\n",
      "######## PROCESSING REFERENCE DATA ########\n",
      "\n",
      "\n",
      "processing ../Reference_database/Fish/RhamphochromisEsox_mt.gb (containing 1 records)\n",
      "\n",
      "total number of valid records: 1\n",
      "\n",
      "\n",
      "processing ../Reference_database/Mammals/12S_UKmammals_SATIVA_cleaned.gb (containing 1046 records)\n",
      "\n",
      "total number of valid records: 1047\n",
      "\n",
      "\n",
      "processing ../Reference_database/Fish/20161026_INBO_12S_fishrefs_hfj_edit.fasta (containing 85 records)\n",
      "\n",
      "total number of valid records: 1132\n",
      "\n",
      "\n",
      "processing ../Reference_database/Fish/custom_extended_12S_edit_10_2016.gb (containing 271 records)\n",
      "\n",
      "total number of valid records: 1403\n",
      "\n",
      "\n",
      "processing ../Reference_database/Birds/12S_UKbirds_SATIVA_cleaned.gb (containing 622 records)\n",
      "\n",
      "total number of valid records: 2025\n",
      "\n",
      "\n",
      "processing ../Reference_database/Amphibians/12S_UKamphibians_SATIVA_cleaned.gb (containing 198 records)\n",
      "\n",
      "total number of valid records: 2223\n",
      "\n",
      "\n",
      "processing ../Reference_database/Reptiles/12S_UKreptiles_SATIVA_cleaned.gb (containing 112 records)\n",
      "\n",
      "total number of valid records: 2335\n",
      "\n",
      "write out reference sequences to refs.fasta\n",
      "\n",
      "\n",
      "Tue Feb 20 16:54:40 2018\n",
      "\n",
      "\n",
      "### GLOBAL CLUSTERING ###\n",
      "\n",
      "\n",
      "Concatenating for global clustering\n",
      "\n",
      "\n",
      "Performing global clustering\n",
      "\n",
      "vsearch --cluster_fast /home/working/Harper_et_al_2018/2-chimera_detection/GLOBAL/global_queries.fasta --id 1.00 --strand both --threads 1 --centroids global_centroids.fasta --uc global.uc\n",
      "vsearch v1.1.0_linux_x86_64, 31.3GB RAM, 8 cores\n",
      "https://github.com/torognes/vsearch\n",
      "\n",
      "\n",
      "\n",
      "parse vsearch uc file\n",
      "\n",
      "\n",
      "writing denovo OTU table\n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Traceback (most recent call last):\n",
      "  File \"/usr/bin/metaBEAT_global.py\", line 2499, in <module>\n",
      "    out.write(BIOM_tables_per_method['OTU_denovo'].to_tsv()) #to_json('generaged by test', direct_io=out)\n",
      "  File \"/usr/local/lib/python2.7/dist-packages/biom/table.py\", line 4027, in to_tsv\n",
      "    observation_column_name)\n",
      "  File \"/usr/local/lib/python2.7/dist-packages/biom/table.py\", line 1268, in delimited_self\n",
      "    raise TableException(\"Cannot delimit self if I don't have data...\")\n",
      "biom.exception.TableException: Cannot delimit self if I don't have data...\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "\n",
    "metaBEAT_global.py \\\n",
    "-R REFmap.txt \\\n",
    "-f \\\n",
    "-@ L.Harper@2015.hull.ac.uk"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "This will produce `refs.fasta`."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      ">NC_018563|163638|Rhamphochromis esox\r\n",
      "GCTAGCGTAGCTTAACTAAAGCATAACACTGAAGATGTTAAGACGGTCCCTAGAAAGGTCCCGTGAGCACAAAGGCTTGGTCCTGACTTTACTGTCAACTTTGGCTAAACTTACACATGCAAGTCTCCGCCCCCCTGTGAGAATGCCCACAGTTTTCTGCCCGAAAACAAGGAGCTGGTATCAGGCACACTTTCCGCAGCCCATGACACCTTGCTTAGCCACACCCTCAAGGGAACTCAGCAGTGATAGACATTAAGCCATAAGTGAAAGCTTGACTTAGTTAAAGCCAAGAGGGCCGGTAAAACTCGTGCCAGCCACCGCGGTTATACGAGAGGCTCAAGTTGACAGACATCGGCGTAAAGGGTGGTTAGGAAGTTTTTAAACTAAAGCCGAACGCCCTCAGAACTGTTATACGTACCCGAGAGCAAGAAGCCCCACTACGAAAGTGGCTTTATATCCCCGACCCCACGAAAGCTGCGAAACAAACTGGGATTAGATACCCCACTATGCCCAGCCCTAAACCTTGATAGCCCCCTACACCCACTATCCGCCCGGGTACTACGAGCACTAGCTTAAAACCCAAAGGACTTGGCGGTGCTTTATATCCACCTAGAGGAGCCTGTTCTAGAACCGATAACCCCCGTTAAACCTCACCCCCTCTTGTTCTTCCCGTCTATATACCGCCGTCGTCAGCTTACCCTATGAAGGAACCACAGTAAGCAAAACTAGTACAACTCAAAACGCCAGGTCGAGGTGTAGCATATGAGGGGGGAAGAAATGAGCTACATTCCCTGCCACAGGGAATACGAACAATGTAATGAAATACACATTAGAAGGAGGATTTAGCAGTAAGCAGAAAATAGAGCGTTCCGCTGAAATTGGCCCTGAAGCGCGCACACACCGCCCGTCACTCTCCCCAAGCCAACAACATCCTATAAATAATATATTTTACCGGTAAAGGGGAGGCAAGTCGTAACATGGTAAGTGTACCGGAAGGTGTACTTGGAAAAATCAGAGTGTAGCTAAGCCAGAAAAGCATCTCCCTTACACTGAGAAGTCGCCCGTGCAAATCGAGCCACCCTGACGCCCAACAGCTAGCCCCTCCATCAACCCCAAATTAACCCTATTTATATCCCCAAATATACCACTCTTTCAACAACAAACCATTTTTCCACCTAAGTACGGGCGACGAAAAAGGACCTAGGAGCAACAGAGAAAGTACCGCAAGGGAACGCTGAAAGAGAAATGAAACAACCCAGTAAAGCACTAAAAAGCAGAGATTACTCCTCGTACCTTTTGCATCATGATTTAGCCAGAAAACCTCAAGCGAAAAGCATTATAGTTTGATACCCCGAAACTAAGCGAGCTACTCCAAGGCAGTCTAATTTATAGGACCCCCCCCCGTCTCTGTGGCAAAAGAGTGGGAAGAACTTCGAGTAGAGGTGACAGACCTACCGAGCCTAGTTATAGCTGGTTGCCTGAAAACTGAATAAAAGTTCAGCCCTTTAAATTCTCCATTCCCATTGGCCTAAGGCCTTCACACCGAGCAAAGAAGTTAAAGGAGTTAGTCAAAGGGGGTACAGCCCCTTTGAAACAAGATACAACTTTCCAAGGAGGGTAAAAATCACAACGAACTAAAAGGTTAAATGTCCTAGTGGGCCTAAAAGCAGCCACCTACCCAGAAAGCGTTAAAGCTCGAGCATCACCTTACCAGCCTTCTAATAAAGACAATACAATTTCACCCCCCTTAAACCTACCAGGCCGTTCCATAAAACTATGGAAGCGTTTATGCTAACATGAGTAATAAGAGGTCCCACCTCTCCCCGCACAAGTGTAACTCGGAACGAACCCTTCACCGACCATTAACGGCCCCAAAACAAAGAGGGCCCTAGGCAAAAAATAAACAACTAGAAAAACCCCTAGTTCCTCACCGTTAACCCCACACTGGTGTGCAAACCAGGAAAGACTAAAAGAAAAAGAAGGAACTCGGCAAACACAAGCCTCGCCTGTTTACCAAAAACATCGCCTCTTGAACCCCTAAATATAAGAGGTCCCGCCTGCCCTGTGACTATAAGTTTAACGGCCGCGGTATTTTGACCGTGCAAAGGTAGCGCAATCACTTGTCTTTTAAATGAAGACCTGTATGAATGGCATAACGAGGGCTTAACTGTCTCCTTTTTCCAGTCAATGAAATTGATCTCCCCGTGCAGAAGCGGGGATATTAACATAAGACGAGAAGACCCTATGGAGCTTTAGACACCAGAACAGACCATGTTAAACACCCCTCAAATAAAGGACAAAACTCATTGGCCCCTGTCCTAATGTCTTTGGTTGGGGCGACCGCGGGGAAACAACAAACCCCCATGTGGACCGGGAGCACACTACCCCCACAACCCAGAGTCACGACTCCAAGTAACAGAACTTCTGACCAACAAGATCCGGCACATGGCCGATCAACGGACCGAGTTACCCTAGGGATAACAGCGCAATCCTCTTCTAGAGCCCATATCGACAAGAGGGTTTACGACCTCGATGTTGGATCAGGACATCCTAATGGTGCAGCCGCTATTAAGGGTTCGTTTGTTCAACGATTAAAGTCCTACGTGATCTGAGTTCAGACCGGAGTAATCCAGGTCAGTTTCTATCTATGTCACGATCTTTTCTAGTACGAAAGGACCGAAAAGAAGGGGCCCCTGTTCTCAATATGCCTCCCCCTCATCTATTGAAGTCAACTAAAATAGATAAAAGGGCGTACCCCCTAGCCATAGAAAATGGCTTGTTAAAGTGGCAGAGCCCGGACATTGCAAAAGACCTAAGCCCTTTCCACGGAGGTTCAAGTCCTCCCTTTAACTATGCTCTCAACACTAGTTTCATCCATCCTCAACCCCTTAATTCTTATCGTGTTTGTCCTTCTAGCCGTTGCACTCCTCACACTTGTCGAACGAAAAGTCCTCGGCTACATACAATTACGAAAGGGTCCAAACATTGTAGGCCCCTACGGCCTCCTCCAACCAGTTGCAGACGGACTTAAGCTCTTTATAAAAGAACCCGTTCGACCCTCCACCTCCTCGCCCATCCTCTTTCTCTTTACCCCCATACTAGCCCTCACCCTGGCCCTCACCCTTTGAACTCCAATGCCCCTCCCCTTCCCAATAGCAGATCTCAACCTTGGCATCCTTTTTATCCTTGCCCTCTCCAGTCTAGCAGTTTATTCTATTCTAGGCTCAGGATGAGCCTCCAATTCAAAATACGCCTTAATCGGGGCCCTTCGAGCCGTAGCACAGACTATCTCCTATGAAGTTAGCCTAGGGCTAATCCTTCTTAATGCTATTATTTTTACTGGAGGCTTCACCCTGCAAACATTCAGCATCGCCCAAGAAAGTGTCTGATTAATCCTCCCCGCCTGGCCCTTAGCCGCTATATGGTACATTTCCACACTTGCAGAAACAAACCGAGCCCCCTTTGACCTCACAGAGGGTGAGTCCGAACTCGTCTCCGGCTTTAACGTAGAATACGCAGGAGGACCTTTCGCCCTTTTTTTCCTCGCTGAATACGCCAACATTCTCCTAATAAATACTCTCTCTGCAACTCTATTCCTAGGCGCCTCTATCTCTCACACCACCCCTGAAATTACAACAACAAATCTTATAATCAAAGCAGCTCTCCTATCCGTCCTCTTCCTATGAGTTCGTGCTTCCTACCCACGATTTCGTTATGATCAACTCATGCACCTAATTTGAAAAAACTTCCTACCACTAACCCTTGCCCTGGTAATTTGACACCTCTCCCTCCCAATTGCCCTGACAGGCCTACCCCCTCAACTATAGTCCGGAGCTGTACCTAAAATAAAGGACCACTTTGATAGAGTGAATCATGAGGGTTAAAATCCCTCCAACTCCTTAGAAAGAAGGGACTTGAACCCTACCTGAAGAGATCAAAACTCTTAGTGCTTCCTCTACACCACTTCCTAGTAAAGTCAGCTAAATAAGCTTTTGGGCCCATACCCCAAACATGTTGGTTAAACTCCTTCCTTCACTAATGAACCCCTACATCTTAGCCATTCTTCTCTTTGGCTTAGGCCTTGGCACAACAATCACATTTGCTAGTTCCCACTGACTTCTCGCCTGAATAGGCCTTGAAATAAATACACTAGCCATTATTCCCCTAATAACCCAACATCACCACCCACGCGCAGTCGAAGCTACAACCAAATATTTTTTGACTCAAGCCACTGCTGCAGCTACCCTTCTATTTGCAAGTGTAACTAACGCCTGACTAACAGGCCAGTGAGAAATTCAACAAATTACGCACCCCCTCCCAAGTACCATAATTACTCTTGCACTTGCTCTCAAAATTGGTCTAGCCCCTCTTCATGCTTGACTCCCCGAAGTCCTGCAAGGCCTAGACCTCACCACAGGCTTAATTCTTTCAACCTGACAAAAGCTTGCCCCCTTCGCCCTGATTCTTCAACTTCAACCTTCAAGCTCAACACTCCTTATCATCTTAGGTCTTACATCCACACTTATTGGGGGCTGAGGCGGATTAAACCAAACACAACTCCGCAAGATTCTTGCATATTCATCAATCGCCCATTTAGGCTGAATAATTCTTATCCTACAATTCTCCCCCTCCATCACCCTCCTTACCCTCCTAACCTATTTCATTATAACATTCTCAACATTTCTTGTATTTAAACTCAACGAATCTACAAATATCAACACCCTTGCCACATCCTGAGCAAAAGCCCCCGCCCTGACAGCTCTCATACCCCTCATTCTACTCTCCCTAGGAGGCCTCCCCCCTCTTACAGGCTTCATGCCAAAATGACTAATCCTTCAAGAACTAACCAAACAAGGGCTTGCTCCAACCGCAACCCTAGCTGCCCTTTCAGCCCTACTTAGCCTGTATTTTTACCTACGCCTCTCTTACGCAATAACCCTCACTATTTCCCCTAACAATCTCACAGGTTCAACCCCCTGACGCTTACCTTCCACTCAACTAACCTACCCCCTCGCCACTTCAACTGCAATAACAATTTGCCTCCTACCTCTCACCCCTGCCATCTCCGCCTTATTAACCTCCTAAGGGGCTTAGGATAACACCCAGACCAAGGGCCTTCAAAGCCCTAAGCGGGAGTGAAAATCTCCCAGCCCCTGCTAAAACTTGCAGGATACTAACCCACATCTTCTGTATGCAAAACAGACACTTTAATTAAGCTAAAGCCTTGCTAGACAGGAAGGCCTCGATCCTACAAACTCTTAGTTAACAGCTAAGCGCTTAAACCAACAAGCATCTGTCTAACCTTTCCCCCGCCCGCCTCCAAAAGGGCGGGGGAAAGCCCCGGCAGGGTTCTAACCTGCTGCTTCGGATTTGCAATCTGATATGTATAACACCTCGAGGCTTGATAAGAAGAGGATTTGAACCTCTGTTCGTGGGGCTACAATCCACCGCTTAACACTCAGCCATCTTACCTGTGGCAATCACTCGTTGATTCTTCTCAACTAATCACAAAGACATCGGTACCCTCTATCTAGTATTTGGTGCTTGGGCCGGAATAGTAGGAACTGCACTTAGCCTCCTAATTCGGGCAGAACTAAGTCAGCCCGGCGCGCTCCTCGGAGATGACCAGATTTATAATGTAATTGTTACGGCACATGCTTTTGTAATAATTTTCTTTATAGTAATACCAATTATGATTGGAGGCTTTGGGAACTGACTAGTGCCTCTTATAATTGGTGCACCCGACATAGCTTTCCCTCGAATAAACAATATAAGCTTCTGACTCCTCCCCCCTTCATTCCTTCTCCTCCTTGCCTCTTCCGGAGTCGAAGCAGGGGCTGGCACAGGATGAACTGTTTATCCTCCGCTCTCAGGCAATCTCGCCCACGCGGGACCTTCTGTCGATTTAACCATCTTCTCCCTCCATTTGGCCGGGGTGTCGTCTATTCTTGGCGCAATTAATTTTATTACAACAATTATTAACATAAAACCCCCTGCCATCTCTCAATACCAGACACCTCTGTTTGTGTGATCTGTCCTAATTACCGCAGTATTGCTTTTATTATCCCTACCCGTGCTTGCTGCCGGCATCACAATACTTCTCACAGACCGAAACCTTAATACAACCTTCTTTGACCCTGCTGGAGGAGGGGACCCCATCCTTTACCAACATTTATTCTGATTCTTTGGTCATCCTGAAGTCTATATCCTTATCCTCCCTGGCTTCGGTATAATCTCCCACATTGTTGCGTACTACGCGGGTAAAAAAGAACCCTTCGGATATATGGGCATGGTCTGGGCCATAATGGCCATTGGCCTCCTAGGGTTTATTGTCTGAGCTCACCACATGTTTACTGTAGGAATGGACGTAGACACACGAGCTTACTTTACTTCCGCCACAATAATTATTGCTATCCCAACCGGGGTAAAAGTTTTCAGCTGGCTGGCCACTCTGCACGGCGGCTCAATTAAATGAGAAACTCCACTCTTATGAGCACTCGGCTTCATTTTCCTATTTACTGTTGGGGGACTAACCGGAATTGTCCTGGCCAACTCTTCTCTAGATATTATGCTTCACGACACATATTACGTCGTTGCCCACTTCCACTATGTTCTCTCGATAGGGGCTGTATTTGCCATCGTTGCCGGCTTCGTCCACTGATTCCCCTTATTCTCAGGATACACGCTCCACAACACCTGAACCAAAATCCACTTCGGGGTGATGTTTGCTGGTGTTAATATAACTTTCTTCCCACAACATTTCCTGGGGCTGGCAGGAATACCTCGCCGATACTCCGACTATCCCGACGCCTACACCCTTTGAAACACAGTTTCCTCCATTGGCTCAATAGTTTCCCTAATCGCAGTAATTATATTTTTATTCATTATTTGAGAAGCATTCGCCGCTAAACGAGAAGTTTCGTCAGTGGAACTCACGGCAACAAACGTAGAATGACTTCACGGTTGCCCTCCTCCTTACCACACCTTCGAAGAACCTGCCTTCGTTCAAGTTCAAACTTGATCAAACTACGAAAAATCTGCTTCCGCCCCATCAAACCCCTAGTAACGAGAAAGGGAGGAATTGAACCCCCATAAACTGGTTTCAAGCCAGCCACATAACCACTCTGTCACTTTCTTCATAAGACACTAGTAAAGTTGACCATTACATTGCCTTGTCAAGGCAAAATTGCGGGTTTAAGCCCCGCGTGTCTTACAACAATGGCACATCCCTCCCAACTAGGTTTTCAAGATGCAGCTTCACCTGTAATAGAAGAACTTCTTCACTTCCACGACCATGCCTTAATAATTGTTTTTCTAATTAGCACATTCGTGCTTTATATTATTGTGGCTATAGTAACAACCAAGTTAACTAATAAATTTATCCTAGACTCCCAAGAAATCGAAATCATTTGAACCCTGCTCCCAGCTATTATTCTGATTCTCATTGCCCTCCCCTCCCTACGTATTCTTTATCTTATAGATGAAATTAACGACCCTCATCTTACAATTAAAGCAATAGGCCACCAATGATACTGAAGTTACGAGTATACTGACTATGAAGACCTCGCCTTTGACTCATACATAGTTCCCACACAAGACTTAGCCCCAGGCCAATTCCGCCTTCTAGAAGCAGACCATCGAATAGTAGTACCAGTTGAATCCCCCATCCGGGTCCTAATTTCTGCCGAAGATGTACTTCACTCCTGAGCCGTCCCAAGCCTGGGGGTAAAAATAGACGCCGTCCCAGGGCGCTTAAACCAAACAGCATTTATTGCCTCCCACCCCGGGATCTTTTATGGCCAATGCTCTGAAATTTGCGGCGCAAACCACAGCTTTATGCCTATTGTGGTAGAAGCAGTACCACTAGAACACTTTGAAAACTGATCCTCCTTAATACTTGAAGACACTTCGCTAAGAAGCTAAATAGGGAATAGCGTTAGCCTTTTAAGCTAAAGACTGGTGACCCCCGCCCACCCCTAGCGAAATGCCACAACTTAACCCCGCCCCTTGATTTGCTATTCTAGTCTTCTCCTGATTAGTTTTCCTAACAGTCATTCCCCCAAAAGTTCTAGCACACACCTTCCCAAACGACCCAACACTCCAAAGCACAGAGAAACCCAAAACAGAGCCCTGAAGTTGACCATGACACTAAGCTTTTTTGACCAATTTATGAGCCCCACATACCTGGGAGTCCCCCTCATTGCCCTTGCCCTCAGTTTACCCTGAATCCTTTACCCCAAACCTACCCCCCGTTGACTAAACAACCGCCTCATTACACTCCAAGGATGGTTTATTAACCGCTTTACCCAACAAATTTTTCAACCTCTAAGTTTAGGGGGCCATAAATGAGCAGCCCTTCTCGCCTCCCTCATACTTTTCCTCATTACCTTAAACATACTTGGTCTCCTACCCTACACCTTTACCCCCACAACACAACTCTCCCTCAACATGGCCTTCGCCGTCCCCCTCTGACTTGCAACAGTCATTATTGGTATACGAAACCAGCCTACCCATGCCCTAGGCCACCTGTTACCAGAAGGTACCCCTACCCTCTTAATCCCTGTCCTAATTATCATCGAGACAATTAGCCTATTTATTCGCCCCCTCGCACTTGGCGTGCGATTAACCGCAAACCTTACAGCCGGTCACCTTTTAATTCAACTCATTGCCACCGCCGCCTTCGTCCTCCTCCCCCTAATACCCACAGTAGCCATTCTGACCGCAGTACTACTATTCCTCCTGACCCTTCTAGAAGTTGCAGTGGCCATAATTCAAGCCTATGTCTTTGTCCTTCTCTTAAGCCTCTACCTACAAGAAAACGTTTAATGGCCCATCAAGTACACGCATATCACATAGTTGACCCCAGCCCATGACCCCTAACAGGCGCAGTAGGCGCCCTTCTACTAACCTCCGGTTTAGCAATCTGAATGCATTTCCATAATATAACCTTATTAACACTAGGTCTTATCCTTCTTCTTCTAACTATATATCAATGATGACGGGATATTGTCCGAGAAGGAACATTCCAAGGGCACCACACCCCTCCTGTCCAAAAAGGCCTCCGATACGGAATGATCCTCTTTATTACCTCAGAAGTGTTCTTCTTTCTAGGTTTCTTCTGAGCCTTTTATCACTCCAGCCTCGCCCCAACTCCGGAACTAGGAGGCTGCTGACCCCCTACAGGAATTACCCCTCTAGACCCCTTCGAAGTCCCCCTGCTCAATACAGCCGTCCTACTAGCCTCAGGAGTCACAGTCACCTGGGCACACCACAGCATCATAGAAGGGCATCGAAAAGAAGCAATCCAATCCCTCACTTTAACTATCCTTCTAGGCTTCTACTTTACCTTCCTTCAAGCAATAGAATACTACGAAGCCCCCTTCACTATTGCAGACGGAGTTTATGGTGCCACCTTCTTCGTAGCAACCGGCTTCCACGGACTCCACGTAATCATTGGCTCCACCTTCCTAGCCATCTGCCTTCTACGACAAGTCCTATATCATTTCACCTCCGAACACCACTTTGGTTTTGAAGCAGCCGCCTGATACTGACACTTTGTAGACGTTGTCTGACTATTCCTTTATATCTCAATTTACTGATGAGGCTCATATCTTTCTAGTATTAAAGCTAGTACCTGTGACTTCCAATCACCTAGTCTTGGTTAAACCCCAAGGAAAGATAATGAATTTAATCACAACAATACTCATTATTTCTATCGCCCTCTCCACTATCCTCGCTATTGTTTCTTTTTGACTCCCCCAAATAACACCTGACCATGAAAAACTTTCCCCCTACGAATGTGGCTTTGATCCCCTAGGTTCCGCTCGCCTGCCCTTCTCTCTCCGCTTCTTCCTTGTTGCAATCCTTTTCCTCCTATTCGATTTGGAAATTGCACTGCTCCTTCCCCTTCCTTGAGGGGATCAACTTTCTTCCCCTCTCACAACATTCACCTGAGCCTTCGCTGTTCTTATATTATTAACCCTCGGCCTAATTTATGAATGAACTCAAGGCGGTCTAGAATGGGCTGAATAGACTGTTAGTTTAAGAAAAACCCTTGATTTCGGCTCAAGAGCTTGTGGTTAAAGTCCATAACCGTCTAATGACCCCCACACATTTCGCCTTTTCCTCTACCTTTCTTCTAGGCCTAGCAGGCCTGGCATTCCACCGAACCCATCTTCTTTCCGCTCTTTTATGTTTAGAGGGTATAATACTCTCACTCTTTATTGCTCTCTCCATGTGAACCCTTCAACTTAACTCCGTCAACTTCTCAGCCTCCCCCATACTTCTCCTGGCTTTCTCAGCCTGTGAAGCAAGCGCCGGCCTCGCACTACTTGTTGCCACTGCCCGCACTCACGGAACAGACCGACTCCAAAACCTAAATCTTCTACAATGCTAAAAATTCTACTCCCTACTATTATGCTTGTCCCCACTATTTGAGCCGTCCCGGCCAAACACCTCTGATCCACCGCCCTTTCCTACAGTCTAATCATCTCCTTAACCAGCCTAACCTGACTAAAATCATCCGCTGAATCAGGCTGATCTTTTCTCAGCCCTTACATAGCAACAGACCCCCTCTCCACCCCCCTTCTTGCACTAACCTGTTGGCTTCTCCCTCTGATAATTCTTGCAAGCCAAAACCACACAGCGTCCGAACCTTCCTCCCGCCAACGAACCTACATTACCCTCCTTACATCCTTACAAATCTTCCTCATTATAGCCTTCGGCGCAACCGAAATAATTATATTTTATATTATGTTTGAAGCTACCCTTATTCCAACCCTTGTAATCATTACTCGTTGAGGAAATCAAACAGAACGACTAAATGCGGGCACTTATTTTTTATTCTATACACTAGCAGGCTCACTCCCTCTGCTTGTCGCTCTCCTACTGCTCCAAAACAGCACTGGCACCCTATCCCTTCTAACACTACAATATGCCCCCTCCCTACAACTCTCTTCTCTCGCCGATAAACTATGGTGAGCCGGTTGCTTGCTCGCCTTTCTAGTAAAAATACCCCTCTATGGCGCCCACCTCTGACTTCCCAAGGCACACGTTGAAGCCCCAATCGCGGGTTCCATAGTACTAGCCGCAGTCTTACTAAAACTAGGAGGCTATGGAATAATACGTATAATAATTATACTAGAACCACTCACCAAAGAACTCAGCTACCCCTTCATTATCTTCGCCCTCTGAGGGGTAATTATAACTGGCTCAATCTGCCTCCGCCAAACAGACTTAAAATCCCTAATTGCTTATTCCTCAGTAAGTCATATGGGGCTCGTAGCAGCAGGCATTCTAATCCAAACCCCCTGAGGTTTCACAGGCGCCCTCATTCTAATAATCGCGCACGGTTTAACTTCCTCCGCCCTCTTTTGCCTAGCTAACACAAACTACGAACGAACCCACAGCCGAACCATAGTATTAGCCCGAGGACTCCAAATGGCTTTACCTCTAATAACCGCATGATGATTCATCGCCAGCCTTGCAAACCTTGCCCTCCCTCCTCTTCCAAATCTGATAGGAGAACTCATAATCATTACCTCCCTGCTCCACTGATCCTGATGAACAATTGCACTCACGGGGGCTGGGACCCTAATCACTGCAGGCTACTCCCTATATATATTTCTTATAACACAACGGGGGCCCCTACCAGCACATATTATATCCCTCGACCCAACACATTCCCGAGAACATCTTCTCCTAGCCCTTCATCTCCTACCTCTAATTCTTCTAATCACCAAGCCCGAATTAATTTGAGGGTGGACCGCCTGTAGATATAGTTTAACAAAAACATTAGATTGTGATTCTAAAGACAGAGGTTAAAACCCCCTTATTCACCGAGAGAGGTTGACAACAACAAAGACTGCTAATTTTTGCCTCTTAGGTTAGACTCCTAAGCTCACTCGCCCCGCTTCTAAAGGATAACAGCTCATCCGTTGGTCTTAGGAACCAAAAACTCTGTGGTGCAAATCCAAGTAGCAGCTATGCACCTCACCTCCACCATAATAACCACTAGTCTAATTCTTATTTTTTTATTACTTACATTCCCCATCCTTTCTTCCTTTTCCCCCACCCCCCTCCCTCCCAACTGAGCACTAACCCAGGTTAAAACAGCAGTAAAATGAGCCTTCTTTACTAGCATCCTCCCTCTCTGCCTCTTCCTTAACGAAGGCACAGAAACAATTATTACCAGCTGAACTTGAATAAACACCCACACATTTGATATTAATATTAGTCTTAAATTTGATATCTACTCAATTATCTTCACCCCTGTCGCCCTCTATGTCACCTGGTCAATCCTAGAATTTGCCTCCTGATACATACATGCCGACCCGAACATAAATCGGTTTTTTAAATACCTCCTAATCTTCCTTATTGCCATGATCATTCTTGTTACCGCAAACAATATATTTCAACTATTCATCGGCTGAGAAGGTGTCGGAATTATATCTTTTCTCCTCATCGGCTGATGATACGGCCGTGCAGACGCAAACACCGCTGCCCTCCAGGCCGTAATCTATAACCGAGTAGGGGACATCGGCCTAATTTTTGCTATAGCCTGAATTGCAACCTCCCTTAACTCCTGAGAAATACAACAAATATTTACCTTATCCAAAGACTTTGATCTAACTTACCCCCTCATTGGCCTCATCATTGCTGCCACTGGTAAGTCCGCCCAATTTGGCCTCCACCCCTGGCTTCCTTCTGCCATAGAAGGTCCTACACCGGTCTCTGCCCTACTGCACTCAAGCACCATGGTCGTAGCAGGCATCTTCCTCCTCATCCGCATAAGCCCAATGCTAGAAAACAATCAAACCACCTTAACCATTTGCCTTTGCTTAGGGGCCCTCACCACCCTCTTTACTGCAACCTGCGCCCTCACCCAAAATGATATCAAAAAAATCGTTGCTTTCTCAACCTCAAGTCAACTGGGTTTAATAATAGTAACAATTGGGCTTAACCAACCCCAACTTGCCTTCCTTCACATCTGCACTCACGCATTCTTTAAAGCTATGCTTTTCCTCTGCTCAGGGTCTATTATTCATAGCCTAAACGACGAGCAAGACATCCGGAAAATAGGAGGTATACATCACCTGACTCCTTTTACATCTTCCTGCCTAACCATCGGAAGCCTAGCTCTCACAGGAACCCCCTTCTTAGCAGGTTTCTTTTCAAAAGATGCTATTATTGAGCCCTTAAACACATCTTACCTTAACGCCTGGGCCCTATTCCTCACCCTCCTAGCCACTTCCTTCACCGCTATCTACAGTCTTCGGGTAATTTTCTTCGTCTCAATAGGCCACCCCCGCTTCAACCCTCTTTCCCCAATTAACGAAAACAATTCAACAGTTATTAACCCCATCAAACGCCTAGCCTGAGGAAGTATTATCGCCGGCCTCCTAATTACCTCTAATATCACTCCCCTGAAAACACCAGTTATATCCATACCGTGCCTTCTCAAAGTTGCCGCCCTAATAGTGACTATTGTCGGCCTTCTCACCGCACTAGAACTCGCCTCACTAACCAATAAGCAATATAAACCAACACCAAACCTTTCTCCCCACCATTTTTCTAACATACTAGGTTTCTTCCCAATAGTTATTCATCGCCTCATCCCCAAACTAAACCTGATTTTAGGACAAACCATCGCCTCCCAAACAGTCGACCAATCATGATTAGAAAAAATGGGCCCAAAAGCAACTGCTACCCTTAACCTCCCTCTAATTACCACAACTAACAACATTCAACAGGGTATAATCAAAACCTATCTTTCCCTCTTCTTCTTCACCTTCGGTTTAGCTCTTCTACTACTACTTTATTAAACAGCTCGAAGGGCCCCCCGACTTAAACCCCGCGTTAACTCCAACACCACAAACAACGTTAACAGCAACACTCAAGCCCCTATCACCAAAACACCCCCACCCATCGAGTACATCAAAGCCACCCCTCCAACATCCCCCCGAAAAACCGAAAACTCAACAAATTCATCAGCAGACACTCAAGCCCCCTCGTATCACCCCCCTCAAAATAACACCGCCGCCACCCCCACCCCTAGTACATACGCCACTATTACCCCCAATACAGGCCAACTCCCTCAACCCTCAGGATAAGGCTCAGCAGCCAATGCTGCTGAATATGCAAACACTACTAATATTCCCCCCAAATAAATCAAAAACAAAATTAAAGATAAAAAAGACCCCCCATGCCCCACCAACACCCCACACCCTATACCTGCTACCACAACCAGCCCTAACGCCGCAAAATACGGCGAAGGATTAGAAGCAACCGCCGCAAGACCTACTACCAGCCCTAGTAAAAATATAAACATAATATAAACCATAGTTTCTGCCAGGACTTTAACCAGGACTAATGACTTGAAAAACCACCGTTGTTATTCAACTACAAAAACAGTAATGGCCAACCTCCGAAAAACCCACCCCCTCCTAAAAATTGCAAACGACGCACTAGTTGATCTCCCAGCCCCTTCAAACATTTCTGTTTGATGAAACTTTGGCTCTCTGCTAGGGCTCTGCCTAGCTGCCCAAATCCTGACAGGCCTTTTCCTAGCCATACACTACACCTCCGATATCGCCACCGCCTTCTCCTCCGTTGCCCACATCTGCCGTGATGTTAACTACGGCTGACTCATTCGAAACATGCACGCTAACGGCGCATCCTTTTTCTTCATTTGTATTTATCTACACATCGGCCGAGGCCTGTACTACGGCTCCTACCTCTATAAAGAAACCTGAAACATTGGAGTAATTCTACTCCTTTTAACTATGATAACAGCTTTTGTGGGCTATGTCCTCCCATGAGGTCAAATATCGTTCTGAGGCGCCACTGTCATCACAAACCTTCTCTCCGCAGTCCCTTATATTGGCAACTCTTTAGTCCAGTGAATCTGAGGAGGGTTTTCCGTAGACAACGCCACCTTAACCCGCTTCTTCGCCTTCCATTTCCTCCTTCCCTTCATTATTGCAGCTGCAACAATAGTACACCTTATTTTCCTCCACGAAACCGGCTCCAACAACCCCACAGGCCTAAACTCAGACGCCGACAAAGTCTCTTTCCACCCTTACTTCTCCTACAAAGACTTATTAGGTTTCGCAATCCTTCTAATCGCCCTCATTTCTCTCGCCCTCTTCTCCCCCAACCTGCTTGGGGACCCCGACAACTTCACCCCTGCAAACCCCTTAGTTACCCCGCCTCATATTAAACCCGAATGATACTTCCTATTTGCCTATGCCATCCTCCGCTCAATCCCTAACAAACTTGGCGGGGTTCTCGCCCTCCTGTTCTCAATCCTTGTCTTGATAGTTATTCCTATGCTCCACACCTCTAAACAACGAGGCTTAACATTTCGCCCTATCACGCAATTCCTCTTCTGACTCTTAATTGCAAACGTCGCCATCCTCACCTGAATTGGAGGCATACCTGTTGAACACCCCTTCGTCATTATTGGGCAAATTGCATCTTTCCTCTACTTCTTCCTCTTTCTCGTCCTCGCCCCTCTCGCCGGCTGACTAGAAAACAAAATCCTTGAATGATCCTGCACTAGTAGCTCAGCGCCAGAGCGCCGGTCTTGTAAACCGGACGTCGAAGGTTAAAGCCCTTCCTACTGCTTCAAACAAAGGGGATTTTAACCCCCGCCCCTAACTCCCAAAGCTAGGATCCTAATTTAGACTATTGTTTGCCGGGCTCTGCCTTTCATGTAAACGCAGTGCATATATGTATTATCACCATTATTTTATATCAAACATATCCTATATATTAATACATATCATTTAAAAGACATAGACAGATATACCACATATTTGTTTAAACCATTTTAACTAAGGGTTACATAAACCATAACTGAACACATCCAATAAAGATTTATTGTCCACTGAACGATAGTTTAAGACCGACCACAACTCTCACTAGTTAAGTTATACCAAGTACCCACCATCCTATTTACTACCCATGTTTAATGTAGTAAGAGCCCACCATCAGTTGATTCCTTAATGTTAACGGTTCTTGAAGGTCAAGGACAAATATTTGTGGGGGTTTCACTTAGTGAATTATTCCTGGCATCTGGTTCCTATTTCAGGTCCTATAATTGTTATAATCCCCCATACTTTCATCGACGCTTGCATAAGTTAATGGTGGTAATACATACTCCTCGTTACCCACCATGCCGAGCATTCTCTCCAGCGTGTGGGGGGTTCTCTTTTTTTTTTTCCTTTCATTTGACATCTCAGAGTGCACACAGAAATGACAGACAAGGTTGAACATTTTCCTTGCATGAAATAAATAGTATGAATGGTGATTAGATATTGATAGAAGAATTGCATAACTGATATCAAGAGCATAAAGTTTAATCAGATATTTAATTTTCCCCAAACTTTCCTATTATCACCCCCGGTTTTTGCGCGTAAACCCCCCCTACCCCCCCAAACTCCTAAGATCTCTAAGACTCCTGTAAACCCCCCGGAAACAGGAAAAGCTCTAGAAGTGTTTTTCGCACTCGTAATGTACAGACATGATTGTTATTCATAAATTGTTTGATGTGTATATTATACTATTGCAATATTGCACAT\r\n",
      ">AP013078|9796|Equus caballus\r\n",
      "CATAGGCTTGGTCCTAGCCTTTTTATTAGTTATTAATAGAATTACACATGCAAGTATCCGCACCCCAGTGAGAATGCCCTCTAAATCGCGTCTCTACGATTAAAAGGAGCAGGTATCAAGCACACTAGAAAGTAGCTCATAACACCTTGCTCAGCCACACCCCCACGGGACACAGCAGTGATAAAAATTAAGCTATGAACGAAAGTTCGACTAAGTTATATTAAATAAGGGTTGGTAAATTTCGTGCCAGCCACCGCGGTCATACGATTAACCCAAATTAATAAACCCTCCGGCGTAAAGCGTGTCAAAGACTAATACCAAAATAAAGTTAAAACCCAGTTAAGCCGTAAAAAGCTACAACCAAAGTAAAATAGACTACGAAAGTGACTTTAATACCTCTGACTACACGATAGCTAAGACCCAAACTGGGATTAGATACCCCACTATGCTTAGCCCTAAACTAAAATAGCTTACCACAACAAAGCTATTCGCCAGAGTACTACTAGCAACAGCCTAAAACTCAAAGGACTTGGCGGTGCTTTACATCCCTCTAGAGGAGCCTGTTCCATAATCGATAAACCCCGATAAACCCCACCATCCCTTGCTAATTCAGCCTATATACCGCCATCTTCAGCAAACCCTAAACAAGGTACCAAAGTAAGCACAAATATCCAACATAAAAACGTTAGGTCAAGGTGTAGCCCATGGGATGGAGAGAAATGGGCTACATTTTCTACCCTAAGAACAAGAACTTTAACCCGGACGAAAGTCTCCATGAAACTGGAGACTAAAGGAGGATTTAGCAGTAAATTAAGAATAGAGAGCTTAATTGAATCAGGCCATGAAGCGCGCACACACCGCCCGTCACCCTCCTTAAATATCACAAATCACAACATAACATAAAACCGTGACCCAAACATATGAAAGGAGACAAGTCGTAACAAGGTAAGTATACCGGAAGGTGTACTTGGATAAC\r\n",
      ">AY879429|9940|Ovis aries\r\n",
      "ATATAGGTTTGGTCCCAGCCTTCCTGTTAACTTTCAATAGACTTATACATGCAAGCATCCACGCCCCGGTGAGTAACGCCCTTCGAATCACACAGGACTAAAAGGAGCAGGTATCAAGCACACACTCTTGTAGCTCACAACGCCTTGCTTAACCACACCCCCACGGGAGACAGCAGTAACAAAAATTAAGCCATAAACGAAAGTTTGACTAAGCCATATTGACCAGGGTTGGTAAATCTCGTGCCAGCCACCGCGGTCATACGATTGACCCAAGCTAACAGGAGTACGGCGTAAAGTGTGTTAAAGCATCATACTAAATAGAGTTAAATTTTAATTAAACTGTAAAAAGCCATAATTATAACAAAAATAAATGACGAAAGTAACCCTACAATAGCTGATACACCATAGCTAAGACCCAAACTGGGATTAGATACCCCACTATGCTTAGCCCTAAACACAAATAATTAT\r\n",
      ">AY962041|9668|Mustela putorius\r\n",
      "CAAAGGTTTGGTCCTAGCCTTCCTATTGATTATTAACAGAATTACACATGCAAGTCTCTACACCCCAGTGAGAATGCCCTCYAAATCTATATGTTGATTAAAAGGAGCGGGTATCAAGCACACTAAATTAGTAGCTCATAACGCCTTGCTCAACCACACCCCCACGGGATACAGCAGTGATAAAAATTAAGCCATAAACGAAAGTTTGACTAAGCCATGTTAACAAAGAGCTGGTAAATTTCGTGCCAGCCACCGCGGTCATACGATTAGCCCGAATCAATAGGCAAACGGCGTAAAACGTGTTAAGGATTATATTATACTAAAGTTAAAATTTGACAAGGCTGTAAAAAGCTACTGTTAATATAAGATAAACCACGAAAGTGACTTTATTACTTCCATCAACACGATAGCTGAGACCCAAACTGGGATTAGATACCCC\r\n",
      ">KC572710|9770|Balaenoptera physalus\r\n",
      "TAAAGGTTTGGTCCCAGCCTTTCTATTAGTTCTTAGCAGACTTACACATGCAAGTATCCACATCCCAGTGAGAACGCCCTCTAAATCATAAAGATTAAAAGGAGCGGGTATCAAGCACGCTAGCATTAGCAGCTCACAACGCCTCGCTTAGCCACGCCCCCACGGGACACAGCAGTGATAAAAATTAAGCTATAAACGAAAGTTCGACTAAGTCATGTTAATTTAAGGGTTGGTAAACTTCGTGCCAGCCACCGCGGTCATACGATCGACCCAAATTAATAGAAGCACGGCGTAAAGAGTGTTAAGGAGTCACATGGAATAAAGTCAAACCTTAATTAAGCTGTAAAAAGCCCTAATTAAAATTAAGCCAAACTACGAAAGTGACTTTAATATAATCTGATCACACGACAGCTAAGATCCAAACTGGGATTAGATACCCCACTATGCTTAGTCGTAAACCCCAATAGTCACAAAACAAGACTATTCGCCAGAGTACTACTAGCAACAGCCTAAAACTCAAAGGACTTGGCGGTGCCTCATACCCCTCTAGAGGAGCCTGTTCTGTAACCGATAAACCCCGATCAACCTCACCAACCCTTGCTACTTCAGTCTATATACCGCCATCTTCAGCAAACCCTAAAGGGAGCAAAAGTAAGCATAACCATCCTACATAAAAACGTTAGGTCAAGGTGTAACCCATGGGTTGGGAAGAAATGGGCTACATTTTCTAAGCTAAGAACATCCCCTATACTCACACGAAAGTTTTTATGAAACTTAAAAACTAAAGGAGGATTTAGTAGTAAATCAAGAGCAGAGTGCTTGATTGAATAAGGCCATGAGGCACGCACACACCGCCCGTCACCCTCCTCAAGTACCCCAGCTATAAACCCCAGTTCGTTAACTCAGGCCAAGCAATTATACAAGAGGAGACAAGTCGTAACAAGGTAAGCATACCGGAAGGTGTGCTTGGACAAAA\r\n"
     ]
    }
   ],
   "source": [
    "!head refs.fasta"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Now run chimera detection."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "### Detecting chimeras in F1 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F10 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F100 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F101 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F102 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F103 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F104 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F105 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F106 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F107 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F108 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F109 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F11 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F110 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F111 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F112 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F113 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F114 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F115 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F116 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F117 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F118 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F119 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F12 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F120 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F121 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F122 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F123 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F124 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F125 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F126 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F127 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F128 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F129 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F13 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F130 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F131 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F132 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F133 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F134 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F135 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F136 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F137 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F138 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F139 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F14 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F140 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F141 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F142 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F143 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F144 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F145 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F146 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F147 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F148 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F149 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F15 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F150 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F151 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F152 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F153 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F154 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F155 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F156 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F157 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F158 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F159 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F16 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F160 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F161 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F162 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F163 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F164 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F165 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F166 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F167 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F168 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F169 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F17 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F170 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F171 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F172 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F173 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F174 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F175 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F176 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F177 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F178 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F179 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F18 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F180 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F181 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F182 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F183 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F184 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F185 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F186 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F187 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F188 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F189 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F19 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F190 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F191 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F192 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F193 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F194 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F195 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F196 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F197 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F198 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F199 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F2 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F20 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F200 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F201 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F202 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F203 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F204 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F205 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F206 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F207 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F208 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F209 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F21 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F210 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F211 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F212 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F213 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F214 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F215 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F216 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F217 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F218 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F219 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F22 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F220 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F221 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F222 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F223 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F224 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F225 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F226 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F227 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F228 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F229 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F23 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F230 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F231 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F232 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F233 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F234 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F235 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F236 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F237 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F238 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F239 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F24 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F240 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F241 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F242 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F243 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F244 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F245 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F246 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F247 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F248 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F249 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F25 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F250 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F251 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F252 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F253 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F254 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F255 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F256 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F257 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F258 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F259 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F26 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F260 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F261 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F262 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F263 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F264 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F265 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F266 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F267 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F268 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F269 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F27 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F270 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F271 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F272 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F273 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F274 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F275 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F276 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F277 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F278 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F279 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F28 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F280 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F281 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F282 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F283 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F284 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F285 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F286 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F287 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F288 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F289 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F29 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F290 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F291 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F292 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F293 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F294 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F295 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F296 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F297 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F298 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F299 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F3 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F30 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F300 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F301 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F302 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F303 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F304 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F305 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F306 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F307 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F308 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F309 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F31 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F310 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F311 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F312 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F313 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F314 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F315 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F316 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F317 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F318 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F319 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F32 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F320 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F321 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F322 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F323 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F324 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F325 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F326 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F327 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F328 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F329 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F33 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F330 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F331 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F332 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F333 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F334 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F335 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F336 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F337 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F338 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F339 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F34 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F340 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F341 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F342 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F343 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F344 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F345 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F346 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F347 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F348 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F349 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F35 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F350 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F351 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F352 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F353 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F354 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F355 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F356 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F357 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F358 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F359 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F36 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F360 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F361 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F362 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F363 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F364 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F365 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F366 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F367 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F368 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F369 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F37 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F370 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F371 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F372 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F373 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F374 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F375 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F376 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F377 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F378 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F379 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F38 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F380 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F381 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F382 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F383 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F384 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F385 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F386 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F387 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F388 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F389 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F39 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F390 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F391 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F392 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F393 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F394 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F395 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F396 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F397 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F398 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F399 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F4 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F40 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F400 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F401 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F402 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F403 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F404 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F405 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F406 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F407 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F408 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F409 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F41 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F410 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F411 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F412 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F413 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F414 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F415 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F416 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F417 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F418 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F419 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F42 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F420 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F421 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F422 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F423 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F424 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F425 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F426 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F427 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F428 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F429 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F43 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F430 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F431 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F432 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F433 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F434 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F435 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F436 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F437 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F438 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F439 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F44 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F440 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F441 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F442 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F443 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F444 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F445 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F446 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F447 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F448 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F449 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F45 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F450 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F451 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F452 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F453 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F454 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F455 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F456 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F457 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F458 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F459 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F46 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F460 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F461 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F462 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F463 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F464 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F465 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F466 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F467 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F468 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F469 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F47 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F470 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F471 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F472 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F473 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F474 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F475 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F476 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F477 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F478 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F479 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F48 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F480 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F481 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F482 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F483 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F484 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F485 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F486 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F487 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F488 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F489 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F49 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F490 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F491 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F492 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F493 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F494 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F495 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F496 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F497 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F498 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F499 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F5 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F50 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F500 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F501 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F502 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F503 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F504 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F505 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F506 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F507 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F508 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F51 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F52 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F53 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F54 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F55 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F56 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F57 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F58 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F59 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F6 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F60 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F61 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F62 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F63 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F64 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F65 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F66 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F67 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F68 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F69 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F7 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F70 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F71 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F72 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F73 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F74 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F75 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F76 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F77 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F78 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F79 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F8 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F80 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F81 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F82 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F83 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F84 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F85 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F86 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F87 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F88 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F89 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F9 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F90 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F91 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F92 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F93 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F94 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F95 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F96 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F97 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F98 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in F99 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in N10-1 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in N10-2 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in N10-3 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in N10-4 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in N10-5 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in N10-6 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in N1-1 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in N11-1 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in N11-2 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in N11-3 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in N11-4 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in N11-5 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in N11-6 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in N1-2 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in N12-1 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in N12-2 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in N12-3 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in N12-4 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in N12-5 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in N12-6 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in N1-3 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in N13-1 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in N13-2 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in N13-3 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in N13-4 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in N13-5 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in N13-6 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in N1-4 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in N14-1 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in N14-2 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in N14-3 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in N14-4 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in N14-5 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in N14-6 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in N1-5 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in N15-1 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in N15-2 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in N15-3 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in N15-4 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in N15-5 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in N15-6 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in N1-6 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in N16-1 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in N16-2 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in N16-3 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in N16-4 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in N16-5 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in N16-6 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in N17-1 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in N17-2 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in N17-3 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in N17-4 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in N17-5 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in N17-6 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in N18-1 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in N18-2 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in N18-3 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in N18-4 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in N18-5 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in N18-6 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in N19-1 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in N19-2 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in N19-3 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in N19-4 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in N19-5 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in N19-6 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in N2-1 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in N2-2 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in N2-3 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in N2-4 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in N2-5 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in N2-6 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in N3-1 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in N3-2 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in N3-3 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in N3-4 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in N3-5 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in N3-6 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in N4-1 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in N4-2 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in N4-3 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in N4-4 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in N4-5 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in N4-6 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in N5-1 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in N5-2 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in N5-3 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in N5-4 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in N5-5 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in N5-6 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in N6-1 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in N6-2 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in N6-3 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in N6-4 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in N6-5 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in N6-6 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in N7-1 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in N7-2 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in N7-3 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in N7-4 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in N7-5 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in N7-6 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in N8-1 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in N8-2 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in N8-3 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in N8-4 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in N8-5 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in N8-6 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in N9-1 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in N9-2 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in N9-3 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in N9-4 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in N9-5 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in N9-6 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in NU1 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in NU10 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in NU11 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in NU12 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in NU13 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in NU14 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in NU15 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in NU16 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in NU17 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in NU18 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in NU19 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in NU2 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in NU20 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in NU21 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in NU22 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in NU23 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in NU24 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in NU3 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in NU4 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in NU5 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in NU6 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in NU7 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in NU8 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in NU9 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in P10-1 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in P10-2 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in P10-3 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in P10-4 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in P10-5 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in P10-6 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in P1-1 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in P11-1 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in P11-2 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in P11-3 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in P11-4 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in P11-5 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in P11-6 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in P1-2 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in P12-1 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in P12-2 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in P12-3 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in P12-4 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in P12-5 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in P12-6 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in P1-3 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in P13-1 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in P13-2 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in P13-3 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in P13-4 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in P13-5 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in P13-6 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in P1-4 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in P14-1 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in P14-2 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in P14-3 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in P14-4 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in P14-5 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in P14-6 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in P1-5 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in P15-1 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in P15-2 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in P15-3 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in P15-4 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in P15-5 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in P15-6 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in P1-6 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in P16-1 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in P16-2 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in P16-3 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in P16-4 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in P16-5 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in P16-6 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in P17-1 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in P17-2 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in P17-3 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in P17-4 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in P17-5 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in P17-6 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in P18-1 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in P18-2 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in P18-3 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in P18-4 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in P18-5 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in P18-6 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in P19-1 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in P19-2 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in P19-3 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in P19-4 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in P19-5 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in P19-6 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in P2-1 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in P2-2 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in P2-3 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in P2-4 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in P2-5 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in P2-6 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in P3-1 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in P3-2 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in P3-3 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in P3-4 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in P3-5 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in P3-6 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in P4-1 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in P4-2 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in P4-3 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in P4-4 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in P4-5 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in P4-6 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in P5-1 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in P5-2 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in P5-3 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in P5-4 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in P5-5 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in P5-6 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in P6-1 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in P6-2 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in P6-3 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in P6-4 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in P6-5 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in P6-6 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in P7-1 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in P7-2 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in P7-3 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in P7-4 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in P7-5 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in P7-6 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in P8-1 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in P8-2 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in P8-3 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in P8-4 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in P8-5 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in P8-6 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in P9-1 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in P9-2 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in P9-3 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in P9-4 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in P9-5 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in P9-6 ###\n",
      "\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "\n",
    "\n",
    "for a in $(cut -f 1 ../1-trimming/Querymap.txt)\n",
    "do\n",
    "    if [ -s ../1-trimming/$a/$a\\_trimmed.fasta ]\n",
    "    then\n",
    "        echo -e \"\\n### Detecting chimeras in $a ###\\n\"\n",
    "        mkdir $a\n",
    "        cd $a\n",
    "        vsearch --uchime_ref ../../1-trimming/$a/$a\\_trimmed.fasta --db ../refs.fasta \\\n",
    "        --nonchimeras $a-nonchimeras.fasta --chimeras $a-chimeras.fasta &> log \n",
    "        cd ..\n",
    "\n",
    "    else\n",
    "        echo -e \"$a is empty\"\n",
    "    fi\n",
    "done"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 2",
   "language": "python",
   "name": "python2"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 2
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython2",
   "version": "2.7.6"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 0
}
