{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Prepare input files.\n",
    "\n",
    "The ailgnment has been produced using Reprophylo (see notebook). We'll just cleanup the sequence headers and create a local copy."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "!cat ../fetch_clean_align_tree/trimmed_mammal_align_ALL.phy | sed 's/_f[0-9] / /' > alignment.phy"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "The tax file has been generated from a Genbank file using this notebook."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "EF027265.1\tEukaryota;Chordata;Mammalia;Rodentia;Cricetidae;Microtus;Microtus arvalis\r\n",
      "KP702285.1\tEukaryota;Chordata;Mammalia;unknown;Bovidae;Ovis;Ovis aries\r\n",
      "AY879429.1\tEukaryota;Chordata;Mammalia;unknown;Bovidae;Ovis;Ovis aries\r\n",
      "AY879349.1\tEukaryota;Chordata;Mammalia;unknown;Bovidae;Ovis;Ovis aries\r\n",
      "EF056451.1\tEukaryota;Chordata;Mammalia;unknown;Bovidae;Ovis;Ovis aries\r\n",
      "EF056393.1\tEukaryota;Chordata;Mammalia;unknown;Bovidae;Ovis;Ovis aries\r\n",
      "EF056448.1\tEukaryota;Chordata;Mammalia;unknown;Bovidae;Ovis;Ovis aries\r\n",
      "EF056453.1\tEukaryota;Chordata;Mammalia;unknown;Bovidae;Ovis;Ovis aries\r\n",
      "EF056455.1\tEukaryota;Chordata;Mammalia;unknown;Bovidae;Ovis;Ovis aries\r\n",
      "AJ000846.1\tEukaryota;Chordata;Mammalia;unknown;Bovidae;Ovis;Ovis aries\r\n"
     ]
    }
   ],
   "source": [
    "!head tax_for_SATIVA.tax"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Let's now run SATIVA."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {
    "collapsed": false,
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "SATIVA v0.9-55-g0cbb090, released on 2016-06-28. Last version: https://github.com/amkozlov/sativa \n",
      "By A.Kozlov and J.Zhang, the Exelixis Lab. Based on RAxML 8.2.3 by A.Stamatakis.\n",
      "\n",
      "SATIVA was called as follows:\n",
      "\n",
      "./sativa/sativa.py -s alignment.phy -t tax_for_SATIVA.tax -x zoo -n 12S -o ./ -T 5 -v\n",
      "\n",
      "Mislabels search is running with the following parameters:\n",
      " Alignment:                        alignment.phy\n",
      " Taxonomy:                         tax_for_SATIVA.tax\n",
      " Output directory:                 /home/working/Reference_Alignment/Mammals/SATIVA\n",
      " Job name / output files prefix:   12S\n",
      " Model of rate heterogeneity:      AUTO\n",
      " Confidence cut-off:               0.000000\n",
      " Number of threads:                5\n",
      "\n",
      "*** STEP 1: Building the reference tree using provided alignment and taxonomic annotations ***\n",
      "\n",
      "=> Loading taxonomy from file: tax_for_SATIVA.tax ...\n",
      "\n",
      "==> Loading reference alignment from file: alignment.phy ...\n",
      "\n",
      "Guessing input format: not fasta\n",
      "Guessing input format: not phylip_relaxed\n",
      "===> Validating taxonomy and alignment ...\n",
      "\n",
      "NOTE: Following sequences are identical: AY337520.1, AY337538.1, AY337540.1\n",
      "NOTE: Following sequences are identical: EF056406.1, EF056401.1\n",
      "NOTE: Following sequences are identical: DQ191149.1, DQ153244.1, DQ191145.1\n",
      "NOTE: Following sequences are identical: KC776716.1, KC776696.1\n",
      "NOTE: Following sequences are identical: KT757763.1, KT757751.1, KT757753.1, JQ340102.1, KT368731.1, KT757740.1, KT985984.1, JQ340127.1, JQ340108.1, KT757757.1, KT998647.1, HQ439467.1, HQ439445.1, HQ439444.1, AP013089.1, JQ340093.1, KT757744.1, KT757748.1, HQ439485.1, JQ340145.1, AP013078.1, JQ340098.1, HQ439493.1, KT757750.1, HQ439446.1, AP013091.1, HQ439487.1, JQ340125.1, HQ439449.1, JQ340113.1, FJ718998.1, KT757739.1, AY584828.1, AP013084.1, KC203000.1, JQ340092.1, HQ439453.1, AP013087.1, HQ439460.1, AP013090.1, HQ439458.1, JQ340154.1, KT757743.1, JQ340099.1, HQ439462.1, AP013088.1, JQ340151.1\n",
      "NOTE: Following sequences are identical: HQ832482.1, AY184433.1, GU457433.1, HQ191428.1, DQ191150.1\n",
      "NOTE: Following sequences are identical: EF373884.1, EF373883.1\n",
      "NOTE: Following sequences are identical: AJ972680.1, KJ681480.1, KJ681481.1\n",
      "NOTE: Following sequences are identical: M10217.1, Y10943.1\n",
      "NOTE: Following sequences are identical: KU290497.1, KU290618.1\n",
      "NOTE: Following sequences are identical: JX088676.1, JX088680.1\n",
      "NOTE: Following sequences are identical: AY962029.1, AY962023.1\n",
      "NOTE: Following sequences are identical: AF527537.1, M35877.1, EF035447.1\n",
      "NOTE: Following sequences are identical: FR691688.1, AJ389542.1\n",
      "NOTE: Following sequences are identical: JF342844.1, KU291066.1, FJ817364.1, JF342890.1, KJ637115.1, JF342861.1, EU408260.1, EU789769.1, AY656739.1\n",
      "NOTE: Following sequences are identical: KT368739.1, AB859014.1, AP013080.1, HQ439443.1, KT757761.1, KT757764.1, KC202960.1, EF597512.1\n",
      "NOTE: Following sequences are identical: HM236183.1, HM236182.1, HM236178.1\n",
      "NOTE: Following sequences are identical: KM892823.1, KM892833.1, KM892832.1\n",
      "NOTE: Following sequences are identical: KM520138.1, KJ737423.1, KJ737418.1, AY243484.1, KJ746664.1, KC493609.1, JN601070.1, JN601075.1, KC493610.1, KM998967.1, KJ746663.1\n",
      "NOTE: Following sequences are identical: EF362717.1, EF362708.1\n",
      "NOTE: Following sequences are identical: KM892817.1, KM892812.1, KM892809.1\n",
      "NOTE: Following sequences are identical: AY879419.1, AY879380.1, JN415745.1, EF056418.1, GU350285.1, EF056425.1, DQ852152.1, EF056449.1, JN415734.1, JQ286189.1, EF056459.1, JN415726.1, JN415731.1, JN415737.1, DQ852163.1, GU350339.1, AY879347.1, GU350392.1, AY879457.1, JQ286225.1, JQ286239.1\n",
      "NOTE: Following sequences are identical: EF122793.1, JN860636.1\n",
      "NOTE: Following sequences are identical: KP223728.1, KM520132.1, KM520142.1, KM520133.1, DQ268530.1, KM520143.1, AF034253.1, KM520135.1, AF486858.1, FJ236992.1, GQ338938.1, AY574046.1, EU333163.1, AY243485.1, AP003428.1, JN601072.1, AF304203.1, KP294522.1, KT965278.1, DQ334861.1, AF486856.1, EF545586.1, DQ207755.1, DQ444705.1, EF545571.1, EF545568.1, EF545588.1, EF545567.1, DQ334860.1, DQ207753.1, KJ746665.1, AY243483.1, KP681245.1, AF304202.1, KM044240.1, AF486870.1, KC493612.1, DQ207754.1, KP681242.1, KP681243.1, AY574047.1, KM520134.1, KM520140.1, KM520137.1, KM520141.1, KM520136.1, KM520144.1, KM520145.1\n",
      "NOTE: Following sequences are identical: KP942534.1, KP942529.1, KP942563.1, KP942570.1, KP942548.1, KP942530.1, GU174596.1, KP942543.1, KP942537.1, KP942531.1\n",
      "NOTE: Following sequences are identical: EU408291.1, KM061533.1\n",
      "NOTE: Following sequences are identical: KF163077.1, KF163062.1, AY676863.1, JN817339.1, KJ709685.1, AB074962.1, AF492351.1, AY676871.1, AY676858.1, JN817317.1, JN817309.1\n",
      "NOTE: Following sequences are identical: KM016851.1, KR180298.1, GU187153.1, HQ405755.1, GU187154.1\n",
      "NOTE: Following sequences are identical: KC747665.1, KP273591.1\n",
      "NOTE: Following sequences are identical: KM892826.1, KM892827.1, KM892821.1\n",
      "NOTE: Following sequences are identical: KM488625.1, HM106322.1\n",
      "NOTE: Following sequences are identical: KC993007.1, KC993004.1\n",
      "NOTE: Following sequences are identical: KT261430.1, KT372134.1\n",
      "NOTE: Following sequences are identical: X84384.1, AY012112.1\n",
      "NOTE: Following sequences are identical: KC776706.1, KC776705.1, KC776709.1\n",
      "NOTE: Following sequences are identical: KC572709.1, X61145.1\n",
      "NOTE: Following sequences are identical: AJ885205.1, AJ885206.1\n",
      "NOTE: Following sequences are identical: HQ529663.1, HQ529670.1\n",
      "NOTE: Following sequences are identical: HM026002.1, HM026052.1, HM026045.1, HM025990.1, HM026009.1, HM026024.1, HM026055.1\n",
      "NOTE: Following sequences are identical: AJ554054.1, AP006468.1\n",
      "NOTE: Following sequences are identical: AF520669.1, AY012123.1\n",
      "NOTE: Following sequences are identical: EF362643.1, EF362639.1\n",
      "NOTE: Following sequences are identical: KP981378.1, KP702285.1, AF010406.1, KF977847.1\n",
      "NOTE: Following sequences are identical: JN695757.1, KP234022.1\n",
      "NOTE: Following sequences are identical: KP172593.1, AY184430.1, AY184431.1, DQ191148.1, DQ191146.1\n",
      "NOTE: Following sequences are identical: HM026012.1, HM026014.1\n",
      "NOTE: Following sequences are identical: KM091450.1, AB119066.1\n",
      "NOTE: Following sequences are identical: FR691686.1, FR691685.1\n",
      "NOTE: Following sequences are identical: KM892825.1, KM892840.1, KM892838.1, KM892814.1, KM892837.1, KM892820.1\n",
      "NOTE: Following sequences are identical: HQ529681.1, HQ529674.1, HQ529680.1, HQ529673.1\n",
      "NOTE: Following sequences are identical: KT757745.1, AP013098.1, HQ439442.1, KT757762.1, KT757760.1, AP013092.1, HQ439455.1, JQ340122.1\n",
      "NOTE: Following sequences are identical: AY337528.1, AY337525.1, AY337522.1, AY337531.1, AY337524.1, AY337533.1, AY337523.1, AY337542.1, AY337541.1\n",
      "NOTE: Following sequences are identical: AB058610.1, KJ397605.1, KR013248.1\n",
      "NOTE: Following sequences are identical: KU891850.1, KT033901.1\n",
      "NOTE: Following sequences are identical: EF056433.1, EF056436.1\n",
      "NOTE: Following sequences are identical: AY337536.1, AY337543.1, AY337527.1, AY337545.1\n",
      "NOTE: Following sequences are identical: DQ191151.1, JN389443.1\n",
      "NOTE: Following sequences are identical: AB058611.1, AY745154.1, AY745152.1\n",
      "NOTE: Following sequences are identical: KT316285.1, KT316284.1, KT316286.1, KT316287.1\n",
      "NOTE: Following sequences are identical: AJ000849.1, AJ000850.1, AJ000848.1\n",
      "NOTE: Following sequences are identical: JN632649.1, EU315254.1, M35876.1, JX254914.2, JF802125.1\n",
      "NOTE: Following sequences are identical: AY184438.1, KM506758.1\n",
      "NOTE: Following sequences are identical: AM181027.1, GU174598.1\n",
      "NOTE: Following sequences are identical: JQ621986.1, KF938318.1, HE577850.1, KP998471.1, JQ621985.1, KF938319.1, AY670663.1, HE577847.1, KF938326.1\n",
      "NOTE: Following sequences are identical: AJ238588.1, KC993000.1\n",
      "NOTE: Following sequences are identical: KM520148.1, EF545585.1\n",
      "NOTE: Following sequences are identical: HQ529638.1, HQ529647.1\n",
      "NOTE: Following sequences are identical: AP003321.1, FJ610338.1, FJ610339.1, AY235570.1, GU261698.1, GU261674.1, AP003317.1, GU261680.1, GU261678.1, GU261693.1, AB086102.1, KT626848.1, KP211419.1, GU261694.1, AP003318.1\n",
      "NOTE: Following sequences are identical: U13083.1, U13100.1\n",
      "NOTE: Following sequences are identical: AP006467.1, AY012141.1\n",
      "NOTE: Following sequences are identical: EF056393.1, EF056402.1\n",
      "NOTE: Following sequences are identical: HM236177.1, HM236174.1\n",
      "NOTE: Following sequences are identical: KF317941.1, KF317935.1\n",
      "NOTE: Following sequences are identical: HM236176.1, HM236181.1\n",
      "NOTE: Following sequences are identical: EF027292.1, EF027290.1\n",
      "NOTE: Following sequences are identical: AF263219.1, AF203774.1\n",
      "NOTE: Following sequences are identical: KF038159.1, HQ439450.1, JQ340141.1, HQ439448.1, JQ340104.1, EF597514.1, EU939445.3\n",
      "NOTE: Following sequences are identical: EU789655.1, EU789654.1, KJ139388.1\n",
      "NOTE: Following sequences are identical: KF011917.1, DQ673916.1, KM577634.1\n",
      "NOTE: Following sequences are identical: KF781660.1, KF781662.1\n",
      "NOTE: Following sequences are identical: KF570389.1, KF570319.1, KU991990.1, EU557093.1, KF570322.1, KT601198.1, KF570371.1, KF570315.1\n",
      "NOTE: Following sequences are identical: DQ874614.2, AY057793.1\n",
      "NOTE: Following sequences are identical: KR180364.1, GU187173.1, KR180300.1, GU187155.1, GU187156.1\n",
      "NOTE: Following sequences are identical: AY184440.1, KJ681489.1\n",
      "NOTE: Following sequences are identical: AY121990.1, KJ681482.1\n",
      "NOTE: Following sequences are identical: AM181036.1, GU174593.1\n",
      "NOTE: Following sequences are identical: AY027823.1, AY027824.1\n",
      "NOTE: Following sequences are identical: HQ996541.1, HQ996539.1, HQ996553.1, HQ996533.1\n",
      "NOTE: Following sequences are identical: KF918859.1, KM892810.1\n",
      "NOTE: Following sequences are identical: HM106318.1, AB119069.1\n",
      "NOTE: Following sequences are identical: KM092492.1, GU981053.1\n",
      "NOTE: Following sequences are identical: KF570379.1, KF570380.1, AY012142.1\n",
      "NOTE: Following sequences are identical: KP342452.1, KT448287.1, AM181037.1\n",
      "NOTE: Following sequences are identical: EF056432.1, EF056443.1\n",
      "NOTE: Following sequences are identical: GU174597.1, AM181028.1\n",
      "NOTE: Following sequences are identical: KT693382.1, Y08515.1\n",
      "NOTE: Following sequences are identical: EU789764.1, AY012152.1, DQ480489.1\n",
      "NOTE: Following sequences are identical: FJ688100.1, FJ688101.1\n",
      "NOTE: Following sequences are identical: AB245426.1, AY121989.1\n",
      "NOTE: Following sequences are identical: JN632610.1, KJ681485.1, KJ681486.1, KJ681487.1\n",
      "NOTE: Following sequences are identical: KJ192209.1, KJ192210.1, KR059188.1, KR059213.1, KR059159.1, GU229281.1, KR059197.1, GU229280.1, M55541.1, KR059154.1, KM244714.1, KJ789954.1, KR349363.1, KP231536.1\n",
      "NOTE: Following sequences are identical: KX512321.1, KM886936.1\n",
      "NOTE: Following sequences are identical: AM181032.1, X63726.1\n",
      "NOTE: Following sequences are identical: JF342816.1, KF661083.1, EU408288.1, KJ139384.1, DQ480492.1, KU290662.1, KU290520.1, KU290785.1, KU290461.1, KU290793.1, KU290401.1, KU290464.1, EU789673.1, EU789640.1, EU789662.1, AB048590.1, FJ817358.1, DQ480499.1, KU290597.1, JF342832.1, KU290812.1, KU290423.1, KU290428.1, KU290811.1, KU290822.1, KU290815.1, KJ637099.1, JF342836.1, KU291057.1, KU290814.1, EU408300.1, EU789688.1, KU290402.1, KU290404.1, KU290731.1, EU789717.1, KU290642.1, KU290622.1, KU290609.1, AY656749.1, AY656737.1, KU290453.1, EU789744.1, KM061586.1, KU291054.1, EU408262.1, KU290907.1, EU789741.1, KU290668.1, JX088685.1, JX088672.1, EU789663.1, KU290857.1, FJ817359.1, KU290971.1, KU290850.1, KU290620.1, EU789710.1, JF342891.1, EU789686.1, JF342893.1, JF342876.1, EU789759.1, KU290557.1, KU290905.1, KU290462.1, KU290939.1, KF661086.1, KU290550.1, AB499816.1\n",
      "NOTE: Following sequences are identical: EF056448.1, EF056419.1\n",
      "NOTE: Following sequences are identical: DQ673917.1, KM820833.1, DQ673908.1\n",
      "NOTE: Following sequences are identical: KU290680.1, KU290717.1, KU290831.1, KU290711.1\n",
      "NOTE: Following sequences are identical: AB211429.1, AB210267.2\n",
      "NOTE: Following sequences are identical: AY962031.1, AY962030.1\n",
      "NOTE: Following sequences are identical: AP013079.1, KC202962.1\n",
      "NOTE: Following sequences are identical: KM009113.1, KM009112.1, AY172581.1, KM820836.1\n",
      "NOTE: Following sequences are identical: AY245607.1, JN003394.1\n",
      "NOTE: Following sequences are identical: KP202275.1, KP202273.1, Y08503.1\n",
      "NOTE: Following sequences are identical: HM623877.1, HM623881.1\n",
      "NOTE: Following sequences are identical: AJ001562.1, X84385.1\n",
      "NOTE: Following sequences are identical: KC572849.1, KC572792.1, KC572832.1, KC572745.1, KC572846.1, KC572715.1, KC572735.1, KC572789.1, KC572710.1, KC572712.1, KC572759.1, KC572807.1, KC572747.1, KC572756.1, KC572720.1, KC572817.1, KC572785.1, KC572726.1, KC572718.1, KC572711.1\n",
      "NOTE: Following sequences are identical: FM165432.1, AJ630114.1\n",
      "NOTE: Following sequences are identical: KJ789952.1, EF545592.1\n",
      "NOTE: Following sequences are identical: AY027831.1, AY027827.1\n",
      "NOTE: Following sequences are identical: JN415740.1, JN415741.1\n",
      "NOTE: Following sequences are identical: GU174594.1, KU963302.1, NC_001602.1\n",
      "NOTE: Following sequences are identical: KP205314.1, KP205310.1, KP205309.1, KP205285.1, KP205318.1, KP205313.1, KP205316.1\n",
      "NOTE: Following sequences are identical: AJ311131.1, AJ311126.1\n",
      "NOTE: Following sequences are identical: LN997430.1, KC776700.1, KC776697.1, KC776708.1, KC776712.1, KC776710.1, KC776713.1\n",
      "NOTE: Following sequences are identical: AF091699.1, JQ622004.1, KF938320.1, KF302448.1, HE577848.1, AY858379.1, KF938350.1, KF938325.1, KP998470.1, KF938321.1\n",
      "NOTE: Following sequences are identical: KR180348.1, KR180337.1, GU187175.1, GU187187.1, KF418378.1, HQ405754.1, GU187176.1, HQ405752.3, KR180336.1, GU187216.1, KF164610.1, KF418375.1, KF418377.1\n",
      "NOTE: Following sequences are identical: JQ287752.1, JQ287756.1, JQ287754.1\n",
      "NOTE: Following sequences are identical: AY962041.1, AY962024.1, AY962036.1\n",
      "\n",
      "NOTE: Found 510 sequence duplicates\n",
      "====> Building a multifurcating tree from taxonomy with 1080 seqs ...\n",
      "\n",
      "Processed nodes:  1000 , added:  999 , skipped:  1\n",
      "Total nodes in resulting tree: 1080\n",
      "=====> Building the reference alignment ...\n",
      "\n",
      "======> Saving the outgroup for later re-rooting ...\n",
      "\n",
      "Outgroup for rooting was saved to: /home/working/Reference_Alignment/Mammals/SATIVA/sativa/tmp/12S_1485190715106/12S_outgr.tre, outgroup size: 4\n",
      "=======> Resolving multifurcation: choosing the best topology from 1 independent RAxML runs ...\n",
      "\n",
      "\n",
      "Reducing the alignment: \n",
      "\n",
      "/home/working/Reference_Alignment/Mammals/SATIVA/sativa/raxml//run_raxml.sh -w /home/working/Reference_Alignment/Mammals/SATIVA/sativa/tmp/12S_1485190715106 -T 5 -f c -s /home/working/Reference_Alignment/Mammals/SATIVA/sativa/tmp/12S_1485190715106/12S_matrix.afa --no-dup-check -m GTRCAT -n reduce --verbose -p 13332\n",
      "\n",
      "\n",
      "Constrained ML inference: \n",
      "\n",
      "/home/working/Reference_Alignment/Mammals/SATIVA/sativa/raxml//run_raxml.sh -w /home/working/Reference_Alignment/Mammals/SATIVA/sativa/tmp/12S_1485190715106 -T 5 -s /home/working/Reference_Alignment/Mammals/SATIVA/sativa/tmp/12S_1485190715106/12S_matrix.afa -g /home/working/Reference_Alignment/Mammals/SATIVA/sativa/tmp/12S_1485190715106/12S_mfu.tre --no-seq-check -N 1 -m GTRCAT -n mfresolv_12S --verbose -p 326\n",
      "\n",
      "\n",
      "CAT-based logLH of the reference tree: -10929.585844\n",
      "\n",
      "========> Calling RAxML-EPA to obtain branch labels ...\n",
      "\n",
      "/home/working/Reference_Alignment/Mammals/SATIVA/sativa/raxml//run_raxml.sh -w /home/working/Reference_Alignment/Mammals/SATIVA/sativa/tmp/12S_1485190715106 -T 5 -s /home/working/Reference_Alignment/Mammals/SATIVA/sativa/tmp/12S_1485190715106/12S_lblq.fa -t /home/working/Reference_Alignment/Mammals/SATIVA/sativa/tmp/12S_1485190715106/12S_bfu.tre --no-seq-check -f y --epa-accumulated-threshold 0.999 -G 0.462962962963 -R /home/working/Reference_Alignment/Mammals/SATIVA/sativa/tmp/12S_1485190715106/12S.opt -m GTRCAT -n epalbl_12S --verbose -p 26407\n",
      "\n",
      "=========> Post-processing the EPA tree (re-rooting, taxonomic labeling etc.) ...\n",
      "\n",
      "\n",
      "==========> Checking branch labels ...\n",
      "shared rank names before training: set(['Eukaryota', 'Chordata'])\n",
      "shared rank names after  training: set(['Eukaryota', 'unknown', 'Chordata'])\n",
      "\n",
      "==========> Saving the reference JSON file: ./12S.refjson\n",
      "\n",
      "Calculating the speciation rate...\n",
      "\n",
      "Writing down the reference file...\n",
      "\n",
      "*** STEP 2: Searching for mislabels ***\n",
      "\n",
      "Running the leave-one-sequence-out test...\n",
      "\n",
      "/home/working/Reference_Alignment/Mammals/SATIVA/sativa/raxml//run_raxml.sh -w /home/working/Reference_Alignment/Mammals/SATIVA/sativa/tmp/12S_1485190715106 -T 5 -s /home/working/Reference_Alignment/Mammals/SATIVA/sativa/tmp/12S_1485190715106/12S.refaln -t /home/working/Reference_Alignment/Mammals/SATIVA/sativa/tmp/12S_1485190715106/ref_12S.tre --no-seq-check -f O --epa-accumulated-threshold 0.999 -G 0.462962962963 -R /home/working/Reference_Alignment/Mammals/SATIVA/sativa/tmp/12S_1485190715106/12S.opt -m GTRCAT -n l1out_seq_12S --verbose -p 13332\n",
      "\n",
      "Leave-one-out test identified 35 suspicious sequences; running final EPA test to check them...\n",
      "\n",
      "/home/working/Reference_Alignment/Mammals/SATIVA/sativa/raxml//run_raxml.sh -w /home/working/Reference_Alignment/Mammals/SATIVA/sativa/tmp/12S_1485190715106 -T 5 -s /home/working/Reference_Alignment/Mammals/SATIVA/sativa/tmp/12S_1485190715106/12S.refaln -t /home/working/Reference_Alignment/Mammals/SATIVA/sativa/tmp/12S_1485190715106/final_ref_12S.tre --no-seq-check -f v --epa-accumulated-threshold 0.999 -G 0.462962962963 -m GTRCAT -n final_epa_12S --verbose -p 326\n",
      "\n",
      "HM637771.1\tClass\tAves\tMammalia\t0.995\tEukaryota;Chordata;Aves;Galliformes;Phasianidae;Gallus;Gallus gallus\tEukaryota;Chordata;Mammalia;Perissodactyla;Equidae;Equus;Equus caballus\t0.999;0.999;0.995;0.184;0.184;0.184;0.184\n",
      "\n",
      "KF471518.1\tClass\tAves\tMammalia\t0.991\tEukaryota;Chordata;Aves;Galliformes;Phasianidae;Gallus;Gallus gallus\tEukaryota;Chordata;Mammalia;Carnivora;Canidae;Canis;Canis lupus\t0.999;0.999;0.991;0.418;0.181;0.176;0.176\n",
      "\n",
      "KF471523.1\tClass\tAves\tMammalia\t0.951\tEukaryota;Chordata;Aves;Galliformes;Phasianidae;Gallus;Gallus gallus\tEukaryota;Chordata;Mammalia;unknown;Bovidae;Ovis;Ovis aries\t1.000;1.000;0.951;0.367;0.216;0.175;0.175\n",
      "\n",
      "KF484729.1\tClass\tAves\tMammalia\t0.946\tEukaryota;Chordata;Aves;Galliformes;Phasianidae;Gallus;Gallus gallus\tEukaryota;Chordata;Mammalia;Carnivora;Mustelidae;Martes;Martes martes\t1.000;1.000;0.946;0.425;0.362;0.100;0.100\n",
      "\n",
      "M13046.1\tClass\tAmphibia\tMammalia\t0.776\tEukaryota;Chordata;Amphibia;Anura;Pipidae;Xenopus;Xenopus laevis\tEukaryota;Chordata;Mammalia;Carnivora;Canidae;Canis;Canis lupus\t0.999;0.999;0.776;0.351;0.260;0.246;0.246\n",
      "\n",
      "JQ287755.1\tOrder\tRodentia\tInsectivora\t0.898\tEukaryota;Chordata;Mammalia;Rodentia;Muridae;Rattus;Rattus rattus\tEukaryota;Chordata;Mammalia;Insectivora;Talpidae;Talpa;Talpa europaea\t0.999;0.999;0.999;0.898;0.871;0.871;0.871\n",
      "\n",
      "JQ287761.1\tOrder\tRodentia\tunknown\t0.319\tEukaryota;Chordata;Mammalia;Rodentia;Muridae;Mus;Mus musculus\tEukaryota;Chordata;Mammalia;unknown;Bovidae;Ovis;Ovis aries\t0.999;0.999;0.999;0.319;0.307;0.196;0.196\n",
      "\n",
      "AF320512.1\tOrder\tCetacea\tCarnivora\t0.275\tEukaryota;Chordata;Mammalia;Cetacea;Physeteridae;Physeter;Physeter catodon\tEukaryota;Chordata;Mammalia;Carnivora;Canidae;Canis;Canis lupus\t0.999;0.999;0.951;0.275;0.258;0.250;0.250\n",
      "\n",
      "FM165432.1\tOrder\tPerissodactyla\tCarnivora\t0.254\tEukaryota;Chordata;Mammalia;Perissodactyla;Equidae;Equus;Equus caballus\tEukaryota;Chordata;Mammalia;Carnivora;Canidae;Canis;Canis lupus\t1.000;1.000;0.971;0.254;0.183;0.180;0.180\n",
      "\n",
      "AY830134.1\tOrder\tunknown\tCarnivora\t0.254\tEukaryota;Chordata;Mammalia;unknown;Bovidae;Bos;Bos taurus\tEukaryota;Chordata;Mammalia;Carnivora;Canidae;Canis;Canis lupus\t1.000;1.000;0.971;0.254;0.182;0.179;0.179\n",
      "\n",
      "AJ630114.1\tOrder\tunknown\tCarnivora\t0.254\tEukaryota;Chordata;Mammalia;unknown;Bovidae;Bos;Bos taurus\tEukaryota;Chordata;Mammalia;Carnivora;Canidae;Canis;Canis lupus\t1.000;1.000;0.971;0.254;0.183;0.180;0.180\n",
      "\n",
      "AJ554063.1\tFamily\tPhocoenidae\tDelphinidae\t0.901\tEukaryota;Chordata;Mammalia;Cetacea;Phocoenidae;Phocoena;Phocoena phocoena\tEukaryota;Chordata;Mammalia;Cetacea;Delphinidae\t0.999;0.999;0.999;0.999;0.901\n",
      "\n",
      "JQ287757.1\tFamily\tMuridae\tCricetidae\t0.831\tEukaryota;Chordata;Mammalia;Rodentia;Muridae;Rattus;Rattus rattus\tEukaryota;Chordata;Mammalia;Rodentia;Cricetidae;Myodes;Myodes glareolus\t0.999;0.999;0.999;0.977;0.831;0.773;0.773\n",
      "\n",
      "HM623883.1\tFamily\tBovidae\tCervidae\t0.505\tEukaryota;Chordata;Mammalia;unknown;Bovidae;Capra;Capra hircus\tEukaryota;Chordata;Mammalia;unknown;Cervidae;Capreolus;Capreolus capreolus\t0.999;0.999;0.999;0.998;0.505;0.456;0.456\n",
      "\n",
      "AB018586.1\tGenus\tStenella\tTursiops\t0.980\tEukaryota;Chordata;Mammalia;Cetacea;Delphinidae;Stenella;Stenella coeruleoalba\tEukaryota;Chordata;Mammalia;Cetacea;Delphinidae;Tursiops;Tursiops truncatus\t0.999;0.999;0.999;0.999;0.999;0.980;0.980\n",
      "\n",
      "EU685091.1\tGenus\tDelphinus\tTursiops\t0.980\tEukaryota;Chordata;Mammalia;Cetacea;Delphinidae;Delphinus;Delphinus delphis\tEukaryota;Chordata;Mammalia;Cetacea;Delphinidae;Tursiops;Tursiops truncatus\t0.999;0.999;0.999;0.999;0.999;0.980;0.980\n",
      "\n",
      "U13083.1\tGenus\tDelphinus\tTursiops\t0.979\tEukaryota;Chordata;Mammalia;Cetacea;Delphinidae;Delphinus;Delphinus delphis\tEukaryota;Chordata;Mammalia;Cetacea;Delphinidae;Tursiops;Tursiops truncatus\t0.999;0.999;0.999;0.999;0.999;0.979;0.979\n",
      "\n",
      "Y08208.1\tGenus\tHydropotes\tRangifer\t0.961\tEukaryota;Chordata;Mammalia;unknown;Cervidae;Hydropotes;Hydropotes inermis\tEukaryota;Chordata;Mammalia;unknown;Cervidae;Rangifer;Rangifer tarandus\t1.000;1.000;1.000;1.000;1.000;0.961;0.961\n",
      "\n",
      "HM623877.1\tGenus\tCapra\tOvis\t0.632\tEukaryota;Chordata;Mammalia;unknown;Bovidae;Capra;Capra hircus\tEukaryota;Chordata;Mammalia;unknown;Bovidae;Ovis;Ovis aries\t0.999;0.999;0.999;0.999;0.999;0.632;0.632\n",
      "\n",
      "DQ867007.1\tGenus\tCapra\tOvis\t0.632\tEukaryota;Chordata;Mammalia;unknown;Bovidae;Capra;Capra hircus\tEukaryota;Chordata;Mammalia;unknown;Bovidae;Ovis;Ovis aries\t0.999;0.999;0.999;0.999;0.999;0.632;0.632\n",
      "\n",
      "JQ287756.1\tGenus\tRattus\tMus\t0.553\tEukaryota;Chordata;Mammalia;Rodentia;Muridae;Rattus;Rattus rattus\tEukaryota;Chordata;Mammalia;Rodentia;Muridae;Mus;Mus musculus\t0.999;0.999;0.999;0.999;0.864;0.553;0.553\n",
      "\n",
      "JQ287754.1\tGenus\tRattus\tMus\t0.553\tEukaryota;Chordata;Mammalia;Rodentia;Muridae;Rattus;Rattus rattus\tEukaryota;Chordata;Mammalia;Rodentia;Muridae;Mus;Mus musculus\t0.999;0.999;0.999;0.999;0.864;0.553;0.553\n",
      "\n",
      "JQ287752.1\tGenus\tRattus\tMus\t0.553\tEukaryota;Chordata;Mammalia;Rodentia;Muridae;Rattus;Rattus rattus\tEukaryota;Chordata;Mammalia;Rodentia;Muridae;Mus;Mus musculus\t0.999;0.999;0.999;0.999;0.864;0.553;0.553\n",
      "\n",
      "KF111725.1\tGenus\tEptesicus\tVespertilio\t0.510\tEukaryota;Chordata;Mammalia;Chiroptera;Vespertilionidae;Eptesicus;Eptesicus serotinus\tEukaryota;Chordata;Mammalia;Chiroptera;Vespertilionidae;Vespertilio;Vespertilio murinus\t1.000;1.000;1.000;1.000;1.000;0.510;0.510\n",
      "\n",
      "D34627.1\tGenus\tCervus\tCapreolus\t0.318\tEukaryota;Chordata;Mammalia;unknown;Cervidae;Cervus;Cervus nippon\tEukaryota;Chordata;Mammalia;unknown;Cervidae;Capreolus;Capreolus capreolus\t0.999;0.999;0.999;0.999;0.993;0.318;0.318\n",
      "\n",
      "JF339973.1\tGenus\tGlobicephala\tTursiops\t0.268\tEukaryota;Chordata;Mammalia;Cetacea;Delphinidae;Globicephala;Globicephala melas\tEukaryota;Chordata;Mammalia;Cetacea;Delphinidae;Tursiops;Tursiops truncatus\t0.999;0.999;0.999;0.664;0.605;0.268;0.268\n",
      "\n",
      "NC_001700.1\tSpecies\tFelis catus\tFelis silvestris\t0.930\tEukaryota;Chordata;Mammalia;Carnivora;Felidae;Felis;Felis catus\tEukaryota;Chordata;Mammalia;Carnivora;Felidae;Felis;Felis silvestris\t1.000;1.000;1.000;1.000;0.930;0.930;0.930\n",
      "\n",
      "Y08503.1\tSpecies\tFelis catus\tFelis silvestris\t0.930\tEukaryota;Chordata;Mammalia;Carnivora;Felidae;Felis;Felis catus\tEukaryota;Chordata;Mammalia;Carnivora;Felidae;Felis;Felis silvestris\t1.000;1.000;1.000;1.000;0.930;0.930;0.930\n",
      "\n",
      "HQ529667.1\tSpecies\tMyotis brandtii\tMyotis lucifugus\t0.611\tEukaryota;Chordata;Mammalia;Chiroptera;Vespertilionidae;Myotis;Myotis brandtii\tEukaryota;Chordata;Mammalia;Chiroptera;Vespertilionidae;Myotis;Myotis lucifugus\t0.999;0.999;0.999;0.999;0.997;0.987;0.611\n",
      "\n",
      "HQ529670.1\tSpecies\tMyotis brandtii\tMyotis lucifugus\t0.611\tEukaryota;Chordata;Mammalia;Chiroptera;Vespertilionidae;Myotis;Myotis brandtii\tEukaryota;Chordata;Mammalia;Chiroptera;Vespertilionidae;Myotis;Myotis lucifugus\t0.999;0.999;0.999;0.999;0.997;0.987;0.611\n",
      "\n",
      "HQ529663.1\tSpecies\tMyotis brandtii\tMyotis lucifugus\t0.611\tEukaryota;Chordata;Mammalia;Chiroptera;Vespertilionidae;Myotis;Myotis brandtii\tEukaryota;Chordata;Mammalia;Chiroptera;Vespertilionidae;Myotis;Myotis lucifugus\t0.999;0.999;0.999;0.999;0.997;0.987;0.611\n",
      "\n",
      "KM224283.1\tSpecies\tFelis catus\tFelis silvestris\t0.594\tEukaryota;Chordata;Mammalia;Carnivora;Felidae;Felis;Felis catus\tEukaryota;Chordata;Mammalia;Carnivora;Felidae;Felis;Felis silvestris\t1.000;1.000;1.000;1.000;0.594;0.594;0.594\n",
      "\n",
      "AY495467.1\tSpecies\tEptesicus serotinus\tEptesicus fuscus\t0.516\tEukaryota;Chordata;Mammalia;Chiroptera;Vespertilionidae;Eptesicus;Eptesicus serotinus\tEukaryota;Chordata;Mammalia;Chiroptera;Vespertilionidae;Eptesicus;Eptesicus fuscus\t1.000;1.000;1.000;1.000;1.000;0.516;0.516\n",
      "\n",
      "KP142702.1\tSpecies\tCervus elaphus\tCervus nippon\t0.441\tEukaryota;Chordata;Mammalia;unknown;Cervidae;Cervus;Cervus elaphus\tEukaryota;Chordata;Mammalia;unknown;Cervidae;Cervus;Cervus nippon\t0.999;0.999;0.999;0.999;0.988;0.509;0.441\n",
      "\n",
      "Mislabeled sequences by rank:\n",
      "Class       :\t5\n",
      "Order       :\t6\n",
      "Family      :\t3\n",
      "Genus       :\t12\n",
      "Species     :\t8\n",
      "\n",
      "Total mislabels: 34 / 3.15 %\n",
      "\n",
      "Results were saved to: /home/working/Reference_Alignment/Mammals/SATIVA/12S.mis\n",
      "Execution log was saved to: /home/working/Reference_Alignment/Mammals/SATIVA/12S.log\n",
      "\n",
      "Analysis completed successfully, elapsed time: 433 seconds (151s reftree, 282s leave-one-out)\n",
      "\n"
     ]
    }
   ],
   "source": [
    "!./sativa/sativa.py -s alignment.phy -t tax_for_SATIVA.tax -x zoo -n 12S -o ./ -T 5 -v"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "SATIVA (among other things) writes a report to a text file."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      ";WARNING: The revised taxon name suggested here is not necessarily the one that has priority in nomenclature. \r\n",
      ";Our suggestion should only be taken as indicative of an affiliation to the same group, whose correct name must be determined \r\n",
      ";in an additional step according to the specific rules of nomenclature that apply to the studied organisms.\r\n",
      ";\r\n",
      ";SeqID\tMislabeledLevel\tOriginalLabel\tProposedLabel\tConfidence\tOriginalTaxonomyPath\tProposedTaxonomyPath\tPerRankConfidence\r\n",
      "HM637771.1\tClass\tAves\tMammalia\t0.995\tEukaryota;Chordata;Aves;Galliformes;Phasianidae;Gallus;Gallus gallus\tEukaryota;Chordata;Mammalia;Perissodactyla;Equidae;Equus;Equus caballus\t0.999;0.999;0.995;0.184;0.184;0.184;0.184\r\n",
      "KF471518.1\tClass\tAves\tMammalia\t0.991\tEukaryota;Chordata;Aves;Galliformes;Phasianidae;Gallus;Gallus gallus\tEukaryota;Chordata;Mammalia;Carnivora;Canidae;Canis;Canis lupus\t0.999;0.999;0.991;0.418;0.181;0.176;0.176\r\n",
      "KF471523.1\tClass\tAves\tMammalia\t0.951\tEukaryota;Chordata;Aves;Galliformes;Phasianidae;Gallus;Gallus gallus\tEukaryota;Chordata;Mammalia;unknown;Bovidae;Ovis;Ovis aries\t1.000;1.000;0.951;0.367;0.216;0.175;0.175\r\n",
      "KF484729.1\tClass\tAves\tMammalia\t0.946\tEukaryota;Chordata;Aves;Galliformes;Phasianidae;Gallus;Gallus gallus\tEukaryota;Chordata;Mammalia;Carnivora;Mustelidae;Martes;Martes martes\t1.000;1.000;0.946;0.425;0.362;0.100;0.100\r\n",
      "M13046.1\tClass\tAmphibia\tMammalia\t0.776\tEukaryota;Chordata;Amphibia;Anura;Pipidae;Xenopus;Xenopus laevis\tEukaryota;Chordata;Mammalia;Carnivora;Canidae;Canis;Canis lupus\t0.999;0.999;0.776;0.351;0.260;0.246;0.246\r\n",
      "JQ287755.1\tOrder\tRodentia\tInsectivora\t0.898\tEukaryota;Chordata;Mammalia;Rodentia;Muridae;Rattus;Rattus rattus\tEukaryota;Chordata;Mammalia;Insectivora;Talpidae;Talpa;Talpa europaea\t0.999;0.999;0.999;0.898;0.871;0.871;0.871\r\n",
      "JQ287761.1\tOrder\tRodentia\tunknown\t0.319\tEukaryota;Chordata;Mammalia;Rodentia;Muridae;Mus;Mus musculus\tEukaryota;Chordata;Mammalia;unknown;Bovidae;Ovis;Ovis aries\t0.999;0.999;0.999;0.319;0.307;0.196;0.196\r\n",
      "AF320512.1\tOrder\tCetacea\tCarnivora\t0.275\tEukaryota;Chordata;Mammalia;Cetacea;Physeteridae;Physeter;Physeter catodon\tEukaryota;Chordata;Mammalia;Carnivora;Canidae;Canis;Canis lupus\t0.999;0.999;0.951;0.275;0.258;0.250;0.250\r\n",
      "FM165432.1\tOrder\tPerissodactyla\tCarnivora\t0.254\tEukaryota;Chordata;Mammalia;Perissodactyla;Equidae;Equus;Equus caballus\tEukaryota;Chordata;Mammalia;Carnivora;Canidae;Canis;Canis lupus\t1.000;1.000;0.971;0.254;0.183;0.180;0.180\r\n",
      "AY830134.1\tOrder\tunknown\tCarnivora\t0.254\tEukaryota;Chordata;Mammalia;unknown;Bovidae;Bos;Bos taurus\tEukaryota;Chordata;Mammalia;Carnivora;Canidae;Canis;Canis lupus\t1.000;1.000;0.971;0.254;0.182;0.179;0.179\r\n",
      "AJ630114.1\tOrder\tunknown\tCarnivora\t0.254\tEukaryota;Chordata;Mammalia;unknown;Bovidae;Bos;Bos taurus\tEukaryota;Chordata;Mammalia;Carnivora;Canidae;Canis;Canis lupus\t1.000;1.000;0.971;0.254;0.183;0.180;0.180\r\n",
      "AJ554063.1\tFamily\tPhocoenidae\tDelphinidae\t0.901\tEukaryota;Chordata;Mammalia;Cetacea;Phocoenidae;Phocoena;Phocoena phocoena\tEukaryota;Chordata;Mammalia;Cetacea;Delphinidae\t0.999;0.999;0.999;0.999;0.901\r\n",
      "JQ287757.1\tFamily\tMuridae\tCricetidae\t0.831\tEukaryota;Chordata;Mammalia;Rodentia;Muridae;Rattus;Rattus rattus\tEukaryota;Chordata;Mammalia;Rodentia;Cricetidae;Myodes;Myodes glareolus\t0.999;0.999;0.999;0.977;0.831;0.773;0.773\r\n",
      "HM623883.1\tFamily\tBovidae\tCervidae\t0.505\tEukaryota;Chordata;Mammalia;unknown;Bovidae;Capra;Capra hircus\tEukaryota;Chordata;Mammalia;unknown;Cervidae;Capreolus;Capreolus capreolus\t0.999;0.999;0.999;0.998;0.505;0.456;0.456\r\n",
      "AB018586.1\tGenus\tStenella\tTursiops\t0.980\tEukaryota;Chordata;Mammalia;Cetacea;Delphinidae;Stenella;Stenella coeruleoalba\tEukaryota;Chordata;Mammalia;Cetacea;Delphinidae;Tursiops;Tursiops truncatus\t0.999;0.999;0.999;0.999;0.999;0.980;0.980\r\n",
      "EU685091.1\tGenus\tDelphinus\tTursiops\t0.980\tEukaryota;Chordata;Mammalia;Cetacea;Delphinidae;Delphinus;Delphinus delphis\tEukaryota;Chordata;Mammalia;Cetacea;Delphinidae;Tursiops;Tursiops truncatus\t0.999;0.999;0.999;0.999;0.999;0.980;0.980\r\n",
      "U13083.1\tGenus\tDelphinus\tTursiops\t0.979\tEukaryota;Chordata;Mammalia;Cetacea;Delphinidae;Delphinus;Delphinus delphis\tEukaryota;Chordata;Mammalia;Cetacea;Delphinidae;Tursiops;Tursiops truncatus\t0.999;0.999;0.999;0.999;0.999;0.979;0.979\r\n",
      "Y08208.1\tGenus\tHydropotes\tRangifer\t0.961\tEukaryota;Chordata;Mammalia;unknown;Cervidae;Hydropotes;Hydropotes inermis\tEukaryota;Chordata;Mammalia;unknown;Cervidae;Rangifer;Rangifer tarandus\t1.000;1.000;1.000;1.000;1.000;0.961;0.961\r\n",
      "HM623877.1\tGenus\tCapra\tOvis\t0.632\tEukaryota;Chordata;Mammalia;unknown;Bovidae;Capra;Capra hircus\tEukaryota;Chordata;Mammalia;unknown;Bovidae;Ovis;Ovis aries\t0.999;0.999;0.999;0.999;0.999;0.632;0.632\r\n",
      "DQ867007.1\tGenus\tCapra\tOvis\t0.632\tEukaryota;Chordata;Mammalia;unknown;Bovidae;Capra;Capra hircus\tEukaryota;Chordata;Mammalia;unknown;Bovidae;Ovis;Ovis aries\t0.999;0.999;0.999;0.999;0.999;0.632;0.632\r\n",
      "JQ287756.1\tGenus\tRattus\tMus\t0.553\tEukaryota;Chordata;Mammalia;Rodentia;Muridae;Rattus;Rattus rattus\tEukaryota;Chordata;Mammalia;Rodentia;Muridae;Mus;Mus musculus\t0.999;0.999;0.999;0.999;0.864;0.553;0.553\r\n",
      "JQ287754.1\tGenus\tRattus\tMus\t0.553\tEukaryota;Chordata;Mammalia;Rodentia;Muridae;Rattus;Rattus rattus\tEukaryota;Chordata;Mammalia;Rodentia;Muridae;Mus;Mus musculus\t0.999;0.999;0.999;0.999;0.864;0.553;0.553\r\n",
      "JQ287752.1\tGenus\tRattus\tMus\t0.553\tEukaryota;Chordata;Mammalia;Rodentia;Muridae;Rattus;Rattus rattus\tEukaryota;Chordata;Mammalia;Rodentia;Muridae;Mus;Mus musculus\t0.999;0.999;0.999;0.999;0.864;0.553;0.553\r\n",
      "KF111725.1\tGenus\tEptesicus\tVespertilio\t0.510\tEukaryota;Chordata;Mammalia;Chiroptera;Vespertilionidae;Eptesicus;Eptesicus serotinus\tEukaryota;Chordata;Mammalia;Chiroptera;Vespertilionidae;Vespertilio;Vespertilio murinus\t1.000;1.000;1.000;1.000;1.000;0.510;0.510\r\n",
      "D34627.1\tGenus\tCervus\tCapreolus\t0.318\tEukaryota;Chordata;Mammalia;unknown;Cervidae;Cervus;Cervus nippon\tEukaryota;Chordata;Mammalia;unknown;Cervidae;Capreolus;Capreolus capreolus\t0.999;0.999;0.999;0.999;0.993;0.318;0.318\r\n",
      "JF339973.1\tGenus\tGlobicephala\tTursiops\t0.268\tEukaryota;Chordata;Mammalia;Cetacea;Delphinidae;Globicephala;Globicephala melas\tEukaryota;Chordata;Mammalia;Cetacea;Delphinidae;Tursiops;Tursiops truncatus\t0.999;0.999;0.999;0.664;0.605;0.268;0.268\r\n",
      "NC_001700.1\tSpecies\tFelis catus\tFelis silvestris\t0.930\tEukaryota;Chordata;Mammalia;Carnivora;Felidae;Felis;Felis catus\tEukaryota;Chordata;Mammalia;Carnivora;Felidae;Felis;Felis silvestris\t1.000;1.000;1.000;1.000;0.930;0.930;0.930\r\n",
      "Y08503.1\tSpecies\tFelis catus\tFelis silvestris\t0.930\tEukaryota;Chordata;Mammalia;Carnivora;Felidae;Felis;Felis catus\tEukaryota;Chordata;Mammalia;Carnivora;Felidae;Felis;Felis silvestris\t1.000;1.000;1.000;1.000;0.930;0.930;0.930\r\n",
      "HQ529667.1\tSpecies\tMyotis brandtii\tMyotis lucifugus\t0.611\tEukaryota;Chordata;Mammalia;Chiroptera;Vespertilionidae;Myotis;Myotis brandtii\tEukaryota;Chordata;Mammalia;Chiroptera;Vespertilionidae;Myotis;Myotis lucifugus\t0.999;0.999;0.999;0.999;0.997;0.987;0.611\r\n",
      "HQ529670.1\tSpecies\tMyotis brandtii\tMyotis lucifugus\t0.611\tEukaryota;Chordata;Mammalia;Chiroptera;Vespertilionidae;Myotis;Myotis brandtii\tEukaryota;Chordata;Mammalia;Chiroptera;Vespertilionidae;Myotis;Myotis lucifugus\t0.999;0.999;0.999;0.999;0.997;0.987;0.611\r\n",
      "HQ529663.1\tSpecies\tMyotis brandtii\tMyotis lucifugus\t0.611\tEukaryota;Chordata;Mammalia;Chiroptera;Vespertilionidae;Myotis;Myotis brandtii\tEukaryota;Chordata;Mammalia;Chiroptera;Vespertilionidae;Myotis;Myotis lucifugus\t0.999;0.999;0.999;0.999;0.997;0.987;0.611\r\n",
      "KM224283.1\tSpecies\tFelis catus\tFelis silvestris\t0.594\tEukaryota;Chordata;Mammalia;Carnivora;Felidae;Felis;Felis catus\tEukaryota;Chordata;Mammalia;Carnivora;Felidae;Felis;Felis silvestris\t1.000;1.000;1.000;1.000;0.594;0.594;0.594\r\n",
      "AY495467.1\tSpecies\tEptesicus serotinus\tEptesicus fuscus\t0.516\tEukaryota;Chordata;Mammalia;Chiroptera;Vespertilionidae;Eptesicus;Eptesicus serotinus\tEukaryota;Chordata;Mammalia;Chiroptera;Vespertilionidae;Eptesicus;Eptesicus fuscus\t1.000;1.000;1.000;1.000;1.000;0.516;0.516\r\n",
      "KP142702.1\tSpecies\tCervus elaphus\tCervus nippon\t0.441\tEukaryota;Chordata;Mammalia;unknown;Cervidae;Cervus;Cervus elaphus\tEukaryota;Chordata;Mammalia;unknown;Cervidae;Cervus;Cervus nippon\t0.999;0.999;0.999;0.999;0.988;0.509;0.441\r\n"
     ]
    }
   ],
   "source": [
    "!cat 12S.mis"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 2",
   "language": "python",
   "name": "python2"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 2
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython2",
   "version": "2.7.6"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 0
}
